<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2021-02-16" />

<title>Section 3: Proteomic signatures of trisomy12</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.17/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CLL Proteome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.embl.de/jlu/Proteomics">
    <span class="fas fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Section 3: Proteomic signatures of trisomy12</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2021-02-16</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CLLproteomics_publish_revision/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-25
</li>
<li>
unnamed-chunk-27
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>manuscript_S3_trisomy12_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/manuscript_S1_Overview_cache/
    Ignored:    analysis/manuscript_S2_genomicAssociation_cache/
    Ignored:    analysis/manuscript_S3_trisomy12_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/.trisomy12_norm.pdf
    Untracked:  analysis/bufferComplexViolin.pdf
    Untracked:  analysis/buffer_Tri12vsTri19.pdf
    Untracked:  analysis/cohortComposition_all.pdf
    Untracked:  analysis/heatmap_tri12_circle.pdf
    Untracked:  analysis/manuscript_S1_Overview.Rmd
    Untracked:  analysis/manuscript_S2_genomicAssociation.Rmd
    Untracked:  analysis/manuscript_S3_trisomy12.Rmd
    Untracked:  analysis/manuscript_S4_trisomy19.Rmd
    Untracked:  analysis/manuscript_S5_IGHV.Rmd
    Untracked:  analysis/manuscript_S6_del11q.Rmd
    Untracked:  analysis/manuscript_S7_SF3B1.Rmd
    Untracked:  analysis/manuscript_S8_drugResponse_Outcomes.Rmd
    Untracked:  analysis/manuscript_S9_STAT2.Rmd
    Untracked:  analysis/plot_PC1_PC2.pdf
    Untracked:  analysis/timsTOF_validate.Rmd
    Untracked:  analysis/tri12_transEnrich.pdf
    Untracked:  analysis/trisomy12_chr_summary.pdf
    Untracked:  code/utils.R
    Untracked:  data/Annotation file March 2021.xlsx
    Untracked:  data/CAS9results.xlsx
    Untracked:  data/CNV_onChrom.RData
    Untracked:  data/ComplexParticipantsPubMedIdentifiers_human.txt
    Untracked:  data/Fig1A.png
    Untracked:  data/IGLV321_SupplementalTables_R2.xlsx
    Untracked:  data/MOFAout.RData
    Untracked:  data/MOFAout_atLeast3.RData
    Untracked:  data/STATexprPCR.xlsx
    Untracked:  data/Western_blot_results_20210309_short.csv
    Untracked:  data/Western_blot_results_separate_blots.xlsx
    Untracked:  data/allComplexes.txt
    Untracked:  data/ddsrna_enc.RData
    Untracked:  data/exprCNV_enc.RData
    Untracked:  data/geneAnno.RData
    Untracked:  data/gmts/
    Untracked:  data/ic50.RData
    Untracked:  data/mofaIn.RData
    Untracked:  data/mofaIn_atLeast3.RData
    Untracked:  data/patMeta_enc.RData
    Untracked:  data/pepCLL_lumos_enc.RData
    Untracked:  data/protMOFA.RData
    Untracked:  data/proteins_in_complexes
    Untracked:  data/proteomic_LUMOS_2pep_enc.RData
    Untracked:  data/proteomic_explore_enc.RData
    Untracked:  data/proteomic_independent_all_enc.RData
    Untracked:  data/proteomic_independent_enc.RData
    Untracked:  data/proteomic_timsTOF_enc.RData
    Untracked:  data/screenData_enc.RData
    Untracked:  data/setToPathway.txt
    Untracked:  data/survival_enc.RData
    Untracked:  output/MSH6_splicing.svg
    Untracked:  output/SUGP1_splicing.svg
    Untracked:  output/deResList.RData
    Untracked:  output/deResListBatch2.RData
    Untracked:  output/deResListRNA.RData
    Untracked:  output/deResListRNA_allGene.RData
    Untracked:  output/deResList_WBC.RData
    Untracked:  output/deResList_batch1.RData
    Untracked:  output/deResList_batch3.RData
    Untracked:  output/deResList_independent.RData
    Untracked:  output/deResList_timsTOF.RData
    Untracked:  output/dxdCLL.RData
    Untracked:  output/dxdCLL2.RData
    Untracked:  output/exprCNV.RData
    Untracked:  output/geneAnno.RData
    Untracked:  output/int_pairs.csv
    Untracked:  output/lassoResults_CPS.RData
    Untracked:  output/resOutcome_batch1.RData
    Untracked:  output/resOutcome_batch13.RData
    Untracked:  output/resOutcome_batch2.RData
    Untracked:  output/resOutcome_batch3.RData

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/analysisSF3B1.Rmd
    Deleted:    analysis/comparePlatforms.Rmd
    Deleted:    analysis/compareProteomicsRNAseq.Rmd
    Deleted:    analysis/correlateCLLPD.Rmd
    Deleted:    analysis/correlateGenomic.Rmd
    Deleted:    analysis/correlateGenomic_removePC.Rmd
    Deleted:    analysis/correlateMIR.Rmd
    Deleted:    analysis/correlateMethylationCluster.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/predictOutcome.Rmd
    Deleted:    analysis/processProteomics_LUMOS.Rmd
    Deleted:    analysis/processProteomics_timsTOF.Rmd
    Deleted:    analysis/qualityControl_LUMOS.Rmd
    Deleted:    analysis/qualityControl_timsTOF.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>#Load packages and datasets</p>
<div id="overview-of-differentially-expressed-proteins-realted-to-trisomy12" class="section level1">
<h1>Overview of differentially expressed proteins realted to trisomy12</h1>
<div id="a-table-of-associations-with-5-fdr" class="section level2">
<h2>A table of associations with 5% FDR</h2>
<pre class="r"><code>resList &lt;- filter(resList, Gene == &quot;trisomy12&quot;) %&gt;%
  #mutate(adj.P.Val = adj.P.global) %&gt;% #use global corrected P-value
  mutate(Chr = rowData(protCLL[id,])$chromosome_name)
resList %&gt;% filter(adj.P.Val &lt;= 0.05) %&gt;% 
  select(name, Chr,logFC, P.Value, adj.P.Val) %&gt;%
  mutate_if(is.numeric, formatC, digits=2) %&gt;%
  DT::datatable()</code></pre>
<div id="htmlwidget-16687c8f6ec6bc0aecc6" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-16687c8f6ec6bc0aecc6">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831","832","833","834","835","836","837","838","839","840","841","842","843","844","845","846","847","848","849","850","851","852","853","854","855","856","857","858","859","860","861","862","863","864","865","866","867","868","869","870","871","872","873","874","875","876","877","878","879","880","881","882","883","884","885","886","887","888","889","890","891","892","893","894","895","896","897","898","899","900","901","902","903","904","905","906","907","908","909","910","911","912","913","914","915","916","917","918","919","920","921","922","923","924","925","926","927","928","929","930","931","932","933","934","935","936","937","938","939","940","941","942","943","944","945","946","947","948","949","950","951","952","953","954","955","956","957","958","959","960","961","962","963","964","965","966","967","968","969","970","971","972","973","974","975","976","977","978","979","980","981","982","983","984","985","986","987","988","989","990","991","992","993","994","995","996","997","998","999","1000","1001","1002","1003","1004","1005","1006","1007","1008","1009","1010","1011","1012","1013","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1025","1026","1027","1028","1029","1030","1031","1032","1033","1034","1035","1036","1037","1038","1039","1040","1041","1042","1043","1044","1045","1046","1047","1048","1049","1050","1051","1052","1053","1054","1055"],["PSMD9","PPP1R12A","CS","USP5","ARHGDIB","PTPN11","EIF2B1","EEA1","OSBPL8","TAOK3","MYG1","TIGAR","GCN1","RAB21","SLC25A3","DERA","ANKRD13A","CAND1","ALDH16A1","RPAP3","MCTS1","PTGES3","EML2","STRAP","PA2G4","SHMT2","ESYT1","TSFM","PGAM5","PRKAG1","PTPN6","GOLGA3","DIABLO","LDHB","TRIAP1","ATP2A2","STAT2","UBXN7","ACYP1","TOP1","RAB2B","PYCARD","VAT1","EIF4B","FAM114A2","ANP32E","C11orf68","DGKA","ATP2B1","METTL7A","BROX","CNPY2","DNM1L","VCL","VPS28","RASGRP2","SCCPDH","SRSF9","ATP6V0A1","SMARCA4","CFDP1","UGDH","ALDH5A1","RAN","CUL3","IRAK4","IRF4","ABHD14B","LANCL1","VAV1","PMF1","NEMP1","ITPRIP","PUS1","RECQL","RFXAP","CNOT9","ABCB10","CMAS","SAMHD1","CENPB","PTK2B","GIT2","LEMD3","TRIM25","BSG","NAP1L1","DSN1","CNOT1","PARP12","FLNA","SCRN2","SP1","XPOT","VASP","TBL2","FRYL","RHOG","ERP29","MARS1","HSD17B8","UBE2N","BZW1","NUMA1","NIPSNAP2","CBR4","ARHGAP45","PIH1D1","PNP","NELFA","SLC25A20","CHDH","PGM2","DNPH1","ZNF384","EHD4","DRG1","DENR","CRAT","AKT2","YARS2","DOK2","USP15","CBX5","SLTM","NOL7","BCL10","MLEC","NMNAT1","ARF6","PRRC1","SART3","CASP1","CORO1C","STAT6","METTL3","GCLC","HNRNPLL","ACAP1","RBM4B","GSK3A","LTA4H","MPG","NCOR1","GAPDH","PRKD2","EEF2","PWP1","SACM1L","MYH9","RUNX3","PYM1","TCEAL4","RAB35","RAC2","OAS2","ESYT2","ACLY","UBA6","SCAF11","MYL6","LBR","CENPC","RFX5","WNK1","HDDC3","SNX9","TREX1","GSTZ1","WDR74","PFN1","TERF2IP","GLRX","ZMYM3","SGTB","ZFR","PMVK","GLTP","MAPK3","TIAL1","RBPJ","GSTO1","FLII","RAB5B","PITPNA","ST6GAL1","RBM33","MIER1","ACY1","TES","ARHGAP4","PAXX","RABGAP1L","HNRNPA3","HP1BP3","NHP2","PALM2AKAP2","PDP1","RPRD2","SCYL2","SNX12","CORO1A","TLN1","CD2AP","ME2","SMARCC1","ZZEF1","CHID1","JUND","RIDA","HMGCL","IKBKG","GPD1L","PKM","SRI","CAPZA1","PDPR","LDHA","RAB3GAP2","GMPPB","C2CD5","RBM26","LRRFIP1","MRNIP","PPP4R3A","RAB4B","GNG7","BCL2L13","SH3GL1","SLC25A12","POLD3","PUF60","AGFG2","MPI","DHX16","HERC4","BIN2","SEC24C","ZNRF2","DPP9","ITGB2","IQGAP1","ACAT2","TERF2","RNF213","C9orf64","GM2A","TBL1XR1","ATP2A3","ACTN4","CNN2","GCHFR","MYEF2","RUSF1","GPD2","PGP","ANKFY1","TMA16","RGS14","CMC4","TOMM70","NOLC1","NSL1","CYFIP1","NELFE","TBC1D5","MATR3","ZNF512","CD44","TASOR","MAP2K3","ITGB1","PRKAA1","GMPR2","MALT1","STK38","EPM2AIP1","STRBP","AMPD3","TATDN1","PDXK","PFKP","SMNDC1","TANK","RB1","PPIE","TRIM38","METTL16","CAPZB","RANBP3","ZNF326","EMC2","OVCA2","STX7","ETF1","ENPP4","SF3B4","KPNA1","IFI16","SEPTIN11","SMARCA2","ISY1","GAR1","RBMXL1","GRK6","BCL7C","DCLK2","TMOD3","BCKDHA","MACF1","CAPRIN1","ANXA7","CSTF1","TPM4","SP4","MTX2","TWF1","ITPR2","CAP1","SNX5","STARD7","CAMK1D","WASHC5","EMG1","DENND2D","TKFC","LUC7L2","MAP4","STAM","C1orf35","HSP90B1","UBE2O","CLIP1","RBM39","DDX27","GLO1","GLIPR2","ARHGAP30","BPGM","CD72","ZHX2","NME3","JPT2","ABCF2","ACOT13","MTAP","C2orf49","MSN","NLN","AK1","PCYT2","TRAFD1","TUBA4A","SNX2","BIN1","ZC3H11A","NCL","SERPINH1","PHB2","RUNX1","ARHGEF1","NIT1","BDH1","MAP3K5","SZRD1","RPL5","ARHGEF18","ADNP","ZC3H14","EIF5B","HIRA","CD200","FGR","PHB","CDKN1B","CD27","SBF1","CSTF2","METAP2","DOCK11","TPI1","PTGR2","IAH1","FUBP1","HMOX1","CTSH","GATAD2A","GAK","DDAH2","ARHGEF6","RPS28","DNAJA1","WASHC4","NIT2","STIM2","CTPS1","ALDH18A1","GPATCH8","CCDC86","PCBP2","AIP","ITPA","TLK1","NAPRT","SFPQ","HNRNPH3","NCOA3","SLC25A19","THYN1","LPXN","PCCA","SAFB","TRIM22","CD74","BANK1","SAP130","SFXN3","NR2C2","RALA","PPP4R2","GRAP","DDX24","MMAB","DDX54","GYS1","DDX47","IGF2R","PPP1R18","FGD2","TXNRD1","RAB8A","ELMO2","ISG20","THRAP3","CPSF7","RBM4","FERMT3","CPSF2","FYTTD1","LONP1","ACTR2","DENND1C","NADK","VARS1","TPMT","RAPGEF1","DCPS","DEK","NUDCD2","CYB5R2","AHCY","GNB2","DDX18","DIP2B","ARPP19","LCP1","USP24","CNOT3","HNRNPF","SUN2","SNX3","CD180","PSMC3","INPP5D","HNRNPL","TRMT6","VAMP8","PTPN7","MIS12","RBM34","NOP9","PFKFB2","WASHC3","MTDH","NFKBIB","XPNPEP1","TP53I3","QRICH1","MTA1","PGM2L1","WDR1","DNAJC8","RASA1","CORO7","SMAD2","RIPOR2","DKC1","PRKCB","LSS","C12orf43","RSBN1L","MORC3","CCT2","RRP1","ARHGAP24","CLINT1","CTDP1","CABIN1","NASP","PIK3CD","CYP20A1","CLPP","ISCU","IKZF1","GMFG","NDUFAF2","ZNF740","TOMM22","EXOSC4","ANXA11","HIP1R","TMPO","MAP4K1","TPP2","RBM15","FIBP","SMARCC2","CPPED1","MOB3A","G3BP1","CCT6A","MIDEAS","HSPD1","CAB39","HSP90AA1","ATP5F1B","SEC23A","CREB1","WAPL","APBB1IP","DNM2","JAK1","AIDA","ATXN2L","BMS1","FOXO1","RNMT","PHIP","NCKAP1L","PCBD2","KLHL36","ERGIC1","STAMBPL1","PSMD1","SCAF8","GRB2","HMBS","SERBP1","SCLY","WRNIP1","RBM14","MYCBP2","CPSF3","MYO1G","HTATSF1","PPP3R1","SON","STMN1","HEBP2","INPP5B","RRP15","TBC1D10B","PIP4K2A","RPL6","DDX17","NUB1","RPL15","HNRNPK","OGA","DNAJC7","DNMBP","KAT7","FCHSD1","INTS3","MDC1","SGPP1","ATP5F1C","GNAI3","DLGAP4","SLC25A5","PUM1","SRSF11","ALDH1B1","RPA2","GNAI2","PSMD5","RPLP2","APIP","PCCB","SUB1","GMDS","METTL14","PES1","ISOC1","GDI2","MGST3","ASH2L","CD79A","EPN1","ELAVL1","TSC22D4","NR3C1","RDH11","RPL7L1","ATP1A1","EXOSC5","UGGT1","GPATCH11","RPS7","CCDC47","AGK","NAPA","SIRT3","FCSK","ZNF638","CCT5","TMX1","AQR","CD79B","RFC4","DDX39B","PABPN1","UBR1","ARIH1","PLPBP","TRAPPC11","FNBP4","NARS1","ATF7IP","RPA1","ACIN1","TRMT1L","ECI1","RACK1","CLIC1","HNRNPM","RANBP1","AP3S1","GEMIN5","SSBP1","TPD52","GLS","GMIP","RBMX","EBNA1BP2","ARPC5L","U2SURP","MYO18A","EIF4E","SIN3A","ATG7","MECR","LEF1","SLAMF6","INO80","ALDOC","PPP6C","PGK1","PGD","VPS33A","GADD45GIP1","RNF40","HARS2","RPUSD2","TARDBP","BCKDHB","DOCK2","DBNL","GAPVD1","SRP54","RPS3","PLIN3","ANXA4","LPGAT1","DNPEP","PBRM1","KIAA1143","SUMF2","ARPC4","ADAR","PHYKPL","PARN","AKR7A2","RASAL3","RTF2","FTO","ILF2","RPS12","MYO1F","NMI","SMS","RBM22","OSGEP","SELENOF","MNDA","SLC25A24","ADPRH","POLDIP3","TMEM63A","AKR1A1","PPP3CB","UBA7","SF3B3","RPS13","ARL6IP5","ERO1A","CHTOP","CSTF3","RPS9","MOGS","CDC73","EIF4A1","RBM12","FASN","ATG3","PSIP1","PMPCB","CCT3","TBC1D10C","WDR12","EIF3A","ROCK1","PSMD12","ECI2","PHAX","CHMP4B","PGGT1B","USP9X","PRPF38A","SIPA1L1","CSNK2A2","NEK7","ATP5PF","LYPLA2","CNP","SLC3A2","CYRIA","TRMT61A","ADSS2","OTULIN","RAB43","FKBP15","DPYSL2","FIP1L1","EIF3K","SIPA1","CCDC88B","CIRBP","SRSF1","DCTD","ARHGAP9","NCOA7","DOCK8","CKAP4","ARPC2","RPRD1B","RUFY1","WDR89","PTPRC","CTSZ","CD22","CYRIB","SIT1","S100A4","CNDP2","PPP1CB","BRD2","PCIF1","PARP14","PSMC6","BAX","RAB14","COPA","ACTR8","RABEP1","KIF2A","TP53BP1","CELF1","FAM120A","DUT","CPSF4","WDR46","SND1","ATM","STX5","PAFAH1B3","GCA","EIF3E","IGHM","RIF1","ACTR3","SSU72","TATDN3","VPS4B","ACTBL2","AFG3L2","MAK16","RPS2","AIMP1","RAB29","MARCHF5","KYNU","TOMM40","MARK2","RABGEF1","DYNLL2","CCT8","LIMD1","DNAJC19","STIM1","NDUFB5","MFF","PCYOX1","SRRT","SH3KBP1","CDK9","ANP32A","NT5DC1","SNX6","PSMD2","ACTR1A","WDR81","ARPC1B","HGS","CDC5L","ALDH2","ATP1B3","VDAC2","EIF3M","IPO9","ARFGAP2","TMED8","POLD2","SPTBN2","FAM3C","SMAP2","TSSC4","MRPS16","HDAC6","GNS","DNTTIP1","GRWD1","HMGN4","MDH1","NOP10","HSD17B11","C7orf50","RPL12","C1QBP","RPL23A","PTPRCAP","SLC9A3R1","EXOSC6","GDAP1","ALDH6A1","GLG1","SCAF1","MIA2","OLA1","EZR","NELFB","CEP250","ECPAS","MSH2","SRPK2","HDDC2","CHERP","GET3","ZMYND11","PSMD11","COPB2","ADD1","PEPD","PSAP","NRDC","ADO","DHX9","NOP56","EHD1","EIF3B","TFRC","FNTA","ATP5F1A","BRD4","NIPBL","GPX4","SLC4A1AP","THAP11","PHF5A","POLR1G","SEPTIN9","IL16","DYNC1H1","ERLIN2","EIF3F","CZIB","GATAD2B","TNPO2","SFXN1","POFUT1","RABL6","PEF1","HNRNPA2B1","PYGB","CUL4B","DOCK10","CAPN2","ZNF800","RPL21","CBX7","NNT","NFATC2","GLOD4","PRCC","SRSF10","RPS21","UBA2","ATOX1","NSD3","VPS16","GABPA","CBX3","FLNB","PITHD1","EXOSC1","YTHDC1","RANGAP1","SHOC2","YWHAQ","RPS15A","DUSP23","RGS10","HECTD3","NCF4","LYAR","IKBKB","SQOR","SEC23B","DCXR","RCL1","MS4A1","ZBTB4","IQGAP2","PLEKHO1","ORC5","NIP7","PSMD13","GMFB","PSMC2","HNRNPR","CCDC12","PPP1R2","HAT1","PPIG","DSTN","CRYL1","HNRNPUL2","CRIPT","SYMPK","TP53RK","PCBP1","PPP3CC","TIMM8A","DDX50","NECAP2","LXN","RAD23B","ZNF22","TRMT10C","RPL9","CYFIP2","SDCBP","PPP6R2","SF3B6","FH","SERPINB6","EEF1AKNMT","MYO9B","PLEKHO2","LIMS1","EIF2S2","GNL3","RPS11","ARPC3","ELP3","HNRNPH1","MAP2K1","UBXN1","RPL14","PRPS2","SRSF5","DDX23","CWC15","POGLUT1","RPS6","PDXP","GRPEL1","AK3","LCMT1","COPG1","PAPOLA","C17orf49","RPL7A","DIMT1","ZGPAT","BLOC1S4","NME1","PPAT","TMED7","SRP68","MAD1L1","SRRM2","PPP3CA","RAB3GAP1","HNRNPU","PPP5C","RPS4X","YTHDF2","MPHOSPH10","XAB2","OSBPL10","ZNF428","PRDX2","PSMC4","PPP1R7","RPL17","YARS1","UBAP2L","SYNE1","RRM2B","GABARAPL2","EML4","RPLP1","SNRPA1","EMC1","TIMM44","FCRLA","TGOLN2","EFHD2","BOP1","THOC6","NELFCD","LAMP1","CCDC127","RPA3","QARS1","LYPLAL1","TMUB1","EDF1","ARF5","NPM1","ADH5","TCP1","RFC1","IPO5"],["12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","12","19","12","X","12","19","12","12","12","12","12","12","12","12","12","12","12","12","12","12","3","14","20","14","16","17","12","5","1","11","12","12","12","1","12","12","10","8","11","1","12","17","19","16","4","6","12","2","12","6","3","2","19","1","12","10","12","12","13","2","1","12","20","20","8","12","12","17","19","12","20","16","7","X","17","12","12","19","7","4","11","12","12","6","12","2","11","7","4","19","19","14","4","3","3","4","6","12","15","22","12","9","19","12","8","12","12","15","6","1","12","1","14","5","12","11","12","12","14","6","2","17","11","19","12","16","17","12","19","19","12","3","22","1","12","X","12","22","12","7","17","4","12","12","1","4","1","12","15","6","3","14","11","17","16","5","X","5","5","1","12","16","10","4","10","17","12","17","3","7","1","3","7","X","9","1","2","1","5","9","8","1","12","X","16","9","6","18","3","17","11","19","8","1","X","3","15","7","1","16","11","1","3","12","13","2","5","14","19","19","22","19","2","11","8","7","15","6","10","12","10","7","19","21","15","6","16","17","9","5","3","17","19","19","15","15","16","2","16","17","4","5","X","3","10","1","15","6","3","5","2","11","3","17","10","5","14","18","6","3","9","11","8","21","10","10","2","13","1","6","17","1","19","1","8","17","6","5","6","1","3","1","4","9","3","4","1","5","16","4","15","19","1","11","10","20","19","7","2","12","12","1","20","2","10","8","12","1","11","7","3","10","1","12","17","12","20","20","6","9","1","7","9","8","16","16","7","6","9","2","X","5","9","17","12","2","5","2","1","2","11","12","21","19","1","3","6","1","1","19","20","14","2","22","3","1","17","12","12","22","X","12","X","12","14","2","1","22","15","19","4","6","X","19","9","12","3","4","1","10","17","11","12","11","20","2","8","1","10","20","17","11","11","13","19","11","5","4","2","10","3","7","3","17","14","12","12","19","12","6","6","6","12","19","20","15","1","11","11","11","14","3","19","2","19","1","6","6","9","11","6","5","11","20","7","2","12","15","13","1","19","10","22","6","5","11","2","19","20","2","1","17","1","14","1","12","8","19","10","2","3","14","11","4","1","5","16","18","6","X","16","21","12","7","21","12","21","4","5","18","22","1","1","2","19","12","7","19","5","12","22","8","10","12","12","19","13","1","11","12","16","19","5","7","14","2","2","14","12","14","2","10","10","19","1","1","16","10","13","18","6","12","5","16","5","10","2","6","17","11","1","2","6","11","13","2","7","X","2","21","1","6","1","1","16","10","12","22","7","3","9","10","17","10","17","5","1","6","14","10","1","20","X","1","1","9","1","3","9","11","11","3","5","6","4","22","5","10","1","8","19","19","19","7","5","14","6","1","19","2","2","2","17","7","19","11","16","2","5","14","15","17","3","6","14","15","15","8","4","11","18","12","17","14","1","16","5","6","19","22","5","5","7","8","2","19","X","1","9","3","17","4","15","3","1","4","1","15","17","9","X","1","12","19","16","5","15","1","6","5","7","9","14","11","19","2","1","2","3","3","7","3","1","5","16","1","19","20","16","1","6","19","2","X","5","14","1","1","1","3","22","1","1","10","3","16","11","3","14","1","11","19","2","1","17","20","17","3","9","7","1","11","2","10","18","17","6","5","20","5","X","1","14","16","1","21","1","17","11","2","14","1","5","3","9","8","4","19","11","11","19","17","4","12","6","9","12","2","20","5","14","1","20","19","8","9","1","18","2","6","20","3","14","19","9","1","3","17","5","15","11","9","15","7","6","7","11","11","19","2","8","14","2","2","1","1","18","5","18","8","16","4","1","10","2","19","11","7","17","21","3","3","11","3","2","2","7","X","9","15","6","14","3","10","17","7","17","6","12","3","10","11","1","11","14","7","11","7","1","11","10","X","12","20","19","6","2","15","4","7","9","17","17","11","17","16","8","14","16","19","14","2","6","9","20","9","2","7","6","19","19","10","17","3","4","19","10","1","10","1","20","11","7","3","8","18","19","5","19","2","16","22","19","17","15","14","8","11","1","1","19","5","20","9","1","7","20","X","2","1","7","13","22","5","20","17","1","1","20","19","5","8","20","21","7","3","1","10","4","22","10","2","16","1","10","1","22","4","8","15","20","17","9","11","17","5","1","7","16","11","14","7","1","3","3","2","2","20","13","11","2","19","20","2","8","X","10","1","3","9","10","3","4","5","8","22","2","1","6","1","19","15","2","20","3","19","12","8","5","15","11","3","X","14","12","11","3","9","22","4","9","16","3","14","17","9","5","20","4","17","4","5","17","7","16","4","2","1","19","X","1","2","19","3","19","19","19","2","18","1","1","6","8","16","2","15","15","1","19","1","2","1","8","16","20","13","5","7","3","1","7","9","7","5","4","6","4","13"],["0.58","0.65","0.67","0.5","0.79","0.57","0.72","0.77","0.75","0.75","0.71","  1","0.5","0.48","0.57","0.69","0.86","0.45","0.54","0.43","0.49","0.46","-0.73","0.4","0.55","0.9","0.66","0.63","0.7","0.5","0.79","0.8","0.54","0.76","0.94","0.56","0.67","-0.46","-0.73","-0.45","-0.56","0.79","-0.84","0.53","-0.33","-0.38","-0.4","0.48","0.94","0.6","-0.61","0.46","0.54","0.99","0.92","0.8","-0.89","0.35","-0.99","0.54","-0.47","-0.73","-0.71","0.27","0.39","0.58","-0.69","-0.62","-0.46","0.63","-0.46","0.71","0.92","0.57","0.61","-0.58","0.24","-0.88","0.43","1.3","-0.41","0.6","0.58","0.57","0.63","0.52","0.48","-0.39","0.29","0.67","0.56","-0.73","0.6","0.41","0.58","-0.53","0.55","0.44","0.52","0.38","-0.66","0.42","0.4","-0.39","-0.4","-0.68","0.36","0.37","0.93","-0.27","0.68","-1.1","-0.32","-0.97","0.49","-0.48","0.28","0.45","-0.74","0.39","0.54","1.1","0.58","0.46","-0.26","-0.42","0.58","0.32","-0.44","0.46","-0.43","0.29","0.98","0.61","0.4","-0.31","-0.59","-0.82","0.54","-0.5","0.27","0.5","-0.33","-0.31","0.4","0.52","0.22","0.46","0.34","0.38","0.56","0.53","-0.78","0.62","0.56","0.5","0.43","-0.37","0.29","0.4","0.35","0.67","-0.42","-0.47","0.52","-0.61","-0.53","-0.46","-0.75","-0.28","0.33","-0.29","1.3","-0.34","-0.72","-0.35","-0.39","0.47","-0.35","-0.23","-0.46","-0.36","0.4","0.41","0.44","-0.96","-0.29","-0.41","-0.88","0.98","0.35","0.83","-0.64","-0.22","-0.31","-0.3","-0.64","-0.37","-0.25","0.44","-0.34","0.33","0.29","-0.25","0.43","-0.41","0.42","-0.41","-0.47","-0.48","-0.35","0.27","0.73","0.31","-0.41","0.29","-0.37","0.37","-0.32","0.33","0.39","-0.26","0.44","-0.53","-0.32","0.38","0.94","-0.41","0.36","-0.3","-0.39","-0.19","-0.55","-0.36","-0.24","-0.3","0.68","-0.2","0.39","0.42","0.86","0.39","-0.4","-0.3","0.3","-0.35","-0.71","-0.32","0.47","0.28","0.36","-0.72"," -1","-0.48","-0.51","0.36","0.41","-0.46","0.38","-0.44","-0.26","-0.38","-0.29","0.42","-0.21","-0.21","-0.26","-0.39","-0.41","-0.29","0.35","0.84","0.32","-0.31","0.45","0.41","-0.65","-0.47","0.7","-0.37","0.62","-0.56","-0.28","-0.29","0.9","-0.22","0.4","-0.35","0.23","-0.23","-0.3","-0.32","0.32","-0.34","0.2","-0.47","-0.29","0.37","-0.22","-0.49","-0.19","-0.2","-0.26","-0.38","0.53","-0.48","-0.87","0.26","-0.33","0.48","-0.16","-0.25","-0.28","0.43","-0.46","-0.39","0.67","0.7","0.22","0.31","-0.3","0.63","0.45","0.2","0.54","-0.35","-0.23","-0.4","-0.35","-0.36","0.3","0.28","0.42","-0.15","-0.28","-0.57","0.55","0.28","-0.58","0.91","-0.31","-0.41","-0.23","0.29","-0.42","-0.3","-0.63","0.26","-0.68","-0.89","-0.28","0.49","-0.33","0.31","0.47","-0.28","-0.23","-0.55","0.24","-0.31","0.31","-0.23","-0.43","-0.48","0.31","0.2","0.44","-0.23","-0.24","0.24","-0.46","-0.78","0.53","0.22","0.36","0.52","0.32","-0.19","0.42","0.27","0.24","-0.61","-0.32","-0.18","-0.61","-0.66","0.22","0.22","0.42","0.61","0.26","0.18","0.51","-0.27","-0.5","0.34","-0.27","-0.31","-0.4","0.2","-0.24","-0.68","-0.59","-0.66","-0.15","-0.24","-0.3","-0.46","-0.27","0.33","-0.35","-0.2","0.28","0.4","0.87","-0.24","-0.39","-0.27","0.26","-0.3","0.57","-0.37","0.55","0.31","0.38","0.27","0.35","0.4","0.4","0.38","0.27","0.48","0.4","-0.21","-0.18","-0.26","0.3","-0.2","-0.28","0.24","0.22","0.31","0.37","0.23","-0.6","-0.3","-0.26","-0.2","-0.26","-0.95","-0.46","0.39","-0.2","0.38","-0.3","0.33","0.28","0.32","-0.18","0.43","0.22","0.78","0.17","0.31","-0.14","-0.21","-0.29","-0.3","-0.36","-0.48","-0.26","0.66","0.38","-0.38","0.38","0.29","-0.58","-0.26","-0.29","-0.47","0.21","-0.21","-0.43","0.33","0.22","0.22","-0.22","-0.26","-0.6","0.32","-0.34","-0.26","0.15","-0.29","-0.37","-0.18","0.27","-0.36","-0.27","0.29","-0.29","0.29","0.29","0.42","0.28","-0.37","0.41","0.29","-0.15","-0.25","1.1","0.25","0.32","0.4","-0.18","0.35","0.2","0.48","0.34","-0.14","0.14","-0.38","0.26","0.28","0.18","0.2","-0.19","-0.25","-0.17","-0.43","0.26","0.23","0.33","-0.17","-0.22","-0.39","-0.19","-0.24","0.27","-0.24","-0.43","-0.37","-0.41","0.16","-0.2","-0.2","-0.33","0.39","-0.29","-0.36","-0.16","0.26","-0.15","0.52","-0.18","0.25","-0.15","-0.36","-0.29","-0.39","-0.35","0.24","0.35","0.18","-0.12","0.31","0.2","-0.14","-0.23","0.22","-0.38","-0.21","-0.3","0.22","-0.23","-0.44","0.22","0.23","-0.38","0.29","-0.28","-0.15","-0.57","-0.3","0.3","-0.26","0.17","0.34","-0.29","0.24","-0.33","-0.27","-0.21","-0.19","0.24","-0.49","-0.22","0.52","0.35","-0.15","0.19","-0.33","0.33","-0.26","0.29","-0.18","0.17","-0.37","0.29","-0.2","0.36","0.21","-0.32","-0.36","-0.21","0.17","-0.22","-0.16","0.54","-0.16","-0.19","-0.29","0.24","0.25","-0.18","0.22","-0.21","0.29","-0.29","-0.26","-0.16","-0.24","-0.28","0.18","0.19","-0.14","0.15","0.18","-0.28","-0.33","0.46","-0.21","0.18","-0.2","-0.2","0.39","-0.13","0.21","-0.17","-0.18","0.29","-0.25","-0.55","-0.36","-0.26","-0.34","0.19","0.17","0.23","0.24","-0.39","-0.16","0.26","-0.32","-0.16","-0.25","0.22","0.24","0.24","0.19","0.15","-0.22","0.95","0.32","-0.2","0.43","-0.43","-0.34","0.19","-0.15","-0.36","-0.19","-0.27","0.26","-0.21","-0.18","-0.14","0.18","0.54","0.37","-0.38","-0.22","-0.17","0.25","0.52","-0.28","-0.26","-0.19","-0.33","-0.24","0.31","0.59","-0.11","0.18","0.27","0.23","-0.2","-0.13","0.18","-0.22","-0.15","0.27","-0.16","0.4","0.19","0.31","0.23","0.13","0.27","-0.25","0.13","0.21","0.14","-0.22","-0.18","0.21","0.26","0.18","-0.1","-0.55","0.15","0.21","0.16","-0.2","0.3","0.33","0.62","-0.25","0.23","0.36","-0.44","0.28","0.74","-0.14","0.16","0.24","0.37","-0.4","-0.12","-0.41","0.27","-0.51","0.19","0.5","0.2","-0.16","0.17","-0.42","0.27","-0.31","0.71","0.25","0.35","0.63","0.34","0.19","-0.24","-0.15","0.37","0.13","0.19","0.15","0.12","-0.27","0.24","0.16","-0.21","-0.2","-0.13","-0.22","-0.13","-0.31","-0.16","0.45","-0.27","-0.36","-0.27","0.15","-0.98","-0.2","0.18","-0.16","-0.35","0.27","0.33","0.17","-0.28","0.16","0.13","0.33","-0.24","0.32","0.23","0.19","0.19","-0.24","0.12","-0.18","-0.24","0.26","-0.27","-0.24","-0.28","-0.14","0.24","0.15","-0.17","-0.26","0.15","0.16","0.23","0.24","0.17","-0.15","-0.097","0.68","0.31","-0.24","0.18","0.28","-0.13","-0.25","-0.25","-0.78","-0.34","-0.18","-0.19","-0.28","-0.23","0.31","-0.21","0.27","-0.41","-0.15","-0.19","-0.18","-0.25","0.15","0.37","0.13","0.3","0.28","-0.12","-0.36","-0.37","0.35","-0.17","0.39","0.14","-0.18","-0.15","0.67","0.22","-0.15","0.28","-0.24","-0.16","0.14","-0.33","0.1","0.11","0.4","-0.19","0.31","0.25","-0.23","-0.1","-0.2","-0.21","0.14","-0.5","0.17","0.15","-0.16","-0.17","-0.2","-0.16","-0.34","-0.12","-0.18","-0.22","-0.25","-0.12","-0.34","0.13","0.19","-0.13","0.28","-0.23","-0.24","0.23","-0.17","-0.16","0.23","-0.23","-0.24","0.34","-0.3","0.17","-0.53","-0.24","0.22","-0.18","-0.23","-0.15","0.16","-0.16","-0.32","-0.29","0.15","-0.16","-0.12","0.6","-0.21","-0.14","-0.18","0.11","0.21","-0.14","0.14","-0.27","-0.2","0.5","0.28","-0.39","0.18","0.25","-0.18","-0.24","-0.15","0.51","-0.29","0.58","-0.39","-0.27","0.19","0.14","-0.24","0.11","-0.15","-0.2","-0.42","-0.18","-0.14","0.33","-0.34","-0.14","-0.28","-0.12","-0.14","-0.15","0.21","-0.24","-0.2","0.19","-0.36","0.14","-0.26","0.21","0.18","0.21","-0.38","-0.16","-0.11","0.14","0.67","0.16","0.17","0.22","-0.28","0.19","-0.19","0.14","0.17","0.17","-0.19","0.24","0.16","0.13","-0.21","-0.23","0.1","-0.16","-0.22","0.13","0.22","0.26","-0.2","-0.23","0.11","0.18","-0.24","0.13","-0.13","0.16","0.25","0.27","-0.23","0.19","0.13","-0.14","-0.11","0.23","-0.21","-0.12","0.18","0.18","-0.16","-0.21","-0.14","0.27","-0.22","0.61","0.15","-0.12","0.16","0.18","-0.1","-0.31","-0.21","0.42","0.32","0.22","-0.097","-0.24","0.19","-0.49","0.33","0.31","-0.17","-0.11","-0.15","0.19","-0.27","-0.2","0.14","-0.2","-0.18","-0.19","-0.2","-0.16","-0.13","0.13","-0.14","0.16"],["  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","2.2e-16","2.2e-16","4.4e-16","4.4e-16","6.7e-16","8.9e-16","8.9e-16","8.9e-16","8.9e-16","1.3e-15","1.4e-15","1.6e-15","2.7e-15","1.3e-14","2.1e-14","1.4e-13","2e-13","3.7e-13","4.2e-13","5.7e-13","7e-13","7.7e-13","8.5e-13","9.3e-13","1.5e-12","1.8e-12","2.7e-12","4.2e-12","4.5e-12","6.9e-12","8.7e-12","8.9e-12","1.2e-11","1.2e-11","1.5e-11","1.5e-11","1.7e-11","2.1e-11","2.1e-11","2.1e-11","2.2e-11","2.9e-11","4e-11","4.3e-11","5.9e-11","7.4e-11","1e-10","1.1e-10","1.3e-10","1.8e-10","1.9e-10","2.1e-10","2.2e-10","2.4e-10","3.8e-10","3.8e-10","4.7e-10","5.2e-10","5.8e-10","6.5e-10","7.9e-10","8.1e-10","8.6e-10","8.7e-10","1e-09","1.6e-09","1.6e-09","2e-09","2e-09","2e-09","2.2e-09","2.4e-09","2.7e-09","3.2e-09","3.4e-09","3.6e-09","3.7e-09","4e-09","4.5e-09","4.6e-09","4.8e-09","5e-09","5.7e-09","5.9e-09","6e-09","6.2e-09","8.5e-09","9.6e-09","9.6e-09","1.1e-08","1.1e-08","1.1e-08","1.2e-08","1.3e-08","1.4e-08","1.6e-08","1.6e-08","1.8e-08","1.8e-08","2e-08","2.1e-08","2.6e-08","2.7e-08","2.8e-08","3.2e-08","3.2e-08","3.6e-08","3.7e-08","4.1e-08","4.9e-08","5e-08","5.1e-08","5.3e-08","5.9e-08","6.5e-08","7.4e-08","8.5e-08","9.5e-08","1.1e-07","1.1e-07","1.2e-07","1.3e-07","1.3e-07","1.3e-07","1.3e-07","1.5e-07","1.5e-07","1.5e-07","1.6e-07","1.7e-07","1.7e-07","1.8e-07","2e-07","2.5e-07","2.6e-07","2.7e-07","2.9e-07","2.9e-07","2.9e-07","3e-07","3.2e-07","3.3e-07","4e-07","4.3e-07","4.4e-07","4.4e-07","5e-07","5e-07","5.2e-07","5.5e-07","5.5e-07","5.7e-07","5.9e-07","6.1e-07","6.3e-07","6.3e-07","6.7e-07","7e-07","7.1e-07","7.1e-07","7.6e-07","7.8e-07","8.2e-07","8.8e-07","8.9e-07","9.1e-07","9.2e-07","9.9e-07","1.1e-06","1.1e-06","1.1e-06","1.2e-06","1.2e-06","1.2e-06","1.3e-06","1.3e-06","1.4e-06","1.4e-06","1.4e-06","1.5e-06","1.6e-06","1.7e-06","1.9e-06","1.9e-06","1.9e-06","1.9e-06","1.9e-06","1.9e-06","2e-06","2e-06","2.1e-06","2.2e-06","2.3e-06","2.3e-06","2.4e-06","2.5e-06","2.7e-06","2.8e-06","2.9e-06","3e-06","3e-06","3.1e-06","3.2e-06","3.3e-06","3.4e-06","3.4e-06","3.5e-06","3.6e-06","3.6e-06","3.7e-06","3.8e-06","3.9e-06","3.9e-06","3.9e-06","4e-06","4e-06","4.1e-06","4.9e-06","4.9e-06","5e-06","5.3e-06","5.4e-06","5.4e-06","5.6e-06","6e-06","6e-06","6e-06","6.5e-06","6.5e-06","6.5e-06","6.7e-06","6.8e-06","7.3e-06","7.4e-06","7.6e-06","7.8e-06","7.8e-06","8.5e-06","8.9e-06","8.9e-06","9e-06","9.1e-06","9.1e-06","9.3e-06","9.4e-06","9.8e-06","1e-05","1e-05","1.1e-05","1.1e-05","1.1e-05","1.2e-05","1.2e-05","1.2e-05","1.3e-05","1.3e-05","1.3e-05","1.4e-05","1.4e-05","1.4e-05","1.4e-05","1.5e-05","1.5e-05","1.5e-05","1.5e-05","1.5e-05","1.5e-05","1.5e-05","1.5e-05","1.6e-05","1.6e-05","1.8e-05","1.8e-05","1.8e-05","1.8e-05","1.9e-05","1.9e-05","2e-05","2e-05","2.1e-05","2.1e-05","2.1e-05","2.2e-05","2.2e-05","2.3e-05","2.3e-05","2.3e-05","2.4e-05","2.4e-05","2.4e-05","2.4e-05","2.4e-05","2.5e-05","2.5e-05","2.5e-05","2.5e-05","2.6e-05","2.7e-05","2.7e-05","3e-05","3.1e-05","3.1e-05","3.2e-05","3.2e-05","3.3e-05","3.4e-05","3.4e-05","3.4e-05","3.4e-05","3.5e-05","3.8e-05","3.9e-05","3.9e-05","4e-05","4e-05","4e-05","4.1e-05","4.2e-05","4.2e-05","4.2e-05","4.2e-05","4.3e-05","4.4e-05","4.5e-05","4.5e-05","4.6e-05","4.6e-05","4.8e-05","4.9e-05","4.9e-05","4.9e-05","4.9e-05","4.9e-05","4.9e-05","5e-05","5e-05","5.2e-05","5.2e-05","5.4e-05","5.5e-05","5.7e-05","5.9e-05","5.9e-05","5.9e-05","6e-05","6.1e-05","6.1e-05","6.3e-05","6.3e-05","6.3e-05","6.3e-05","6.5e-05","6.5e-05","6.6e-05","6.7e-05","6.8e-05","6.8e-05","7.8e-05","7.8e-05","8.1e-05","8.1e-05","8.2e-05","8.5e-05","8.5e-05","8.7e-05","8.8e-05","8.8e-05","9e-05","9e-05","9.1e-05","9.2e-05","9.3e-05","9.5e-05","9.5e-05","9.6e-05","9.7e-05","9.8e-05","0.0001","0.0001","0.0001","0.00011","0.00011","0.00011","0.00011","0.00012","0.00012","0.00012","0.00012","0.00012","0.00013","0.00013","0.00013","0.00013","0.00013","0.00013","0.00013","0.00014","0.00014","0.00014","0.00014","0.00014","0.00014","0.00014","0.00014","0.00015","0.00015","0.00016","0.00016","0.00016","0.00016","0.00016","0.00017","0.00017","0.00017","0.00017","0.00017","0.00017","0.00018","0.00019","0.00019","0.00019","0.00019","0.00019","0.00019","0.0002","0.0002","0.0002","0.0002","0.00021","0.00021","0.00022","0.00022","0.00022","0.00022","0.00022","0.00022","0.00022","0.00023","0.00023","0.00023","0.00023","0.00023","0.00024","0.00024","0.00024","0.00025","0.00025","0.00025","0.00025","0.00025","0.00025","0.00027","0.00027","0.00027","0.00028","0.00028","0.00028","0.00028","0.00028","0.00028","0.00029","0.00029","0.00029","0.0003","0.00031","0.00031","0.00031","0.00031","0.00031","0.00032","0.00032","0.00032","0.00033","0.00033","0.00034","0.00035","0.00036","0.00037","0.00037","0.00037","0.00038","0.00039","0.0004","0.0004","0.0004","0.0004","0.00041","0.00041","0.00042","0.00042","0.00043","0.00043","0.00044","0.00044","0.00044","0.00045","0.00045","0.00046","0.00048","0.00048","0.00048","0.00048","0.00048","0.00049","0.00049","0.00049","0.00051","0.00051","0.00051","0.00051","0.00052","0.00053","0.00053","0.00054","0.00055","0.00055","0.00055","0.00056","0.00056","0.00057","0.00057","0.00057","0.00058","0.00059","0.00059","0.0006","0.0006","0.00061","0.00063","0.00064","0.00065","0.00065","0.00065","0.00066","0.00067","0.00068","0.00068","0.00068","0.00069","0.00071","0.00071","0.00072","0.00072","0.00072","0.00074","0.00075","0.00076","0.00077","0.00079","0.00079","0.00079","0.0008","0.0008","0.0008","0.00081","0.00082","0.00082","0.00082","0.00082","0.00083","0.00083","0.00084","0.00084","0.00085","0.00087","0.00088","0.00089","0.00091","0.00091","0.00092","0.00093","0.00095","0.00096","0.00097","0.00098","0.00099","0.001","0.001","0.001","0.001","0.001","0.001","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0016","0.0016","0.0016","0.0016","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.0019","0.002","0.002","0.002","0.002","0.0021","0.0021","0.0021","0.0022","0.0022","0.0022","0.0022","0.0022","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0024","0.0024","0.0024","0.0024","0.0024","0.0024","0.0024","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0025","0.0026","0.0026","0.0026","0.0026","0.0026","0.0026","0.0027","0.0027","0.0027","0.0027","0.0027","0.0027","0.0027","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.003","0.003","0.003","0.003","0.003","0.003","0.0031","0.0031","0.0031","0.0031","0.0032","0.0032","0.0032","0.0032","0.0032","0.0033","0.0033","0.0033","0.0034","0.0034","0.0035","0.0036","0.0036","0.0036","0.0036","0.0036","0.0036","0.0037","0.0037","0.0038","0.0038","0.0038","0.0038","0.0039","0.0039","0.0039","0.0039","0.0039","0.004","0.004","0.004","0.004","0.0041","0.0041","0.0041","0.0041","0.0042","0.0042","0.0042","0.0042","0.0043","0.0045","0.0046","0.0046","0.0046","0.0046","0.0046","0.0046","0.0046","0.0047","0.0047","0.0048","0.0048","0.0048","0.0049","0.0049","0.005","0.0051","0.0051","0.0052","0.0052","0.0052","0.0052","0.0052","0.0053","0.0053","0.0053","0.0054","0.0054","0.0054","0.0055","0.0055","0.0055","0.0056","0.0057","0.0058","0.0058","0.0058","0.0058","0.0058","0.0059","0.0059","0.0059","0.0059","0.006","0.006","0.006","0.006","0.006","0.006","0.0061","0.0061","0.0061","0.0061","0.0062","0.0063","0.0063","0.0063","0.0063","0.0064","0.0064","0.0065","0.0065","0.0065","0.0066","0.0066","0.0066","0.0067","0.0067","0.0068","0.0069","0.0069","0.0069","0.007","0.007","0.0071","0.0071","0.0071","0.0072","0.0072","0.0072","0.0072","0.0072","0.0072","0.0073","0.0073","0.0074","0.0074","0.0075","0.0075","0.0075","0.0075","0.0076","0.0076","0.0076","0.0076","0.0077","0.0077","0.0077","0.0078","0.0078","0.0078","0.0079","0.0079","0.0079","0.008","0.008","0.008","0.008","0.008","0.008","0.008","0.008","0.0081","0.0082","0.0082","0.0082","0.0082","0.0082","0.0083","0.0083","0.0083","0.0084","0.0084","0.0085","0.0086","0.0086","0.0086","0.0086","0.0086","0.0088","0.0088","0.0088","0.0089","0.0089","0.0089","0.0091","0.0091","0.0093","0.0093","0.0093","0.0094","0.0094","0.0094","0.0094","0.0095","0.0095","0.0095","0.0095","0.0095","0.0095","0.0095","0.0096","0.0097","0.0097","0.0098","0.0098","0.0099","0.0099","0.01","0.01","0.01","0.01","0.01","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.016","0.016","0.016","0.016","0.016","0.016","0.016"],["  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","  0","5.3e-14","5.3e-14","9.2e-14","9.2e-14","1.3e-13","1.4e-13","1.4e-13","1.4e-13","1.4e-13","2e-13","2e-13","2.1e-13","3.5e-13","1.7e-12","2.6e-12","1.7e-11","2.2e-11","4.1e-11","4.5e-11","5.9e-11","7e-11","7.5e-11","8e-11","8.5e-11","1.4e-10","1.6e-10","2.3e-10","3.5e-10","3.7e-10","5.4e-10","6.7e-10","6.7e-10","8.6e-10","8.6e-10","1e-09","1.1e-09","1.1e-09","1.3e-09","1.3e-09","1.4e-09","1.4e-09","1.8e-09","2.4e-09","2.5e-09","3.4e-09","4.2e-09","5.8e-09","6.2e-09","7.1e-09","9.5e-09","9.8e-09","1.1e-08","1.1e-08","1.2e-08","1.9e-08","1.9e-08","2.3e-08","2.5e-08","2.7e-08","3e-08","3.6e-08","3.6e-08","3.8e-08","3.8e-08","4.5e-08","6.6e-08","6.9e-08","8.2e-08","8.2e-08","8.2e-08","8.9e-08","9.5e-08","1.1e-07","1.2e-07","1.3e-07","1.4e-07","1.4e-07","1.5e-07","1.6e-07","1.6e-07","1.7e-07","1.8e-07","2e-07","2e-07","2e-07","2.1e-07","2.8e-07","3.2e-07","3.2e-07","3.5e-07","3.5e-07","3.6e-07","3.7e-07","4e-07","4.4e-07","5e-07","5e-07","5.3e-07","5.5e-07","5.9e-07","6.3e-07","7.5e-07","7.7e-07","8e-07","9.1e-07","9.1e-07","9.9e-07","1e-06","1.1e-06","1.3e-06","1.3e-06","1.4e-06","1.4e-06","1.6e-06","1.7e-06","1.9e-06","2.2e-06","2.4e-06","2.8e-06","2.8e-06","3e-06","3.1e-06","3.1e-06","3.1e-06","3.1e-06","3.6e-06","3.7e-06","3.7e-06","3.7e-06","4e-06","4e-06","4.1e-06","4.5e-06","5.6e-06","5.9e-06","6.1e-06","6.4e-06","6.4e-06","6.4e-06","6.5e-06","6.8e-06","7e-06","8.5e-06","9.2e-06","9.2e-06","9.3e-06","1e-05","1e-05","1.1e-05","1.1e-05","1.1e-05","1.1e-05","1.2e-05","1.2e-05","1.2e-05","1.2e-05","1.3e-05","1.4e-05","1.4e-05","1.4e-05","1.5e-05","1.5e-05","1.5e-05","1.7e-05","1.7e-05","1.7e-05","1.7e-05","1.8e-05","2e-05","2e-05","2.1e-05","2.1e-05","2.1e-05","2.1e-05","2.2e-05","2.3e-05","2.4e-05","2.4e-05","2.5e-05","2.6e-05","2.7e-05","2.8e-05","3.2e-05","3.2e-05","3.2e-05","3.2e-05","3.2e-05","3.2e-05","3.3e-05","3.4e-05","3.5e-05","3.5e-05","3.7e-05","3.7e-05","3.9e-05","4e-05","4.3e-05","4.4e-05","4.5e-05","4.7e-05","4.7e-05","4.8e-05","5e-05","5.1e-05","5.2e-05","5.2e-05","5.3e-05","5.4e-05","5.5e-05","5.5e-05","5.6e-05","5.8e-05","5.8e-05","5.8e-05","5.8e-05","5.9e-05","5.9e-05","7e-05","7.1e-05","7.1e-05","7.6e-05","7.6e-05","7.7e-05","7.8e-05","8.3e-05","8.3e-05","8.3e-05","8.9e-05","8.9e-05","8.9e-05","9.2e-05","9.2e-05","9.8e-05","0.0001","0.0001","0.0001","0.0001","0.00011","0.00012","0.00012","0.00012","0.00012","0.00012","0.00012","0.00012","0.00013","0.00013","0.00013","0.00014","0.00014","0.00014","0.00014","0.00015","0.00015","0.00016","0.00016","0.00016","0.00017","0.00017","0.00017","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00019","0.00019","0.00021","0.00021","0.00021","0.00021","0.00021","0.00022","0.00022","0.00023","0.00023","0.00023","0.00024","0.00024","0.00025","0.00025","0.00025","0.00026","0.00026","0.00026","0.00026","0.00026","0.00026","0.00027","0.00027","0.00027","0.00027","0.00028","0.00028","0.00028","0.00032","0.00033","0.00033","0.00033","0.00033","0.00034","0.00035","0.00035","0.00035","0.00036","0.00036","0.00039","0.0004","0.0004","0.0004","0.00041","0.00041","0.00041","0.00042","0.00042","0.00042","0.00042","0.00043","0.00044","0.00044","0.00045","0.00045","0.00045","0.00047","0.00047","0.00047","0.00047","0.00047","0.00047","0.00047","0.00047","0.00048","0.0005","0.0005","0.00051","0.00051","0.00054","0.00055","0.00055","0.00055","0.00056","0.00056","0.00056","0.00058","0.00058","0.00058","0.00058","0.00059","0.00059","0.00059","0.0006","0.00061","0.00061","0.00069","0.0007","0.00072","0.00072","0.00073","0.00075","0.00075","0.00076","0.00077","0.00077","0.00078","0.00078","0.00079","0.00079","0.0008","0.00081","0.00081","0.00082","0.00083","0.00083","0.00085","0.00086","0.00087","0.00089","0.00093","0.00095","0.00096","0.00096","0.00096","0.00098","0.00099","0.001","0.001","0.001","0.001","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0012","0.0012","0.0012","0.0012","0.0012","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0014","0.0014","0.0014","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0016","0.0016","0.0016","0.0016","0.0016","0.0016","0.0016","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0019","0.0019","0.002","0.002","0.002","0.002","0.002","0.002","0.002","0.002","0.002","0.002","0.0021","0.0021","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0023","0.0023","0.0023","0.0024","0.0024","0.0025","0.0025","0.0025","0.0026","0.0026","0.0026","0.0027","0.0027","0.0027","0.0027","0.0027","0.0028","0.0028","0.0028","0.0028","0.0029","0.0029","0.0029","0.0029","0.0029","0.003","0.0031","0.0031","0.0031","0.0031","0.0031","0.0031","0.0031","0.0032","0.0032","0.0033","0.0033","0.0033","0.0033","0.0033","0.0033","0.0034","0.0034","0.0034","0.0034","0.0035","0.0035","0.0035","0.0035","0.0036","0.0036","0.0036","0.0036","0.0037","0.0037","0.0037","0.0038","0.0039","0.004","0.004","0.004","0.004","0.0041","0.0041","0.0041","0.0041","0.0041","0.0042","0.0042","0.0043","0.0043","0.0043","0.0044","0.0044","0.0045","0.0046","0.0046","0.0046","0.0046","0.0047","0.0047","0.0047","0.0047","0.0048","0.0048","0.0048","0.0048","0.0048","0.0048","0.0048","0.0048","0.0049","0.005","0.0051","0.0051","0.0052","0.0052","0.0053","0.0053","0.0054","0.0054","0.0055","0.0056","0.0056","0.0057","0.0058","0.0059","0.0059","0.0059","0.0059","0.006","0.006","0.0062","0.0062","0.0062","0.0062","0.0062","0.0062","0.0063","0.0063","0.0065","0.0065","0.0066","0.0066","0.0066","0.0067","0.0068","0.0069","0.0071","0.0071","0.0071","0.0072","0.0073","0.0074","0.0075","0.0075","0.0076","0.0077","0.0077","0.0077","0.0079","0.0079","0.008","0.008","0.0081","0.0081","0.0081","0.0081","0.0081","0.0082","0.0085","0.0086","0.0087","0.0087","0.0087","0.0087","0.0088","0.0088","0.0089","0.0089","0.0089","0.009","0.009","0.009","0.009","0.009","0.009","0.0091","0.0091","0.0091","0.0091","0.0093","0.0093","0.0094","0.0094","0.0095","0.0095","0.0095","0.0095","0.0095","0.0096","0.0096","0.0096","0.0096","0.0096","0.0097","0.0098","0.0098","0.0098","0.01","0.01","0.01","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.015","0.015","0.015","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.018","0.018","0.018","0.018","0.018","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.02","0.02","0.02","0.02","0.02","0.02","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.022","0.022","0.022","0.022","0.022","0.022","0.022","0.022","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.024","0.025","0.025","0.025","0.025","0.025","0.025","0.025","0.025","0.025","0.026","0.026","0.026","0.026","0.026","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.027","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.028","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.03","0.03","0.03","0.03","0.03","0.03","0.03","0.03","0.03","0.03","0.03","0.031","0.031","0.031","0.031","0.031","0.031","0.031","0.031","0.031","0.032","0.032","0.032","0.032","0.032","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.033","0.034","0.034","0.034","0.034","0.034","0.034","0.034","0.035","0.035","0.036","0.036","0.036","0.036","0.036","0.037","0.037","0.037","0.037","0.037","0.037","0.037","0.038","0.038","0.038","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.039","0.04","0.04","0.04","0.04","0.041","0.041","0.041","0.041","0.041","0.041","0.041","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.042","0.043","0.043","0.043","0.043","0.044","0.044","0.044","0.044","0.044","0.045","0.045","0.045","0.046","0.046","0.046","0.046","0.046","0.046","0.046","0.046","0.047","0.047","0.047","0.047","0.047","0.047","0.047","0.047","0.047","0.048","0.048","0.048","0.048","0.048","0.048","0.048","0.048","0.048","0.048","0.048","0.049","0.049","0.049","0.049","0.05","0.05","0.05","0.05","0.05"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>Chr<\/th>\n      <th>logFC<\/th>\n      <th>P.Value<\/th>\n      <th>adj.P.Val<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-differentially-expressed-proteins" class="section level2">
<h2>Heatmap of differentially expressed proteins</h2>
<pre class="r"><code>proList &lt;- filter(resList, !is.na(name), adj.P.Val &lt; 0.01) %&gt;% distinct(name, .keep_all = TRUE) %&gt;% pull(id)
plotMat &lt;- assays(protCLL)[[&quot;QRILC_combat&quot;]][proList,]
rownames(plotMat) &lt;- rowData(protCLL[proList,])$hgnc_symbol

colAnno &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  select(Patient.ID, trisomy12, IGHV.status) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

rowAnno &lt;- rowData(protCLL)[proList,c(&quot;chromosome_name&quot;,&quot;hgnc_symbol&quot;),drop=FALSE] %&gt;% 
  data.frame(stringsAsFactors = FALSE) %&gt;%
  mutate(onChr12 = ifelse(chromosome_name == &quot;12&quot;,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  select(hgnc_symbol, onChr12) %&gt;% data.frame() %&gt;% remove_rownames() %&gt;% column_to_rownames(&quot;hgnc_symbol&quot;)

plotMat &lt;- jyluMisc::mscale(plotMat, censor = 5)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV.status = c(M = colList[4], U = colList[3]),
                onChr12 = c(yes = colList[1],no = &quot;white&quot;))

pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = &quot;none&quot;,
                   annotation_row = rowAnno,
                   clustering_method = &quot;complete&quot;, clustering_distance_cols = &quot;euclidean&quot;,
                   color = colorRampPalette(c(colList[2],&quot;white&quot;,colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = FALSE, show_colnames = FALSE,
                   treeheight_row = 0)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_heatmap-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="summary-of-the-chromosome-distribution-of-genes-that-encode-those-de-proteins" class="section level2">
<h2>Summary of the chromosome distribution of genes that encode those DE proteins</h2>
<pre class="r"><code>plotTab &lt;- filter(resList, adj.P.Val &lt;=0.05) %&gt;% mutate(change = ifelse(logFC&gt;0,&quot;Up regulated&quot;,&quot;Down regulated&quot;),
                              chromosome = ifelse(Chr %in% &quot;12&quot;,&quot;chr12&quot;,&quot;other&quot;)) %&gt;%
  group_by(change, chromosome) %&gt;% summarise(n = length(id))

sigNumPlot &lt;- ggplot(plotTab, aes(x=change, y=n, fill = chromosome)) + 
  geom_bar(stat = &quot;identity&quot;, width = 0.8,
           position = position_dodge2(width = 6),
           col = &quot;black&quot;) +
  geom_text(aes(label=n), 
            position = position_dodge(width = 0.9),
            size=4, hjust=-0.1)  +
  scale_fill_manual(name = &quot;&quot;, labels = c(&quot;chr12&quot;,&quot;other&quot;), values = colList) +
  coord_flip(ylim = c(0,700), expand = FALSE) + xlab(&quot;&quot;) + ylab(&quot;Number of associations (5% FDR)&quot;) + theme_half +
  theme(legend.text = element_text(size=12))
sigNumPlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_chr_summary-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-plot-for-differentially-expressed-proteins." class="section level2">
<h2>Volcano plot for differentially expressed proteins.</h2>
<pre class="r"><code>plotTab &lt;- resList 
nameList &lt;- c(&quot;PTPN11&quot;, &quot;BCL10&quot;, &quot;SMAD2&quot;, &quot;PYCARD&quot;, &quot;STAT2&quot;, &quot;CD72&quot;)
tri12Vocano &lt;- plotVolcano(plotTab, fdrCut =0.05, x_lab=bquote(&quot;log&quot;[2]*&quot;(fold change)&quot;), posCol = colList[1], negCol = colList[2],
            plotTitle = &quot;trisomy12 (present versus absent)&quot;, ifLabel = TRUE, labelList = nameList)
tri12Vocano</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_volcano-1.png" width="1152" style="display: block; margin: auto;" /> The proteins discussed in the manuscript are labelled.</p>
</div>
<div id="boxplot-plot-of-selected-genes" class="section level2">
<h2>Boxplot plot of selected genes</h2>
<pre class="r"><code>protTab &lt;- sumToTidy(protCLL, rowID = &quot;uniprotID&quot;, colID = &quot;patID&quot;) %&gt;%
  mutate(count = count_combat)
plotTab &lt;- protTab %&gt;% filter(hgnc_symbol %in% nameList) %&gt;%
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(status = ifelse(trisomy12 %in% 1,&quot;trisomy12&quot;,&quot;other&quot;),
         name = hgnc_symbol) %&gt;%
  mutate(status = factor(status, levels = c(&quot;other&quot;,&quot;trisomy12&quot;)))
pList &lt;- plotBox(plotTab, pValTabel = resList, y_lab = &quot;Protein expression&quot;)
protBoxplot &lt;- cowplot::plot_grid(plotlist= pList, ncol=3)
protBoxplot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_selected_protein-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;trisomy12_selected_protein.pdf&quot;, height = 6, width = 10)</code></pre>
</div>
<div id="gene-set-enrichment-analysis-to-explore-the-pathway-changes-related-to-trisomy12" class="section level2">
<h2>Gene set enrichment analysis to explore the pathway changes related to trisomy12</h2>
<div id="for-all-differentially-expressed-proteins" class="section level3">
<h3>For all differentially expressed proteins</h3>
<div id="barplot-of-enriched-pathways-using-hallmark-gene-set" class="section level4">
<h4>Barplot of enriched pathways using Hallmark gene set</h4>
<pre class="r"><code>gmts = list(H= &quot;../data/gmts/h.all.v7.3.symbols.gmt&quot;,
            KEGG = &quot;../data/gmts/c2.cp.kegg.v7.3.symbols.gmt&quot;,
            combine = &quot;../data/gmts/hallmark_kegg_combine.gmt&quot;)
inputTab &lt;- resList %&gt;% filter(adj.P.Val &lt; 0.05, Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)
enRes &lt;- list()
enRes[[&quot;Cancer Hallmark gene sets&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)

tri12Enrich &lt;- plotEnrichmentBar(enRes[[1]], pCut =0.05, ifFDR= TRUE, setName = &quot;&quot;, 
                       title = names(enRes)[1], removePrefix = &quot;HALLMARK_&quot;, insideLegend=TRUE, setMap = setMap) +
  theme(legend.position = &quot;none&quot;)
tri12Enrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="barplot-of-enriched-pathways-using-kegg-gene-set" class="section level4">
<h4>Barplot of enriched pathways using KEGG gene set</h4>
<pre class="r"><code>inputTab &lt;- resList %&gt;% filter(adj.P.Val &lt; 0.05, Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)
enRes &lt;- list()
enRes[[&quot;KEGG gene sets&quot;]] &lt;- runGSEA(inputTab, gmts$KEGG, &quot;page&quot;)

tri12EnrichKEGG &lt;- plotEnrichmentBar(enRes[[1]], pCut =0.05, ifFDR= TRUE, setName =  &quot;&quot;, 
                       title = names(enRes)[1], removePrefix = &quot;KEGG_&quot;, insideLegend=TRUE, setMap = setMap) +
  theme(legend.position = c(0.8,0.2))</code></pre>
</div>
</div>
<div id="for-de-proteins-encoded-by-genes-on-chromosome-12-cis-effect" class="section level3">
<h3>For DE proteins encoded by genes on chromosome 12 (cis-effect)</h3>
<p>We will use Fisher test to identify geneset that are over-rerepresented.</p>
<p>All protein coding genes in the genome will be used as background</p>
<pre class="r"><code>#using all quantified proteins
rnaAll &lt;- dds[rowData(dds)$biotype %in% &quot;protein_coding&quot; &amp; !rowData(dds)$symbol %in% c(&quot;&quot;,NA),] #all protein coding gene as background
backList &lt;- rowData(rnaAll)$symbol
resListTri12 &lt;- mutate(resList, chr = rowData(protCLL[id,])$chromosome_name)</code></pre>
<div id="enrichment-results" class="section level4">
<h4>Enrichment results</h4>
<pre class="r"><code>protList &lt;- filter(resListTri12, adj.P.Val &lt; 0.05, t&gt;0, chr == &quot;12&quot;)$name

enRes &lt;- runFisher(protList, backList, gmts$combine, pCut =0.05, ifFDR = TRUE,removePrefix = &quot;KEGG_|HALLMARK_&quot;,
                   plotTitle = &quot;trisomy12 cis-effect&quot;, insideLegend = TRUE,
                   setName = &quot;&quot;, setMap = setMap, topN=5)
cisComEnrich &lt;- enRes$enrichPlot + theme(plot.margin = margin(1,3,1,1, unit = &quot;cm&quot;))
cisComEnrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;tri12_cisEnrich.pdf&quot;, height = 5, width = 7)</code></pre>
</div>
<div id="for-proteins-not-on-chr12-trans-effect" class="section level4">
<h4>For proteins not on chr12 (trans-effect)</h4>
<pre class="r"><code>protList &lt;- filter(resListTri12, adj.P.Val &lt; 0.05, t&gt;0, chr != &quot;12&quot;)$name

enRes &lt;- runFisher(protList, backList, gmts$combine, pCut =0.05, ifFDR = TRUE,removePrefix = &quot;KEGG_|HALLMARK_&quot;,
                   plotTitle = &quot;trisomy12 trans-effect&quot;, insideLegend = TRUE,
                   setName = &quot;&quot;, setMap = setMap, topN=5)
transComEnrich &lt;- enRes$enrichPlot + theme(plot.margin = margin(1,3,1,1, unit = &quot;cm&quot;))
transComEnrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;tri12_transEnrich.pdf&quot;, height = 5, width = 7)</code></pre>
</div>
</div>
<div id="heatmaps-of-protein-expression-in-enriched-pathways" class="section level3">
<h3>Heatmaps of protein expression in enriched pathways</h3>
<pre class="r"><code>resList.sig &lt;- filter(resList, !is.na(name), adj.P.Val &lt; 0.05, t&gt;0) %&gt;% distinct(name, .keep_all = TRUE) 
upRNAlist &lt;- filter(resListRNA, Gene == &quot;trisomy12&quot;) %&gt;% filter(t&gt;0, adj.P.Val &lt;= 0.05) %&gt;% pull(name)
plotMat &lt;- assays(protCLL)[[&quot;QRILC_combat&quot;]][resList.sig$id,]
rownames(plotMat) &lt;- rowData(protCLL[rownames(plotMat),])$hgnc_symbol
colAnno &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  select(Patient.ID, trisomy12) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

rowAnno &lt;- rowData(protCLL)[resList.sig$id,c(&quot;chromosome_name&quot;,&quot;hgnc_symbol&quot;),drop=FALSE] %&gt;% 
  data.frame(stringsAsFactors = FALSE) %&gt;%
  mutate(`on Chr12` = ifelse(chromosome_name == &quot;12&quot;,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  mutate(`RNA up-regulated` = ifelse(hgnc_symbol %in% upRNAlist,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  select(hgnc_symbol, `on Chr12`, `RNA up-regulated`) %&gt;% remove_rownames() %&gt;% column_to_rownames(&quot;hgnc_symbol&quot;)

plotMat &lt;- jyluMisc::mscale(plotMat, censor = 5)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey90&quot;),
                IGHV.status = c(M = colList[3], U = colList[4]),
                `on Chr12` = c(yes = colList[1],no = &quot;grey90&quot;),
                `RNA up-regulated` = c(&quot;yes&quot; = colList[5], no = &quot;grey90&quot;))</code></pre>
<pre class="r"><code>#pdf(&quot;trisomy12_PI3K_heatmap.pdf&quot;, height=5, width=10)
plotSetHeatmap(resList.sig, gmts$H, &quot;HALLMARK_PI3K_AKT_MTOR_SIGNALING&quot;, plotMat, 
               colAnno, rowAnno = rowAnno, annoCol = annoCol, highLight = nameList, plotName = &quot;PI3K-AKT-mTOR signaling&quot;)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_PI3K_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>plotSetHeatmap(resList.sig, gmts$H, &quot;HALLMARK_MTORC1_SIGNALING&quot;, plotMat, colAnno, rowAnno = rowAnno, annoCol = annoCol, plotName = &quot;MTORC1 signaling&quot;,
                highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_MTOR_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotSetHeatmap(resList.sig, gmts$H, &quot;HALLMARK_INTERFERON_ALPHA_RESPONSE&quot;, plotMat, colAnno, rowAnno = rowAnno, annoCol = annoCol, plotName = &quot;Interferon alpha response&quot;,
                highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_IFNA_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#pdf(&quot;trisomy12_BCR_heatmap.pdf&quot;, height=6, width=10)
plotSetHeatmap(resList.sig, gmts$KEGG, &quot;KEGG_B_CELL_RECEPTOR_SIGNALING_PATHWAY&quot;, plotMat, highLight = nameList,
               colAnno, rowAnno = rowAnno, annoCol = annoCol, plotName = &quot;B-cell receptor signaling&quot;) </code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_BCR_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
</div>
</div>
</div>
<div id="compare-protein-expression-with-rna-expression-related-to-trisomy12" class="section level1">
<h1>Compare protein expression with RNA expression related to trisomy12</h1>
<div id="differentially-expressed-rna-related-to-trisomy12" class="section level2">
<h2>Differentially expressed RNA related to trisomy12</h2>
<p>Prepare RNA sequencing data</p>
<pre class="r"><code>dds$trisomy12 &lt;- factor(patMeta[match(dds$PatID, patMeta$Patient.ID),]$trisomy12)
dds$IGHV &lt;- factor(patMeta[match(dds$PatID, patMeta$Patient.ID),]$IGHV.status)

ddsCLL &lt;- dds[rownames(dds) %in% rowData(protCLL)$ensembl_gene_id, !is.na(dds$IGHV) &amp; !is.na(dds$trisomy12)]
ddsCLL.vst &lt;- ddsCLL
assay(ddsCLL.vst) &lt;- log2(counts(ddsCLL, normalized = TRUE) + 1)
#ddsCLL.vst &lt;- varianceStabilizingTransformation(ddsCLL)</code></pre>
<p>Differential expression</p>
<pre class="r"><code>resTab &lt;- filter(resListRNA, Gene == &quot;trisomy12&quot;)</code></pre>
<div id="boxplot-of-selected-genes" class="section level3">
<h3>Boxplot of selected genes</h3>
<pre class="r"><code>plotTab &lt;- assay(ddsCLL.vst[match(nameList, rowData(ddsCLL.vst)$symbol),]) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
  mutate(name = rowData(ddsCLL.vst[id,])$symbol) %&gt;%
  gather(key = &quot;patID&quot;, value = &quot;count&quot;, -id, -name) %&gt;% 
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(status = ifelse(trisomy12 %in% 1,&quot;trisomy12&quot;,&quot;other&quot;))  %&gt;%
  mutate(status = factor(status, levels = c(&quot;other&quot;,&quot;trisomy12&quot;)))


pList &lt;- plotBox(plotTab, pValTabel = resTab, y_lab = &quot;RNA expression&quot;)
cowplot::plot_grid(plotlist= pList, ncol=2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_selected_RNA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlations-between-rna-and-protein-expression" class="section level3">
<h3>Correlations between RNA and protein expression</h3>
<pre class="r"><code>rnaMat &lt;- assay(ddsCLL.vst)
protMat &lt;- assays(protCLL)[[&quot;log2Norm_combat&quot;]]
rownames(protMat) &lt;- rowData(protCLL)$ensembl_gene_id
overSample &lt;- intersect(colnames(rnaMat), colnames(protMat))
rnaMat &lt;- rnaMat[,overSample]
protMat &lt;- protMat[,overSample]

plotList &lt;- lapply(nameList, function(n) {
  geneId &lt;- rownames(ddsCLL.vst)[match(n, rowData(ddsCLL.vst)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab &lt;- tibble(x=rnaMat[geneId,],y=protMat[geneId,])
  coef &lt;- cor(plotTab$x, plotTab$y, use=&quot;pairwise.complete&quot;)
  annoPos &lt;- ifelse (coef &gt; 0, &quot;left&quot;,&quot;right&quot;)
  plotCorScatter(plotTab, &quot;x&quot;,&quot;y&quot;, showR2 = FALSE, annoPos = annoPos, x_lab = &quot;RNA expression&quot;,
                 y_lab =&quot;Protein expression&quot;, title = n,dotCol = colList[4], textCol = colList[1])
})
cowplot::plot_grid(plotlist = plotList, ncol =2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_protein_RNA_cor-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="gene-dosage-effect-and-protein-buffering-related-to-trisomy12" class="section level1">
<h1>Gene dosage effect and protein buffering related to trisomy12</h1>
<div id="visualizing-gene-dosage-effect-on-protein-and-rna-level" class="section level2">
<h2>Visualizing gene dosage effect on protein and RNA level</h2>
<p>Processing protein and RNA count data</p>
<pre class="r"><code>protExprTab &lt;- sumToTidy(protCLL) %&gt;%
  filter(chromosome_name == &quot;12&quot;) %&gt;%
  mutate(id = ensembl_gene_id, patID = colID, expr = log2Norm_combat, type = &quot;Protein&quot;) %&gt;%
  select(id, patID, expr, type)

rnaExprTab &lt;- counts(dds[rownames(dds) %in% protExprTab$id,
                            colnames(dds) %in% protExprTab$patID], normalized= TRUE) %&gt;%
  as_tibble(rownames = &quot;id&quot;) %&gt;%
  pivot_longer(-id, names_to = &quot;patID&quot;, values_to = &quot;count&quot;) %&gt;%
  mutate(expr = log2(count)) %&gt;%
  select(id, patID, expr) %&gt;% mutate(type = &quot;RNA&quot;) 

comExprTab &lt;- bind_rows(rnaExprTab, protExprTab) %&gt;%
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  filter(!is.na(trisomy12)) %&gt;% mutate(cnv = ifelse(trisomy12 %in% 1, &quot;trisomy12&quot;,&quot;other&quot;))</code></pre>
<div id="proteinsrnas-on-chr12-have-higher-expressions-in-trisomy12-samples-compared-to-other-samples" class="section level3">
<h3>Proteins/RNAs on Chr12 have higher expressions in trisomy12 samples compared to other samples</h3>
<pre class="r"><code>plotTab &lt;- comExprTab %&gt;%
  group_by(id,type) %&gt;% mutate(zscore = (expr-mean(expr))/sd(expr)) %&gt;%
  group_by(id, cnv, type) %&gt;% summarise(meanExpr = mean(zscore, na.rm=TRUE)) %&gt;%
  ungroup()

dosagePlot &lt;- ggplot(plotTab, aes(x=meanExpr, fill = cnv, col=cnv)) + 
  geom_histogram(position = &quot;identity&quot;, alpha=0.5, bins=30) + facet_wrap(~type, scale = &quot;fixed&quot;) +
  scale_fill_manual(values = c(other = &quot;grey80&quot;, trisomy12 = colList[1]), name = &quot;&quot;) +
  scale_color_manual(values = c(other = &quot;grey80&quot;, trisomy12 = colList[1]), name = &quot;&quot;) +
  xlim(-1,1.5) +
  theme_full + xlab(&quot;Mean Z-score&quot;) +
  theme(strip.text = element_text(size =20), 
        legend.position = c(0.1,0.9), legend.background = element_rect(fill = NA),
        legend.text = element_text(size=15))

dosagePlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/dosage_effect-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="analyzing-protein-buffering-effect" class="section level2">
<h2>Analyzing protein buffering effect</h2>
<div id="detect-buffered-and-non-buffered-proteins" class="section level3">
<h3>Detect buffered and non-buffered proteins</h3>
<p>Preprocessing protein and RNA data</p>
<pre class="r"><code>#subset samples and genes
overSampe &lt;- intersect(colnames(ddsCLL), colnames(protCLL))
overGene &lt;- intersect(rownames(ddsCLL), rowData(protCLL)$ensembl_gene_id)
ddsSub &lt;- ddsCLL[overGene, overSampe]
protSub &lt;- protCLL[match(overGene, rowData(protCLL)$ensembl_gene_id),overSampe]
rowData(ddsSub)$uniprotID &lt;- rownames(protSub)[match(rownames(ddsSub),rowData(protSub)$ensembl_gene_id)]

#vst
ddsSub.vst &lt;- ddsSub
assay(ddsSub.vst) &lt;- log2(counts(ddsSub, normalized=TRUE) +1)

rnaRes &lt;- resListRNA %&gt;% filter(Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(chrom = rowData(dds[id,])$chromosome) %&gt;%
  dplyr::rename(geneID = id, log2FC.rna = log2FC, 
                pvalue.rna = P.Value, padj.rna = adj.P.Val, stat.rna= t) %&gt;%
  arrange(pvalue.rna) %&gt;% distinct(name, .keep_all = TRUE) %&gt;%
  select(geneID, name, log2FC.rna, pvalue.rna, padj.rna, stat.rna, chrom)

protRes &lt;- resList %&gt;% filter(Gene == &quot;trisomy12&quot;) %&gt;%
    mutate(Chr = rowData(protCLL[id,])$chromosome_name) %&gt;%
    #filter(Chr == &quot;12&quot;) %&gt;%
    #mutate(adj.P.Val = p.adjust(P.Value, method= &quot;BH&quot;)) %&gt;%
    dplyr::rename(uniprotID = id, 
                  pvalue = P.Value, padj = adj.P.Val) %&gt;% 
    #mutate(geneID = rowData(protCLL[uniprotID,])$ensembl_gene_id) %&gt;%
    select(name, uniprotID, log2FC, pvalue, padj, t) %&gt;%
    dplyr::rename(stat =t) %&gt;%
    arrange(pvalue) %&gt;% distinct(name,.keep_all = TRUE) %&gt;%
  as_tibble() 

allRes &lt;- left_join(protRes, rnaRes, by = &quot;name&quot;) %&gt;%
  filter(!is.na(stat), !is.na(stat.rna))</code></pre>
<p>Identify buffered and non-buffered proteins by comparing protein expression changes and RNA expression changes</p>
<pre class="r"><code>fdrCut &lt;- 0.05

bufferTab &lt;- allRes %&gt;% filter(chrom == &quot;12&quot;, stat.rna &gt; 0) %&gt;%
  ungroup() %&gt;%
  mutate(stat.prot.sqrt = sqrt(stat),
         stat.prot.center = stat.prot.sqrt - mean(stat.prot.sqrt,na.rm= TRUE)) %&gt;%
  mutate(score = -stat.prot.center*stat.rna,
         diffFC = log2FC.rna-log2FC) %&gt;%
  mutate(ifBuffer = case_when(
    padj &lt;= fdrCut &amp; padj.rna &lt;= fdrCut &amp; stat &gt; 0 ~ &quot;non-Buffered&quot;,
    padj &gt; fdrCut &amp; padj.rna &lt;= fdrCut ~ &quot;Buffered&quot;,
    padj &lt; fdrCut &amp; padj.rna &gt; fdrCut &amp; stat &gt; 0 ~ &quot;Enhanced&quot;,
    TRUE ~ &quot;Undetermined&quot;
  )) %&gt;%
  arrange(desc(score))</code></pre>
<p>Table of comparison</p>
<pre class="r"><code>bufferTab %&gt;% select(name, geneID, ifBuffer, score, log2FC, padj, log2FC.rna, padj.rna) %&gt;%
  mutate_if(is.numeric, formatC, digits=2) %&gt;%
  DT::datatable()</code></pre>
<div id="htmlwidget-e0d2c6e79f3c926924f1" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-e0d2c6e79f3c926924f1">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167"],["SUDS3","NDUFA9","PPHLN1","SMARCD1","AAAS","NUP107","COPS7A","FGFR1OP2","PIP4K2C","SNRPF","INTS13","NUP37","SARNP","ZCCHC8","CHD4","KRR1","PPP1CC","ENO2","DDX23","DCTN2","FKBP4","TBK1","PPTC7","ARPC3","BAZ2A","NOP2","HNRNPA1","RPLP0","BLOC1S1","NCKAP1L","HVCN1","WBP11","PITPNM2","COPZ1","VPS33A","PFKM","M6PR","ACADS","TBC1D15","SPRYD4","DYNLL1","OAS1","CKAP4","HIP1R","RPL6","NDUFA12","GNS","HEBP1","RBM19","ISCU","ARHGAP9","CCT2","DIP2B","DDX54","TESC","SMARCC2","ALDH2","WASHC4","DDX47","C12orf43","PCBP2","TXNRD1","WASHC3","METAP2","CDKN1B","TMPO","PHB2","TRAFD1","CLIP1","EMG1","TPI1","MMAB","TWF1","ITPR2","HSP90B1","PYM1","BIN2","RAB5B","METTL7A","SCYL2","MYL6","OAS2","C2CD5","CMAS","WNK1","YARS2","GAPDH","CBX5","GLTP","CORO1C","SCAF11","STAT6","XPOT","SART3","RAB35","LTA4H","ERP29","PWP1","MLEC","PA2G4","SP1","DENR","NEMP1","USP15","NAP1L1","UBE2N","PUS1","SHMT2","DIABLO","RAN","EIF4B","LEMD3","ESYT1","RAB21","PTPN6","ZNF384","TRIAP1","DGKA","PTGES3","ATP2B1","ATP2A2","IRAK4","PGAM5","SRSF9","LDHB","CNPY2","RECQL","DNM1L","TSFM","STRAP","USP5","TIGAR","PSMD9","CAND1","GOLGA3","EEA1","PRKAG1","GIT2","RPAP3","SLC25A3","STAT2","OSBPL8","DERA","ARHGDIB","GCN1","PPP1R12A","ANKRD13A","PTPN11","EIF2B1","TAOK3","CS","CPSF6","RSRC2","HDAC7","GATC","EP400","YBX3","TMEM263","COQ5","UTP20","VPS29","PFDN5","CNOT2","ARL6IP4","KDM5A","RFC5","OAS3"],["ENSG00000111707","ENSG00000139180","ENSG00000134283","ENSG00000066117","ENSG00000094914","ENSG00000111581","ENSG00000111652","ENSG00000111790","ENSG00000166908","ENSG00000139343","ENSG00000064102","ENSG00000075188","ENSG00000205323","ENSG00000033030","ENSG00000111642","ENSG00000111615","ENSG00000186298","ENSG00000111674","ENSG00000174243","ENSG00000175203","ENSG00000004478","ENSG00000183735","ENSG00000196850","ENSG00000111229","ENSG00000076108","ENSG00000111641","ENSG00000135486","ENSG00000089157","ENSG00000135441","ENSG00000123338","ENSG00000122986","ENSG00000084463","ENSG00000090975","ENSG00000111481","ENSG00000139719","ENSG00000152556","ENSG00000003056","ENSG00000122971","ENSG00000121749","ENSG00000176422","ENSG00000088986","ENSG00000089127","ENSG00000136026","ENSG00000130787","ENSG00000089009","ENSG00000184752","ENSG00000135677","ENSG00000013583","ENSG00000122965","ENSG00000136003","ENSG00000123329","ENSG00000166226","ENSG00000066084","ENSG00000123064","ENSG00000088992","ENSG00000139613","ENSG00000111275","ENSG00000136051","ENSG00000213782","ENSG00000157895","ENSG00000197111","ENSG00000198431","ENSG00000120860","ENSG00000111142","ENSG00000111276","ENSG00000120802","ENSG00000215021","ENSG00000135148","ENSG00000130779","ENSG00000126749","ENSG00000111669","ENSG00000139428","ENSG00000151239","ENSG00000123104","ENSG00000166598","ENSG00000170473","ENSG00000110934","ENSG00000111540","ENSG00000185432","ENSG00000136021","ENSG00000092841","ENSG00000111335","ENSG00000111731","ENSG00000111726","ENSG00000060237","ENSG00000139131","ENSG00000111640","ENSG00000094916","ENSG00000139433","ENSG00000110880","ENSG00000139218","ENSG00000166888","ENSG00000184575","ENSG00000075856","ENSG00000111737","ENSG00000111144","ENSG00000089248","ENSG00000136045","ENSG00000110917","ENSG00000170515","ENSG00000185591","ENSG00000139726","ENSG00000166881","ENSG00000135655","ENSG00000187109","ENSG00000177889","ENSG00000177192","ENSG00000182199","ENSG00000184047","ENSG00000132341","ENSG00000063046","ENSG00000174106","ENSG00000139641","ENSG00000080371","ENSG00000111679","ENSG00000126746","ENSG00000170855","ENSG00000065357","ENSG00000110958","ENSG00000070961","ENSG00000174437","ENSG00000198001","ENSG00000247077","ENSG00000111786","ENSG00000111716","ENSG00000257727","ENSG00000004700","ENSG00000087470","ENSG00000123297","ENSG00000023734","ENSG00000111667","ENSG00000078237","ENSG00000110801","ENSG00000111530","ENSG00000090615","ENSG00000102189","ENSG00000181929","ENSG00000139436","ENSG00000005175","ENSG00000075415","ENSG00000170581","ENSG00000091039","ENSG00000023697","ENSG00000111348","ENSG00000089154","ENSG00000058272","ENSG00000076513","ENSG00000179295","ENSG00000111361","ENSG00000135090","ENSG00000062485","ENSG00000111605","ENSG00000111011","ENSG00000061273","ENSG00000257218","ENSG00000183495","ENSG00000060138","ENSG00000151135","ENSG00000110871","ENSG00000120800","ENSG00000111237","ENSG00000123349","ENSG00000111596","ENSG00000182196","ENSG00000073614","ENSG00000111445","ENSG00000111331"],["Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Undetermined","non-Buffered","Undetermined","Undetermined","Buffered","Buffered","non-Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","non-Buffered","Undetermined","Buffered","Undetermined","Buffered","non-Buffered","Buffered","Buffered","Undetermined","Undetermined","Undetermined","Undetermined","Undetermined","non-Buffered","non-Buffered","non-Buffered","Undetermined","Enhanced","Undetermined","Undetermined","non-Buffered","Enhanced","non-Buffered","non-Buffered","non-Buffered","Undetermined","non-Buffered","Enhanced","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","Enhanced","non-Buffered","non-Buffered","Enhanced","non-Buffered","non-Buffered","non-Buffered","non-Buffered","Enhanced","Enhanced","non-Buffered","non-Buffered","Enhanced","Enhanced","non-Buffered","Enhanced","Enhanced","non-Buffered","non-Buffered","non-Buffered","non-Buffered","Enhanced","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","Enhanced","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","non-Buffered","Undetermined","Buffered","Undetermined","Buffered","Buffered","Buffered","Undetermined","Undetermined","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Buffered","Undetermined"],[" 13"," 13"," 12","8.9","8.4","8.4","  7","5.9","5.8","5.5","4.9","4.8","4.6","4.3","4.2","4.2","3.2","3.2","3.2","3.1","  3","  3","2.9","2.8","2.7","2.7","2.6","2.5","2.5","2.5","2.3","2.2","2.1","  2","  2","1.9","1.9","1.8","1.8","1.8","1.8","1.8","1.4","1.4","1.3","1.3","  1","  1","0.97","0.93","0.89","0.88","0.87","0.85","0.83","0.76","0.75","0.74","0.69","0.69","0.65","0.54","0.43","0.43","0.43","0.42","0.4","0.37","0.19","0.18","0.17","0.16","0.15","0.099","0.088","-0.13","-0.21","-0.39","-0.63","-0.65","-0.7","-0.71","-0.75","-0.85","-0.9"," -1"," -1"," -1","-1.1","-1.1","-1.1","-1.1","-1.2","-1.2","-1.3","-1.4","-1.4","-1.4","-1.5","-1.6","-1.8","-1.8","-1.8","-1.9"," -2","-2.4","-2.6","-2.6","-2.6","-2.6","-2.7","-2.7","-2.8","-2.9","-2.9"," -3","-3.1","-3.4","-3.5","-3.6","-3.7","-3.7","-3.8","-3.8","-3.8"," -4","-4.1","-4.2","-4.3","-4.4","-4.4","-4.4","-4.8","-4.9","-4.9","-4.9","-5.1","-5.3","-5.3","-5.8","-6.8","-6.8","-6.8","-7.3","-7.3","-7.3","-7.4","-8.8","-9.4","-9.8","-13","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN","NaN"],["0.016","0.037","0.052","0.04","0.023","0.047","0.1","0.055","-0.0074","0.11","0.18","0.06","0.068","0.094","0.036","0.12","0.11","0.065","0.11","0.046","0.065","0.12","0.21","0.19","0.28","0.17","0.15","0.21","0.17","0.29","0.18","0.15","0.21","0.11","0.26","0.28","0.21","0.28","0.13","0.33","0.12","0.78","0.53","1.6","0.2","0.049","0.33","0.082","0.14","0.3","0.29","0.17","0.42","0.32","0.25","0.22","0.76","0.55","0.28","0.32","0.22","0.41","0.41","0.44","0.38","0.27","0.26","0.5","0.44","0.21","0.28","0.6","0.74","0.76","0.33","0.56","0.75","0.44","0.62","0.45","0.38","0.54","0.41","0.46","0.56","0.56","0.43","0.49","0.5","0.65","0.42","0.41","0.44","0.3","0.67","0.51","0.55","0.49","0.34","0.58","0.62","0.48","0.72","0.61","0.51","0.44","0.6","0.92","0.56","0.3","0.56","0.6","0.69","0.5","0.82","0.5","0.98","0.5","0.49","  1","0.58","0.61","0.74","0.36","0.79","0.47","0.64","0.56","0.66","0.41","0.52","  1","0.6","0.47","0.84","0.81","0.52","0.61","0.44","0.59","0.7","0.79","0.72","0.82","0.52","0.67","0.9","0.6","0.74","0.79","0.7","-0.094","-0.099","-0.18","-0.13","-0.064","-0.17","-0.19","-0.095","-0.07","-0.023","-0.013","-0.029","-0.0088","-0.027","-0.00039","-0.037"],["0.99","0.76","0.58","0.73","0.77","0.51","0.45","0.66","0.81","0.25","0.082","0.72","0.63","0.23","0.75","0.3","0.26","0.8","0.041","0.74","0.66","0.11","0.14","0.04","0.063","0.06","0.063","0.068","0.051","0.0037","0.58","0.064","0.2","0.077","0.0095","0.11","0.11","0.18","0.4","0.12","0.28","0.13","0.014","0.0031","0.0047","0.73","0.024","0.77","0.2","0.0028","0.014","0.0026","0.0017","0.0013","0.39","0.0031","0.022","0.00089","0.0013","0.0025","0.00099","0.0014","0.002","0.00077","0.00075","0.0031","0.00056","0.00051","0.00042","0.0004","0.00078","0.0013","0.00034","0.00035","0.00042","6.5e-06","8.3e-05","2.1e-05","1.3e-09","3.2e-05","1.1e-05","9.2e-06","5.5e-05","6.9e-08","1.2e-05","1.1e-06","4.5e-06","1.4e-06","1.7e-05","3.1e-06","1e-05","3.1e-06","1.8e-07","2.8e-06","7e-06","4e-06","2.8e-07","6.1e-06","1.9e-06","3.5e-13","1.7e-07","9.1e-07","3e-08","1.3e-06","1.3e-07","3.5e-07","3.6e-08","1.7e-12","7e-11","1.1e-08","6.7e-10","9.5e-08","2.6e-12","5.3e-14","4.5e-11","7.7e-07","8e-11","1.1e-09","2e-13","1.1e-09","8.5e-11","1.2e-08","2.2e-11","4.2e-09","7.5e-11","1.4e-09","3.8e-08","1.4e-09","1.7e-11","2.1e-13","  0","  0","  0","1.4e-13","5.9e-11","  0","4.1e-11","8.9e-08","1.4e-13","9.2e-14","1.4e-10","  0","9.2e-14","  0","5.3e-14","  0","1.3e-13","  0","  0","  0","  0","0.055","0.17","0.21","0.21","0.38","0.41","0.42","0.52","0.57","0.64","0.77","0.85","0.94","0.95","0.95","0.97"],["0.5","0.73","0.56","0.5","0.86","0.63","0.57","0.59","0.32","0.52","0.57","0.33","0.5","0.5","0.32","0.67","0.49","0.63","0.47","0.2","0.42","0.48","0.58","0.5","0.58","0.64","0.43","0.42","0.58","0.9","0.27","0.61","0.33","0.35","0.51","0.41","0.31","0.4","0.34","0.33","0.37","0.54","0.62","2.3","0.49","0.13","0.34","0.15","0.12","0.47","0.28","0.6","0.72","0.6","0.29","0.32","0.42","1.1","0.88","0.37","0.55","0.41","0.26","0.43","0.8","0.29","0.5","0.71","0.48","0.41","0.29","0.11","0.77","0.52","0.32","0.089","0.51","0.45","0.22","0.52","0.43","0.5","0.65","0.2","0.5","0.29","0.53","0.42","0.45","0.45","0.54","0.45","0.25","0.52","0.57","0.6","0.43","0.89","0.48","0.16","0.51","0.46","0.38","0.58","0.41","0.62","0.53","0.46","0.61","0.58","0.34","0.59","0.38","0.47","0.69","0.69","0.57","0.49","0.37","0.93","0.5","0.63","0.56","0.45","0.48","0.7","0.64","0.71","0.54","0.69","0.48","0.64","0.47","0.56","0.61","0.76","0.46","0.82","0.54","0.73","  1","0.89","0.7","0.69","0.56","0.67","0.83","0.55","0.47","0.71","0.73","0.24","0.6","0.016","0.32","0.5","0.63","0.018","0.3","0.53","0.61","0.57","0.62","0.48","0.75","0.46","0.28"],["3.5e-09","1.4e-15","4.1e-19","1.6e-08","5e-07","5.3e-11","1.4e-08","0.00016","0.0021","4.7e-08","6.2e-12","0.0054","0.004","2.5e-05","0.022","0.0002","0.004","0.12","6.7e-08","0.11","0.089","0.00037","0.0018","3.3e-06","0.0001","0.00011","0.00022","0.0006","0.00022","3.4e-22","0.19","0.0028","0.065","0.014","2.2e-07","0.034","0.04","0.092","0.23","0.055","0.16","0.068","0.0035","3.4e-08","1.5e-05","0.6","0.086","0.71","0.47","0.00018","0.085","0.00022","3.5e-06","6.2e-08","0.65","0.0054","0.24","6.9e-10","2.2e-05","0.0054","8.7e-07","0.0026","0.075","0.00015","0.0001","0.18","6.2e-07","3.2e-07","0.00066","0.00029","0.23","0.47","3.5e-05","0.015","0.17","0.78","0.028","0.079","0.57","5.5e-05","0.006","0.0094","2.1e-11","0.22","0.00013","0.017","0.00094","0.01","4.7e-08","0.001","3.5e-06","0.00056","0.028","0.00022","5.5e-07","6.5e-06","0.0047","1.5e-07","2.3e-05","0.25","0.00063","5.7e-06","0.0044","2e-07","0.00022","3.5e-08","1.3e-05","0.033","0.0078","6.4e-05","0.0013","2.2e-07","0.015","0.037","0.0029","3e-15","0.00098","9.8e-06","0.0053","2.3e-06","3.9e-05","1.5e-09","9.8e-05","1.6e-08","2.2e-05","4.7e-08","1.5e-13","9.1e-09","1e-05","0.00022","0.00069","0.0089","0.004","3.1e-05","3.1e-08","0.00016","1.6e-08","3.2e-27","4.9e-06","1e-06","1.3e-16","1.4e-07","3.4e-09","3.1e-06","6.2e-10","2.2e-09","1.1e-10","1e-09","3.4e-11","1.6e-12","4.9e-21","0.089","0.00097","0.96","0.023","2.2e-09","0.0042","0.96","0.088","2.5e-05","2.1e-08","7.5e-07","9e-12","0.013","7.6e-09","0.0013","0.6"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>geneID<\/th>\n      <th>ifBuffer<\/th>\n      <th>score<\/th>\n      <th>log2FC<\/th>\n      <th>padj<\/th>\n      <th>log2FC.rna<\/th>\n      <th>padj.rna<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="summary-plot" class="section level3">
<h3>Summary plot</h3>
<pre class="r"><code>sumTab &lt;- bufferTab %&gt;% group_by(ifBuffer) %&gt;%
  summarise(n = length(name))

bufferPlot &lt;- ggplot(sumTab, aes(x=ifBuffer, y = n)) + 
  geom_bar(aes(fill = ifBuffer), stat=&quot;identity&quot;, width = 0.7) + 
  geom_text(aes(label = paste0(n)),vjust=-0.5,col=&quot;black&quot;,size=5) +
  scale_fill_manual(values =c(Buffered = colList[1],
                              Enhanced = colList[4],
                              `non-Buffered` = colList[2],
                              Undetermined = &quot;grey50&quot;)) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5),
                     legend.position = &quot;none&quot;) +
  ylab(&quot;Number of proteins&quot;) + ylim(0,120) +xlab(&quot;&quot;)
bufferPlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/sum_buffer_number-1.png" width="384" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;tri12_sum_buffer_number.pdf&quot;, width =4, height = 4)</code></pre>
</div>
<div id="enrichment-of-buffer-and-non-buffered-proteins" class="section level3">
<h3>Enrichment of buffer and non-buffered proteins</h3>
<div id="non-buffered-prpteins" class="section level4">
<h4>Non-buffered prpteins</h4>
<div id="using-cancer-hallmark-genesets" class="section level5">
<h5>Using cancer hallmark genesets</h5>
<pre class="r"><code>rnaAll &lt;- dds[rowData(dds)$biotype %in% &quot;protein_coding&quot; &amp; !rowData(dds)$symbol %in% c(&quot;&quot;,NA),] #all protein coding gene as background

protList &lt;- filter(bufferTab, ifBuffer == &quot;non-Buffered&quot;)$name
refList &lt;- rowData(rnaAll)$symbol
enRes &lt;- runFisher(protList, refList, gmts$H, pCut =0.01, ifFDR = TRUE,removePrefix = &quot;HALLMARK_&quot;,
                   plotTitle = &quot;Non-buffered proteins&quot;, insideLegend = TRUE,
                   setName = &quot;HALLMARK gene set&quot;, setMap = setMap)
bufferEnrich &lt;- enRes$enrichPlot + theme(plot.margin = margin(1,3,1,1, unit = &quot;cm&quot;))
bufferEnrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_nonBuffer_enrich-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;tri12_nonBuffer_enrich.pdf&quot;, height = 3.5, width = 7)</code></pre>
</div>
</div>
<div id="buffered-proteins" class="section level4">
<h4>Buffered proteins</h4>
<pre class="r"><code>protList &lt;- filter(bufferTab, ifBuffer == &quot;Buffered&quot;)$name
enRes &lt;- runFisher(protList, refList, gmts$H, pCut =0.1, ifFDR = TRUE)</code></pre>
<pre><code>[1] &quot;No sets passed the criteria&quot;</code></pre>
<p>No enrichment</p>
</div>
</div>
</div>
<div id="compare-the-protein-buffering-effect-between-trisomy19-and-trisomy12" class="section level2">
<h2>Compare the protein buffering effect between trisomy19 and trisomy12</h2>
<pre class="r"><code>load(&quot;../output/deResList.RData&quot;)
load(&quot;../output/deResListRNA.RData&quot;)
testTabProt &lt;- resList %&gt;% mutate(chr = rowData(protCLL[id,])$chromosome_name) %&gt;%
  filter(Gene == paste0(&quot;trisomy&quot;,chr)) %&gt;%
  select(name, log2FC, Gene) %&gt;% mutate(type = &quot;Protein&quot;)
testTabRNA &lt;- resListRNA %&gt;% mutate(chr = rowData(dds[id,])$chromosome) %&gt;%
  filter(Gene == paste0(&quot;trisomy&quot;,chr)) %&gt;%
  select(name, log2FC, Gene) %&gt;% mutate(type = &quot;RNA&quot;)
overGene &lt;- intersect(testTabProt$name, testTabRNA$name)
testTab &lt;- bind_rows(testTabProt, testTabRNA) %&gt;%
  filter(name %in% overGene)</code></pre>
<pre class="r"><code>plotTab &lt;- lapply(seq(-2,2, length.out = 50), function(foldCut) {
  filTab &lt;- mutate(testTab, pass = log2FC &gt; foldCut) %&gt;%
    group_by(Gene, type) %&gt;% summarise(n = sum(pass),per = sum(pass)/length(pass)) %&gt;% 
    mutate(cut = foldCut)
}) %&gt;% bind_rows() %&gt;%
  mutate(group =paste0(Gene,&quot;_&quot;,type))</code></pre>
<p>Cummulative curve plot</p>
<pre class="r"><code>ggplot(plotTab, aes(x=cut, y = per))+ 
  geom_line(aes(col = Gene, linetype = type),size=1) +
  scale_color_manual(values = c(trisomy12 = colList[1],trisomy19=colList[2]), name = &quot;&quot;) +
  scale_linetype_discrete(name = &quot;&quot;) +
  coord_cartesian(xlim=c(-0.5,1)) +
  ylab(&quot;Cumulative fraction&quot;) +
  xlab(bquote(&quot;log&quot;[2]*&quot;(fold change)&quot;)) +
  theme_full +
  theme(legend.position = c(0.80,0.80),
        axis.text = element_text(size=15),
        axis.title = element_text(size=15),
        legend.text = element_text(size=15),
        legend.background = element_rect(fill = NA))</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/buffer_Tri12vsTri19-1.png" width="460.8" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;buffer_Tri12vsTri19.pdf&quot;, height = 5, width = 4.8)</code></pre>
<div id="ks-test-to-compare-the-rna-expression-changes-and-protein-expression-changes-between-trisomy12-cases-and-trisomy19-cases" class="section level3">
<h3>KS test to compare the RNA expression changes and protein expression changes between trisomy12 cases and trisomy19 cases</h3>
<p>RNA level</p>
<pre class="r"><code>testTab &lt;- plotTab %&gt;% filter(type == &quot;RNA&quot;) %&gt;%
  select(Gene, per,cut) %&gt;% pivot_wider(names_from = Gene, values_from = per)
ks.test(testTab$trisomy12, testTab$trisomy19)</code></pre>
<pre><code>
    Two-sample Kolmogorov-Smirnov test

data:  testTab$trisomy12 and testTab$trisomy19
D = 0.22, p-value = 0.1777
alternative hypothesis: two-sided</code></pre>
<p>Protein level</p>
<pre class="r"><code>testTab &lt;- plotTab %&gt;% filter(type == &quot;Protein&quot;) %&gt;%
  select(Gene, per,cut) %&gt;% pivot_wider(names_from = Gene, values_from = per)
ks.test(testTab$trisomy12, testTab$trisomy19)</code></pre>
<pre><code>
    Two-sample Kolmogorov-Smirnov test

data:  testTab$trisomy12 and testTab$trisomy19
D = 0.42, p-value = 0.0002955
alternative hypothesis: two-sided</code></pre>
</div>
</div>
<div id="plot-protein-and-rna-expression-of-genes-togehter-with-their-genomic-coordinate-information" class="section level2">
<h2>Plot protein and RNA expression of genes togehter with their genomic coordinate information</h2>
<pre class="r"><code>patBack &lt;- dplyr::filter(patMeta, Patient.ID %in% unique(allProtTab$patID)) %&gt;%
  dplyr::select(Patient.ID, trisomy12) %&gt;%
  dplyr::rename(patID = Patient.ID) %&gt;%
  mutate_all(as.character) %&gt;%
  mutate_at(vars(-patID),str_replace, &quot;1&quot;,&quot;yes&quot;) %&gt;%
  mutate_at(vars(-patID),str_replace, &quot;0&quot;,&quot;no&quot;)</code></pre>
<pre class="r"><code>plotExprVar &lt;- function(gene, chr, patBack, allBand, allLine, allProtTab, allRnaTab, protLine = NULL,
                        region = c(-Inf,Inf),ifTrend = FALSE, normalize = TRUE, maxVal =2, minVal=-2) {
  
  #table for cyto band
  bandTab &lt;- filter(allBand, ChromID == chr, chromStart &gt;= region[1], chromEnd &lt;= region[2]) %&gt;%
    mutate(chromMid = chromMid)
  
  #table for expression
  plotProtTab &lt;- filter(allProtTab, ChromID == chr, start_position &gt;= region[1], end_position &lt;= region[2]) %&gt;%
    mutate_if(is.factor,as.character)
  plotRnaTab &lt;- filter(allRnaTab, ChromID == chr, start_position &gt;= region[1], end_position &lt;= region[2]) %&gt;%
    mutate_if(is.factor,as.character)
  
  
  #summarise group mean
  plotProtTab &lt;- plotProtTab %&gt;% 
    mutate(group = patBack[match(patID, patBack$patID),][[gene]]) %&gt;%
    filter(!is.na(group)) %&gt;%
    group_by(id, group) %&gt;% mutate(meanExpr = mean(expr, na.rm=TRUE)) %&gt;%
    distinct(group, id,.keep_all = TRUE) %&gt;% ungroup()

  plotRnaTab &lt;- plotRnaTab %&gt;%
    mutate(group = patBack[match(patID, patBack$patID),][[gene]]) %&gt;%
    filter(!is.na(group)) %&gt;%
    group_by(id, group) %&gt;% mutate(meanExpr = mean(expr, na.rm=TRUE)) %&gt;%
    distinct(group, id,.keep_all = TRUE) %&gt;% ungroup()
  
  if (!is.null(protLine)) {
    bufferLineTab &lt;- plotProtTab %&gt;%
      select(symbol, mid_position, meanExpr, group) %&gt;%
      filter(symbol %in% protLine) %&gt;%
      pivot_wider(names_from = group, values_from = meanExpr) %&gt;%
      mutate(lowVal = map2_dbl(yes, no, min),
             highVal = map2_dbl(yes, no, max))
  } else bufferLineTab &lt;- NULL
  

  xMax &lt;- max(bandTab$chromEnd, na.rm = T)
  
  #main plot for Protein
  gPro &lt;- ggplot() + 
    geom_rect(data=bandTab, mapping=aes(xmin=chromStart, xmax=chromEnd, ymin=minVal+0.5, ymax=maxVal-0.5, 
                                        fill=Colour, label = band), alpha=0.1) +
    geom_text(data=bandTab, mapping=aes(label=band, x=chromMid), y=maxVal-0.5, hjust =1, angle = 90, size=2.5) 
    
  if (!is.null(protLine)) {
    gPro &lt;- gPro + geom_segment(data = bufferLineTab, aes(x=mid_position, xend = mid_position, 
                                        y=lowVal, yend = highVal), linetype = &quot;dashed&quot;) 
    
  }
  
  gPro &lt;- gPro + geom_rect(data = plotProtTab, 
            mapping=aes(xmin=start_position,
                        xmax=end_position, ymin=meanExpr, ymax=meanExpr+0.1,
                        fill = group, label = symbol)) +
    scale_x_continuous(expand=c(0,0),limits = c(0,xMax)) +
    xlab(&quot;Genomic position [Mb]&quot;) + 
    ylab(&quot;Expression Z-score&quot;) + 
    scale_fill_manual(values = c(even = &quot;white&quot;,odd = &quot;grey50&quot;,
                                yes = colList[1], no = colList[4])) +
    scale_color_manual(values = c(yes = colList[1],no = colList[4])) +
    ggtitle(paste0(&quot;Protein expression&quot;,&quot;(&quot;,chr,&quot;)&quot;)) +
    theme(plot.title = element_text(face = &quot;bold&quot;, size = 10),
        legend.position = &quot;none&quot;,
        panel.background = element_blank(),
        panel.grid.major = element_line(colour=&quot;grey90&quot;, size=0.1))
  
    if (ifTrend) {
      gPro &lt;- gPro + stat_smooth(data =filter(plotProtTab, expr &gt;0), 
                mapping = aes(y=meanExpr, x= mid_position,
                              color = group), 
                geom=&quot;line&quot;,
                formula = y ~ x, method = &quot;loess&quot;, se=FALSE, span=0.5,
                size =1, alpha=0.5)
    }
    
  #main plot for RNA
  gRna &lt;- ggplot() + 
    geom_rect(data=bandTab, mapping=aes(xmin=chromStart, xmax=chromEnd, ymin=minVal, ymax=maxVal, 
                                        fill=Colour, label = band), alpha=0.1) +
    geom_text(data=bandTab, mapping=aes(label=band, x=chromMid), y=maxVal, hjust =1, angle = 90, size=2.5) +
    geom_rect(data = plotRnaTab, 
            mapping=aes(xmin=start_position,
                        xmax=end_position, ymin=meanExpr, ymax=meanExpr+0.1,
                        fill = group, label = symbol)) +
    scale_x_continuous(expand=c(0,0),limits = c(0,xMax)) +
    xlab(&quot;Genomic position [Mb]&quot;) + 
    ylab(&quot;Expression Z-score&quot;) + 
    scale_fill_manual(values = c(even = &quot;white&quot;,odd = &quot;grey50&quot;,
                                yes = colList[1], no = colList[4])) +
    scale_color_manual(values = c(yes = colList[1], no = colList[4])) +
    ggtitle(paste0(&quot;RNA expression&quot;,&quot;(&quot;,chr,&quot;)&quot;)) +
    theme(plot.title = element_text(face = &quot;bold&quot;, size = 10),
        legend.position = &quot;none&quot;,
        panel.background = element_blank(),
        panel.grid.major = element_line(colour=&quot;grey90&quot;, size=0.1))
  
    if (ifTrend) {
      gRna &lt;- gRna + stat_smooth(data =filter(plotRnaTab), 
                mapping = aes(y=meanExpr, x= mid_position,
                              color = group), 
                geom=&quot;line&quot;,
                formula = y ~ x, method = &quot;loess&quot;, se=FALSE, span=0.2,
                size =1, alpha=0.5)
    }
    #for legend
    ## if the patient has CNV data
    lgTab &lt;- tibble(x= seq(6),y=seq(6),
                    Expression = c(rep(&quot;yes&quot;,3), rep(&quot;no&quot;,3)))
  
    lg &lt;- ggplot(lgTab, aes(x=x,y=y)) +
        geom_point(aes(fill = Expression), shape =22,size=3) +
        scale_fill_manual(values = c(yes = &quot;darkred&quot;, no = &quot;darkgreen&quot;), name = gene) +
        theme(legend.position = &quot;bottom&quot;)
    
    lg &lt;- get_legend(lg)
    
    return(list(plotPro = gPro, plotRNA = gRna, legend = lg))
}</code></pre>
<p>Normalize protein and RNA expression</p>
<pre class="r"><code>normalized &lt;- TRUE
#if perform normalization
if (normalized) {
  #for protein
  exprMat &lt;- select(allProtTab,patID, id,expr) %&gt;% 
    distinct(patID, id, .keep_all = TRUE) %&gt;%
    spread(key = patID, value =expr) %&gt;% data.frame() %&gt;% 
    column_to_rownames(&quot;id&quot;) %&gt;% as.matrix()
  qm &lt;- jyluMisc::mscale(exprMat, useMad = F)
  normTab &lt;- data.frame(qm) %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
    gather(key = &quot;patID&quot;, value = &quot;expr&quot;, -id)
  allProtTab &lt;- select(allProtTab, -expr) %&gt;% left_join(normTab, by = c(&quot;patID&quot;,&quot;id&quot;))


  #for RNA
  exprMat &lt;- select(allRnaTab,patID, id,expr) %&gt;% 
    distinct(patID, id, .keep_all = TRUE) %&gt;%
    spread(key = patID, value =expr) %&gt;% data.frame() %&gt;% 
    column_to_rownames(&quot;id&quot;) %&gt;% as.matrix()
  qm &lt;- jyluMisc::mscale(exprMat, useMad = F)
  normTab &lt;- data.frame(qm) %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
    gather(key = &quot;patID&quot;, value = &quot;expr&quot;, -id)
  allRnaTab &lt;- select(allRnaTab, -expr) %&gt;% left_join(normTab, by = c(&quot;patID&quot;,&quot;id&quot;))
}</code></pre>
<pre class="r"><code>g &lt;- plotExprVar(&quot;trisomy12&quot;,&quot;chr12&quot;,patBack,allBand, allLine, 
                 allProtTab, allRnaTab, ifTrend = TRUE)
plot_grid(g$plotRNA, g$plotPro, g$legend, ncol = 1, rel_heights = c(1,1,0.2))</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_geneCoord-1.png" width="768" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;chr12Plot.pdf&quot;, height = 5, width = 8)</code></pre>
</div>
<div id="using-circular-heatmap-to-visuailize-rna-and-protein-expressions-of-chr12-genes" class="section level2">
<h2>Using circular heatmap to visuailize RNA and protein expressions of chr12 genes</h2>
<div id="prepare-data-for-plotting" class="section level3">
<h3>Prepare data for plotting</h3>
<p>Protein expression data</p>
<pre class="r"><code>protChr12 &lt;- protCLL[rowData(protCLL)$chromosome_name %in% &quot;12&quot;,]

#subset for fast testing
#protChr12 &lt;- protChr12[sample(rownames(protChr12), size = as.integer(nrow(protChr12)/5))]

protMat &lt;- assays(protChr12)[[&quot;QRILC_combat&quot;]] 
rownames(protMat) &lt;- rowData(protChr12)$hgnc_symbol
#column annotation
colAnno &lt;- tibble(patID = colnames(protChr12),
                  trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 %in% &quot;1&quot;,&quot;Tri12&quot;,&quot;other&quot;)) %&gt;%
  arrange(trisomy12) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;patID&quot;)

protMat &lt;- protMat[!duplicated(rownames(protMat)),rownames(colAnno)]</code></pre>
<p>RNA expression data</p>
<pre class="r"><code>rnaChr12 &lt;- ddsCLL.vst[rowData(ddsCLL.vst)$symbol %in% rownames(protMat),]
rnaMat &lt;- assay(rnaChr12)
rownames(rnaMat) &lt;- rowData(rnaChr12)$symbol

colAnnoRNA &lt;- tibble(patID = colnames(rnaChr12),
                  trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 %in% &quot;1&quot;,&quot;Tri12&quot;,&quot;other&quot;)) %&gt;%
  arrange(trisomy12) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;patID&quot;)


rnaMat &lt;- rnaMat[!duplicated(rownames(rnaMat)) ,rownames(colAnnoRNA)]

#subset for commonly detected genes
overGene &lt;- intersect(rownames(rnaMat), rownames(protMat))
protMat &lt;- protMat[overGene,]
rnaMat &lt;- rnaMat[overGene,]</code></pre>
<p>Get chromosome band membership for each protein</p>
<pre class="r"><code>library(biomaRt)
allID &lt;- rowData(protChr12)$ensembl_gene_id
idToSymbol &lt;- rowData(protChr12)[,c(&quot;ensembl_gene_id&quot;,&quot;hgnc_symbol&quot;)] %&gt;% as_tibble()
mart&lt;-useMart(biomart=&quot;ENSEMBL_MART_ENSEMBL&quot;,
              dataset=&quot;hsapiens_gene_ensembl&quot;,
              host=&quot;grch37.ensembl.org&quot;)

locAnno &lt;- getBM(values = allID ,mart = mart, 
      attributes = c(&quot;ensembl_gene_id&quot;,&quot;start_position&quot;,&quot;end_position&quot;),
      filters = &quot;ensembl_gene_id&quot;) %&gt;%
  distinct(ensembl_gene_id, .keep_all = TRUE)

geneAnno &lt;- tibble(id = allID) %&gt;%
  left_join(locAnno, by = c(id= &quot;ensembl_gene_id&quot;)) %&gt;%
  left_join(idToSymbol, by = c(id = &quot;ensembl_gene_id&quot;))

stopifnot(all(!is.na(geneAnno$hgnc_symbol)))

chr12Band &lt;- filter(allBand, ChromID == &quot;chr12&quot;)
bandTab &lt;- lapply(seq(nrow(geneAnno)), function(i) {
  rec &lt;- geneAnno[i,]
  bandName &lt;- mutate(chr12Band, within = chromStart &lt; rec$start_position/10^6 &amp; chromEnd &gt; rec$start_position/10^6) %&gt;%
    filter(within) %&gt;% pull(band)
  tibble(id = rec$id, band = as.character(bandName))
}) %&gt;% bind_rows()

geneAnno &lt;- left_join(geneAnno, bandTab, by = &quot;id&quot;) %&gt;%
  arrange(start_position) %&gt;%
  distinct(hgnc_symbol, .keep_all = TRUE) %&gt;%
  mutate(band = factor(band, levels = unique(band)))


detach(&quot;package:biomaRt&quot;, unload=TRUE)</code></pre>
<p>Prepare row annotation</p>
<pre class="r"><code>#for annotate protein regulation
upProtList &lt;- filter(resList, Gene == &quot;trisomy12&quot;, t &gt;0, adj.P.Val &lt;= 0.05)$name
upRNAList &lt;- filter(resListRNA, Gene == &quot;trisomy12&quot;, t &gt;0, adj.P.Val &lt;= 0.05)$name

rowAnno &lt;- tibble(symbol = rownames(protMat)) %&gt;%
  left_join(dplyr::select(geneAnno, hgnc_symbol, band, start_position), by = c(symbol = &quot;hgnc_symbol&quot;)) %&gt;%
  mutate(direction = case_when(
    symbol %in% intersect(upProtList, upRNAList) ~ &quot;both up&quot;,
    symbol %in% upProtList &amp; !symbol %in% upRNAList ~ &quot;protein up&quot;,
    symbol %in% upRNAList &amp; !symbol %in% upProtList ~ &quot;RNA up&quot;,
    TRUE ~ &quot;none&quot;
  )) %&gt;%
  mutate(`protUp` = ifelse(direction %in% c(&quot;both up&quot;,&quot;protein up&quot;),&quot;yes&quot;,&quot;no&quot;),
         `rnaUp` = ifelse(direction %in% c(&quot;both up&quot;,&quot;RNA up&quot;), &quot;yes&quot;,&quot;no&quot;)) %&gt;%
  arrange(start_position) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;symbol&quot;)
protMat &lt;- protMat[rownames(rowAnno),]
rnaMat &lt;- rnaMat[rownames(rowAnno),]</code></pre>
</div>
<div id="plot-heatmap-using-circos-package-with-gene-names" class="section level3">
<h3>Plot heatmap using circos package (with gene names)</h3>
<pre class="r"><code>library(circlize)
library(ComplexHeatmap)

protMat &lt;- mscale(protMat, censor = 5)
rnaMat &lt;- mscale(rnaMat, censor =5)
pdf(&quot;heatmap_tri12_circle.pdf&quot;, height=10, width=10)

#global gap size
circos.par(gap.after = c(rep(2, length(unique(rowAnno$band))-1),6))


#protein heatmap
col_funProt = colorRamp2(c(-5, 0, 5), c(colList[2], &quot;white&quot;, colList[1]))
circos.heatmap(protMat, split = rowAnno$band, cluster = FALSE, col = col_funProt, show.sector.labels = FALSE, rownames.side = &quot;outside&quot;)

#column annotation of sample type for protein
nTri12 &lt;- table(colAnno$trisomy12)[&quot;Tri12&quot;]
nWT &lt;- table(colAnno$trisomy12)[[&quot;other&quot;]]
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    if(CELL_META$sector.numeric.index == length(unique(rowAnno$band))) { # the last sector
        circos.rect(CELL_META$cell.xlim[2] + convert_x(1, &quot;mm&quot;), 0,
                    CELL_META$cell.xlim[2] + convert_x(7, &quot;mm&quot;), nWT,
                    col = NA, border = &quot;grey60&quot;)
        circos.text(CELL_META$cell.xlim[2] + convert_x(4, &quot;mm&quot;), nWT/2,
                    &quot;other&quot;, cex = 0.5, facing = &quot;clockwise&quot;, font=2)

        circos.rect(CELL_META$cell.xlim[2] + convert_x(1, &quot;mm&quot;), nWT,
                    CELL_META$cell.xlim[2] + convert_x(7, &quot;mm&quot;), nWT+nTri12,
                    col = NA, border = &quot;grey60&quot;)
        circos.text(CELL_META$cell.xlim[2] + convert_x(4, &quot;mm&quot;), nWT + nTri12/2,
                    &quot;tri12&quot;, cex = 0.5, facing = &quot;clockwise&quot;, font=2)
    }
}, bg.border = NA)

#annotation of prtein regulation
col_ProtDir = c(yes = colList[3], no = &quot;grey90&quot;)
circos.heatmap(rowAnno$protUp, col = col_ProtDir, track.height = 0.02, bg.border = &quot;grey50&quot;)

#annotation of rna regulation
col_RnaDir = c(yes = colList[5], no = &quot;grey90&quot;)
circos.heatmap(rowAnno$rnaUp, col = col_RnaDir, track.height = 0.02, bg.border = &quot;grey50&quot;)

#rna heatmap
col_funRna = colorRamp2(c(-5, 0, 5), c(colList[4], &quot;white&quot;, colList[1]))
circos.heatmap(rnaMat, split = rowAnno$band, cluster = FALSE, col = col_funRna)

#column annotation of sample type for rna
nTri12 &lt;- table(colAnnoRNA$trisomy12)[&quot;Tri12&quot;]
nWT &lt;- table(colAnnoRNA$trisomy12)[[&quot;other&quot;]]
circos.track(track.index = get.current.track.index(), panel.fun = function(x, y) {
    if(CELL_META$sector.numeric.index == length(unique(rowAnno$band))) { # the last sector
        circos.rect(CELL_META$cell.xlim[2] + convert_x(1, &quot;mm&quot;), 0,
                    CELL_META$cell.xlim[2] + convert_x(5, &quot;mm&quot;), nWT,
                    col = NA, border = &quot;grey60&quot;)
        circos.text(CELL_META$cell.xlim[2] + convert_x(3, &quot;mm&quot;), nWT/2,
                    &quot;other&quot;, cex = 0.5, facing = &quot;clockwise&quot;, font=2)

        circos.rect(CELL_META$cell.xlim[2] + convert_x(1, &quot;mm&quot;), nWT,
                    CELL_META$cell.xlim[2] + convert_x(5, &quot;mm&quot;), nWT+nTri12,
                    col = NA, border = &quot;grey60&quot;)
        circos.text(CELL_META$cell.xlim[2] + convert_x(3, &quot;mm&quot;), nWT + nTri12/2,
                    &quot;tri12&quot;, cex = 0.5, facing = &quot;clockwise&quot;, font=2)
    }
}, bg.border = NA)

circos.track(ylim = c(0, 1), panel.fun = function(x, y) {
    circos.text(CELL_META$xcenter, CELL_META$ycenter, 
        unique(rowAnno$band)[CELL_META$sector.numeric.index],
        facing = &quot;clockwise&quot;, niceFacing = TRUE,  cex = 0.7, col = &quot;grey50&quot;)
}, bg.border = NA, track.height = 0.06)


circos.clear()



dev.off()</code></pre>
<pre><code>pdf 
  2 </code></pre>
<p>Plot legend separately</p>
<pre class="r"><code>#draw legend
#pdf(&quot;heatmap_tri12_circle_legend.pdf&quot;, height=5, width=5)

#draw legend

lgd_Prot = Legend(title = &quot;Protein expression&quot;, col_fun = col_funProt, title_position = &quot;lefttop-rot&quot;)
lgd_Rna = Legend(title = &quot;RNA expression&quot;, col_fun = col_funRna, title_position = &quot;lefttop-rot&quot;)
lgd_ProtDir = Legend(title = &quot;Protein up-regulated &quot;, at = names(col_ProtDir), legend_gp = gpar(fill = col_ProtDir))
lgd_RnaDir = Legend(title = &quot;RNA up-regulated&quot;, at = names(col_RnaDir), legend_gp = gpar(fill = col_RnaDir))

lgd_list = packLegend(lgd_Prot, lgd_Rna, lgd_ProtDir, lgd_RnaDir, direction = &quot;vertical&quot;)
grid.draw(lgd_list) </code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="protein-complex-analysis" class="section level1">
<h1>Protein complex analysis</h1>
<div id="data-preprocessing" class="section level2">
<h2>Data preprocessing</h2>
<p>Read stable protein complex pairs</p>
<pre class="r"><code>int_pairs &lt;- read_csv2(&quot;../output/int_pairs.csv&quot;)</code></pre>
</div>
<div id="whether-complex-formation-affects-protein-buffering" class="section level2">
<h2>Whether complex formation affects protein buffering?</h2>
<div id="compare-log2-fold-changes-difference" class="section level3">
<h3>Compare log2 fold changes difference</h3>
<pre class="r"><code>testTab &lt;- bufferTab %&gt;% mutate(inComplex = ifelse(uniprotID %in% c(int_pairs$ProtA,int_pairs$ProtB),
                                                   &quot;complex in&quot;,&quot;complex out&quot;)) %&gt;%
  group_by(inComplex) %&gt;% mutate(n = length(name)) %&gt;%
  mutate(xLabel = sprintf(&quot;%s\n(N=%s)&quot;,inComplex,n))

tRes &lt;- t.test(diffFC~inComplex, testTab)

pVal &lt;- formatC(tRes$p.value, digits = 1, format=&quot;e&quot;)
pLab &lt;- bquote(italic(&quot;P&quot;)~&quot;=&quot;~.(pVal))

ggplot(testTab, aes(x=xLabel, y=diffFC)) + 
  geom_violin(aes(fill = xLabel), trim=FALSE) + 
  stat_summary(fun.data=&quot;mean_sdl&quot;, mult=1, 
                 geom=&quot;errorbar&quot;, width=0.05) +
  stat_summary(fun.y=mean, geom=&quot;point&quot;) +
  scale_fill_manual(values = colList[4:5]) +
  theme_full +
  xlab(&quot;&quot;) + ylab(&quot;log2(RNA fold change) - log2(protein fold change)&quot;) +
  annotate(&quot;text&quot;, label = pLab , x=Inf, y=Inf, hjust=1.2, vjust=2,size=6) +
  theme(legend.position = &quot;none&quot;,
        axis.text.y = element_text(size=15),
        axis.text.x = element_text(size=18),
        axis.title = element_text(size=15)) </code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-30-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;bufferComplexViolin.pdf&quot;, height = 6, width = 4.5)</code></pre>
</div>
</div>
<div id="construct-protein-protein-interaction-network-by-connecting-chr12-proteins-and-non-chr12-protein" class="section level2">
<h2>Construct protein-protein interaction network by connecting chr12 proteins and non-chr12 protein</h2>
<pre class="r"><code>comTab &lt;- int_pairs %&gt;% select(ProtA, ProtB, database) %&gt;%
  mutate(chrA = rowData(protCLL[ProtA,])$chromosome_name,
         chrB = rowData(protCLL[ProtB,])$chromosome_name) %&gt;%
  filter(!is.na(chrA), !is.na(chrB)) %&gt;%
  filter((chrA == &quot;12&quot; &amp; chrB != &quot;12&quot;) | (chrA !=&quot;12&quot; &amp; chrB == &quot;12&quot;)) %&gt;%
  mutate(source = ifelse(chrA == 12, ProtA, ProtB),
         target = ifelse(chrA == 12, ProtB, ProtA)) %&gt;%
  select(source, target, database)</code></pre>
<pre class="r"><code>fdrCut &lt;- 0.05
resTab &lt;- select(allRes, name, uniprotID, chrom, padj, padj.rna, log2FC,log2FC.rna) %&gt;%
  mutate(sigProt = padj &lt;= fdrCut,
         sigRna = padj.rna &lt;=fdrCut,
         upProt = sigProt &amp; log2FC &gt; 0,
         upRna = sigRna &amp; log2FC.rna &gt; 0)
comTab &lt;- comTab %&gt;% 
  left_join(resTab, by = c(source = &quot;uniprotID&quot;)) %&gt;%
  left_join(resTab, by = c(target = &quot;uniprotID&quot;)) %&gt;%
  rename_all(funs(str_replace(., &quot;x&quot;, &quot;source&quot;))) %&gt;%
  rename_all(funs(str_replace(., &quot;y&quot;, &quot;target&quot;))) </code></pre>
<pre class="r"><code>comTab.filter &lt;- filter(comTab, upRna.source, upProt.source, upProt.target)</code></pre>
<pre class="r"><code>#get node list
allNodes &lt;- union(comTab.filter$name.source, comTab.filter$name.target) 

nodeList &lt;- data.frame(id = seq(length(allNodes))-1, name = allNodes, stringsAsFactors = FALSE) %&gt;%
  mutate(chromosome = ifelse(rowData(protCLL[match(name, rowData(protCLL)$hgnc_symbol),])$chromosome_name %in% &quot;12&quot;, 
                          &quot;chr12&quot;,&quot;other&quot;))

#get edge list
edgeList &lt;- select(comTab.filter, name.source, name.target, database, upRna.target) %&gt;%
  dplyr::rename(Source = name.source, Target = name.target) %&gt;% 
  mutate(Source = nodeList[match(Source,nodeList$name),]$id,
         Target = nodeList[match(Target, nodeList$name),]$id,
         sigRNA = ifelse(upRna.target,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  data.frame(stringsAsFactors = FALSE)

net &lt;- graph_from_data_frame(vertices = nodeList, d=edgeList, directed = FALSE)</code></pre>
<pre class="r"><code>tidyNet &lt;- as_tbl_graph(net)
complexNet &lt;- ggraph(tidyNet, layout = &quot;igraph&quot;, algorithm = &quot;nicely&quot;) + 
  geom_edge_link(color = colList[3], width=1) + 
  geom_node_point(aes(color =chromosome, shape = chromosome), size=6) + 
  geom_node_text(aes(label = name), repel = TRUE, size=6) +
  #scale_edge_linetype_manual(values = c(no = &quot;dotted&quot;, yes = &quot;solid&quot;), name = &quot;RNA up-regulated&quot;)+
  scale_color_manual(values = c(chr12 = colList[1],other = colList[2])) +
  scale_edge_color_brewer(palette = &quot;Set2&quot;) +
  theme_graph(base_family = &quot;sans&quot;) + theme(legend.position = &quot;bottom&quot;) 
complexNet</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12Complex-1.png" width="1440" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#ggsave(&quot;trisomy12Complex.pdf&quot;, height = 15, width = 15)</code></pre>
</div>
</div>
<div id="associate-gene-expression-protein-expression-and-pathways-on-chr12-and-chr19" class="section level1">
<h1>Associate gene expression, protein expression and pathways on Chr12 and Chr19</h1>
<pre class="r"><code>load(&quot;../output/deResListRNA_allGene.RData&quot;)
load(&quot;../output/deResList.RData&quot;)
rnaRes &lt;- resListRNA_allGene %&gt;% filter(Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(Chr = rowData(dds[id,])$chromosome) %&gt;%
  filter(Chr == &quot;12&quot;, adj.P.Val &lt;= 0.05, t &gt; 0) %&gt;%
  distinct(name)

protRes &lt;- resList %&gt;% filter(Gene == &quot;trisomy12&quot;) %&gt;%
    mutate(Chr = rowData(protCLL[id,])$chromosome_name) %&gt;%
    filter(Chr == &quot;12&quot;, adj.P.Val &lt;=0.05, t &gt;0) %&gt;%
    distinct(name)</code></pre>
</div>
<div id="associations-between-pi3kaktmtor-inhibitor-responses-and-trisomy12" class="section level1">
<h1>Associations between PI3K/Akt/mTOR inhibitor responses and trisomy12</h1>
<div id="prepare-table" class="section level3">
<h3>Prepare table</h3>
<pre class="r"><code>load(&quot;../data/screenData_enc.RData&quot;)
screenData &lt;- screenData %&gt;% filter(patientID %in% colnames(protCLL),
                                        Drug %in% c(&quot;Duvelisib&quot;,&quot;MK-2206&quot;,&quot;Rapamycin&quot;)) %&gt;%
  group_by(patientID, Drug) %&gt;% summarise(viab = mean(normVal.cor_auc)) %&gt;%
  mutate(trisomy12 = patMeta[match(patientID, patMeta$Patient.ID),]$trisomy12,
         IGHV = patMeta[match(patientID, patMeta$Patient.ID),]$IGHV.status) %&gt;%
  mutate(status = ifelse(trisomy12==1,&quot;trisomy12&quot;,&quot;other&quot;),
         IGHV = ifelse(IGHV==&quot;M&quot;,&quot;M-CLL&quot;,&quot;U-CLL&quot;),
         Drug = as.character(Drug))</code></pre>
<p>Sample size</p>
<pre class="r"><code>table(distinct(screenData, patientID, trisomy12)$trisomy12)</code></pre>
<pre><code>
 0  1 
59 23 </code></pre>
</div>
<div id="t-test" class="section level3">
<h3>T-test</h3>
<p>All samples</p>
<pre class="r"><code>tRes &lt;- group_by(screenData, Drug) %&gt;% nest() %&gt;%
  mutate(m = map(data, ~t.test(viab~trisomy12,.))) %&gt;%
  mutate(res = map(m, broom::tidy)) %&gt;%
  unnest(res) %&gt;%
  select(Drug, estimate, p.value)
tRes</code></pre>
<pre><code># A tibble: 3 x 3
# Groups:   Drug [3]
  Drug      estimate p.value
  &lt;chr&gt;        &lt;dbl&gt;   &lt;dbl&gt;
1 Duvelisib   0.0787 0.0117 
2 MK-2206     0.0324 0.00631
3 Rapamycin   0.0568 0.0139 </code></pre>
<p>IGHV stratified</p>
<pre class="r"><code>tRes.ighv &lt;- group_by(screenData, Drug, IGHV) %&gt;% nest() %&gt;%
  mutate(m = map(data, ~t.test(viab~trisomy12,.))) %&gt;%
  mutate(res = map(m, broom::tidy)) %&gt;%
  unnest(res) %&gt;%
  select(Drug,IGHV, estimate, p.value)
tRes.ighv</code></pre>
<pre><code># A tibble: 6 x 4
# Groups:   Drug, IGHV [6]
  Drug      IGHV  estimate  p.value
  &lt;chr&gt;     &lt;chr&gt;    &lt;dbl&gt;    &lt;dbl&gt;
1 Duvelisib M-CLL   0.0438 0.340   
2 MK-2206   M-CLL   0.0111 0.406   
3 Rapamycin M-CLL   0.0507 0.217   
4 Duvelisib U-CLL   0.103  0.000474
5 MK-2206   U-CLL   0.0517 0.00422 
6 Rapamycin U-CLL   0.0637 0.0122  </code></pre>
</div>
<div id="visualization" class="section level3">
<h3>Visualization</h3>
<div id="all-samples" class="section level4">
<h4>All samples</h4>
<p>Function for plot drug effect as boxplots</p>
<pre class="r"><code>plotDrugBox &lt;- function(screenData, tRes, y_lab = &quot;Viability&quot;) {
  ymax &lt;- max(screenData$viab)
  ymin &lt;- min(screenData$viab)
  plotList &lt;- lapply(unique(screenData$Drug), function(n) {
    eachTab &lt;- filter(screenData, Drug == n) %&gt;%
      group_by(status) %&gt;% mutate(n=n()) %&gt;% ungroup() %&gt;%
      mutate(group = sprintf(&quot;%s\n(N=%s)&quot;,status,n)) %&gt;%
      arrange(status) %&gt;% mutate(group = factor(group, levels = unique(group)))
  
    pval &lt;- formatNum(filter(tRes, Drug == n)$p.value, digits = 1, format=&quot;e&quot;)
    annoText &lt;- bquote(.(n)~&quot; (&quot;~italic(&quot;P&quot;)~&quot;=&quot;~.(pval)~&quot;)&quot;)
      
    
    ggplot(eachTab, aes(x=group, y = viab)) +
      geom_beeswarm(aes(col=IGHV), size =2.5,cex = 2, alpha=0.8) +
      geom_boxplot(fill = NA, width=0.3, outlier.shape = NA) +
      ggtitle(annoText)+
      #ggtitle(sprintf(&quot;%s (p = %s)&quot;,geneName, formatNum(pval, digits = 1, format = &quot;e&quot;))) +
      ylab(y_lab) + xlab(&quot;&quot;) +
      scale_color_manual(values = colList[2:3]) +
      scale_y_continuous(limits=c(ymin,ymax),labels = scales::number_format(accuracy = 0.1))+
      theme_full +
      theme(legend.position = &quot;bottom&quot;,
            plot.title = element_text(hjust = 0.5),
            plot.margin = margin(0,20,0,20))

   })
  return(plotList)
}</code></pre>
<p>All samples</p>
<pre class="r"><code>pList &lt;- plotDrugBox(screenData, tRes)
plot_grid(plotlist = pList, ncol=3)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-41-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="ighv-stratified" class="section level4">
<h4>IGHV stratified</h4>
<p>Function for plot drug effect as boxplots</p>
<pre class="r"><code>plotDrugBoxIGHV &lt;- function(screenData, tRes.ighv, y_lab = &quot;Viability&quot;) {
  screenData &lt;- mutate(screenData, drugIGHV =paste0(Drug,&quot;_&quot;,IGHV))
  tRes.ighv &lt;- mutate(tRes.ighv, drugIGHV = paste0(Drug, &quot;_&quot;,IGHV))
  ymax &lt;- max(screenData$viab)
  ymin &lt;- min(screenData$viab)
  
  plotList &lt;- lapply(unique(screenData$drugIGHV), function(n) {
    eachTab &lt;- filter(screenData, drugIGHV == n) %&gt;%
      group_by(status) %&gt;% mutate(n=n()) %&gt;% ungroup() %&gt;%
      mutate(group = sprintf(&quot;%s\n(N=%s)&quot;,status,n)) %&gt;%
      arrange(status) %&gt;% mutate(group = factor(group, levels = unique(group)))
    drug &lt;- unique(eachTab$Drug)
    ighv &lt;- unique(eachTab$IGHV)
    pval &lt;- formatNum(filter(tRes.ighv, drugIGHV == n)$p.value, digits = 1, format=&quot;e&quot;)
    annoText &lt;- bquote(.(drug)~&quot; (&quot;~italic(&quot;P&quot;)~&quot;=&quot;~.(pval)~&quot;,&quot;~.(ighv)~&quot;)&quot;)
      
    
    ggplot(eachTab, aes(x=group, y = viab)) +
      geom_beeswarm(aes(col=IGHV), size =2.5,cex = 2, alpha=0.8) +
      geom_boxplot(fill = NA, width=0.3, outlier.shape = NA) +
      ggtitle(annoText)+
      #ggtitle(sprintf(&quot;%s (p = %s)&quot;,geneName, formatNum(pval, digits = 1, format = &quot;e&quot;))) +
      ylab(y_lab) + xlab(&quot;&quot;) +
      scale_color_manual(values = c(`M-CLL` = colList[2], `U-CLL` = colList[3])) +
      scale_y_continuous(limits=c(ymin,ymax), labels = scales::number_format(accuracy = 0.1))+
      theme_full +
      theme(legend.position = &quot;none&quot;,
            plot.title = element_text(hjust = 0.5),
            plot.margin = margin(20,20,20,20))

   })
  return(plotList)
}</code></pre>
<pre class="r"><code>pList &lt;- plotDrugBoxIGHV(screenData, tRes.ighv)
plot_grid(plotlist = pList, ncol=3)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-43-1.png" width="1152" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] circlize_0.4.12             piano_2.4.0                
 [3] latex2exp_0.4.0             forcats_0.5.1              
 [5] stringr_1.4.0               dplyr_1.0.5                
 [7] purrr_0.3.4                 readr_1.4.0                
 [9] tidyr_1.1.3                 tibble_3.1.0               
[11] tidyverse_1.3.0             ggbeeswarm_0.6.0           
[13] ComplexHeatmap_2.4.3        pheatmap_1.0.12            
[15] proDA_1.2.0                 ggraph_2.0.5               
[17] ggplot2_3.3.3               igraph_1.2.6               
[19] cowplot_1.1.1               tidygraph_1.2.0            
[21] DESeq2_1.28.1               SummarizedExperiment_1.18.2
[23] DelayedArray_0.14.1         matrixStats_0.58.0         
[25] Biobase_2.48.0              GenomicRanges_1.40.0       
[27] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[29] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[31] limma_3.44.3               

loaded via a namespace (and not attached):
  [1] rappdirs_0.3.3         exactRankTests_0.8-31  bit64_4.0.5           
  [4] knitr_1.31             multcomp_1.4-16        data.table_1.14.0     
  [7] rpart_4.1-15           RCurl_1.98-1.2         generics_0.1.0        
 [10] TH.data_1.0-10         RSQLite_2.2.3          bit_4.0.4             
 [13] xml2_1.3.2             lubridate_1.7.10       httpuv_1.5.5          
 [16] assertthat_0.2.1       viridis_0.5.1          xfun_0.21             
 [19] hms_1.0.0              jquerylib_0.1.3        evaluate_0.14         
 [22] promises_1.2.0.1       fansi_0.4.2            progress_1.2.2        
 [25] caTools_1.18.1         dbplyr_2.1.0           readxl_1.3.1          
 [28] km.ci_0.5-2            DBI_1.1.1              geneplotter_1.66.0    
 [31] htmlwidgets_1.5.3      ellipsis_0.3.1         jyluMisc_0.1.5        
 [34] crosstalk_1.1.1        ggpubr_0.4.0           backports_1.2.1       
 [37] annotate_1.66.0        vctrs_0.3.6            abind_1.4-5           
 [40] cachem_1.0.4           withr_2.4.1            ggforce_0.3.3         
 [43] checkmate_2.0.0        prettyunits_1.1.1      cluster_2.1.1         
 [46] crayon_1.4.1           drc_3.0-1              relations_0.6-9       
 [49] genefilter_1.70.0      pkgconfig_2.0.3        slam_0.1-48           
 [52] labeling_0.4.2         tweenr_1.0.1           nlme_3.1-152          
 [55] vipor_0.4.5            nnet_7.3-15            rlang_0.4.10          
 [58] lifecycle_1.0.0        sandwich_3.0-0         BiocFileCache_1.12.1  
 [61] modelr_0.1.8           cellranger_1.1.0       rprojroot_2.0.2       
 [64] polyclip_1.10-0        Matrix_1.3-2           KMsurv_0.1-5          
 [67] carData_3.0-4          zoo_1.8-9              reprex_1.0.0          
 [70] base64enc_0.1-3        beeswarm_0.3.1         GlobalOptions_0.1.2   
 [73] png_0.1-7              viridisLite_0.3.0      rjson_0.2.20          
 [76] bitops_1.0-6           shinydashboard_0.7.1   KernSmooth_2.23-18    
 [79] visNetwork_2.0.9       blob_1.2.1             workflowr_1.6.2       
 [82] shape_1.4.5            maxstat_0.7-25         jpeg_0.1-8.1          
 [85] rstatix_0.7.0          ggsignif_0.6.1         scales_1.1.1          
 [88] memoise_2.0.0          magrittr_2.0.1         gplots_3.1.1          
 [91] zlibbioc_1.34.0        compiler_4.0.2         RColorBrewer_1.1-2    
 [94] plotrix_3.8-1          clue_0.3-58            cli_2.3.1             
 [97] XVector_0.28.0         htmlTable_2.1.0        Formula_1.2-4         
[100] MASS_7.3-53.1          mgcv_1.8-34            tidyselect_1.1.0      
[103] stringi_1.5.3          highr_0.8              yaml_2.2.1            
[106] askpass_1.1            locfit_1.5-9.4         latticeExtra_0.6-29   
[109] ggrepel_0.9.1          survMisc_0.5.5         sass_0.3.1            
[112] fastmatch_1.1-0        tools_4.0.2            rio_0.5.26            
[115] rstudioapi_0.13        foreign_0.8-81         git2r_0.28.0          
[118] gridExtra_2.3          farver_2.1.0           digest_0.6.27         
[121] shiny_1.6.0            Rcpp_1.0.6             car_3.0-10            
[124] broom_0.7.5            later_1.1.0.1          httr_1.4.2            
[127] survminer_0.4.9        AnnotationDbi_1.50.3   colorspace_2.0-0      
[130] rvest_1.0.0            XML_3.99-0.5           fs_1.5.0              
[133] splines_4.0.2          graphlayouts_0.7.1     xtable_1.8-4          
[136] jsonlite_1.7.2         marray_1.66.0          R6_2.5.0              
[139] sets_1.0-18            Hmisc_4.5-0            pillar_1.5.1          
[142] htmltools_0.5.1.1      mime_0.10              glue_1.4.2            
[145] fastmap_1.1.0          DT_0.17                BiocParallel_1.22.0   
[148] codetools_0.2-18       fgsea_1.14.0           mvtnorm_1.1-1         
[151] utf8_1.1.4             lattice_0.20-41        bslib_0.2.4           
[154] curl_4.3               gtools_3.8.2           zip_2.1.1             
[157] shinyjs_2.0.0          openxlsx_4.2.3         openssl_1.4.3         
[160] survival_3.2-7         rmarkdown_2.7          munsell_0.5.0         
[163] GetoptLong_1.0.5       GenomeInfoDbData_1.2.3 haven_2.3.1           
[166] gtable_0.3.0          </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
