<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2020-10-09" />

<title>Section 9: Explaining STAT2 protein expression using multi-omic data</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CLL Proteome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.embl.de/jlu/Proteomics">
    <span class="fas fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Section 9: Explaining STAT2 protein expression using multi-omic data</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2020-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CLLproteomics_publish_revision/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-19
</li>
<li>
unnamed-chunk-26
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>manuscript_S9_STAT2_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/manuscript_S1_Overview_cache/
    Ignored:    analysis/manuscript_S2_genomicAssociation_cache/
    Ignored:    analysis/manuscript_S3_trisomy12_cache/
    Ignored:    analysis/manuscript_S4_IGHV_cache/
    Ignored:    analysis/manuscript_S5_trisomy19_cache/
    Ignored:    analysis/manuscript_S6_del11q_cache/
    Ignored:    analysis/manuscript_S8_drugResponse_Outcomes_cache/
    Ignored:    analysis/manuscript_S9_STAT2_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/.trisomy12_norm.pdf
    Untracked:  analysis/IGHV_box.pdf
    Untracked:  analysis/IGHV_enrich.pdf
    Untracked:  analysis/IGHV_volcano.pdf
    Untracked:  analysis/bufferComplexViolin.pdf
    Untracked:  analysis/buffer_Tri12vsTri19.pdf
    Untracked:  analysis/cohortComposition_all.pdf
    Untracked:  analysis/drugBar.pdf
    Untracked:  analysis/dubelisib_tri12.pdf
    Untracked:  analysis/heatmap_tri12_circle.pdf
    Untracked:  analysis/manuscript_S1_Overview.Rmd
    Untracked:  analysis/manuscript_S2_genomicAssociation.Rmd
    Untracked:  analysis/manuscript_S3_trisomy12.Rmd
    Untracked:  analysis/manuscript_S4_IGHV.Rmd
    Untracked:  analysis/manuscript_S5_trisomy19.Rmd
    Untracked:  analysis/manuscript_S6_del11q.Rmd
    Untracked:  analysis/manuscript_S7_SF3B1.Rmd
    Untracked:  analysis/manuscript_S8_drugResponse_Outcomes.Rmd
    Untracked:  analysis/manuscript_S9_STAT2.Rmd
    Untracked:  analysis/plot_PC1_PC2.pdf
    Untracked:  analysis/protDrugTP53.pdf
    Untracked:  analysis/timsTOF_validate.Rmd
    Untracked:  analysis/tri12_transEnrich.pdf
    Untracked:  analysis/tri19_dosage_effect.pdf
    Untracked:  analysis/tri19_sum_buffer_number.pdf
    Untracked:  analysis/trisomy12_chr_summary.pdf
    Untracked:  code/utils.R
    Untracked:  data/Annotation file March 2021.xlsx
    Untracked:  data/CAS9results.xlsx
    Untracked:  data/CNV_onChrom.RData
    Untracked:  data/ComplexParticipantsPubMedIdentifiers_human.txt
    Untracked:  data/Fig1A.png
    Untracked:  data/IGLV321_SupplementalTables_R2.xlsx
    Untracked:  data/MOFAout.RData
    Untracked:  data/MOFAout_atLeast3.RData
    Untracked:  data/STATexprPCR.xlsx
    Untracked:  data/Western_blot_results_20210309_short.csv
    Untracked:  data/Western_blot_results_separate_blots.xlsx
    Untracked:  data/allComplexes.txt
    Untracked:  data/ddsrna_enc.RData
    Untracked:  data/exprCNV_enc.RData
    Untracked:  data/geneAnno.RData
    Untracked:  data/gmts/
    Untracked:  data/ic50.RData
    Untracked:  data/mofaIn.RData
    Untracked:  data/mofaIn_atLeast3.RData
    Untracked:  data/patMeta_enc.RData
    Untracked:  data/pepCLL_lumos_enc.RData
    Untracked:  data/protMOFA.RData
    Untracked:  data/proteins_in_complexes
    Untracked:  data/proteomic_LUMOS_2pep_enc.RData
    Untracked:  data/proteomic_explore_enc.RData
    Untracked:  data/proteomic_independent_all_enc.RData
    Untracked:  data/proteomic_independent_enc.RData
    Untracked:  data/proteomic_timsTOF_enc.RData
    Untracked:  data/screenData_enc.RData
    Untracked:  data/setToPathway.txt
    Untracked:  data/survival_enc.RData
    Untracked:  output/MSH6_splicing.svg
    Untracked:  output/SUGP1_splicing.svg
    Untracked:  output/deResList.RData
    Untracked:  output/deResListBatch2.RData
    Untracked:  output/deResListRNA.RData
    Untracked:  output/deResListRNA_allGene.RData
    Untracked:  output/deResList_WBC.RData
    Untracked:  output/deResList_batch1.RData
    Untracked:  output/deResList_batch3.RData
    Untracked:  output/deResList_independent.RData
    Untracked:  output/deResList_timsTOF.RData
    Untracked:  output/dxdCLL.RData
    Untracked:  output/dxdCLL2.RData
    Untracked:  output/exprCNV.RData
    Untracked:  output/geneAnno.RData
    Untracked:  output/int_pairs.csv
    Untracked:  output/lassoResults_CPS.RData
    Untracked:  output/resOutcome_batch1.RData
    Untracked:  output/resOutcome_batch13.RData
    Untracked:  output/resOutcome_batch2.RData
    Untracked:  output/resOutcome_batch3.RData

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/analysisSF3B1.Rmd
    Deleted:    analysis/comparePlatforms.Rmd
    Deleted:    analysis/compareProteomicsRNAseq.Rmd
    Deleted:    analysis/correlateCLLPD.Rmd
    Deleted:    analysis/correlateGenomic.Rmd
    Deleted:    analysis/correlateGenomic_removePC.Rmd
    Deleted:    analysis/correlateMIR.Rmd
    Deleted:    analysis/correlateMethylationCluster.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/predictOutcome.Rmd
    Deleted:    analysis/processProteomics_LUMOS.Rmd
    Deleted:    analysis/processProteomics_timsTOF.Rmd
    Deleted:    analysis/qualityControl_LUMOS.Rmd
    Deleted:    analysis/qualityControl_timsTOF.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
<pre class="r"><code>library(cowplot)
library(piano)
library(pheatmap)
library(ComplexHeatmap)
library(jyluMisc)
library(limma)
library(gtable)
library(ggbeeswarm)
library(glmnet)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load(&quot;../data/patMeta_enc.RData&quot;)
load(&quot;../data/ddsrna_enc.RData&quot;)
load(&quot;../data/proteomic_explore_enc.RData&quot;)
load(&quot;../output/deResList.RData&quot;) #precalculated differential expression
load(&quot;../data/screenData_enc.RData&quot;)

# source 
source(&quot;../code/utils.R&quot;)</code></pre>
</div>
<div id="feature-selection-with-lasso-on-multi-omic-data-to-explain-stat2-protein-expression" class="section level1">
<h1>Feature selection with LASSO on multi-omic data to explain STAT2 protein expression</h1>
<div id="preprocessing-multi-omic-data" class="section level2">
<h2>Preprocessing multi-omic data</h2>
<p>Proteomics data</p>
<pre class="r"><code>expVar &lt;- &quot;STAT2&quot;

protMat &lt;- assays(protCLL)[[&quot;QRILC_combat&quot;]]
rownames(protMat) &lt;- rowData(protCLL)$hgnc_symbol

yVec &lt;- protMat[expVar,]
protMat &lt;- protMat[rownames(protMat) != expVar,]

## Pre-filter for significant associations
designMat &lt;- model.matrix(~yVec)
fit &lt;- lmFit(protMat, design = designMat)
fit2 &lt;- eBayes(fit)
resTab &lt;- topTable(fit2, number = Inf)
keepProt &lt;- filter(resTab, adj.P.Val &lt; 0.05)$ID
protMat &lt;- t(protMat[keepProt, ])
dim(protMat)</code></pre>
<pre><code>[1]  91 474</code></pre>
<pre class="r"><code>responseList &lt;- list()
responseList[[&quot;STAT2&quot;]] &lt;- yVec

colnames(protMat) &lt;- paste0(colnames(protMat),&quot;_protein&quot;)</code></pre>
<p>RNAseq</p>
<pre class="r"><code>#subset
ddsSub &lt;- dds[,dds$PatID %in% colnames(protCLL)]

#only keep protein coding genes with symbol
ddsSub &lt;- ddsSub[rowData(ddsSub)$biotype %in% &quot;protein_coding&quot; &amp; rowData(ddsSub)$symbol %in% rowData(protCLL)$hgnc_symbol,]

#remove lowly expressed genes
ddsSub &lt;- ddsSub[rowSums(counts(ddsSub, normalized = TRUE)) &gt; 100,]

#voom transformation
ddsSub.vst &lt;- varianceStabilizingTransformation(ddsSub)
exprMat &lt;- assay(ddsSub.vst)

rnaMat &lt;- exprMat
rownames(rnaMat) &lt;- rowData(ddsSub.vst)$symbol
# Prefiltering
overSampe &lt;- intersect(names(yVec), colnames(rnaMat))
designMat &lt;- model.matrix(~ yVec[overSampe])
fit &lt;- lmFit(rnaMat[,overSampe], design = designMat)
fit2 &lt;- eBayes(fit)
resTab &lt;- topTable(fit2, number = Inf) %&gt;% data.frame() %&gt;% rownames_to_column(&quot;ID&quot;)
keepRna &lt;- filter(resTab, adj.P.Val &lt; 0.05)$ID
rnaMat &lt;- t(rnaMat[keepRna, ])
dim(rnaMat)</code></pre>
<pre><code>[1]  82 156</code></pre>
<pre class="r"><code>colnames(rnaMat) &lt;- paste0(colnames(rnaMat),&quot;_rna&quot;)</code></pre>
<p>Genomic data</p>
<pre class="r"><code>ighvMap &lt;- c(M = 1, U=0)
methMap &lt;- c(LP= 0, IP=0.5, HP=1 )

#genetics
genData &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  dplyr::rename(IGHV = IGHV.status) %&gt;%
  mutate_at(vars(-Patient.ID), as.character) %&gt;%
  mutate(IGHV = ighvMap[IGHV]) %&gt;%
  mutate_at(vars(-Patient.ID), as.numeric) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)

#remove gene with higher than 40% missing values
genData &lt;- genData[,colSums(is.na(genData))/nrow(genData) &lt;= 0.4]

#remove genes with less than 5 mutated cases
genData &lt;- genData[,colSums(genData, na.rm = TRUE) &gt;= 5]  

#fill the missing value with majority
genData &lt;- apply(genData, 2, function(x) {
  xVec &lt;- x
  avgVal &lt;- mean(x,na.rm= TRUE)
  if (avgVal &gt;= 0.5) {
    xVec[is.na(xVec)] &lt;- 1
  } else xVec[is.na(xVec)] &lt;- 0
  xVec
})</code></pre>
<p>Drug responses</p>
<pre class="r"><code>#choose the first sample
viabMat &lt;- arrange(screenData, screenDate) %&gt;%
  filter(diagnosis == &quot;CLL&quot;, patientID %in% colnames(protCLL)) %&gt;%
  distinct(patientID, Drug, concIndex, .keep_all = TRUE) %&gt;%
  filter(! Drug %in% c(&quot;DMSO&quot;,&quot;PBS&quot;)) %&gt;%
  mutate(id = paste0(Drug,&quot;_&quot;,concIndex)) %&gt;%
  select(patientID, id, normVal.adj.sigm) %&gt;%
  spread(key = patientID, value = normVal.adj.sigm) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;id&quot;) %&gt;% 
  as.matrix() %&gt;% t()</code></pre>
</div>
<div id="feature-selection-with-lasso-l1-penalty" class="section level2">
<h2>Feature selection with LASSO (L1) penalty</h2>
<pre class="r"><code>#Functions for running glm
runGlm &lt;- function(X, y, method = &quot;ridge&quot;, repeats=20, folds = 3, lambda = &quot;lambda.1se&quot;) {
  modelList &lt;- list()
  lambdaList &lt;- c()
  varExplain &lt;- c()
  coefMat &lt;- matrix(NA, ncol(X), repeats)
  rownames(coefMat) &lt;- colnames(X)

  if (method == &quot;lasso&quot;){
    alpha = 1
  } else if (method == &quot;ridge&quot;) {
    alpha = 0
  }
  
  for (i in seq(repeats)) {
    if (ncol(X) &gt; 2) {
      res &lt;- cv.glmnet(X,y, type.measure = &quot;mse&quot;, family=&quot;gaussian&quot;, 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      lambdaList &lt;- c(lambdaList, res[[lambda]])
      modelList[[i]] &lt;- res
      
      coefModel &lt;- coef(res, s = lambda)[-1] #remove intercept row
      coefMat[,i] &lt;- coefModel
      
      #calculate variance explained
      y.pred &lt;- predict(res, s = lambda, newx = X)
      varExp &lt;- cor(as.vector(y),as.vector(y.pred))^2
      varExplain[i] &lt;- ifelse(is.na(varExp), 0, varExp) 
      
    } else {
      fitlm&lt;-lm(y~., data.frame(X))
      varExp &lt;- summary(fitlm)$r.squared
      varExplain &lt;- c(varExplain, varExp)
      
    }

  }
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}</code></pre>
<pre class="r"><code>#function for scaling predictors
dataScale &lt;- function(x, censor = NULL, robust = FALSE) {
        #function to scale different variables
        if (length(unique(na.omit(x))) &lt;=3){
          #a binary variable, change to -0.5 and 0.5 for 1 and 2
          x - 0.5
        } else {
          if (robust) {
          #continuous variable, centered by median and divied by 2*mad
          mScore &lt;- (x-median(x,na.rm=TRUE))/mad(x,na.rm=TRUE)
            if (!is.null(censor)) {
              mScore[mScore &gt; censor] &lt;- censor
              mScore[mScore &lt; -censor] &lt;- -censor
            }
          mScore/2
          } else {
            mScore &lt;- (x-mean(x,na.rm=TRUE))/(sd(x,na.rm=TRUE))
              if (!is.null(censor)) {
                mScore[mScore &gt; censor] &lt;- censor
                mScore[mScore &lt; -censor] &lt;- -censor
              }
          mScore/2
          }
        }
      }</code></pre>
<pre class="r"><code>#function to generate response vector and explainatory variable for each seahorse measurement
generateData &lt;- function(responseList, inclSet,  onlyCombine = FALSE, censor = NULL, robust = FALSE) {
    
    allResponse &lt;- list()
    allExplain &lt;- list()

    for (measure in names(responseList)) {
      y &lt;- responseList[[measure]]
      y &lt;- y[!is.na(y)]
      
      #get overlapped samples for each dataset 
      overSample &lt;- names(y)
      
      for (eachSet in inclSet) {
        overSample &lt;- intersect(overSample,rownames(eachSet))
      }
      
      y &lt;- dataScale(y[overSample], censor = censor, robust = robust)
      
      expTab &lt;- list()
      
      if (&quot;Gene&quot; %in% names(inclSet)) {
        geneTab &lt;- inclSet$Gene[overSample,]
        #at least 3 mutated sample
        geneTab &lt;- geneTab[, colSums(geneTab) &gt;= 3]
        vecName &lt;- sprintf(&quot;genetic(%s)&quot;, ncol(geneTab))
        expTab[[vecName]] &lt;- apply(geneTab,2,dataScale)
      }
      
      
      if (&quot;RNA&quot; %in% names(inclSet)){
        rnaMat &lt;- inclSet$RNA[overSample, ]
        colnames(rnaMat) &lt;- paste0(&quot;con.&quot;,colnames(rnaMat), sep = &quot;&quot;)
        vecName &lt;- sprintf(&quot;RNA(%s)&quot;, ncol(rnaMat))
        expTab[[vecName]] &lt;- apply(rnaMat,2,dataScale, censor = censor, robust = robust)
        
      }
      
      if (&quot;Protein&quot; %in% names(inclSet)){
        protMat &lt;- inclSet$Protein[overSample, ]
        colnames(protMat) &lt;- paste0(&quot;con.&quot;,colnames(protMat), sep = &quot;&quot;)
        vecName &lt;- sprintf(&quot;Protein(%s)&quot;, ncol(protMat))
        expTab[[vecName]] &lt;- apply(protMat,2,dataScale, censor = censor, robust = robust)
        
      }
        
      if (&quot;Drug&quot; %in% names(inclSet)){
        drugMat &lt;- inclSet$Drug[overSample, ]
        colnames(drugMat) &lt;- paste0(&quot;con.&quot;,colnames(drugMat), sep = &quot;&quot;)
        vecName &lt;- sprintf(&quot;Drug(%s)&quot;, ncol(drugMat))
        expTab[[vecName]] &lt;- apply(drugMat,2,dataScale, censor = censor, robust = robust)
        
      }
      
      comboTab &lt;- c()
      for (eachSet in names(expTab)){
        comboTab &lt;- cbind(comboTab, expTab[[eachSet]])
      }
      vecName &lt;- sprintf(&quot;all(%s)&quot;, ncol(comboTab))
      expTab[[vecName]] &lt;- comboTab
      
      allResponse[[measure]] &lt;- y
      allExplain[[measure]] &lt;- expTab
    }
  if (onlyCombine) {
    #only return combined results, for feature selection
    allExplain &lt;- lapply(allExplain, function(x) x[length(x)])
  }
    
  return(list(allResponse=allResponse, allExplain=allExplain))

}</code></pre>
<p>Clean and integrate multi-omics data</p>
<pre class="r"><code>inclSet&lt;-list(Gene=genData, RNA = rnaMat, Protein = protMat)
cleanData &lt;- generateData(responseList, inclSet, censor = 5)</code></pre>
<pre class="r"><code>#Function for multi-variate regression

runGlm &lt;- function(X, y, method = &quot;ridge&quot;, repeats=20, folds = 3) {
  modelList &lt;- list()
  lambdaList &lt;- c()
  varExplain &lt;- c()
  coefMat &lt;- matrix(NA, ncol(X), repeats)
  rownames(coefMat) &lt;- colnames(X)

  if (method == &quot;lasso&quot;){
    alpha = 1
  } else if (method == &quot;ridge&quot;) {
    alpha = 0
  }
  
  for (i in seq(repeats)) {
    if (ncol(X) &gt; 2) {
      res &lt;- cv.glmnet(X,y, type.measure = &quot;mse&quot;, family=&quot;gaussian&quot;, 
                       nfolds = folds, alpha = alpha, standardize = FALSE)
      lambdaList &lt;- c(lambdaList, res$lambda.min)
      modelList[[i]] &lt;- res
      
      coefModel &lt;- coef(res, s = &quot;lambda.min&quot;)[-1] #remove intercept row
      coefMat[,i] &lt;- coefModel
      
      #calculate variance explained
      y.pred &lt;- predict(res, s = &quot;lambda.min&quot;, newx = X)
      
      varExp &lt;- 1-min(res$cvm)/res$cvm[1]
      #varExp &lt;- cor(as.vector(y),as.vector(y.pred))^2
      varExplain[i] &lt;- ifelse(is.na(varExp), 0, varExp) 
      
    } else {
      fitlm&lt;-lm(y~., data.frame(X))
      varExp &lt;- summary(fitlm)$r.squared
      varExplain &lt;- c(varExplain, varExp)
      
    }

  }
  list(modelList = modelList, lambdaList = lambdaList, varExplain = varExplain, coefMat = coefMat)
}</code></pre>
<pre class="r"><code>set.seed(2021)
lassoResults &lt;- list()
for (eachMeasure in names(cleanData$allResponse)) {
  dataResult &lt;- list()
  for (eachDataset in names(cleanData$allExplain[[eachMeasure]])) {
    y &lt;- cleanData$allResponse[[eachMeasure]]
    X &lt;- cleanData$allExplain[[eachMeasure]][[eachDataset]]
  
   
    glmRes &lt;- runGlm(X, y, method = &quot;lasso&quot;, repeats = 50, folds = 3)
    dataResult[[eachDataset]] &lt;- glmRes 
  }
  lassoResults[[eachMeasure]] &lt;- dataResult
  
}</code></pre>
</div>
<div id="variance-explained-for-stat2-expression-by-multi-omics-datasets" class="section level2">
<h2>Variance explained for STAT2 expression by multi-omics datasets</h2>
<p><img src="figure/manuscript_S9_STAT2.Rmd/varExpLASSO-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmap-of-selected-features" class="section level2">
<h2>Heatmap of selected features</h2>
<pre class="r"><code>library(gtable)
lassoPlot &lt;- function(lassoOut, cleanData, freqCut = 1, coefCut = 0.01, setNumber = &quot;last&quot;, legend = TRUE, labSuffix = &quot; protein expression&quot;, scaleFac =1) {
  plotList &lt;- list()
  if (setNumber == &quot;last&quot;) {
    setNumber &lt;- length(lassoOut[[1]])
  } else {
    setNumber &lt;- setNumber
  }
  for (seaName in names(lassoOut)) {
    #for the barplot on the left of the heatmap
    barValue &lt;- rowMeans(lassoOut[[seaName]][[setNumber]]$coefMat)
    freqValue &lt;- rowMeans(abs(sign(lassoOut[[seaName]][[setNumber]]$coefMat)))
    barValue &lt;- barValue[abs(barValue) &gt;= coefCut &amp; freqValue &gt;= freqCut] # a certain threshold
    barValue &lt;- barValue[order(barValue)]
    if(length(barValue) == 0) {
      plotList[[seaName]] &lt;- NA
      next
    }

    #for the heatmap and scatter plot below the heatmap
    allData &lt;- cleanData$allExplain[[seaName]][[setNumber]]
    seaValue &lt;- cleanData$allResponse[[seaName]]*2 #back to Z-score
    
    tabValue &lt;- allData[, names(barValue),drop=FALSE]
    ord &lt;- order(seaValue)
    seaValue &lt;- seaValue[ord]
    tabValue &lt;- tabValue[ord, ,drop=FALSE]
    sampleIDs &lt;- rownames(tabValue)
    tabValue &lt;- as.tibble(tabValue)
    
    #change scaled binary back to catagorical
    for (eachCol in colnames(tabValue)) {
      if (strsplit(eachCol, split = &quot;[.]&quot;)[[1]][1] != &quot;con&quot;) {
        tabValue[[eachCol]] &lt;- as.integer(as.factor(tabValue[[eachCol]]))
      }
      else {
        tabValue[[eachCol]] &lt;- tabValue[[eachCol]]*2 #back to Z-score
      }
    }
    
    tabValue$Sample &lt;- sampleIDs
    #Mark different rows for different scaling in heatmap
    matValue &lt;- gather(tabValue, key = &quot;Var&quot;,value = &quot;Value&quot;, -Sample)
    matValue$Type &lt;- &quot;mut&quot;
    
    #For continuious value
    matValue$Type[grep(&quot;con.&quot;,matValue$Var)] &lt;- &quot;con&quot;
    
    #for methylation_cluster
    matValue$Type[grep(&quot;ConsCluster&quot;,matValue$Var)] &lt;- &quot;meth&quot;
    
    #change the scale of the value, let them do not overlap with each other
    matValue[matValue$Type == &quot;mut&quot;,]$Value = matValue[matValue$Type == &quot;mut&quot;,]$Value + 10
    matValue[matValue$Type == &quot;meth&quot;,]$Value = matValue[matValue$Type == &quot;meth&quot;,]$Value + 20
    
    
    #color scale for viability
    idx &lt;- matValue$Type == &quot;con&quot;
    
    myCol &lt;- colorRampPalette(c(colList[2],&#39;white&#39;,colList[1]), 
                   space = &quot;Lab&quot;)
    if (sum(idx) != 0) {
      matValue[idx,]$Value = round(matValue[idx,]$Value,digits = 2)
      minViab &lt;- min(matValue[idx,]$Value)
      maxViab &lt;- max(matValue[idx,]$Value)
      limViab &lt;- max(c(abs(minViab), abs(maxViab)))
      scaleSeq1 &lt;- round(seq(-limViab, limViab,0.01), digits=2)
      color4viab &lt;- setNames(myCol(length(scaleSeq1+1)), nm=scaleSeq1)
    } else {
      scaleSeq1 &lt;- round(seq(0,1,0.01), digits=2)
      color4viab &lt;- setNames(myCol(length(scaleSeq1+1)), nm=scaleSeq1)
    }
    

    
    #change continues measurement to discrete measurement
    matValue$Value &lt;- factor(matValue$Value,levels = sort(unique(matValue$Value)))
    
    #change order of heatmap
    names(barValue) &lt;-  gsub(&quot;con.&quot;, &quot;&quot;, names(barValue))
    matValue$Var &lt;- gsub(&quot;con.&quot;,&quot;&quot;,matValue$Var)
    matValue$Var &lt;- factor(matValue$Var, levels = names(barValue))
    matValue$Sample &lt;- factor(matValue$Sample, levels = names(seaValue))
    #plot the heatmap
    p1 &lt;- ggplot(matValue, aes(x=Sample, y=Var)) + geom_tile(aes(fill=Value), color = &quot;gray&quot;) + 
      theme_bw() + scale_y_discrete(expand=c(0,0),position = &quot;right&quot;) + 
      theme(axis.text.y=element_text(hjust=0, size=10*scaleFac), axis.text.x=element_blank(), 
            axis.title = element_blank(),
            axis.ticks=element_blank(), panel.border=element_rect(colour=&quot;gainsboro&quot;),  
            plot.title=element_blank(), panel.background=element_blank(), 
            panel.grid.major=element_blank(), panel.grid.minor=element_blank(),
            plot.margin = margin(0,0,0,0)) +  
      scale_fill_manual(name=&quot;Mutated&quot;,  values=c(color4viab, `11`=&quot;gray96&quot;, `12`=&#39;black&#39;, `21`=&#39;lightgreen&#39;, 
                                                       `22`=&#39;green&#39;,`23` = &#39;green4&#39;),guide=FALSE) #+ ggtitle(seaName)
    
    #Plot the bar plot on the left of the heatmap
    barDF = data.frame(barValue, nm=factor(names(barValue),levels=names(barValue)))
    
    p2 &lt;- ggplot(data=barDF, aes(x=nm, y=barValue)) + 
      geom_bar(stat=&quot;identity&quot;, fill=colList[6], colour=&quot;black&quot;, position = &quot;identity&quot;, width=.66, size=0.2) + 
      theme_bw() + geom_hline(yintercept=0, size=0.3) + scale_x_discrete(expand=c(0,0.5)) + 
      scale_y_continuous(expand=c(0,0)) + coord_flip() + 
      theme(panel.grid.major=element_blank(), panel.background=element_blank(), axis.ticks.y = element_blank(),
            panel.grid.minor = element_blank(), 
            axis.text.x =element_text(size=8*scaleFac), 
            axis.text.y = element_blank(),
            axis.title = element_blank(),
            panel.border=element_blank(),plot.margin = margin(0,0,0,0))  + geom_vline(xintercept=c(0.5), color=&quot;black&quot;, size=0.6)
    
    #Plot the scatter plot under the heatmap
    
    # scatterplot below
    scatterDF = data.frame(X=factor(names(seaValue), levels=names(seaValue)), Y=seaValue)
    
    p3 &lt;- ggplot(scatterDF, aes(x=X, y=Y)) + geom_point(shape=21, fill=&quot;dimgrey&quot;, colour=&quot;black&quot;, size=1.2) + 
      xlab(paste0(seaName, labSuffix)) + ylab(&quot;Z-score&quot;) +
      theme_bw() + 
      theme(panel.grid.minor=element_blank(), panel.grid.major.x=element_blank(), 
            axis.title=element_text(size=10*scaleFac), 
            axis.text.x=element_blank(), axis.ticks.x=element_blank(), 
            axis.text.y=element_text(size=8*scaleFac), 
            panel.border=element_rect(colour=&quot;dimgrey&quot;, size=0.1), 
            panel.background=element_rect(fill=&quot;gray96&quot;),plot.margin = margin(0,0,0,0))
    
     
    dummyGrob &lt;- ggplot() + theme_void()
    
    #Scale bar for continuous variable
    if (legend) {
       Vgg = ggplot(data=data.frame(x=1, y=as.numeric(names(color4viab))), aes(x=x, y=y, color=y)) + geom_point() + 
      scale_color_gradientn(name=&quot;Z-score&quot;, colours =color4viab) + 
      theme(legend.title=element_text(size=12*scaleFac), legend.text=element_text(size=10*scaleFac))
    barLegend &lt;- plot_grid(gtable_filter(ggplotGrob(Vgg), &quot;guide-box&quot;))
    #Assemble all the plots togehter
    } else {
     barLegend &lt;- dummyGrob  
    }
   
    
   
    gt &lt;- egg::ggarrange(p2,p1,barLegend,dummyGrob, p3, dummyGrob, ncol=3, nrow=2, 
                         widths = c(0.6,2,0.3), padding = unit(0,&quot;line&quot;), clip = &quot;off&quot;,
                         heights = c(length(unique(matValue$Var))/2,2),draw = FALSE)
    plotList[[seaName]] &lt;- gt
  }
  return(plotList)
}</code></pre>
<div id="genetics-only" class="section level3">
<h3>Genetics only</h3>
<pre class="r"><code>heatMaps &lt;- lassoPlot(lassoResults, cleanData, freqCut = 1,setNumber = 1, legend = FALSE, scaleFac = 1)</code></pre>
<pre class="r"><code>heatMaps &lt;- heatMaps[!is.na(heatMaps)]
geneLasso &lt;- plot_grid(plotlist=heatMaps, ncol=1)
geneLasso</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-15-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="combined" class="section level3">
<h3>Combined</h3>
<pre class="r"><code>heatMaps &lt;- lassoPlot(lassoResults, cleanData, freqCut = 1,setNumber = 4, legend = TRUE,  scaleFac = 1)</code></pre>
<pre class="r"><code>heatMaps &lt;- heatMaps[!is.na(heatMaps)]
comLasso &lt;- plot_grid(plotlist=heatMaps, ncol=1)
comLasso</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-17-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="lasso-model-with-only-protein-and-rna-combined" class="section level2">
<h2>LASSO model with only protein and RNA combined</h2>
<p>Clean and integrate multi-omics data</p>
<pre class="r"><code>inclSet&lt;-list(RNA = rnaMat, Protein = protMat)
cleanData &lt;- generateData(responseList, inclSet, censor = 5)</code></pre>
<pre class="r"><code>set.seed(2020)
lassoResults &lt;- list()
for (eachMeasure in names(cleanData$allResponse)) {
  dataResult &lt;- list()
  for (eachDataset in names(cleanData$allExplain[[eachMeasure]])) {
    y &lt;- cleanData$allResponse[[eachMeasure]]
    X &lt;- cleanData$allExplain[[eachMeasure]][[eachDataset]]
  
   
    glmRes &lt;- runGlm(X, y, method = &quot;lasso&quot;, repeats = 50, folds = 3)
    dataResult[[eachDataset]] &lt;- glmRes 
  }
  lassoResults[[eachMeasure]] &lt;- dataResult
  
}</code></pre>
<pre class="r"><code>heatMaps &lt;- lassoPlot(lassoResults, cleanData, freqCut = 1,setNumber = 3, legend = TRUE,  scaleFac = 1)</code></pre>
<pre class="r"><code>heatMaps &lt;- heatMaps[!is.na(heatMaps)]
comLasso &lt;- plot_grid(plotlist=heatMaps, ncol=1)
comLasso</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-21-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="ighv-mutational-status-and-trisomy12-status-jointly-determine-stat2-protein-expression" class="section level1">
<h1>IGHV mutational status and trisomy12 status jointly determine STAT2 protein expression</h1>
<div id="anova-test-on-ighvtrisomy12-status-to-stat2-protein-expression" class="section level3">
<h3>ANOVA test on IGHV/trisomy12 status to STAT2 protein expression</h3>
<pre class="r"><code>plotTab &lt;- tibble(patID = colnames(protCLL),
                  STAT2 = assays(protCLL)[[&quot;count_combat&quot;]][rowData(protCLL)$hgnc_symbol == &quot;STAT2&quot;,],
                  trisomy12 = protCLL$trisomy12,
                  IGHV=protCLL$IGHV.status) %&gt;%
  filter(!is.na(IGHV), !is.na(trisomy12)) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 == 0, &quot;non-tri12&quot;,&quot;tri12&quot;)) %&gt;%
  mutate(group = paste0(IGHV, &quot;_&quot;, trisomy12))
stat2BoxProt &lt;- ggplot(plotTab, aes(group, y=STAT2, fill = group)) + geom_boxplot() + geom_point() + theme_full +
  scale_fill_manual(values = colList) + theme(legend.position = &quot;none&quot;) +
  xlab(&quot;&quot;) + ggtitle(&quot;STAT2 protein expression&quot;) +ylab(&quot;Normalized expression&quot;)

summary(lm(STAT2 ~ IGHV * trisomy12, plotTab))</code></pre>
<pre><code>
Call:
lm(formula = STAT2 ~ IGHV * trisomy12, data = plotTab)

Residuals:
    Min      1Q  Median      3Q     Max 
-0.9044 -0.2317 -0.0275  0.2222  0.8272 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          15.79174    0.06256 252.427  &lt; 2e-16 ***
IGHVU                 0.18836    0.09073   2.076 0.040846 *  
trisomy12tri12        0.43065    0.11074   3.889 0.000196 ***
IGHVU:trisomy12tri12  0.41597    0.15789   2.634 0.009974 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3539 on 87 degrees of freedom
Multiple R-squared:  0.5157,    Adjusted R-squared:  0.499 
F-statistic: 30.87 on 3 and 87 DF,  p-value: 1.097e-13</code></pre>
<p><strong>STAT2 protein expression is strongly affected by IGHV and trisomy12 status, U-CLLs with trisomy12 shows significant up-regulation of STAT2</strong></p>
</div>
<div id="anova-test-on-ighvtrisomy12-status-to-stat2-rna-expression" class="section level3">
<h3>ANOVA test on IGHV/trisomy12 status to STAT2 RNA expression</h3>
<pre class="r"><code>plotTab &lt;- tibble(patID = colnames(ddsSub.vst),
                  STAT2 = assay(ddsSub.vst)[rowData(ddsSub.vst)$symbol == &quot;STAT2&quot;,],
                  trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12,
                  IGHV=patMeta[match(patID, patMeta$Patient.ID),]$IGHV.status) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 == 0, &quot;non-tri12&quot;,&quot;tri12&quot;)) %&gt;%
  mutate(group = paste0(IGHV, &quot;_&quot;, trisomy12))

stat2BoxRNA &lt;- ggplot(plotTab, aes(group, y=STAT2, fill = group)) + geom_boxplot() + geom_point() + theme_full +
  scale_fill_manual(values = colList) + theme(legend.position = &quot;none&quot;) +
  xlab(&quot;&quot;) + ggtitle(&quot;STAT2 RNA expression&quot;) +ylab(&quot;Normalized expression&quot;)

summary(lm(STAT2 ~ IGHV * trisomy12, plotTab))</code></pre>
<pre><code>
Call:
lm(formula = STAT2 ~ IGHV * trisomy12, data = plotTab)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.73477 -0.23395 -0.03465  0.24535  0.75047 

Coefficients:
                     Estimate Std. Error t value Pr(&gt;|t|)    
(Intercept)          13.63685    0.06136 222.229  &lt; 2e-16 ***
IGHVU                -0.23968    0.08994  -2.665  0.00935 ** 
trisomy12tri12        0.78685    0.11616   6.774 2.12e-09 ***
IGHVU:trisomy12tri12  0.48696    0.16596   2.934  0.00439 ** 
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

Residual standard error: 0.3417 on 78 degrees of freedom
Multiple R-squared:  0.6752,    Adjusted R-squared:  0.6627 
F-statistic: 54.05 on 3 and 78 DF,  p-value: &lt; 2.2e-16</code></pre>
</div>
<div id="visualize-differences-using-boxplots" class="section level3">
<h3>Visualize differences using boxplots</h3>
<pre class="r"><code>stat2Box &lt;- plot_grid(stat2BoxProt, stat2BoxRNA)
stat2Box</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-24-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pathway-enrichment-for-rna-expressions-correlated-with-stat2-protein-expression" class="section level1">
<h1>Pathway enrichment for RNA expressions correlated with STAT2 protein expression</h1>
<pre class="r"><code>expVar &lt;- &quot;STAT2&quot;
protMat &lt;- assays(protCLL)[[&quot;QRILC_combat&quot;]]
rownames(protMat) &lt;- rowData(protCLL)$hgnc_symbol
yVec &lt;- protMat[expVar,]</code></pre>
<p>Prepare data</p>
<pre class="r"><code>#subset
ddsSub &lt;- dds[,dds$PatID %in% names(yVec)]

#only keep protein coding genes with symbol
ddsSub &lt;- ddsSub[rowData(ddsSub)$biotype %in% &quot;protein_coding&quot; &amp; !rowData(ddsSub)$symbol %in% c(&quot;&quot;,NA),]

#remove lowly expressed genes
ddsSub &lt;- ddsSub[rowSums(counts(ddsSub, normalized = TRUE)) &gt; 100,]

#voom transformation
exprMat &lt;- limma::voom(counts(ddsSub), lib.size = ddsSub$sizeFactor)$E
ddsSub.voom &lt;- ddsSub
assay(ddsSub.voom) &lt;- exprMat

rnaMat &lt;- exprMat
rownames(rnaMat) &lt;- rowData(ddsSub.voom)$symbol

overSampe &lt;- intersect(names(yVec), colnames(rnaMat))

rnaMat &lt;- rnaMat[,overSampe]
yVec &lt;- yVec[overSampe]</code></pre>
<p>Test</p>
<pre class="r"><code>#buid design matirx
ighv &lt;- patMeta[match(names(yVec),patMeta$Patient.ID),]$IGHV.status
tri12 &lt;- patMeta[match(names(yVec),patMeta$Patient.ID),]$trisomy12

d0 &lt;- model.matrix(~yVec)
d1 &lt;- model.matrix(~ighv+tri12+yVec)</code></pre>
<div id="without-blocking-for-ighv-or-trisomy12" class="section level2">
<h2>Without blocking for IGHV or trisomy12</h2>
<pre class="r"><code>fit &lt;- lmFit(rnaMat, design = d0)
fit2 &lt;- eBayes(fit)
resTab.noBlock &lt;- topTable(fit2, number = Inf, coef = &quot;yVec&quot;) %&gt;% data.frame() %&gt;% rownames_to_column(&quot;name&quot;)
resTab.noBlock.sig &lt;- filter(resTab.noBlock, adj.P.Val &lt; 0.1)</code></pre>
<pre class="r"><code>plotCorScatter &lt;- function(plotTab, x_lab = &quot;X&quot;, y_lab = &quot;Y&quot;, title = &quot;&quot;,
                           showR2 = FALSE, annoPos = &quot;right&quot;,
                           dotCol = colList, textCol=&quot;darkred&quot;) {

  #prepare annotation values
  corRes &lt;- cor.test(plotTab$x, plotTab$y)
  pval &lt;- formatNum(corRes$p.value, digits = 1, format = &quot;e&quot;)
  Rval &lt;- formatNum(corRes$estimate, digits = 1, format = &quot;e&quot;)
  R2val &lt;- formatNum(corRes$estimate^2, digits = 1, format = &quot;e&quot;)
  Nval &lt;- nrow(plotTab)
  annoP &lt;- bquote(italic(&quot;P&quot;)~&quot;=&quot;~.(pval))

  if (showR2) {
    annoCoef &lt;-  bquote(R^2~&quot;=&quot;~.(R2val))
  } else {
    annoCoef &lt;- bquote(R~&quot;=&quot;~.(Rval))
  }
  annoN &lt;- bquote(N~&quot;=&quot;~.(Nval))

  corPlot &lt;- ggplot(plotTab, aes(x = x, y = y)) + geom_point(aes(col = trisomy12, shape = IGHV), size=5) +
    scale_shape_manual(values = c(M = 19, U = 1)) + 
    scale_color_manual(values = c(yes = colList[2], no = colList[3])) +
    geom_smooth(formula = y~x,method = &quot;lm&quot;, se=FALSE, color = &quot;grey50&quot;, linetype =&quot;dashed&quot; ) 

  if (annoPos == &quot;right&quot;) {

    corPlot &lt;- corPlot + annotate(&quot;text&quot;, x = max(plotTab$x), y = Inf, label = annoN,
                                  hjust=1, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate(&quot;text&quot;, x = max(plotTab$x), y = Inf, label = annoP,
               hjust=1, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate(&quot;text&quot;, x = max(plotTab$x), y = Inf, label = annoCoef,
               hjust=1, vjust =6, size = 5, parse = FALSE, col= textCol)

  } else if (annoPos== &quot;left&quot;) {
    corPlot &lt;- corPlot + annotate(&quot;text&quot;, x = min(plotTab$x), y = Inf, label = annoN,
                                  hjust=0, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate(&quot;text&quot;, x = min(plotTab$x), y = Inf, label = annoP,
               hjust=0, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate(&quot;text&quot;, x = min(plotTab$x), y = Inf, label = annoCoef,
               hjust=0, vjust =6, size = 5, parse = FALSE, col= textCol)
  }
  corPlot &lt;- corPlot + ylab(y_lab) + xlab(x_lab) + ggtitle(title) +
    scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
    scale_x_continuous(labels = scales::number_format(accuracy = 0.1)) +
    theme_full + theme(legend.position = &quot;bottom&quot;, plot.margin = margin(12,12,12,12))
  corPlot
}</code></pre>
<div id="correlation-between-selected-rnas-and-stat2-protein-expression" class="section level3">
<h3>Correlation between selected RNAs and STAT2 protein expression</h3>
<pre class="r"><code>geneList &lt;- c(&quot;OAS2&quot;, &quot;IFI44&quot;)
rnaSTAT2cor &lt;- lapply(geneList, function(n) {
  plotTab &lt;- tibble(x = yVec, y = rnaMat[n,], IGHV = ighv, tri12 = tri12) %&gt;%
    mutate(trisomy12 = ifelse(tri12==1,&quot;yes&quot;,&quot;no&quot;))
  plotCorScatter(plotTab, annoPos = &quot;left&quot;, x_lab = &quot;STAT2 protein expression&quot;, y_lab = sprintf(&quot;%s RNA expression&quot;, n))
})
names(rnaSTAT2cor) &lt;- geneList
plotRNAcor &lt;- plot_grid(plotlist = rnaSTAT2cor)
plotRNAcor</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-30-1.png" width="1152" style="display: block; margin: auto;" /></p>
<p>Visualize RNA expression using boxplots</p>
<pre class="r"><code>geneList &lt;- c(&quot;OAS2&quot;, &quot;IFI44&quot;)
rnaSTAT2corBox &lt;- lapply(geneList, function(n) {
  plotTab &lt;- tibble(expr = rnaMat[n,], IGHV = ighv, tri12 = tri12) %&gt;%
  mutate(trisomy12 = ifelse(tri12==1,&quot;tri12&quot;,&quot;non-tri12&quot;)) %&gt;%
  mutate(group = paste0(IGHV, &quot;_&quot;, trisomy12))

ggplot(plotTab, aes(group, y=expr, fill = group)) + geom_boxplot() + geom_point() + theme_full +
  scale_fill_manual(values = colList) + theme(legend.position = &quot;none&quot;) +
  xlab(&quot;&quot;) + ylab(sprintf(&quot;%s RNA expression&quot;,n)) + ggtitle(n)
})
names(rnaSTAT2corBox) &lt;- geneList
plotRNAbox &lt;- plot_grid(plotlist = rnaSTAT2corBox)
plotRNAbox</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-31-1.png" width="1344" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-enrichment-analysis-for-rna-expressions-associated-with-stat2-protein-expression-level" class="section level3">
<h3>Gene enrichment analysis for RNA expressions associated with STAT2 protein expression level</h3>
<pre class="r"><code>gmts &lt;- list(H = &quot;../data/gmts/h.all.v6.2.symbols.gmt&quot;,
             KEGG = &quot;../data/gmts/c2.cp.kegg.v6.2.symbols.gmt&quot;)

enRes &lt;-  runCamera(rnaMat, d0, gmts$H,
            removePrefix = &quot;HALLMARK_&quot;, pCut = 0.1, ifFDR = TRUE, setMap=setMap)
enRes$enrichPlot</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/rnaEnrichNoBlock-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="heatmap-showing-the-rna-expression-of-genes-belong-to-the-enriched-pathways" class="section level4">
<h4>Heatmap showing the RNA expression of genes belong to the enriched pathways</h4>
<pre class="r"><code>colAnno &lt;- tibble(id = names(yVec),  STAT2= yVec, IGHV = ighv, trisomy12 = tri12) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 ==1,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;id&quot;)
annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV = c(M = colList[4], U = colList[3]),
                STAT2 = circlize::colorRamp2(c(min(colAnno$STAT2),max(colAnno$STAT2)), 
                                             c(&quot;white&quot;, &quot;green&quot;)))

nameList &lt;- c(&quot;STAT2&quot;,&quot;IFI44&quot;,&quot;OAS1&quot;,&quot;OAS2&quot;, &quot;IFI30&quot;)
plotSetHeatmap(resTab.noBlock.sig, gmts$H, &quot;HALLMARK_INTERFERON_ALPHA_RESPONSE&quot;, rnaMat, colAnno = colAnno, annoCol = annoCol, highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/IFNA_heatmap_RNA-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotSetHeatmap(resTab.noBlock.sig, gmts$H, &quot;HALLMARK_INTERFERON_GAMMA_RESPONSE&quot;, rnaMat, colAnno = colAnno, annoCol = annoCol,highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/IFNG_heatmap_RNA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="with-blocking-for-ighv-and-trisomy12" class="section level2">
<h2>With blocking for IGHV and trisomy12</h2>
<pre class="r"><code>fit &lt;- lmFit(rnaMat, design = d1)
fit2 &lt;- eBayes(fit)
resTab.block &lt;- topTable(fit2, number = Inf, coef = &quot;yVec&quot;) %&gt;% data.frame() %&gt;% rownames_to_column(&quot;name&quot;)
resTab.block.sig &lt;- filter(resTab.block, P.Value &lt; 0.01)</code></pre>
<p>Gene set enrichment analysis</p>
<pre class="r"><code>enRes &lt;-  runCamera(rnaMat, d1, gmts$H, contrast  = &quot;yVec&quot;,
            removePrefix = &quot;HALLMARK_&quot;, pCut = 0.05, ifFDR = TRUE, plotTitle = &quot;RNA enrichment&quot;, setMap = setMap)
rnaEnrich &lt;- enRes$enrichPlot
rnaEnrich</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/rnaEnrichBlock-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="pathway-enrichment-for-protein-expressions-correlated-with-stat2-protein-level" class="section level1">
<h1>Pathway enrichment for protein expressions correlated with STAT2 protein level</h1>
<div id="identify-protein-expression-that-are-correlated-with-stat2-protein-expression" class="section level2">
<h2>Identify protein expression that are correlated with STAT2 protein expression</h2>
<pre class="r"><code>expVar &lt;- &quot;STAT2&quot;
protMat &lt;- assays(protCLL)[[&quot;QRILC_combat&quot;]]
rownames(protMat) &lt;- rowData(protCLL)$hgnc_symbol
yVec &lt;- protMat[expVar,]
protMat &lt;- protMat[rownames(protMat) != expVar,]</code></pre>
<pre class="r"><code>#buid design matirx
ighv &lt;- patMeta[match(names(yVec),patMeta$Patient.ID),]$IGHV.status
tri12 &lt;- patMeta[match(names(yVec),patMeta$Patient.ID),]$trisomy12

d0 &lt;- model.matrix(~yVec)
d1 &lt;- model.matrix(~ighv+tri12+yVec)</code></pre>
<div id="without-blocking-for-ighv-status-and-trisomy12" class="section level3">
<h3>Without blocking for IGHV status and trisomy12</h3>
<pre class="r"><code>fit &lt;- lmFit(protMat, design = d0)
fit2 &lt;- eBayes(fit)
resTab.noBlock &lt;- topTable(fit2, number = Inf, coef = &quot;yVec&quot;) %&gt;% data.frame() %&gt;% mutate(name = ID)
resTab.noBlock.sig &lt;- filter(resTab.noBlock, adj.P.Val &lt; 0.1)</code></pre>
</div>
<div id="correlation-between-selected-proteins-and-stat2-protein-expression" class="section level3">
<h3>Correlation between selected proteins and STAT2 protein expression</h3>
<pre class="r"><code>geneList &lt;- c(&quot;OAS1&quot;, &quot;OAS2&quot;, &quot;IFI30&quot; )
geneList &lt;- c(&quot;OAS2&quot;)

protSTAT2cor &lt;- lapply(geneList, function(n) {
  plotTab &lt;- tibble(x = yVec, y = protMat[n,], IGHV = ighv, tri12 = tri12) %&gt;%
    mutate(trisomy12 = ifelse(tri12==1,&quot;yes&quot;,&quot;no&quot;))
  plotCorScatter(plotTab, annoPos = &quot;left&quot;, x_lab = &quot;STAT2 protein expression&quot;, y_lab = sprintf(&quot;%s protein expression&quot;, n)) 
})
names(protSTAT2cor) &lt;- geneList
plot_grid(plotlist = protSTAT2cor, ncol=1)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/corSTAT2prot-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Boxplots</p>
<pre class="r"><code>protSTAT2corBox &lt;- lapply(geneList, function(n) {
  plotTab &lt;- tibble(expr = protMat[n,], IGHV = ighv, tri12 = tri12) %&gt;%
  mutate(trisomy12 = ifelse(tri12==1,&quot;tri12&quot;,&quot;non-tri12&quot;)) %&gt;%
  filter(!is.na(IGHV), !is.na(trisomy12)) %&gt;%
  mutate(group = paste0(IGHV, &quot;_&quot;, trisomy12))

  ggplot(plotTab, aes(group, y=expr, fill = group)) + geom_boxplot() + geom_point() + theme_full +
    scale_fill_manual(values = colList) + theme(legend.position = &quot;none&quot;) +
    xlab(&quot;&quot;) + ylab(sprintf(&quot;%s protein expression&quot;,n)) + ggtitle(n)
})
names(protSTAT2corBox) &lt;- geneList
plot_grid(plotlist = protSTAT2corBox, ncol=3)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-36-1.png" width="2016" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-set-enrichment-analysis" class="section level3">
<h3>Gene set enrichment analysis</h3>
<pre class="r"><code>enRes &lt;-  runCamera(protMat, d0, gmts$H,
            removePrefix = &quot;HALLMARK_&quot;, pCut = 0.1, ifFDR = TRUE, setMap = setMap)
enRes$enrichPlot</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/proteinEnrichNoBlock-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotSetHeatmap(resTab.noBlock.sig, gmts$H, &quot;HALLMARK_INTERFERON_ALPHA_RESPONSE&quot;, protMat, colAnno = colAnno, annoCol = annoCol, highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/IFNA_heatmap_protein-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotSetHeatmap(resTab.noBlock.sig, gmts$H, &quot;HALLMARK_INTERFERON_GAMMA_RESPONSE&quot;, protMat, colAnno = colAnno, annoCol = annoCol, highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/IFNG_heatmap_Protein-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="with-blocking-for-ighv-and-trisomy12-1" class="section level2">
<h2>With blocking for IGHV and trisomy12</h2>
<pre class="r"><code>fit &lt;- lmFit(protMat, design = d1)
fit2 &lt;- eBayes(fit)
resTab.block &lt;- topTable(fit2, number = Inf, coef = &quot;yVec&quot;) %&gt;% data.frame() %&gt;% mutate(name=ID)
hist(resTab.block$P.Value)</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-37-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>resTab.block.sig &lt;- filter(resTab.block, P.Value &lt; 0.01)</code></pre>
<pre class="r"><code>enRes &lt;-  runCamera(protMat, d1, gmts$H, contrast  = &quot;yVec&quot;,
            removePrefix = &quot;HALLMARK_&quot;, pCut = 0.05, ifFDR = TRUE, plotTitle = &quot;Protein enrichment&quot;, setMap = setMap)
protEnrich &lt;- enRes$enrichPlot
protEnrich</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-38-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="western-blot-validation-of-stat2-expression" class="section level1">
<h1>Western blot validation of STAT2 expression</h1>
<p>Read in western results</p>
<pre class="r"><code>wesTab &lt;- readxl::read_xlsx(&quot;../data/Western_blot_results_separate_blots.xlsx&quot;) %&gt;%
  separate(`IGHV_Tri12`, c(&quot;IGHV&quot;,&quot;trisomy12&quot;),&quot;_&quot;) %&gt;%
  mutate(trisomy12 = ifelse(trisomy12 == &quot;WT&quot;,&quot;non-tri12&quot;,&quot;tri12&quot;)) %&gt;%
  dplyr::rename(intensity = `STAT2 Total`) %&gt;%
  mutate(logIntensity = log10(intensity))</code></pre>
<div id="associations-with-ighv-and-trisomy12" class="section level2">
<h2>Associations with IGHV and trisomy12</h2>
<div id="ighv" class="section level3">
<h3>IGHV</h3>
<pre class="r"><code>tRes &lt;- car::Anova(lm(intensity~IGHV + Blot, wesTab))
pval &lt;- tRes$`Pr(&gt;F)`[1]</code></pre>
<p>Boxplot</p>
<pre class="r"><code>plotTab &lt;- filter(wesTab) %&gt;%
  mutate(status = ifelse(IGHV==&quot;M&quot;,&quot;M-CLL&quot;,&quot;U-CLL&quot;)) %&gt;%
  group_by(status) %&gt;% mutate(n=n()) %&gt;% ungroup() %&gt;%
  mutate(group = sprintf(&quot;%s\n(N=%s)&quot;,status,n)) %&gt;%
  arrange(status) %&gt;% mutate(group = factor(group, levels = unique(group)))

pval &lt;- formatNum(pval, digits = 2)
titleText &lt;- bquote(&quot;STAT2 protein expression by Western Blot analysis&quot;~&quot; ( ANOVA &quot;~italic(&quot;P&quot;)~&quot;=&quot;~.(pval)~&quot;)&quot;)
pValText &lt;-  bquote(italic(&quot;P&quot;)~&quot;-value =&quot;~.(pval))
ggplot(plotTab, aes(x=group, y = intensity)) +
  geom_boxplot(width=0.3, aes(fill = status), outlier.shape = NA) +
  annotate(geom = &quot;text&quot; ,label = pValText, x= 1.5, y=Inf, vjust=2, col = colList[1]) +
  geom_beeswarm(aes(shape = Blot), size =2.5,cex = 2, alpha=0.5) +
  ggtitle(&quot;STAT2 protein expression by Western Blot analysis&quot;)+
  #ggtitle(sprintf(&quot;%s (p = %s)&quot;,geneName, formatNum(pval, digits = 1, format = &quot;e&quot;))) +
  ylab(&quot;Normalized intensity&quot;) + xlab(&quot;&quot;) +
  scale_fill_manual(values = colList[3:5], name = &quot;&quot;) +
  theme_full +
  theme(legend.position = &quot;bottom&quot;,
        plot.title = element_text(hjust = 0.5, size=13),
        plot.margin = margin(10,10,10,10))</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-41-1.png" width="480" style="display: block; margin: auto;" /></p>
</div>
<div id="trisomy12" class="section level3">
<h3>trisomy12</h3>
<pre class="r"><code>tRes &lt;- car::Anova(lm(intensity~trisomy12 + Blot, wesTab))
pval &lt;- tRes$`Pr(&gt;F)`[1]</code></pre>
<p>Boxplot</p>
<pre class="r"><code>plotTab &lt;- filter(wesTab) %&gt;%
  mutate(status = trisomy12) %&gt;%
  group_by(status) %&gt;% mutate(n=n()) %&gt;% ungroup() %&gt;%
  mutate(group = sprintf(&quot;%s\n(N=%s)&quot;,status,n)) %&gt;%
  arrange(status) %&gt;% mutate(group = factor(group, levels = unique(group)))

pval &lt;- formatNum(pval, digits = 2)
titleText &lt;- bquote(&quot;STAT2 protein expression by Western Blot analysis&quot;~&quot; ( ANOVA &quot;~italic(&quot;P&quot;)~&quot;=&quot;~.(pval)~&quot;)&quot;)
pValText &lt;-  bquote(italic(&quot;P&quot;)~&quot;-value =&quot;~.(pval))
ggplot(plotTab, aes(x=group, y = intensity)) +
  geom_boxplot(width=0.3, aes(fill = status), outlier.shape = NA) +
  annotate(geom = &quot;text&quot; ,label = pValText, x= 1.5, y=Inf,vjust=2, col = colList[1]) +
  geom_beeswarm(aes(shape = Blot), size =2.5,cex = 2, alpha=0.5) +
  ggtitle(&quot;STAT2 protein expression by Western Blot analysis&quot;)+
  #ggtitle(sprintf(&quot;%s (p = %s)&quot;,geneName, formatNum(pval, digits = 1, format = &quot;e&quot;))) +
  ylab(&quot;Normalized intensity&quot;) + xlab(&quot;&quot;) +
  scale_fill_manual(values = colList[3:5], name = &quot;&quot;) +
  theme_full +
  theme(legend.position = &quot;bottom&quot;,
        plot.title = element_text(hjust = 0.5, size=13),
        plot.margin = margin(10,10,10,10))</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/STAT2_trisomy12_western-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Joint IGHV and trisomy12</p>
<pre class="r"><code>plotTab &lt;- wesTab %&gt;%
  mutate(group = paste0(IGHV, &quot;_&quot;, trisomy12))

stat2BoxWest &lt;- ggplot(plotTab, aes(group, y=intensity, fill = group)) + geom_boxplot() + 
  geom_beeswarm(aes(shape = Blot), size =2.5,cex = 2, alpha=0.5) +
  theme_full +
  scale_fill_manual(values = colList, name = &quot;&quot;) + theme(legend.position = &quot;bottom&quot;) +
  xlab(&quot;&quot;) + ggtitle(&quot;STAT2 protein expression by Western Blot analysis&quot;) +ylab(&quot;Normalized intensity&quot;)

stat2BoxWest</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/STAT2_western_tri12_IGHV-1.png" width="624" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="analysis-of-the-pcr-results-of-cll-cell-lines-with-and-without-stat2-knockouts" class="section level1">
<h1>Analysis of the PCR results of CLL cell lines with and without STAT2 knockouts</h1>
<p>Load datasets</p>
<pre class="r"><code>tab1 &lt;- readxl::read_xlsx(&quot;../data/CAS9results.xlsx&quot;, sheet = 1) %&gt;% mutate(control = &quot;ACTB&quot;) 
tab2 &lt;- readxl::read_xlsx(&quot;../data/CAS9results.xlsx&quot;, sheet = 2) %&gt;% mutate(control = &quot;GAPDH&quot;)
pcrTab &lt;- bind_rows(tab1, tab2) %&gt;% separate(name, into = c(&quot;cellLine&quot;,&quot;sgRNA&quot;,&quot;treatment&quot;,&quot;IFN&quot;), sep = &quot;[_ ]&quot;, remove = FALSE) %&gt;%
  pivot_longer(contains(&quot;fold change&quot;), names_to = &quot;replicate&quot;,values_to = &quot;foldChange&quot;) %&gt;%
  mutate(replicate=str_replace(replicate, &quot;fold change replicate &quot;,&quot;R&quot;),
         IFN = ifelse(treatment == &quot;+&quot;, &quot;IFN&quot;, &quot;no IFN&quot;),
         trisomy12 = ifelse(cellLine %in% c(&quot;MEC-1&quot;,&quot;HG-3&quot;), &quot;no&quot; ,&quot;yes&quot;),
         sgTreat= paste0(sgRNA,&quot; &quot;,treatment,&quot; IFN&quot;)) %&gt;%
  select(-treatment) %&gt;%
  mutate(sgRNA = factor(sgRNA, levels = c(&quot;NTC&quot;,&quot;34A&quot;,&quot;36D&quot;)),
         Gene = factor(Gene, levels = c(&quot;STAT2&quot;,&quot;OAS1&quot;,&quot;OAS2&quot;))) %&gt;%
  arrange(sgRNA, Gene) %&gt;%
  mutate(sgTreat = factor(sgTreat, levels = unique(sgTreat))) %&gt;%
  mutate(log2foldChange = log2(foldChange))
geneSymbol &lt;- c(bquote(italic(&quot;STAT2&quot;)),bquote(italic(&quot;OAS1&quot;)), bquote(italic(&quot;OAS2&quot;)))</code></pre>
<p>GAPDH as control</p>
<pre class="r"><code>plotTab &lt;- pcrTab %&gt;% filter(control == &quot;GAPDH&quot;)

barGAPDH &lt;- ggplot(plotTab, aes(x=sgTreat, y=log2foldChange)) +
  geom_bar(aes(fill = Gene), position = &quot;dodge&quot;, stat = &quot;summary&quot;, fun.y = &quot;mean&quot;) +
  geom_point(aes(dodge=Gene), col = &quot;black&quot;, position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = colList, labels = geneSymbol) +
  facet_wrap(~cellLine, scale = &quot;free_x&quot;, ncol=2 ) +
  xlab(&quot;&quot;) + ylab(bquote(&quot;log&quot;[2]*&quot;(fold change) relative to control in RNA expression&quot;)) +
  #scale_y_continuous() +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, col = &quot;grey50&quot;) +
  theme_half +
  theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5),
        strip.text = element_text(size=12, face= &quot;bold&quot;),
        legend.position = &quot;right&quot;)
barGAPDH</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-44-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>ACTB as control</p>
<pre class="r"><code>plotTab &lt;- pcrTab %&gt;% filter(control == &quot;ACTB&quot;)

barACTB &lt;- ggplot(plotTab, aes(x=sgTreat, y=log2foldChange)) +
  geom_bar(aes(fill = Gene), position = &quot;dodge&quot;, stat = &quot;summary&quot;, fun.y = &quot;mean&quot;) +
  geom_point(aes(dodge=Gene), col = &quot;black&quot;, position = position_dodge(width = 0.9)) +
  scale_fill_manual(values = colList, labels = geneSymbol) +
  facet_wrap(~cellLine, scale = &quot;free_x&quot;, ncol=2) +
  xlab(&quot;&quot;) + ylab(bquote(&quot;log&quot;[2]*&quot;(fold change) relative to control in RNA expression&quot;)) +
  #scale_y_continuous() +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, col = &quot;grey50&quot;) +
  theme_half +
  theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5),
        strip.text = element_text(size=12, face= &quot;bold&quot;),
        legend.position = &quot;right&quot;)
barACTB</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-45-1.png" width="960" style="display: block; margin: auto;" /></p>
<div id="compare-trisomy12-cell-lines-with-non-trisomy12-cell-lines" class="section level2">
<h2>Compare trisomy12 cell lines with non-trisomy12 cell lines</h2>
<pre class="r"><code>pcrTab &lt;- readxl::read_excel(&quot;../data/STATexprPCR.xlsx&quot;) %&gt;%
  separate(name, into = c(&quot;cellLine&quot;, &quot;sgRNA&quot;,&quot;IFN&quot;,&quot;other&quot;),sep = &quot; &quot;) %&gt;%
  #mutate(IFN = ifelse(IFN == &quot;+&quot;,&quot;with &quot;,&quot;no &quot;)) %&gt;%
  mutate(IFN = paste0(IFN, &quot; &quot;,other),
         trisomy12 = str_replace(trisomy12, &quot;Non&quot;,&quot;non&quot;)) %&gt;% 
  arrange( trisomy12, IFN) %&gt;%
  mutate(group = paste0(trisomy12,&quot; &quot;, IFN)) %&gt;%
  mutate(group =factor(group, levels=unique(group)),
         sgRNA = factor(sgRNA, levels = c(&quot;NTC&quot;,&quot;34A&quot;,&quot;36D&quot;)),
         Gene = factor(Gene, levels = c(&quot;STAT2&quot;,&quot;OAS1&quot;,&quot;OAS2&quot;))) %&gt;%
  group_by(cellLine, sgRNA, group, Gene) %&gt;%
  summarise(fc = mean(foldChange)) %&gt;%
  ungroup()</code></pre>
<pre class="r"><code>geneSymbol &lt;- c(bquote(italic(&quot;STAT2&quot;)),bquote(italic(&quot;OAS1&quot;)), bquote(italic(&quot;OAS2&quot;)))


meanTab &lt;- group_by(pcrTab, group, sgRNA, Gene) %&gt;%
  summarise(meanVal = mean(fc))
  
ggplot(pcrTab, aes(x=group, y = fc, col = Gene, group = Gene)) +
  geom_point(aes(shape = cellLine),position = position_dodge(width = 0.8), size=3) +
  xlab(&quot;&quot;) + ylab(&quot;fold change&quot;) +
  #ggrepel::geom_text_repel(aes(label = cellLine)) +
  scale_color_manual(values = colList, labels = geneSymbol, name = &quot;gene&quot;) +
  scale_shape_discrete(name = &quot;cell line&quot;) +
  geom_point(data=meanTab, aes(x=group, y=meanVal), position = position_dodge(width = 0.8), color= &quot;grey50&quot;, shape = &quot;—&quot;, size=6) +
  #scale_y_continuous(limits = c(-3,6)) +
  #ggtitle(sprintf(&quot;%s expression (%s)&quot;, geneName, condi)) +
  facet_wrap(~sgRNA, ncol=3, scale = &quot;free_x&quot;) +
  theme_full + theme(legend.position = &quot;right&quot;,
                     plot.title = element_text(face = &quot;bold&quot;),
                     strip.text = element_text(size=12), 
                     axis.text.x = element_text(angle = 45, hjust=1, vjust=1))</code></pre>
<p><img src="figure/manuscript_S9_STAT2.Rmd/unnamed-chunk-47-1.png" width="1056" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] parallel  stats4    grid      stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] DESeq2_1.28.1               latex2exp_0.4.0            
 [3] forcats_0.5.1               stringr_1.4.0              
 [5] dplyr_1.0.5                 purrr_0.3.4                
 [7] readr_1.4.0                 tidyr_1.1.3                
 [9] tibble_3.1.0                tidyverse_1.3.0            
[11] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
[13] matrixStats_0.58.0          Biobase_2.48.0             
[15] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
[17] IRanges_2.22.2              S4Vectors_0.26.1           
[19] BiocGenerics_0.34.0         glmnet_4.1-1               
[21] Matrix_1.3-2                ggbeeswarm_0.6.0           
[23] ggplot2_3.3.3               gtable_0.3.0               
[25] limma_3.44.3                jyluMisc_0.1.5             
[27] ComplexHeatmap_2.4.3        pheatmap_1.0.12            
[29] piano_2.4.0                 cowplot_1.1.1              

loaded via a namespace (and not attached):
  [1] utf8_1.1.4             shinydashboard_0.7.1   tidyselect_1.1.0      
  [4] RSQLite_2.2.3          AnnotationDbi_1.50.3   htmlwidgets_1.5.3     
  [7] BiocParallel_1.22.0    maxstat_0.7-25         munsell_0.5.0         
 [10] codetools_0.2-18       DT_0.17                withr_2.4.1           
 [13] colorspace_2.0-0       highr_0.8              knitr_1.31            
 [16] rstudioapi_0.13        ggsignif_0.6.1         labeling_0.4.2        
 [19] git2r_0.28.0           slam_0.1-48            GenomeInfoDbData_1.2.3
 [22] KMsurv_0.1-5           bit64_4.0.5            farver_2.1.0          
 [25] rprojroot_2.0.2        vctrs_0.3.6            generics_0.1.0        
 [28] TH.data_1.0-10         xfun_0.21              sets_1.0-18           
 [31] R6_2.5.0               clue_0.3-58            locfit_1.5-9.4        
 [34] cachem_1.0.4           bitops_1.0-6           fgsea_1.14.0          
 [37] assertthat_0.2.1       promises_1.2.0.1       scales_1.1.1          
 [40] multcomp_1.4-16        beeswarm_0.3.1         egg_0.4.5             
 [43] sandwich_3.0-0         workflowr_1.6.2        rlang_0.4.10          
 [46] genefilter_1.70.0      GlobalOptions_0.1.2    splines_4.0.2         
 [49] rstatix_0.7.0          broom_0.7.5            yaml_2.2.1            
 [52] abind_1.4-5            modelr_0.1.8           backports_1.2.1       
 [55] httpuv_1.5.5           tools_4.0.2            relations_0.6-9       
 [58] ellipsis_0.3.1         gplots_3.1.1           jquerylib_0.1.3       
 [61] RColorBrewer_1.1-2     Rcpp_1.0.6             visNetwork_2.0.9      
 [64] zlibbioc_1.34.0        RCurl_1.98-1.2         ggpubr_0.4.0          
 [67] GetoptLong_1.0.5       zoo_1.8-9              haven_2.3.1           
 [70] cluster_2.1.1          exactRankTests_0.8-31  fs_1.5.0              
 [73] magrittr_2.0.1         data.table_1.14.0      openxlsx_4.2.3        
 [76] circlize_0.4.12        reprex_1.0.0           survminer_0.4.9       
 [79] mvtnorm_1.1-1          hms_1.0.0              shinyjs_2.0.0         
 [82] mime_0.10              evaluate_0.14          xtable_1.8-4          
 [85] XML_3.99-0.5           rio_0.5.26             readxl_1.3.1          
 [88] gridExtra_2.3          shape_1.4.5            compiler_4.0.2        
 [91] KernSmooth_2.23-18     crayon_1.4.1           htmltools_0.5.1.1     
 [94] mgcv_1.8-34            later_1.1.0.1          geneplotter_1.66.0    
 [97] lubridate_1.7.10       DBI_1.1.1              dbplyr_2.1.0          
[100] MASS_7.3-53.1          car_3.0-10             cli_2.3.1             
[103] marray_1.66.0          igraph_1.2.6           pkgconfig_2.0.3       
[106] km.ci_0.5-2            foreign_0.8-81         xml2_1.3.2            
[109] foreach_1.5.1          annotate_1.66.0        vipor_0.4.5           
[112] bslib_0.2.4            XVector_0.28.0         drc_3.0-1             
[115] rvest_1.0.0            digest_0.6.27          rmarkdown_2.7         
[118] cellranger_1.1.0       fastmatch_1.1-0        survMisc_0.5.5        
[121] curl_4.3               shiny_1.6.0            gtools_3.8.2          
[124] rjson_0.2.20           nlme_3.1-152           lifecycle_1.0.0       
[127] jsonlite_1.7.2         carData_3.0-4          fansi_0.4.2           
[130] pillar_1.5.1           lattice_0.20-41        fastmap_1.1.0         
[133] httr_1.4.2             plotrix_3.8-1          survival_3.2-7        
[136] glue_1.4.2             zip_2.1.1              png_0.1-7             
[139] iterators_1.0.13       bit_4.0.4              stringi_1.5.3         
[142] sass_0.3.1             blob_1.2.1             memoise_2.0.0         
[145] caTools_1.18.1        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
