<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2020-10-09" />

<title>Section 3: Proteomic signatures of trisomy12</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.14/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.1.0.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.0.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CLL Proteome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.embl.de/jlu/Proteomics">
    <span class="fa fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">

<div class="btn-group pull-right">
<button type="button" class="btn btn-default btn-xs dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Section 3: Proteomic signatures of trisomy12</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2020-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-10-09
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CLLproteomics_publish/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you'll want to first commit it to the Git repo. If you're still working on the analysis, you can ignore this warning. When you're finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it's best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>manuscript_S3_trisomy12_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/manuscript_S1_Overview_cache/
    Ignored:    analysis/manuscript_S3_trisomy12_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/.trisomy12_norm.pdf
    Untracked:  analysis/manuscript_S1_Overview.Rmd
    Untracked:  analysis/manuscript_S2_genomicAssociation.Rmd
    Untracked:  analysis/manuscript_S3_trisomy12.Rmd
    Untracked:  analysis/manuscript_S4_IGHV.Rmd
    Untracked:  analysis/manuscript_S5_trisomy19.Rmd
    Untracked:  analysis/manuscript_S6_del11q.Rmd
    Untracked:  analysis/manuscript_S7_SF3B1.Rmd
    Untracked:  analysis/manuscript_S8_drugResponse_Outcomes.Rmd
    Untracked:  analysis/manuscript_S9_STAT2.Rmd
    Untracked:  code/utils.R
    Untracked:  data/Fig1A.png
    Untracked:  data/ddsrna_enc.RData
    Untracked:  data/exprCNV_enc.RData
    Untracked:  data/gmts/
    Untracked:  data/patMeta_enc.RData
    Untracked:  data/pepCLL_lumos_enc.RData
    Untracked:  data/pheno1000_enc.RData
    Untracked:  data/proteins_in_complexes
    Untracked:  data/proteomic_LUMOS_enc.RData
    Untracked:  data/proteomic_timsTOF_enc.RData
    Untracked:  data/survival_enc.RData
    Untracked:  output/deResList.RData
    Untracked:  output/deResList_timsTOF.RData
    Untracked:  output/dxdCLL.RData
    Untracked:  output/dxdCLL2.RData
    Untracked:  output/splicingResults.RData

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/analysisSF3B1.Rmd
    Deleted:    analysis/comparePlatforms.Rmd
    Deleted:    analysis/compareProteomicsRNAseq.Rmd
    Deleted:    analysis/correlateCLLPD.Rmd
    Deleted:    analysis/correlateGenomic.Rmd
    Deleted:    analysis/correlateGenomic_removePC.Rmd
    Deleted:    analysis/correlateMIR.Rmd
    Deleted:    analysis/correlateMethylationCluster.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/predictOutcome.Rmd
    Deleted:    analysis/processProteomics_LUMOS.Rmd
    Deleted:    analysis/processProteomics_timsTOF.Rmd
    Deleted:    analysis/qualityControl_LUMOS.Rmd
    Deleted:    analysis/qualityControl_timsTOF.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
</div>
<div id="overview-of-differentially-expressed-proteins" class="section level1">
<h1>Overview of differentially expressed proteins</h1>
<div id="a-table-of-associations-with-10-fdr" class="section level2">
<h2>A table of associations with 10% FDR</h2>
<pre class="r"><code>resList &lt;- filter(resList, Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(adj.P.Val = adj.P.IHW) %&gt;% #use IHW corrected P-value
  mutate(Chr = rowData(protCLL[id,])$chromosome_name)
resList %&gt;% filter(adj.P.Val &lt;= 0.1) %&gt;% 
  select(name, Chr,logFC, P.Value, adj.P.Val) %&gt;%
  mutate_if(is.numeric, formatC, digits=2) %&gt;%
  DT::datatable()</code></pre>
<div id="htmlwidget-687c8f6ec6bc0aecc6e0" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-687c8f6ec6bc0aecc6e0">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214","215","216","217","218","219","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","239","240","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","276","277","278","279","280","281","282","283","284","285","286","287","288","289","290","291","292","293","294","295","296","297","298","299","300","301","302","303","304","305","306","307","308","309","310","311","312","313","314","315","316","317","318","319","320","321","322","323","324","325","326","327","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574","575","576","577","578","579","580","581","582","583","584","585","586","587","588","589","590","591","592","593","594","595","596","597","598","599","600","601","602","603","604","605","606","607","608","609","610","611","612","613","614","615","616","617","618","619","620","621","622","623","624","625","626","627","628","629","630","631","632","633","634","635","636","637","638","639","640","641","642","643","644","645","646","647","648","649","650","651","652","653","654","655","656","657","658","659","660","661","662","663","664","665","666","667","668","669","670","671","672","673","674","675","676","677","678","679","680","681","682","683","684","685","686","687","688","689","690","691","692","693","694","695","696","697","698","699","700","701","702","703","704","705","706","707","708","709","710","711","712","713","714","715","716","717","718","719","720","721","722","723","724","725","726","727","728","729","730","731","732","733","734","735","736","737","738","739","740","741","742","743","744","745","746","747","748","749","750","751","752","753","754","755","756","757","758","759","760","761","762","763","764","765","766","767","768","769","770","771","772","773","774","775","776","777","778","779","780","781","782","783","784","785","786","787","788","789","790","791","792","793","794","795","796","797","798","799","800","801","802","803","804","805","806","807","808","809","810","811","812","813","814","815","816","817","818","819","820","821","822","823","824","825","826","827","828","829","830","831","832","833","834","835","836","837","838","839","840","841","842","843","844","845","846","847","848","849","850","851","852","853","854","855","856","857","858","859","860","861","862","863","864","865","866","867","868","869","870","871","872","873","874","875","876","877","878","879","880","881","882","883","884","885","886","887","888","889","890","891","892","893","894","895","896","897","898","899","900","901","902","903","904","905","906","907","908","909","910","911","912","913","914","915","916","917","918","919","920","921","922","923","924","925","926","927","928","929","930","931","932","933","934","935","936","937","938","939","940","941","942","943","944","945","946","947","948","949","950","951","952","953","954","955","956","957","958","959","960","961","962","963","964","965","966","967","968","969","970","971","972","973","974","975","976","977","978","979","980","981","982","983","984","985","986","987","988","989","990","991","992","993","994","995","996","997","998","999","1000","1001","1002","1003","1004","1005","1006","1007","1008","1009","1010","1011","1012","1013","1014","1015","1016","1017","1018","1019","1020","1021","1022","1023","1024","1025","1026","1027","1028","1029","1030","1031","1032","1033","1034","1035","1036","1037","1038","1039","1040","1041","1042","1043","1044","1045","1046","1047","1048","1049","1050","1051","1052","1053","1054","1055","1056","1057","1058","1059","1060","1061","1062","1063","1064","1065","1066","1067","1068","1069","1070","1071","1072","1073"],["RAP1B","TCEAL3","PSMD9","PTPN11","CS","SLC25A3","TAOK3","ARHGDIB","EEA1","ACYP1","OSBPL8","RPAP3","GCN1L1","EIF2B1","ALDH16A1","USP5","PGAM5","MCTS1","DERA","PPP1R12A","TOP1","C12orf5","RAB35","HIP1R","STAT2","ANKRD13A","C12orf10","LDHB","C11orf68","FAM114A2","TSFM","PYCARD","ESYT1","EML2","SHMT2","ABCB10","ATP2A2","RAB21","PRKAG1","RAB2B","STRAP","CAND1","GIT2","MYL9","GBAS","SCCPDH","VASP","PTGES3","RAP1A","PA2G4","CBR4","TRIM25","NHP2","CASP1","HSD17B8","DNM1L","VCL","VPS28","BROX","TBL2","PNP","EHD4","LDHA","NELFA","UNG","MIER1","UGDH","GOLGA3","UBXN7","ZFP64","RASGRP2","PTPN6","GSTZ1","DGKA","RAN","ALDH5A1","CFDP1","DSN1","VAV1","MARS","XPOT","SLC25A12","PFN1","RBM4B","ACY1","ATP6V0A1","ANP32E","PDPR","SRXN1","RBM38","CHID1","RFX5","VAT1","IRF4","DIABLO","PTK2B","RECQL","STK38","UBE2N","RPRD2","SLC25A20","MYL6","EIF4B","GNG7","C9orf142","JADE3","BSG","CHDH","IQGAP1","LANCL1","ATP2B1","CUL3","TES","FAM208A","SMARCA4","RQCD1","C16orf58","PDP1","CORO1C","TRIM38","C5orf45","SRSF9","LEMD3","RRP1","GCLC","NUMA1","PPM1K","USP15","SMAD2","CNN2","PFN2","METTL7A","ALDH18A1","SPTBN2","NUDT16L1","LPCAT3","PITPNA","CMAS","ABHD14B","PCBD1","IRAK4","BCL10","HMHA1","PKM","OAS2","EPM2AIP1","VPS37B","SACM1L","SLC25A19","SCAF11","RFXAP","CBX5","HRSP12","WNK1","ARHGEF6","CARD16","ARRB2","METTL3","RHOG","TREX1","NOLC1","STARD7","DOK2","NELFE","MPG","TBL1XR1","LUC7L","PMF1","CSTF3","LRRFIP1","TWF1","NSL1","SAMHD1","SLTM","WDR74","CNOT1","NME3","DRG1","ARHGAP4","TCEAL4","GSK3A","LY9","FLNA","ACAP1","GLRX","RBPJ","HP1BP3","SNX9","SP1","CD2AP","CRAT","DCPS","STAMBPL1","LPXN","MLEC","CENPB","DHX16","ITGB1","GAR1","GM2A","PDXK","RABGAP1L","CAP1","RB1","FLII","PPP1R18","MYO1F","NMNAT1","DENR","C1orf35","DDX18","PPP3CA","FRYL","ITPA","IFI30","DDX27","THYN1","ERP29","CYFIP1","SGTB","PGP","C7orf55-LUC7L2","CYP20A1","ABCD1","ARF6","SARM1","MYH9","CCT5","DKC1","DNPH1","ZFR","SMEK1","CBX7","METTL16","PRKD2","PPIE","SMARCC2","HDDC3","HSP90B1","EEF2","AGAP2","MAP2K3","HNRNPLL","CCDC86","RAB4B","SLC25A6","PGM2","MTA1","SCRN2","BZW1","HERC4","CPPED1","AQR","TLN1","OXLD1","PMVK","CENPC","SAMD9","RBMX","PSMC3","GAPDH","MYO18A","ZNF384","KAT7","ST6GAL1","ITGB7","CSTF1","ME2","MPV17","YARS2","MOGS","FAM49A","STX7","TRIAP1","PARP14","CORO1A","UBA6","TPI1","ORC4","AGFG2","RALA","ALDH6A1","TOMM70A","NCOR1","STAM","PUS1","ATP1A1","AK1","TMEM194A","CAPZB","RNF213","RPA1","SH3GL1","HIRIP3","HSP90AA1","DNMBP","RASAL3","TRIM22","MPI","UBA7","GLTP","SEC24C","SNX12","RAC2","GCHFR","MTX2","PPP3CC","IGF2R","MYEF2","NENF","HNRNPA3","FAM49B","SLC4A1AP","ACOT1","PTPRC","C6orf106","CSTF2","CTPS1","CTSD","ASL","ACLY","CLIC3","KIAA1033","SIPA1L1","BIN2","MAPK3","RGS14","LBR","PCYT2","ZMYM3","SAP130","MPHOSPH10","RUNX3","PIP4K2A","RAB43","SRI","ISY1","GLIPR2","SGPP1","HMGCL","RAC1","CAPZA1","BCL2L13","MSN","RAP2B","ZNF512","CETN3","ATG3","CD27","ITGB2","PWP1","ARHGEF18","BMS1","ZMIZ2","TMOD3","NLN","GPATCH8","RPA2","ITGAL","USP24","DNAJC8","MATR3","FERMT3","AMPD3","LEF1","TATDN1","BANK1","TP53I3","AKAP2","TERF2IP","PRRC1","POLR1E","C11orf30","MIS12","DDAH2","DLGAP4","CNPY2","PTGR2","ANXA7","DCLK2","STX5","SNX2","DPP9","TBXAS1","PIH1D1","SIN3A","PUF60","ARHGAP30","UAP1L1","TBC1D10C","NAP1L1","BIN1","PPP3R1","TRAF1","CD72","CDKN1B","ITPRIP","STIM1","GMFG","FNBP4","RBM28","LTA4H","GPD2","DDX24","IQSEC1","ARHGEF1","NOP9","TFCP2","ARHGAP24","PPP1CB","ZC3H12D","PCBD2","GRB2","TPP2","GEMIN5","SF3B3","ZZEF1","RPAIN","TBL1X","TPM4","DENND1C","RPRD1B","SNX5","RBM33","EEF1A1","GAK","CABIN1","ANKFY1","CRYZ","PFKP","PIP4K2B","CRTC2","ENPP4","HTATSF1","WIBG","GRK6","PROSC","AKR7A2","PCCA","ITPR3","GRAP2","RAB3GAP2","FAIM3","PGK1","POLD3","TLK1","NAPA","GNAI3","CPSF7","EPN1","ZHX2","STRBP","GLO1","RANBP3","CAMK1D","CPSF2","ARPP19","BBC3","CCT2","USP9X","BID","GRAP","ORC2","GDAP1","IKBKG","FLNC","MACF1","COMTD1","TMA16","FUBP1","RCL1","CSTF2T","PHAX","SERPINH1","STAT6","MYO1G","SOD1","MAP4","PREX1","LCP1","MDC1","RBM4","CAB39","FHL1","CHEK2","ETF1","BCKDHA","DGKZ","ACOT13","ZNF326","CLPP","FLNB","CTSH","LSS","RPS12","S100A11","RPL5","ENTPD1","RBMXL1","SCPEP1","TSC22D4","BDH1","PPP1R2","GTF3C2","KIAA0196","CYB5R2","MRPS16","PGM2L1","CD180","UBE2O","ATP2A3","NCKAP1L","NR2C2","C1orf174","RAP1GAP2","PPP4R2","TXNRD2","TMPO","NIT1","CD44","SAMSN1","ELMO2","RPS21","HN1L","RPL7L1","PSIP1","CCDC47","EXOSC5","WAPAL","FAM65B","RPLP0","FAM21A","FYTTD1","TNFAIP8L2","GYG1","PSMD1","CD79B","MPLKIP","ACAT2","GPSM3","CNST","WDR1","CYBA","TMED8","SLC25A5","SMNDC1","DNAJA1","SAFB","ERGIC1","ATP1B3","SCLY","NOL7","KPNA1","INPP5B","PGD","WHSC1L1","ZC3H11A","INO80","PUM1","HSPD1","PLEKHO2","C2CD5","SUMF2","TERF2","RNF40","MECR","RNMT","BRK1","NMT2","LAT2","ACTN4","F13A1","PKN1","GIMAP4","DIP2B","C9orf64","CLEC17A","MNDA","EIF3E","SSSCA1","CCDC53","RBM34","SEMA4A","PCYT1A","MGME1","CCNL1","GMIP","C2orf49","TMEM120A","RBM39","S100A4","FIP1L1","SUN2","CYBB","WDR89","TNPO2","TRAFD1","CD3EAP","SFPQ","PES1","PSMA4","SH3KBP1","NAP1L4","NCL","GATAD2A","UTP20","FASN","GPATCH11","HSD17B4","CNOT3","ACTR2","MOB3A","EPB41L2","PEX14","HIRA","GSTO1","BRD2","UBR1","HMBS","SLC9A3R1","MALT1","WDR36","EBNA1BP2","TXNRD1","BCAP31","FCER2","DTX3L","WRNIP1","TPD52","TSG101","KIF2A","GMPPB","ELMSAN1","LGALS1","AP3D1","MTAP","UTP14A","BPGM","SERBP1","SYPL1","ERO1L","FKBP15","VARS","ESYT2","EMC1","AHCY","DEK","MTDH","RBM26","CDK12","SFXN3","GTF2A2","ACTBL2","SZRD1","TRMT1L","CCT6A","CARD11","MAP4K1","ZC3H14","IQGAP2","RPS28","HNRNPH3","ZNF800","CRTC3","G3BP1","PHB","KCNAB2","GSPT2","METTL14","FGR","MTX1","USP48","RPS4X","PLCG1","PI4K2A","RPL6","APIP","DDX17","SNX6","CBLB","PSMB1","GYS1","SAR1B","ALDOC","DAK","PRDX2","CPSF1","AKT2","TAPBPL","ANP32A","DDX55","RPS3A","EML4","BCKDHB","MTMR1","TBL3","SND1","SH3BGRL3","STK4","COX15","RHOA","SART3","MDP1","DHRS1","ARFGAP2","RAPGEF1","DNPEP","FYB","MSH2","MEAF6","PARN","RBM14","CLINT1","RASA1","TOMM22","RBBP9","RANBP9","NOP56","RPS7","CD53","SUPT6H","HS2ST1","PF4","TOMM40","CD81","SMARCAL1","PIK3CD","FBXO18","ILF2","AGO2","PSMC2","ATP5B","BMF","SRSF6","RALB","M6PR","SNRPA1","ISG20","NDUFV3","HMOX1","ATP6V0D1","FGD2","RPS2","ANXA11","TP53RK","FAM129A","ANXA1","PSMA6","ERLIN2","EMC2","MRPL19","RBM15","TIMM8A","PSMA5","PHB2","ROCK1","RFC4","VAMP8","GNB2L1","FOXO1","ARIH1","TBC1D5","LONP1","DCAF13","FUK","GRHPR","UBE3A","SYMPK","CR1","C17orf49","JAK1","SQRDL","BAX","HSD17B11","MCEE","CAPRIN1","MAK16","SBDS","IFI16","AIP","ADD1","CRIPT","SEC23A","ECI2","IPO5","CPSF4","LAGE3","CLIC1","RPS6","PAWR","PHIP","GNL1","DFFA","PKIG","DDX49","DBNL","EMC7","NIPBL","TPT1","ATXN2L","PNPT1","YAF2","KLHL36","CPSF3","PSMA2","MRFAP1","JUND","CELF1","NCF4","RPS3","AK3","GAPVD1","CMC4","CORO7","HGS","HTRA2","GIMAP7","LAMP1","SERPINB6","KAT8","YTHDC1","CLIP1","RPLP2","INPP5D","PSME2","BRD3","COTL1","HAT1","MMAB","MORC3","DOCK11","ARID4A","SETD1A","TBCB","SDCBP","TAF6","SP4","CFL1","FAHD2A","GZMA","INF2","DIAPH1","THBS1","TMUB1","PSTPIP1","MBD2","ATP5C1","RRM2B","GDI2","METAP2","ARPC1B","RPS27L","EFHD2","ADAT2","SLC30A7","HNRNPM","PPP6C","NIT2","GMPR2","GLG1","THAP11","TCF4","CD79A","DOCK2","PYCR2","KIAA0368","TUBA4A","DOCK8","HNRNPU","GNS","SIPA1","TMX1","FTO","NARS","LRRK2","RTFDC1","ANXA4","MSL1","BAG2","ACTR5","NFYA","HS1BP3","MYO9B","CCDC127","FAM107B","NDUFS5","PRKACA","GNAI2","PSAP","AP3S1","VPS72","CNP","GNB2","AIDA","MPC2","PSMD2","SMS","NDUFAF2","MARS2","MMGT1","CAPN2","CCR7","NRAS","PRDM2","CIRH1A","RPL15","ATF7IP","ITPR2","ITGA4","ALAD","TRMT112","EXOSC4","SRP68","WDR3","ACSL4","SIRT3","THOC6","FHIT","CMC2","ECI1","THRAP3","ITGB3","NUDCD2","EIF5B","ISOC1","PLEKHO1","GNPDA1","RPS11","ZFX","MOCS3","MS4A1","ENO1","DUT","PPP5C","FH","CTR9","RPA3","ABCF1","ALG1","DNAJC19","MGEA5","WDR12","RBM22","PCCB","CHMP4B","NCEH1","CDC16","HDAC7","DNAJC7","TBC1D10B","TRMT6","YTHDF2","PSMD5","CDK9","RPF2","CRYL1","PSMD12","CDC5L","BDH2","ADNP","FCRL3","RUNX1","GSTP1","NNT","AGK","PDLIM1","UMPS","CDC42","SFXN2","XPNPEP1","NTMT1","NUB1","VPS4B","NKAP","MAP2K1","SUB1","TXNDC5","DNTTIP1","SYNE1","PTPRCAP","DERL1","HNRNPR","MAN2C1","TBC1D13","ZNF428","RPL3","GNAO1","GATAD2B","EIF3K","MFF","ACTN1","HNRNPUL2","SNIP1","AARS2","ATOX1","LHPP","SRSF2","PSMD11","PCIF1","NDUFB5","FAM32A","APOOL","RDH11","TP53BP1","MMAA","CD22","FAM3C","SLC25A24","SEC24A","TLE1","RFC1","ACTR8","PRDX1","AKR1A1","ITGAX","EOGT","SRSF11","MAPK13","RPUSD2","BLOC1S4","GBP1","PRKAA1","SCAF1","RCOR3","LSM14B","TAF5","DCXR","DDT","STXBP2","TMA7","PARP10","ARPC4","GTF3C3","CDC40","SRSF1","AATF","NOC2L","NFATC2IP","PPBP","EIF3A","CHMP6","QRICH1","SLC3A2","DNAJB2","CD48","CWC27","FCHSD2","PFKFB2","ZNRF2","PPP6R1","HNRNPL","DENND2D","GMDS","CD74","NOL11","ACIN1","TFRC","TSNAX","ST13","GPX7","BCAT1","C7orf50","TUBB6","BCL11A","OLA1","ELAVL1","SRP54","RPS19BP1","CCT3","MEN1","RANBP1","GALK2","TAGLN2"],["12","X","12","12","12","12","12","12","12","14","12","12","12","12","19","12","12","X","12","12","20","12","12","12","12","12","12","12","11","5","12","16","12","19","12","1","12","12","12","14","12","12","12","20","7","1","19","12","1","12","4","17","5","11","6","12","10","8","1","7","14","15","11","4","12","1","4","12","3","20","11","12","14","12","12","6","16","20","19","12","12","2","17","11","3","17","1","16","20","20","11","1","17","6","12","8","12","6","12","1","3","12","12","19","9","X","19","3","15","2","12","2","7","3","19","2","16","8","12","6","5","12","12","21","6","11","4","12","18","19","3","12","10","11","16","12","17","12","3","10","12","1","19","15","12","3","12","3","17","12","13","12","8","12","X","11","17","14","11","3","10","2","8","6","16","3","16","1","11","2","12","1","20","15","11","16","16","22","X","X","19","1","X","17","5","4","1","6","12","6","9","11","10","11","12","20","6","10","4","5","21","1","1","13","17","6","19","1","12","1","2","4","4","20","19","20","11","12","15","5","16","7","2","X","14","17","22","5","X","6","5","14","22","17","19","1","12","15","12","19","12","17","2","11","19","X","4","14","17","2","10","16","15","9","17","1","4","7","X","11","12","17","12","17","3","12","20","18","2","12","2","2","6","12","3","16","4","12","2","7","7","14","3","17","10","12","1","9","12","1","17","17","19","16","14","10","19","11","15","3","12","10","X","22","15","2","8","6","15","1","2","8","2","14","1","6","X","1","11","7","17","9","12","14","12","16","5","1","17","X","2","2","1","10","3","7","3","9","14","1","7","1","22","X","3","2","5","3","12","21","12","19","10","7","15","5","17","1","16","1","1","5","11","11","4","8","4","2","9","16","5","9","11","17","6","20","12","14","10","4","11","5","19","7","19","15","8","1","9","11","12","2","2","9","9","12","10","11","19","11","7","12","2","14","3","19","14","12","4","2","6","5","17","13","5","16","17","17","X","19","19","20","20","7","6","4","22","17","1","10","17","1","6","X","12","5","8","1","13","6","22","1","1","X","11","2","19","1","11","19","8","9","6","19","10","14","15","19","12","X","22","17","2","8","X","7","1","10","4","1","9","10","5","11","12","7","21","3","20","13","6","11","2","X","22","5","19","11","6","1","19","3","15","21","6","1","1","10","1","17","7","3","3","2","8","11","10","11","5","17","17","12","3","1","17","3","22","12","1","11","21","20","20","16","6","9","17","19","10","6","12","10","3","1","3","2","17","7","6","6","1","4","16","14","X","10","9","19","5","3","2","6","3","1","1","8","1","15","1","2","15","12","7","16","16","1","18","3","10","7","19","6","19","7","12","9","19","1","8","11","12","1","1","3","20","3","19","2","7","20","1","4","22","X","14","19","12","19","1","22","15","X","11","2","19","12","17","2","5","19","2","19","6","1","22","10","6","15","11","17","18","5","1","12","X","19","3","6","8","11","5","3","14","22","19","9","X","7","1","7","14","9","6","7","1","20","6","8","13","17","10","15","5","1","1","7","7","19","14","5","19","10","7","15","5","17","1","X","4","1","1","1","X","20","10","12","11","22","14","3","6","19","5","17","11","19","8","19","12","15","12","4","2","6","X","16","7","1","20","10","3","12","14","14","11","9","2","5","2","1","16","11","5","5","22","20","6","20","2","1","17","1","4","19","11","2","1","10","1","8","7","12","15","20","2","12","15","15","21","22","16","6","16","10","20","1","9","14","8","8","2","1","X","1","12","18","3","2","5","13","15","3","19","8","16","9","15","19","1","17","1","15","19","4","2","11","8","7","1","11","4","2","14","6","13","7","X","6","9","12","6","6","1","20","19","7","15","5","13","16","2","12","16","2","7","4","19","11","22","11","9","9","X","16","17","2","7","13","6","16","4","12","11","2","14","9","16","2","12","21","X","14","16","19","8","7","7","11","2","5","14","5","15","7","15","18","10","8","10","12","7","15","1","6","1","19","9","3","14","16","16","18","19","5","1","9","2","9","1","12","11","14","16","18","12","20","2","17","6","20","6","2","19","5","10","1","19","3","10","5","1","17","7","1","1","3","X","5","2","X","1","17","1","1","16","3","12","12","2","9","11","8","17","1","X","11","16","3","16","16","1","17","5","2","5","1","5","19","X","20","11","1","15","19","1","11","7","6","16","3","10","2","5","3","20","3","13","12","17","16","20","1","9","9","6","13","17","6","4","20","1","21","11","5","7","10","3","1","10","10","9","7","18","X","15","5","6","20","6","11","8","1","15","9","19","22","16","1","19","2","14","11","1","6","5","10","17","17","20","3","19","X","14","15","4","19","7","1","5","9","4","3","1","1","16","3","1","6","15","4","1","5","19","1","20","10","17","22","19","3","8","3","2","6","17","17","1","16","4","10","17","3","11","2","1","5","11","1","7","19","19","1","6","5","17","14","3","1","22","1","12","7","18","2","2","19","14","22","1","11","22","15","1"],["0.96","-0.93","0.56","0.59","0.61","0.56","0.72","0.81","0.83","-0.73","0.68","0.42","0.44","0.6","0.59","0.49","0.66","0.46","0.68","0.63","-0.46","0.78","0.67","1.3","0.61","0.76","0.64","0.84","-0.42","-0.33","0.63","0.91","0.62","-0.61","0.79","-0.97","0.56","0.41","0.46","-0.6","0.37","0.43","0.52","0.51","-0.44","-0.94","0.67","0.47","0.64","0.58","-0.76","0.56","-0.37","1.1","-0.74","0.53","  1","0.94","-0.66","-0.57","  1","-0.55","0.62","-0.35","0.47","-0.38","-0.67","0.63","-0.39","-1.2","0.78","0.74","-0.93","0.43","0.29","-0.71","-0.44","-0.43","0.62","0.42","0.48","-0.41","0.45","-0.27","-0.82","-0.98","-0.38","-0.44","-0.37","-0.45","-0.47","-0.49","-0.76","-0.63","0.44","0.59","0.62","0.48","0.3","-0.28","0.65","0.42","0.54","  1","0.92","-1.2","0.53","-1.2","0.48","-0.44","0.81","0.37","  1","-0.35","0.48","0.22","-0.64","-0.43","0.5","0.4","-0.59","0.3","0.49","-0.39","-0.67","-0.38","-0.48","0.68","0.33","0.44","-1.4","0.49","-0.3","-1.2","-0.35","0.62","0.46","0.41","-0.58","-0.86","0.55","0.6","0.38","0.39","0.55","-0.78","1.6","0.36","-0.59","0.4","-0.45","0.53","-0.53","0.52","0.48","0.73","0.54","-0.35","0.46","-0.41","-0.47","-0.38","0.97","-0.25","-0.37","-0.31","-0.25","-0.4","-0.24","0.52","0.83","-0.35","1.2","-0.26","-0.29","0.28","-0.48","0.25","0.39","-0.64","0.24","-0.7","0.59","0.53","1.2","-0.44","-0.34","-0.58","0.5","-0.27","-0.63","-0.31","-0.5","0.39","0.28","-0.38","-0.3","0.94","-0.3","-0.78","0.65","-0.64","0.31","1.1","0.41","0.57","0.6","-0.41","0.45","-0.36","-0.24","0.38","0.44","-0.57","0.75","-0.32","-0.32","0.49","0.46","-0.61","0.35","-0.21","-0.37","0.77","0.45","-0.57","0.42","0.31","-0.27","-0.98","-0.25","-0.35","-1.3","-0.4","0.45","-0.24","0.29","-0.59","0.34","0.24","0.58","0.39","-0.68","-0.4","0.37","0.34","-0.26","-0.39","-0.56","0.37","-0.35","0.63","-0.23","0.31","-0.48","-0.38","-0.38","0.28","-0.21","0.2","0.4","0.32","0.34","-0.24","-0.74","0.72","-0.35","0.37","-0.43","0.46","-0.32","0.86","-0.36","0.62","0.49","0.39","0.27","0.34","-0.33","-0.52","0.3","-0.42","-0.26","-0.28","-0.41","0.47","0.42","-1.1","0.44","0.27","0.34","-0.38","0.39","-0.52","0.28","-0.55","0.4","0.31","-0.32","0.7","0.43","-0.21","-0.34","0.6","-0.74","-0.44","0.36","0.35","-0.82","-0.54","-0.2","0.33","-0.26","-0.4","0.4","-0.52","-0.22","0.41","0.55","-0.41","-0.31","0.79","0.45","-0.7","0.65","-0.28","0.39","0.63","-0.29","-0.34","-0.26","-0.34","0.43","0.48","-0.66","-0.37","-0.21","0.6","-0.59","-0.34","0.51","0.29","-0.37","0.33","0.37","-0.39","-0.5","0.29","0.55","  1","0.42","0.46","-0.29","-0.38","0.26","-0.66","-0.35","-0.32","0.75","0.31","-0.22","-0.24","0.38","0.67","-0.65","-0.38","0.88","-0.71","-0.61","-0.26","-0.38","-0.39","0.67","-0.48","0.47","-0.52","0.36","-0.65","-0.25","-0.9","-0.39","0.29","0.37","0.81","0.29","-0.23","-0.19","0.3","-0.64","0.33","0.43","0.47","0.33","-0.59","0.84","0.35","0.62","0.35","0.31","-0.28","-0.27","0.35","-0.44","-0.37","0.74","0.39","-0.34","0.35","-0.46","0.28","0.63","-0.31","-0.25","0.46","-0.36","-0.14","0.37","-0.44","-0.23","0.37","0.34","-0.23","0.33","-0.28","0.21","0.22","-0.5","0.33","-0.49","-0.53","0.37","-0.29","-0.41","-0.21","0.49","0.42","-0.21","-0.27","-0.35","-0.51","0.96","-0.26","-0.99","0.22","-0.33","-0.54","0.27","0.28","-0.21","0.33","-0.33","-0.56","-0.54","-0.22","0.44","-0.22","-0.36","-1.2","0.19","0.21","0.6","0.49","-0.36","-0.48","0.21","0.57","0.46","-0.61","-0.47","-0.2","-0.21","-0.19","-0.23","-0.45","0.31","0.54","-0.31","-0.34","0.48","0.38","-0.27","-0.33","0.41","1.2","-0.49","0.2","-0.31","0.44","-0.46","-0.29","0.35","0.88","-0.59","-0.58","0.16","0.74","0.2","-0.38","-0.34","0.68","0.22","-0.44","-0.24","-0.43","0.42","-1.1","-0.36","-0.53","0.68","0.29","0.41","0.3","-0.27","-0.59","0.63","-0.34","0.23","0.19","-0.22","-0.33","-0.84","0.4","0.23","-0.21","-0.23","0.38","-0.25","-0.23","-0.2","0.25","0.23","0.34","-0.28","0.48","0.27","0.17","0.69","-0.4","-0.35","0.28","0.26","0.26","0.76","-0.35","0.34","-0.27","0.19","-0.21","-0.44","0.38","-0.32","-0.43","0.41","-0.36","0.28","-0.46","-0.3","-0.28","-0.35","0.29","0.28","0.34","-0.4","-0.28","-0.19","-0.29","-0.24","0.36","-0.5","0.73","0.42","0.79","0.25","0.79","0.37","-0.26","-0.66","0.67","0.21","-0.43","0.38","-0.53","-0.55","-0.19","-0.53","-0.35","0.21","-0.55","-0.44","-0.14","0.72","-0.18","0.44","0.93","-0.43","0.32","0.31","-0.25","-0.15","-0.23","0.16","0.34","0.26","-0.24","0.17","-0.22","0.41","-0.45","-0.44","0.24","0.2","0.33","0.7","-0.3","-0.45","-0.23","-0.26","0.21","-0.36","0.4","0.43","-0.19","-0.22","0.4","0.3","-0.79","0.33","-0.24","0.44","1.1","0.2","0.23","-0.44","0.72","0.17","-0.25","-0.22","-0.47","0.31","0.45","0.27","0.28","0.21","0.3","-0.3","-0.51","-0.17","-0.41","-0.21","-0.53","-0.35","-0.19","0.36","0.35","-0.26","0.16","-0.29","0.34","-0.2","0.68","0.27","-0.19","-0.29","0.55","-0.15","0.18","0.32","0.22","-0.34","0.44","-0.21","-0.25","0.2","-0.48","-0.32","0.19","0.38","-0.14","0.21","-0.42","0.2","0.35","-0.2","-0.34","-0.28","0.86","-0.16","0.27","0.72","-0.16","0.32","0.2","0.41","-0.32","0.24","-0.2","-0.21","0.3","-0.21","-0.4","0.19","0.18","-0.21","-0.37","-0.18","-0.23","-0.2","0.76","-0.18","-0.25","-0.22","-0.16","-0.17","-0.43","0.31","-0.53","-0.22","-0.26","0.32","0.43","-0.17","-0.47","0.92","0.23","0.49","-0.28","0.29","-0.51","-0.15","0.35","0.16","0.18","-0.55","-0.17","0.39","0.27","-0.14","0.34","-0.53","-0.48","-0.27","0.33","0.17","-0.25","-0.18","0.8","0.87","0.18","-0.36","-0.26","0.37","-0.19","-0.32","0.15","0.2","0.25","-0.13","-0.25","0.19","-0.32","0.28","-0.17","0.22","-0.24","-0.35","-0.32","-0.19","-0.15","0.81","-0.33","0.29","0.3","0.21","-0.25","-0.52","-0.14","-0.3","0.77","-0.2","-0.24","0.33","-0.42","-0.17","-0.22","0.19","-0.16","-0.31","0.22","0.17","0.39","-0.2","-0.21","-0.13","-0.52","0.88","0.27","-0.3","-0.2","0.45","-0.15","0.21","0.48","-0.47","-0.16","0.18","-0.57","-0.37","-0.25","0.29","0.18","-0.26","0.26","-0.32","0.31","-0.15","-0.28","0.61","0.27","0.81","-0.29","-0.25","0.35","0.2","0.33","0.28","0.27","0.25","-0.19","0.39","-0.24","0.21","-0.3","-0.15","0.22","-0.51","-0.26","-0.37","0.2","-0.27","0.69","0.44","0.2","0.89","-0.23","0.31","-0.15","0.22","-0.28","0.24","0.35","0.2","0.16","0.47","-0.65","0.24","-0.15","0.21","-0.23","-0.22","0.39","-0.46","-0.37","0.56","0.2","-0.36","0.23","-0.25","0.23","-0.16","0.35","0.27","-0.22","-0.17","0.34","0.4","-0.23","0.95","-0.23","-0.37","-0.22","-0.59","-0.25","0.21","-0.29","0.27","-0.16","0.26","0.23","0.39","0.16","-0.44","0.32","0.31","0.27","-0.32","0.15","-0.38","-0.32","0.3","-0.34","0.45","-0.39","0.17","-0.49","-0.19","0.21","-0.29","0.49","0.85","0.25","-0.28","-0.13","0.17","-0.23","0.17","-0.34","-0.14","-0.7","0.24","-0.27","-0.14","0.58","-0.22","0.19","-0.16","-0.42","-0.23","0.16","-0.35","-0.23","0.58","0.17","-0.2","0.22","0.2","-0.24","-0.26","0.18","-0.54","-0.26","-0.23","-0.25","-0.11","-0.26","0.21","-0.31","-0.28","-0.29","0.21","0.22","-0.15","-0.2","-0.24","0.15","-0.43","-0.39","0.14","-0.12","-0.52","-0.17"," -1","-0.21","-0.36","-0.21","0.35","0.22","0.24","0.38","-0.42","0.23","0.22","0.29","0.16","-0.2","0.21","0.24","-0.92","-0.28","-0.59","0.34","0.2","-0.15","0.54","-0.16","-0.27","0.18","-1.1","-0.14","0.17","-0.25","0.63","-0.19","-0.17","-0.19","-0.35","-0.34","-0.14","0.12","-0.16","-0.27","-0.35","-0.18","0.22","-0.23","-0.28","0.64","-0.38","-0.26","-0.29","-0.21","-0.14","-0.32","-0.24","-0.25","0.54","-0.44","-0.14","-0.34","-0.33","0.31","0.55","0.24","-0.19","-0.31","-0.37","-0.31","-0.25","-0.22","0.18","-0.33","0.2","0.17","-0.38","-0.17","-0.11","-0.51","-0.18","-0.38","0.79","0.14","0.28","-0.21","0.29","-0.2","0.3","-0.25","-0.39","0.39","0.28","0.19","-0.12","0.32","-0.3","0.31","-0.27","-0.14","-0.51","-0.17","-0.22","-0.56","0.92","-0.24","0.66","-0.36","0.15","-0.14","0.18","-0.9","0.12","-0.15","0.17","-0.25","0.24"],["5.4e-13","1.5e-12","2e-12","3.3e-12","3.6e-12","3.7e-12","7.9e-12","1.2e-11","2.1e-11","5.9e-11","1.6e-10","2.6e-10","5.5e-10","7.7e-10","7.8e-10","8.3e-10","8.5e-10","1.5e-09","1.7e-09","2e-09","2.3e-09","2.6e-09","3.1e-09","3.3e-09","3.7e-09","4.5e-09","4.8e-09","5e-09","5e-09","5.9e-09","6.3e-09","6.5e-09","6.5e-09","7.6e-09","8.6e-09","9.4e-09","9.5e-09","1e-08","1.4e-08","1.6e-08","1.6e-08","1.7e-08","1.9e-08","3e-08","3.1e-08","3.2e-08","3.7e-08","3.8e-08","4e-08","4e-08","4.4e-08","4.8e-08","6.8e-08","7.9e-08","8.4e-08","8.7e-08","8.8e-08","1.1e-07","1.1e-07","1.5e-07","1.6e-07","1.6e-07","2e-07","2e-07","2.2e-07","2.2e-07","2.3e-07","2.3e-07","2.5e-07","2.5e-07","3e-07","3.7e-07","3.8e-07","3.9e-07","3.9e-07","4.1e-07","4.4e-07","4.5e-07","5.9e-07","6.5e-07","7e-07","7e-07","7.3e-07","7.3e-07","7.3e-07","8.3e-07","9e-07","9.4e-07","1e-06","1.3e-06","1.4e-06","1.4e-06","1.6e-06","1.6e-06","1.7e-06","1.8e-06","1.8e-06","1.8e-06","1.9e-06","2e-06","2e-06","2.2e-06","2.3e-06","2.3e-06","2.4e-06","2.6e-06","2.6e-06","2.6e-06","2.7e-06","2.9e-06","3e-06","3.2e-06","3.2e-06","3.2e-06","3.8e-06","3.8e-06","3.9e-06","4e-06","4.4e-06","4.4e-06","4.6e-06","4.7e-06","4.8e-06","4.8e-06","4.9e-06","5e-06","5.5e-06","5.5e-06","5.5e-06","6.3e-06","6.3e-06","6.4e-06","6.6e-06","7e-06","7.1e-06","8e-06","8.2e-06","8.3e-06","8.4e-06","8.6e-06","8.8e-06","8.9e-06","8.9e-06","8.9e-06","9.2e-06","9.2e-06","9.2e-06","9.4e-06","9.5e-06","9.7e-06","9.8e-06","1e-05","1e-05","1.1e-05","1.1e-05","1.1e-05","1.1e-05","1.2e-05","1.2e-05","1.2e-05","1.2e-05","1.3e-05","1.3e-05","1.3e-05","1.3e-05","1.5e-05","1.6e-05","1.6e-05","1.6e-05","1.7e-05","1.7e-05","1.7e-05","1.8e-05","1.8e-05","1.9e-05","1.9e-05","2e-05","2.1e-05","2.1e-05","2.1e-05","2.4e-05","2.5e-05","2.5e-05","2.5e-05","2.6e-05","2.7e-05","2.7e-05","2.8e-05","2.8e-05","2.8e-05","2.9e-05","3.1e-05","3.1e-05","3.1e-05","3.2e-05","3.2e-05","3.3e-05","3.6e-05","3.6e-05","3.7e-05","3.7e-05","3.7e-05","3.8e-05","3.9e-05","3.9e-05","4e-05","4.3e-05","4.3e-05","4.4e-05","4.5e-05","4.5e-05","4.6e-05","4.6e-05","4.7e-05","4.8e-05","4.9e-05","4.9e-05","5e-05","5.2e-05","5.3e-05","5.7e-05","5.7e-05","5.8e-05","5.8e-05","5.9e-05","6e-05","6e-05","6.5e-05","6.7e-05","6.8e-05","6.9e-05","6.9e-05","7e-05","7e-05","7e-05","7.1e-05","7.1e-05","7.3e-05","7.5e-05","7.7e-05","7.7e-05","7.8e-05","7.8e-05","8.2e-05","8.5e-05","8.5e-05","8.8e-05","9e-05","9.2e-05","9.2e-05","9.4e-05","9.8e-05","0.0001","0.0001","0.0001","0.0001","0.0001","0.00011","0.00011","0.00011","0.00011","0.00011","0.00011","0.00011","0.00011","0.00011","0.00012","0.00012","0.00012","0.00013","0.00013","0.00013","0.00013","0.00013","0.00013","0.00014","0.00014","0.00014","0.00014","0.00014","0.00014","0.00015","0.00015","0.00015","0.00015","0.00015","0.00015","0.00016","0.00017","0.00017","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00018","0.00019","0.00019","0.00019","0.00019","0.00019","0.00019","0.0002","0.0002","0.0002","0.00021","0.00021","0.00021","0.00021","0.00021","0.00022","0.00022","0.00022","0.00023","0.00023","0.00023","0.00023","0.00023","0.00023","0.00024","0.00024","0.00024","0.00024","0.00024","0.00025","0.00026","0.00026","0.00027","0.00027","0.00027","0.00027","0.00027","0.00028","0.00028","0.00028","0.00028","0.00029","0.00029","0.0003","0.00031","0.00032","0.00032","0.00032","0.00032","0.00032","0.00032","0.00033","0.00033","0.00034","0.00034","0.00036","0.00036","0.00037","0.00037","0.00038","0.00039","0.0004","0.0004","0.00041","0.00042","0.00042","0.00043","0.00043","0.00044","0.00044","0.00045","0.00045","0.00045","0.00045","0.00046","0.00046","0.00046","0.00047","0.00047","0.00048","0.00049","0.0005","0.0005","0.0005","0.00051","0.00052","0.00052","0.00053","0.00053","0.00054","0.00054","0.00054","0.00054","0.00055","0.00055","0.00056","0.00056","0.00058","0.00059","0.0006","0.0006","0.0006","0.0006","0.00063","0.00063","0.00064","0.00064","0.00064","0.00065","0.00065","0.00066","0.00068","0.00069","0.00069","0.00069","0.00069","0.00069","0.00071","0.00071","0.00072","0.00072","0.00072","0.00073","0.00074","0.00074","0.00075","0.00077","0.00077","0.00078","0.00079","0.0008","0.00081","0.00082","0.00083","0.00084","0.00086","0.00086","0.00087","0.00089","0.0009","0.00091","0.00091","0.00093","0.00095","0.00095","0.00095","0.00097","0.00099","0.001","0.001","0.001","0.001","0.001","0.001","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0011","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0012","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0013","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0014","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0015","0.0016","0.0016","0.0016","0.0016","0.0016","0.0016","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0017","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0018","0.0019","0.0019","0.0019","0.0019","0.002","0.002","0.002","0.002","0.002","0.002","0.002","0.0021","0.0021","0.0021","0.0021","0.0021","0.0021","0.0021","0.0021","0.0021","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0023","0.0024","0.0024","0.0024","0.0024","0.0025","0.0025","0.0025","0.0025","0.0025","0.0026","0.0026","0.0026","0.0026","0.0026","0.0026","0.0026","0.0026","0.0027","0.0027","0.0027","0.0027","0.0027","0.0027","0.0027","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0028","0.0029","0.0029","0.0029","0.0029","0.0029","0.0029","0.003","0.0031","0.0031","0.0031","0.0032","0.0032","0.0032","0.0032","0.0033","0.0033","0.0033","0.0034","0.0034","0.0034","0.0034","0.0034","0.0034","0.0034","0.0034","0.0035","0.0036","0.0036","0.0036","0.0036","0.0037","0.0037","0.0037","0.0037","0.0037","0.0037","0.0037","0.0037","0.0038","0.0038","0.0038","0.0038","0.0038","0.0039","0.0039","0.0039","0.004","0.004","0.004","0.004","0.004","0.0041","0.0041","0.0041","0.0041","0.0041","0.0041","0.0041","0.0041","0.0042","0.0042","0.0042","0.0042","0.0043","0.0043","0.0044","0.0044","0.0044","0.0045","0.0045","0.0045","0.0046","0.0046","0.0046","0.0047","0.0047","0.0047","0.0047","0.0047","0.0047","0.0047","0.0047","0.0048","0.0048","0.0049","0.005","0.005","0.005","0.0051","0.0051","0.0051","0.0052","0.0052","0.0052","0.0052","0.0052","0.0052","0.0052","0.0054","0.0054","0.0054","0.0054","0.0054","0.0055","0.0056","0.0056","0.0056","0.0056","0.0056","0.0056","0.0056","0.0057","0.0057","0.0059","0.0059","0.0059","0.0059","0.0059","0.0059","0.0059","0.006","0.006","0.006","0.0061","0.0061","0.0061","0.0061","0.0061","0.0062","0.0062","0.0062","0.0063","0.0063","0.0063","0.0063","0.0063","0.0063","0.0064","0.0064","0.0064","0.0064","0.0064","0.0064","0.0064","0.0064","0.0065","0.0065","0.0065","0.0066","0.0067","0.0067","0.0068","0.0068","0.0068","0.0069","0.0069","0.007","0.007","0.0071","0.0071","0.0071","0.0071","0.0072","0.0072","0.0073","0.0073","0.0073","0.0075","0.0075","0.0075","0.0075","0.0075","0.0075","0.0077","0.0077","0.0078","0.0078","0.0078","0.0079","0.008","0.008","0.0081","0.0081","0.0081","0.0082","0.0082","0.0082","0.0083","0.0083","0.0083","0.0083","0.0083","0.0084","0.0084","0.0084","0.0085","0.0085","0.0085","0.0085","0.0085","0.0086","0.0086","0.0087","0.0087","0.0087","0.0088","0.0088","0.0088","0.0088","0.0089","0.0089","0.0089","0.009","0.009","0.009","0.0091","0.0091","0.0091","0.0091","0.0092","0.0092","0.0092","0.0092","0.0092","0.0092","0.0093","0.0094","0.0094","0.0094","0.0095","0.0095","0.0095","0.0096","0.0097","0.0097","0.0097","0.0097","0.0099","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.01","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.011","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.013","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.014","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.015","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.016","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.017","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.018","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.02","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.022","0.022","0.022","0.022","0.022","0.022","0.022","0.022","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.023","0.024","0.024","0.024","0.024"],["2e-09","3.4e-09","3.5e-09","4.4e-09","4.4e-09","4.4e-09","8.4e-09","1.1e-08","1.4e-08","3.7e-08","9.3e-08","1.4e-07","2.7e-07","3.2e-07","3.2e-07","3.2e-07","3.2e-07","5.5e-07","5.8e-07","6.6e-07","6.8e-07","8e-07","8.4e-07","9.2e-07","1.1e-06","1.1e-06","1.2e-06","1.2e-06","1.2e-06","1.3e-06","1.3e-06","1.4e-06","1.3e-06","1.4e-06","1.7e-06","1.7e-06","1.7e-06","2.2e-06","2.4e-06","2.6e-06","2.6e-06","2.6e-06","3.7e-06","5.2e-06","4.9e-06","5.3e-06","5.2e-06","5.3e-06","5.3e-06","5.3e-06","5.6e-06","6.4e-06","8.7e-06","9.4e-06","1e-05","1e-05","1e-05","1.2e-05","1.2e-05","1.6e-05","1.9e-05","1.7e-05","1.9e-05","2e-05","2.3e-05","2.4e-05","2.2e-05","2.2e-05","2.3e-05","2.3e-05","2.6e-05","3.2e-05","3.2e-05","3.3e-05","3.2e-05","3.3e-05","4.3e-05","3.5e-05","4.8e-05","5.1e-05","5.1e-05","5.2e-05","5.3e-05","5.3e-05","6.4e-05","7.2e-05","7.7e-05","7.9e-05","7.7e-05","8.8e-05","9.7e-05","9.4e-05","0.0001","0.00011","0.00013","0.00011","0.00013","0.00012","0.00012","0.00013","0.00013","0.00014","0.00014","0.00014","0.00014","0.00015","0.00017","0.00015","0.00015","0.00019","0.0002","0.0002","0.00017","0.0002","0.0002","0.0002","0.0002","0.00021","0.00023","0.00023","0.00023","0.00023","0.00027","0.00023","0.00025","0.00024","0.00026","0.00027","0.00026","0.00028","0.00028","0.00029","0.00029","0.0003","0.00032","0.00034","0.00034","0.00035","0.00035","0.00035","0.00036","0.00037","0.00036","0.00036","0.00036","0.00037","0.00037","0.00037","0.00037","0.00045","0.00037","0.00038","0.00041","0.00041","0.00048","0.00042","0.00042","0.00045","0.00046","0.00045","0.00046","0.00047","0.00046","0.00048","0.00049","0.00051","0.00054","0.00054","0.00058","0.00056","0.00057","0.00058","0.00059","0.00061","0.00062","0.00063","0.00063","0.00066","0.00068","0.00068","0.00079","0.00088","0.00088","0.00075","0.00081","0.00081","0.00095","0.00082","0.00082","0.00098","0.00089","0.00088","0.001","0.00089","0.00091","0.00096","0.00095","0.001","0.0012","0.001","0.001","0.0012","0.001","0.0011","0.0011","0.0011","0.0011","0.0011","0.0014","0.0014","0.0012","0.0013","0.0013","0.0012","0.0013","0.0013","0.0013","0.0013","0.0013","0.0016","0.0015","0.0014","0.0016","0.0015","0.0014","0.0014","0.0015","0.0016","0.0018","0.0016","0.0019","0.0016","0.0016","0.0016","0.0016","0.0016","0.0016","0.0018","0.0017","0.002","0.0018","0.0018","0.0018","0.0018","0.0019","0.0019","0.0019","0.002","0.002","0.002","0.0021","0.0021","0.0021","0.0022","0.0021","0.0022","0.0022","0.0022","0.0022","0.0022","0.0022","0.0023","0.0023","0.0023","0.0023","0.0023","0.0025","0.0025","0.0026","0.003","0.0025","0.0026","0.0026","0.0026","0.0031","0.0027","0.0028","0.0028","0.0033","0.0028","0.0033","0.0033","0.003","0.0029","0.0029","0.003","0.003","0.0032","0.0037","0.0034","0.0034","0.0034","0.0034","0.0038","0.0034","0.0034","0.0039","0.0034","0.0034","0.0034","0.0035","0.0035","0.0034","0.0035","0.0035","0.0035","0.0035","0.0042","0.0037","0.0038","0.0043","0.0039","0.0038","0.0038","0.0038","0.0045","0.0046","0.004","0.0041","0.0041","0.0046","0.0041","0.004","0.0042","0.0047","0.0041","0.0041","0.0043","0.005","0.0045","0.0044","0.0051","0.0045","0.0045","0.0045","0.0052","0.0053","0.0046","0.0046","0.0046","0.0046","0.0047","0.0057","0.005","0.005","0.005","0.0052","0.005","0.005","0.005","0.0051","0.0052","0.0062","0.0054","0.0054","0.0055","0.0056","0.0057","0.0062","0.006","0.0061","0.0072","0.0065","0.0063","0.0063","0.0066","0.0074","0.0067","0.0065","0.0068","0.0065","0.0077","0.0066","0.0069","0.0067","0.0069","0.0068","0.0068","0.0069","0.0069","0.007","0.0074","0.0083","0.0072","0.0072","0.0076","0.0077","0.0074","0.0086","0.0074","0.0078","0.0079","0.0079","0.0076","0.0076","0.0078","0.0079","0.0081","0.0094","0.0081","0.0081","0.0083","0.0084","0.0084","0.0088","0.0085","0.0084","0.0086","0.0086","0.0088","0.01","0.01","0.0093","0.0089","0.0089","0.009","0.011","0.0096","0.0096","0.011","0.0093","0.0094","0.0098","0.0096","0.0098","0.0098","0.0098","0.0098","0.0098","0.01","0.01","0.01","0.012","0.011","0.012","0.011","0.011","0.011","0.011","0.011","0.012","0.011","0.011","0.011","0.011","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.012","0.014","0.012","0.014","0.014","0.013","0.015","0.013","0.013","0.013","0.015","0.013","0.014","0.015","0.014","0.013","0.015","0.015","0.014","0.015","0.014","0.014","0.014","0.014","0.016","0.014","0.014","0.014","0.014","0.016","0.014","0.015","0.015","0.017","0.015","0.015","0.017","0.017","0.015","0.015","0.015","0.015","0.015","0.017","0.015","0.017","0.017","0.015","0.015","0.016","0.018","0.016","0.016","0.016","0.017","0.016","0.016","0.017","0.016","0.016","0.016","0.017","0.017","0.017","0.017","0.018","0.017","0.017","0.017","0.018","0.017","0.02","0.021","0.018","0.018","0.021","0.021","0.018","0.019","0.019","0.019","0.019","0.019","0.02","0.019","0.019","0.019","0.019","0.019","0.019","0.019","0.02","0.02","0.02","0.02","0.02","0.02","0.021","0.02","0.02","0.02","0.02","0.02","0.021","0.024","0.021","0.021","0.021","0.021","0.021","0.021","0.021","0.022","0.021","0.022","0.022","0.026","0.022","0.023","0.022","0.026","0.022","0.022","0.022","0.026","0.023","0.023","0.023","0.024","0.023","0.023","0.023","0.024","0.024","0.024","0.023","0.027","0.024","0.024","0.023","0.024","0.024","0.025","0.025","0.025","0.025","0.024","0.029","0.025","0.029","0.029","0.027","0.026","0.027","0.027","0.026","0.031","0.027","0.027","0.027","0.027","0.031","0.027","0.028","0.032","0.027","0.027","0.032","0.029","0.029","0.028","0.033","0.029","0.029","0.029","0.029","0.029","0.029","0.029","0.03","0.034","0.03","0.031","0.029","0.029","0.03","0.034","0.029","0.035","0.032","0.03","0.031","0.03","0.035","0.032","0.031","0.031","0.032","0.031","0.036","0.036","0.032","0.033","0.033","0.033","0.037","0.033","0.033","0.032","0.038","0.033","0.034","0.033","0.033","0.039","0.034","0.034","0.039","0.034","0.039","0.039","0.034","0.039","0.035","0.035","0.04","0.035","0.035","0.037","0.035","0.041","0.036","0.036","0.037","0.036","0.037","0.037","0.036","0.037","0.038","0.037","0.039","0.038","0.037","0.038","0.038","0.043","0.038","0.039","0.038","0.038","0.039","0.039","0.044","0.04","0.039","0.039","0.039","0.04","0.04","0.04","0.04","0.046","0.046","0.041","0.042","0.041","0.042","0.047","0.047","0.041","0.041","0.043","0.043","0.041","0.043","0.043","0.043","0.041","0.048","0.043","0.042","0.042","0.042","0.042","0.043","0.043","0.048","0.042","0.044","0.043","0.049","0.043","0.043","0.043","0.043","0.044","0.044","0.046","0.051","0.044","0.046","0.045","0.045","0.051","0.045","0.045","0.052","0.046","0.048","0.047","0.053","0.047","0.046","0.047","0.047","0.054","0.047","0.049","0.048","0.048","0.048","0.049","0.049","0.049","0.056","0.049","0.05","0.051","0.057","0.05","0.049","0.05","0.049","0.049","0.05","0.05","0.052","0.05","0.05","0.05","0.05","0.05","0.052","0.05","0.059","0.051","0.06","0.052","0.06","0.052","0.052","0.06","0.06","0.06","0.052","0.052","0.052","0.055","0.053","0.055","0.061","0.061","0.054","0.053","0.061","0.053","0.053","0.054","0.062","0.054","0.054","0.054","0.054","0.054","0.057","0.055","0.056","0.055","0.056","0.059","0.056","0.066","0.056","0.056","0.059","0.057","0.057","0.057","0.06","0.06","0.057","0.06","0.058","0.059","0.068","0.061","0.061","0.06","0.061","0.059","0.06","0.059","0.06","0.06","0.062","0.06","0.06","0.061","0.064","0.061","0.064","0.061","0.061","0.072","0.063","0.066","0.063","0.063","0.064","0.067","0.063","0.074","0.074","0.068","0.067","0.067","0.066","0.067","0.067","0.067","0.067","0.07","0.076","0.068","0.068","0.067","0.07","0.07","0.07","0.067","0.068","0.068","0.069","0.071","0.078","0.071","0.071","0.068","0.071","0.069","0.078","0.068","0.078","0.078","0.071","0.07","0.069","0.069","0.07","0.07","0.07","0.08","0.07","0.07","0.073","0.073","0.07","0.073","0.074","0.072","0.072","0.074","0.073","0.082","0.073","0.075","0.073","0.075","0.083","0.075","0.074","0.074","0.083","0.074","0.073","0.074","0.074","0.074","0.074","0.074","0.085","0.075","0.085","0.075","0.075","0.086","0.076","0.076","0.076","0.076","0.076","0.078","0.076","0.087","0.079","0.087","0.077","0.076","0.077","0.078","0.081","0.079","0.082","0.09","0.09","0.082","0.079","0.08","0.09","0.082","0.081","0.081","0.08","0.08","0.091","0.081","0.081","0.092","0.082","0.082","0.081","0.081","0.082","0.082","0.082","0.082","0.082","0.082","0.082","0.083","0.083","0.082","0.082","0.083","0.082","0.096","0.084","0.083","0.085","0.084","0.085","0.087","0.085","0.088","0.084","0.085","0.086","0.098","0.085","0.089","0.086","0.089","0.086","0.099","0.09","0.087","0.09","0.087","0.086","0.09","0.087","0.088","0.09","0.091","0.088","0.089","0.089","0.09","0.092","0.089","0.09","0.09","0.09","0.09","0.093","0.09","0.09","0.091","0.09","0.09","0.091","0.09","0.092","0.092","0.095","0.098","0.098","0.094","0.095","0.095","0.099","0.096","0.096","0.096","0.096","0.095","0.097","0.096","0.096","0.096","0.097","0.098","0.098","0.099","0.098","0.098","0.098"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>name<\/th>\n      <th>Chr<\/th>\n      <th>logFC<\/th>\n      <th>P.Value<\/th>\n      <th>adj.P.Val<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"order":[],"autoWidth":false,"orderClasses":false,"columnDefs":[{"orderable":false,"targets":0}]}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="heatmap-of-differentially-expressed-proteins-1-fdr" class="section level2">
<h2>Heatmap of differentially expressed proteins (1% FDR)</h2>
<pre class="r"><code>proList &lt;- filter(resList, !is.na(name), adj.P.Val &lt; 0.01) %&gt;% distinct(name, .keep_all = TRUE) %&gt;% pull(id)
plotMat &lt;- assays(protCLL)[[&quot;QRILC&quot;]][proList,]
rownames(plotMat) &lt;- rowData(protCLL[proList,])$hgnc_symbol

colAnno &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  select(Patient.ID, trisomy12, IGHV.status) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

rowAnno &lt;- rowData(protCLL)[proList,c(&quot;chromosome_name&quot;,&quot;hgnc_symbol&quot;),drop=FALSE] %&gt;% 
  data.frame(stringsAsFactors = FALSE) %&gt;%
  mutate(onChr12 = ifelse(chromosome_name == &quot;12&quot;,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  select(hgnc_symbol, onChr12) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;hgnc_symbol&quot;)

plotMat &lt;- jyluMisc::mscale(plotMat, censor = 5)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV.status = c(M = colList[4], U = colList[3]),
                onChr12 = c(yes = colList[1],no = &quot;white&quot;))

pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = &quot;none&quot;,
                   annotation_row = rowAnno,
                   clustering_method = &quot;ward.D2&quot;,
                   color = colorRampPalette(c(colList[2],&quot;white&quot;,colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = FALSE, show_colnames = FALSE,
                   treeheight_row = 0)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="summary-of-chromosome-distribution-10-fdr" class="section level2">
<h2>Summary of chromosome distribution (10% FDR)</h2>
<pre class="r"><code>plotTab &lt;- filter(resList, adj.P.Val &lt;=0.1) %&gt;% mutate(change = ifelse(logFC&gt;0,&quot;Up regulated&quot;,&quot;Down regulated&quot;),
                              chromosome = ifelse(Chr %in% &quot;12&quot;,&quot;chr12&quot;,&quot;other&quot;)) %&gt;%
  group_by(change, chromosome) %&gt;% summarise(n = length(id))

sigNumPlot &lt;- ggplot(plotTab, aes(x=change, y=n, fill = chromosome)) + 
  geom_bar(stat = &quot;identity&quot;, width = 0.8,
           position = position_dodge2(width = 6),
           col = &quot;black&quot;) +
  geom_text(aes(label=n), 
            position = position_dodge(width = 0.9),
            size=4, hjust=-0.1)  +
  scale_fill_manual(name = &quot;&quot;, labels = c(&quot;chr12&quot;,&quot;other&quot;), values = colList) +
  coord_flip(ylim = c(0,600), expand = FALSE) + xlab(&quot;&quot;) + ylab(&quot;Number of significant associations (10% FDR)&quot;) + theme_half 
sigNumPlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_chr_summary-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="volcano-plot" class="section level2">
<h2>Volcano plot</h2>
<pre class="r"><code>plotTab &lt;- resList 
nameList &lt;- c(&quot;PTPN11&quot;, &quot;PLCG1&quot;, &quot;GRB2&quot;, &quot;SMAD2&quot;, &quot;PYCARD&quot;, &quot;STAT2&quot;)
tri12Vocano &lt;- plotVolcano(plotTab, fdrCut =0.1, x_lab=&quot;log2FoldChange&quot;, posCol = colList[1], negCol = colList[2],
            plotTitle = &quot;trisomy12&quot;, ifLabel = TRUE, labelList = nameList)
tri12Vocano</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_volcano-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="boxplot-plot-of-selected-genes" class="section level2">
<h2>Boxplot plot of selected genes</h2>
<pre class="r"><code>protTab &lt;- sumToTidy(protCLL, rowID = &quot;uniprotID&quot;, colID = &quot;patID&quot;)
plotTab &lt;- protTab %&gt;% filter(hgnc_symbol %in% nameList) %&gt;%
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(status = ifelse(trisomy12 %in% 1,&quot;trisomy12&quot;,&quot;WT&quot;),
         name = hgnc_symbol) %&gt;%
  mutate(status = factor(status, levels = c(&quot;WT&quot;,&quot;trisomy12&quot;)))
pList &lt;- plotBox(plotTab, pValTabel = resList, y_lab = &quot;Protein expression&quot;)
protBoxplot &lt;- cowplot::plot_grid(plotlist= pList, ncol=2)
protBoxplot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_selected_protein-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-analysis" class="section level2">
<h2>Enrichment analysis</h2>
<div id="barplot-of-enriched-pathways" class="section level3">
<h3>Barplot of enriched pathways</h3>
<pre class="r"><code>gmts = list(H= &quot;../data/gmts/h.all.v6.2.symbols.gmt&quot;,
            KEGG = &quot;../data/gmts/c2.cp.kegg.v6.2.symbols.gmt&quot;)
inputTab &lt;- resList %&gt;% filter(adj.P.Val &lt; 0.1, Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)
enRes &lt;- list()
enRes[[&quot;Proteins associated with trisomy12&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)

p &lt;- plotEnrichmentBar(enRes[[1]], pCut =0.1, ifFDR= TRUE, setName = &quot;HALLMARK gene set&quot;, 
                       title = names(enRes)[1], removePrefix = &quot;HALLMARK_&quot;, insideLegend=TRUE)
tri12Enrich &lt;- cowplot::plot_grid(p)
tri12Enrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-3-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="heatmaps-of-protein-expression-in-enriched-pathways" class="section level3">
<h3>Heatmaps of protein expression in enriched pathways</h3>
<pre class="r"><code>resList.sig &lt;- filter(resList, !is.na(name), adj.P.Val &lt; 0.1) %&gt;% distinct(name, .keep_all = TRUE) 
plotMat &lt;- assays(protCLL)[[&quot;QRILC&quot;]][resList.sig$id,]
rownames(plotMat) &lt;- rowData(protCLL[rownames(plotMat),])$hgnc_symbol
colAnno &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  select(Patient.ID, trisomy12) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

rowAnno &lt;- rowData(protCLL)[resList.sig$id,c(&quot;chromosome_name&quot;,&quot;hgnc_symbol&quot;),drop=FALSE] %&gt;% 
  data.frame(stringsAsFactors = FALSE) %&gt;%
  mutate(onChr12 = ifelse(chromosome_name == &quot;12&quot;,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  select(hgnc_symbol, onChr12) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;hgnc_symbol&quot;)

plotMat &lt;- jyluMisc::mscale(plotMat, censor = 5)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV.status = c(M = colList[3], U = colList[4]),
                onChr12 = c(yes = colList[1],no = &quot;white&quot;))</code></pre>
<pre class="r"><code>plotSetHeatmap(resList.sig, gmts$H, &quot;HALLMARK_PI3K_AKT_MTOR_SIGNALING&quot;, plotMat, 
               colAnno, rowAnno = rowAnno, annoCol = annoCol, highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_PI3K_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plotSetHeatmap(resList.sig, gmts$H, &quot;HALLMARK_MTORC1_SIGNALING&quot;, plotMat, colAnno, rowAnno = rowAnno, annoCol = annoCol,
                highLight = nameList)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_MTOR_heatmap-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="compare-with-rna-sequencing-data" class="section level1">
<h1>Compare with RNA sequencing data</h1>
<div id="differentially-expressed-genes-related-to-trisomy12" class="section level2">
<h2>Differentially expressed genes related to trisomy12</h2>
<p>Prepare RNA sequencing data</p>
<pre class="r"><code>dds$diag &lt;- patMeta[match(dds$PatID, patMeta$Patient.ID),]$diagnosis
dds$trisomy12 &lt;- patMeta[match(dds$PatID, patMeta$Patient.ID),]$trisomy12
dds$IGHV &lt;- patMeta[match(dds$PatID, patMeta$Patient.ID),]$IGHV.status

ddsCLL &lt;- dds[rownames(dds) %in% rowData(protCLL)$ensembl_gene_id, dds$diag %in% &quot;CLL&quot; &amp; !is.na(dds$IGHV) &amp; !is.na(dds$trisomy12)]
ddsCLL.vst &lt;- varianceStabilizingTransformation(ddsCLL)</code></pre>
<p>Differential expression</p>
<pre class="r"><code>design(ddsCLL) &lt;- ~ trisomy12 + IGHV
deRes &lt;- DESeq(ddsCLL)
resTab &lt;- results(deRes, contrast = c(&quot;trisomy12&quot;,&quot;1&quot;,&quot;0&quot;), tidy = TRUE) %&gt;%
  mutate(name = rowData(ddsCLL[row,])$symbol) %&gt;%
  dplyr::rename(P.Value = pvalue)</code></pre>
<div id="boxplot-of-selected-genes" class="section level3">
<h3>Boxplot of selected genes</h3>
<pre class="r"><code>plotTab &lt;- assay(ddsCLL.vst[match(nameList, rowData(ddsCLL.vst)$symbol),]) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
  mutate(name = rowData(ddsCLL.vst[id,])$symbol) %&gt;%
  gather(key = &quot;patID&quot;, value = &quot;count&quot;, -id, -name) %&gt;% 
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(status = ifelse(trisomy12 %in% 1,&quot;trisomy12&quot;,&quot;WT&quot;))  %&gt;%
  mutate(status = factor(status, levels = c(&quot;WT&quot;,&quot;trisomy12&quot;)))


pList &lt;- plotBox(plotTab, pValTabel = resTab, y_lab = &quot;RNA expression&quot;)
cowplot::plot_grid(plotlist= pList, ncol=2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_selected_RNA-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="correlations-between-rna-and-protein-expression" class="section level3">
<h3>Correlations between RNA and protein expression</h3>
<pre class="r"><code>rnaMat &lt;- assay(ddsCLL.vst)
protMat &lt;- assays(protCLL)[[&quot;count&quot;]]
rownames(protMat) &lt;- rowData(protCLL)$ensembl_gene_id
overSample &lt;- intersect(colnames(rnaMat), colnames(protMat))
rnaMat &lt;- rnaMat[,overSample]
protMat &lt;- protMat[,overSample]

plotList &lt;- lapply(nameList, function(n) {
  geneId &lt;- rownames(ddsCLL.vst)[match(n, rowData(ddsCLL.vst)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab &lt;- tibble(x=rnaMat[geneId,],y=protMat[geneId,])
  coef &lt;- cor(plotTab$x, plotTab$y, use=&quot;pairwise.complete&quot;)
  annoPos &lt;- ifelse (coef &gt; 0, &quot;left&quot;,&quot;right&quot;)
  plotCorScatter(plotTab, &quot;x&quot;,&quot;y&quot;, showR2 = FALSE, annoPos = annoPos, x_lab = &quot;RNA expression&quot;,
                 y_lab =&quot;Protein expression&quot;, title = n,dotCol = colList[4], textCol = colList[1])
})
cowplot::plot_grid(plotlist = plotList, ncol =2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_protein_RNA_cor-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="buffering-of-gene-dosage-effect" class="section level1">
<h1>Buffering of gene dosage effect</h1>
<div id="visualizing-gene-dosage-effect-on-protein-and-rna-level" class="section level2">
<h2>Visualizing gene dosage effect on protein and RNA level</h2>
<pre class="r"><code>#Load combined copy number and expression data for visualization
load(&quot;../data/exprCNV_enc.RData&quot;)


protExprTab &lt;- allProtTab %&gt;% mutate(type = &quot;Protein&quot;)
rnaExprTab &lt;- filter(allRnaTab, id %in% protExprTab$id) %&gt;% mutate(type = &quot;RNA&quot;)

comExprTab &lt;- bind_rows(rnaExprTab, protExprTab) %&gt;%
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  filter(!is.na(trisomy12)) %&gt;% mutate(cnv = ifelse(trisomy12 %in% 1, &quot;trisomy12&quot;,&quot;WT&quot;))</code></pre>
<div id="proteinsrnas-on-chr12-have-higher-expressions-in-trisomy12-samples-compared-to-other-samples" class="section level3">
<h3>Proteins/RNAs on Chr12 have higher expressions in trisomy12 samples compared to other samples</h3>
<pre class="r"><code>plotTab &lt;- filter(comExprTab, ChromID %in% &quot;chr12&quot;) %&gt;%
  group_by(id,type) %&gt;% mutate(med=mean(expr)) %&gt;% mutate(expr = (expr-med)) %&gt;%
  group_by(id, symbol, cnv, type) %&gt;% summarise(meanExpr = mean(expr, na.rm=TRUE)) %&gt;%
  ungroup()

dosagePlot &lt;- ggplot(plotTab, aes(x=meanExpr, fill = cnv, col=cnv)) + 
  geom_histogram(position = &quot;identity&quot;, alpha=0.5) + facet_wrap(~type, scale = &quot;free&quot;) +
  scale_fill_manual(values = c(WT = &quot;grey80&quot;, trisomy12 = colList[1]), name = &quot;&quot;) +
  scale_color_manual(values = c(WT = &quot;grey80&quot;, trisomy12 = colList[1]), name = &quot;&quot;) +
  theme_full + xlab(&quot;Deviation to mean expression&quot;) +
  theme(strip.text = element_text(size =20))

dosagePlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/dosage_effect-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="the-variation-of-expression-is-higher-in-rna-than-protein" class="section level3">
<h3>The variation of expression is higher in RNA than protein</h3>
<div id="for-proteinsrna-on-chr12" class="section level4">
<h4>For proteins/RNA on chr12</h4>
<pre class="r"><code>plotTab &lt;- filter(comExprTab, ChromID %in% &quot;chr12&quot;) %&gt;%
  group_by(id, symbol, type) %&gt;% summarise(varExp = sd(expr, na.rm=TRUE)) %&gt;%
  ungroup()

ggplot(plotTab, aes(x=varExp,  fill = type, col=type)) + 
  geom_histogram(position = &quot;identity&quot;, alpha=0.5) + 
  scale_fill_manual(values = c(RNA = colList[4], Protein = colList[5]), name = &quot;&quot;) +
  scale_color_manual(values = c(RNA = colList[4], Protein = colList[5]), name = &quot;&quot;) +
  theme_full + xlab(&quot;Standard deviation of expression&quot;)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/expression_variation_chr12-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
<div id="the-same-trend-can-be-oberserved-for-non-chr12-proteinsrnas-but-less-striking" class="section level4">
<h4>The same trend can be oberserved for non-chr12 proteins/RNAs, but less striking</h4>
<pre class="r"><code>plotTab &lt;- filter(comExprTab, !ChromID %in% &quot;chr12&quot;) %&gt;%
  group_by(id, symbol, type) %&gt;% summarise(varExp = sd(expr, na.rm=TRUE)) %&gt;%
  ungroup()

ggplot(plotTab, aes(x=varExp, fill = type, col=type)) + 
  geom_histogram(position = &quot;identity&quot;, alpha=0.5) + 
  scale_fill_manual(values = c(RNA = colList[4], Protein = colList[5]), name = &quot;&quot;) +
  scale_color_manual(values = c(RNA = colList[4], Protein = colList[5]), name = &quot;&quot;) +
  theme_full + xlab(&quot;Standard deviation of expression&quot;)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/expression_variation_other-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="the-overall-scale-of-change-is-higher-in-rna-expression-than-protein-expression" class="section level3">
<h3>The overall scale of change is higher in RNA expression than protein expression</h3>
<pre class="r"><code>plotTab &lt;- filter(comExprTab, ChromID %in% &quot;chr12&quot;) %&gt;%
  group_by(id, symbol, type, cnv) %&gt;% summarise(meanExp = mean(expr, na.rm=TRUE)) %&gt;%
  ungroup() %&gt;% spread(key = cnv, value = meanExp) %&gt;%
  mutate(log2FC = log2(trisomy12/WT))

ggplot(plotTab, aes(x=log2FC, fill = type, col=type)) + 
  geom_histogram(position = &quot;identity&quot;, alpha=0.5, bins = 100) + 
  scale_fill_manual(values = c(RNA = colList[3], Protein = colList[4]), name = &quot;&quot;) +
  scale_color_manual(values = c(RNA = colList[3], Protein = colList[4]), name = &quot;&quot;) +
  coord_cartesian(xlim=c(-0.25,0.25))+
  geom_vline(xintercept = 0, col = colList[1], linetype = &quot;dashed&quot;) +
  theme_full + xlab(&quot;log2 Fold Change&quot;)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/log2FC_chr12-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="analyzing-buffering-effect" class="section level2">
<h2>Analyzing buffering effect</h2>
<div id="detect-buffered-and-non-buffered-proteins" class="section level3">
<h3>Detect buffered and non-buffered proteins</h3>
<p>Preprocessing protein and RNA data</p>
<pre class="r"><code>#subset samples and genes
overSampe &lt;- intersect(colnames(ddsCLL), colnames(protCLL))
overGene &lt;- intersect(rownames(ddsCLL), rowData(protCLL)$ensembl_gene_id)
ddsSub &lt;- ddsCLL[overGene, overSampe]
protSub &lt;- protCLL[match(overGene, rowData(protCLL)$ensembl_gene_id),overSampe]
rowData(ddsSub)$uniprotID &lt;- rownames(protSub)[match(rownames(ddsSub),rowData(protSub)$ensembl_gene_id)]

#vst
ddsSub.vst &lt;- varianceStabilizingTransformation(ddsSub)</code></pre>
<p>Differential expression on RNA level</p>
<pre class="r"><code>ddsSub$trisomy12 &lt;- patMeta[match(ddsSub$PatID,patMeta$Patient.ID),]$trisomy12
ddsSub$IGHV &lt;- patMeta[match(ddsSub$PatID, patMeta$Patient.ID),]$IGHV.status
design(ddsSub) &lt;- ~ trisomy12 + IGHV
deRes &lt;- DESeq(ddsSub, betaPrior = TRUE)
rnaRes &lt;- results(deRes, name = &quot;trisomy121&quot;, tidy = TRUE) %&gt;%
  dplyr::rename(geneID = row, log2FC.rna = log2FoldChange, 
                pvalue.rna = pvalue, padj.rna = padj, stat.rna= stat) %&gt;%
  select(geneID, log2FC.rna, pvalue.rna, padj.rna, stat.rna)</code></pre>
<p>Protein abundance changes related to trisomy12</p>
<pre class="r"><code>fdrCut &lt;- 0.1
protRes &lt;- resList %&gt;% filter(Gene == &quot;trisomy12&quot;) %&gt;%
    dplyr::rename(uniprotID = id, 
                  pvalue = P.Value, padj = adj.P.IHW,
                  chrom = Chr) %&gt;% 
    mutate(geneID = rowData(protCLL[uniprotID,])$ensembl_gene_id) %&gt;%
    select(name, uniprotID, geneID, chrom, logFC, pvalue, padj, t) %&gt;%
    dplyr::rename(stat =t) %&gt;%
    arrange(pvalue) %&gt;% as_tibble() </code></pre>
<p>Combine</p>
<pre class="r"><code>allRes &lt;- left_join(protRes, rnaRes, by = &quot;geneID&quot;)</code></pre>
<p><strong>Only chr12 genes that are up-regulated are considered.</strong></p>
<pre class="r"><code>bufferTab &lt;- allRes %&gt;% filter(chrom %in% 12,stat.rna &gt; 0) %&gt;%
  ungroup() %&gt;%
  mutate(stat.prot.sqrt = sqrt(stat),
         stat.prot.center = stat.prot.sqrt - mean(stat.prot.sqrt)) %&gt;%
  mutate(score = -stat.prot.center*stat.rna) %&gt;%
  mutate(ifBuffer = case_when(
    padj &lt; 0.1 &amp; padj.rna &lt; 0.1 &amp; stat &gt; 0 ~ &quot;non-Buffered&quot;,
    padj &gt; 0.1 &amp; padj.rna &lt; 0.1 ~ &quot;Buffered&quot;,
    padj &lt; 0.1 &amp; padj.rna &gt; 0.1 &amp; stat &gt; 0 ~ &quot;Enhanced&quot;,
    TRUE ~ &quot;Undetermined&quot;
  )) %&gt;%
  arrange(desc(score))</code></pre>
</div>
<div id="summary-plot" class="section level3">
<h3>Summary plot</h3>
<pre class="r"><code>sumTab &lt;- bufferTab %&gt;% group_by(ifBuffer) %&gt;%
  summarise(n = length(name))

bufferPlot &lt;- ggplot(sumTab, aes(x=ifBuffer, y = n)) + 
  geom_bar(aes(fill = ifBuffer), stat=&quot;identity&quot;, width = 0.7) + 
  geom_text(aes(label = paste0(&quot;n=&quot;, n)),vjust=-1,col=colList[1]) +
  scale_fill_manual(values =c(Buffered = colList[1],
                              Enhanced = colList[4],
                              `non-Buffered` = colList[2],
                              Undetermined = &quot;grey50&quot;)) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5),
                     legend.position = &quot;none&quot;) +
  ylab(&quot;Number of proteins&quot;) + ylim(0,120) +xlab(&quot;&quot;)
bufferPlot</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/sum_buffer_number-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-example-cases-of-buffered-and-non-buffered-proteins" class="section level3">
<h3>Plot example cases of buffered and non-buffered proteins</h3>
<pre class="r"><code>protList &lt;- c(&quot;PTPN11&quot;,&quot;DDX51&quot;,&quot;BCAT1&quot;, &quot;METTL7A&quot;)
geneList &lt;- bufferTab[match(protList, bufferTab$name),]$geneID
pList &lt;- lapply(geneList, function(i) {
  tabProt &lt;- allProtTab %&gt;% filter(id == i) %&gt;%
    select(id, patID, symbol,expr) %&gt;% dplyr::rename(protExpr = expr)
  tabRna &lt;- allRnaTab %&gt;% filter(id == i) %&gt;%
    select(id, patID, expr) %&gt;% dplyr::rename(rnaExpr = expr)
  plotTab &lt;- left_join(tabProt, tabRna, by = c(&quot;id&quot;,&quot;patID&quot;)) %&gt;% 
    filter(!is.na(protExpr), !is.na(rnaExpr)) %&gt;%
    mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
    mutate(trisomy12 = ifelse(trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;))
  p &lt;- ggplot(plotTab, aes(x=rnaExpr, y = protExpr)) +
    geom_point(aes(col=trisomy12)) + 
    geom_smooth(formula =  y~x, method=&quot;lm&quot;,se=FALSE, color = &quot;grey50&quot;, linetype =&quot;dashed&quot; ) + 
    ggtitle(unique(plotTab$symbol)) +
    ylab(&quot;Protein expression&quot;) + xlab(&quot;RNA expression&quot;) +
    scale_color_manual(values =c(yes = colList[1],no=colList[2])) +
    theme_full + theme(legend.position = &quot;bottom&quot;)
  ggExtra::ggMarginal(p, type = &quot;histogram&quot;, groupFill = TRUE)
  })
cowplot::plot_grid(plotlist = pList, ncol=2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/buffer_example-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="enrichment-of-buffer-and-non-buffered-proteins" class="section level3">
<h3>Enrichment of buffer and non-buffered proteins</h3>
<div id="non-buffered-prpteins" class="section level4">
<h4>Non-buffered prpteins</h4>
<pre class="r"><code>protList &lt;- filter(bufferTab, ifBuffer == &quot;non-Buffered&quot;)$name
refList &lt;- unique(protExprTab$symbol)
enRes &lt;- runFisher(protList, refList, gmts$H, pCut =0.05, ifFDR = FALSE,removePrefix = &quot;HALLMARK_&quot;,
                   plotTitle = &quot;Non-buffered proteins&quot;, insideLegend = TRUE,
                   setName = &quot;HALLMARK gene set&quot;)
bufferEnrich &lt;- enRes$enrichPlot + theme(plot.margin = margin(1,3,1,1, unit = &quot;cm&quot;))
bufferEnrich</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_nonBuffer_enrich-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="buffered-proteins" class="section level4">
<h4>Buffered proteins</h4>
<pre class="r"><code>protList &lt;- filter(bufferTab, ifBuffer == &quot;Buffered&quot;)$name
enRes &lt;- runFisher(protList, refList, gmts$H, pCut =0.05, ifFDR = FALSE)</code></pre>
<pre><code>[1] &quot;No sets passed the criteria&quot;</code></pre>
<p>No enrichment</p>
</div>
</div>
</div>
<div id="plot-expression-on-genomic-coordiate" class="section level2">
<h2>Plot expression on genomic coordiate</h2>
<pre class="r"><code>load(&quot;../data/exprCNV_enc.RData&quot;)
patBack &lt;- dplyr::filter(patMeta, Patient.ID %in% unique(allProtTab$patID)) %&gt;%
  dplyr::select(Patient.ID, trisomy12) %&gt;%
  dplyr::rename(patID = Patient.ID) %&gt;%
  mutate_all(as.character) %&gt;%
  mutate_at(vars(-patID),str_replace, &quot;1&quot;,&quot;yes&quot;) %&gt;%
  mutate_at(vars(-patID),str_replace, &quot;0&quot;,&quot;no&quot;)</code></pre>
<pre class="r"><code>plotExprVar &lt;- function(gene, chr, patBack, allBand, allLine, allProtTab, allRnaTab, protLine = NULL,
                        region = c(-Inf,Inf),ifTrend = FALSE, normalize = TRUE, maxVal =2, minVal=-2) {
  
  #table for cyto band
  bandTab &lt;- filter(allBand, ChromID == chr, chromStart &gt;= region[1], chromEnd &lt;= region[2]) %&gt;%
    mutate(chromMid = chromMid)
  
  #table for expression
  plotProtTab &lt;- filter(allProtTab, ChromID == chr, start_position &gt;= region[1], end_position &lt;= region[2]) %&gt;%
    mutate_if(is.factor,as.character)
  plotRnaTab &lt;- filter(allRnaTab, ChromID == chr, start_position &gt;= region[1], end_position &lt;= region[2]) %&gt;%
    mutate_if(is.factor,as.character)
  
  
  #summarise group mean
  plotProtTab &lt;- plotProtTab %&gt;% 
    mutate(group = patBack[match(patID, patBack$patID),][[gene]]) %&gt;%
    filter(!is.na(group)) %&gt;%
    group_by(id, group) %&gt;% mutate(meanExpr = mean(expr, na.rm=TRUE)) %&gt;%
    distinct(group, id,.keep_all = TRUE) %&gt;% ungroup()

  plotRnaTab &lt;- plotRnaTab %&gt;%
    mutate(group = patBack[match(patID, patBack$patID),][[gene]]) %&gt;%
    filter(!is.na(group)) %&gt;%
    group_by(id, group) %&gt;% mutate(meanExpr = mean(expr, na.rm=TRUE)) %&gt;%
    distinct(group, id,.keep_all = TRUE) %&gt;% ungroup()
  
  if (!is.null(protLine)) {
    bufferLineTab &lt;- plotProtTab %&gt;%
      select(symbol, mid_position, meanExpr, group) %&gt;%
      filter(symbol %in% protLine) %&gt;%
      pivot_wider(names_from = group, values_from = meanExpr) %&gt;%
      mutate(lowVal = map2_dbl(yes, no, min),
             highVal = map2_dbl(yes, no, max))
  } else bufferLineTab &lt;- NULL
  

  xMax &lt;- max(bandTab$chromEnd, na.rm = T)
  
  #main plot for Protein
  gPro &lt;- ggplot() + 
    geom_rect(data=bandTab, mapping=aes(xmin=chromStart, xmax=chromEnd, ymin=minVal+0.5, ymax=maxVal-0.5, 
                                        fill=Colour, label = band), alpha=0.1) +
    geom_text(data=bandTab, mapping=aes(label=band, x=chromMid), y=maxVal-0.5, hjust =1, angle = 90, size=2.5) 
    
  if (!is.null(protLine)) {
    gPro &lt;- gPro + geom_segment(data = bufferLineTab, aes(x=mid_position, xend = mid_position, 
                                        y=lowVal, yend = highVal), linetype = &quot;dashed&quot;) 
    
  }
  
  gPro &lt;- gPro + geom_rect(data = plotProtTab, 
            mapping=aes(xmin=start_position,
                        xmax=end_position, ymin=meanExpr, ymax=meanExpr+0.1,
                        fill = group, label = symbol)) +
    scale_x_continuous(expand=c(0,0),limits = c(0,xMax)) +
    xlab(&quot;Genomic position [Mb]&quot;) + 
    ylab(&quot;Expression (normalized by length)&quot;) + 
    scale_fill_manual(values = c(even = &quot;white&quot;,odd = &quot;grey50&quot;,
                                yes = &quot;darkred&quot;, no = &quot;darkgreen&quot;)) +
    scale_color_manual(values = c(yes = &quot;darkred&quot;,no = &quot;darkgreen&quot;)) +
    ggtitle(paste0(&quot;Protein expression&quot;,&quot;(&quot;,chr,&quot;)&quot;)) +
    theme(plot.title = element_text(face = &quot;bold&quot;, size = 10),
        legend.position = &quot;none&quot;,
        panel.background = element_blank(),
        panel.grid.major = element_line(colour=&quot;grey90&quot;, size=0.1))
  
    if (ifTrend) {
      gPro &lt;- gPro + geom_smooth(data =filter(plotProtTab, expr &gt;0), 
                mapping = aes(y=meanExpr, x= mid_position,
                              color = group), 
                formula = y ~ x, method = &quot;loess&quot;, se=FALSE, span=0.5,
                size =0.2, alpha=0.5)
    }
    
  #main plot for RNA
  gRna &lt;- ggplot() + 
    geom_rect(data=bandTab, mapping=aes(xmin=chromStart, xmax=chromEnd, ymin=minVal, ymax=maxVal, 
                                        fill=Colour, label = band), alpha=0.1) +
    geom_text(data=bandTab, mapping=aes(label=band, x=chromMid), y=maxVal, hjust =1, angle = 90, size=2.5) +
    geom_rect(data = plotRnaTab, 
            mapping=aes(xmin=start_position,
                        xmax=end_position, ymin=meanExpr, ymax=meanExpr+0.1,
                        fill = group, label = symbol)) +
    scale_x_continuous(expand=c(0,0),limits = c(0,xMax)) +
    xlab(&quot;Genomic position [Mb]&quot;) + 
    ylab(&quot;Expression Z-score&quot;) + 
    scale_fill_manual(values = c(even = &quot;white&quot;,odd = &quot;grey50&quot;,
                                yes = &quot;darkred&quot;, no = &quot;darkgreen&quot;)) +
    scale_color_manual(values = c(yes = &quot;darkred&quot;, no = &quot;darkgreen&quot;)) +
    ggtitle(paste0(&quot;RNA expression&quot;,&quot;(&quot;,chr,&quot;)&quot;)) +
    theme(plot.title = element_text(face = &quot;bold&quot;, size = 10),
        legend.position = &quot;none&quot;,
        panel.background = element_blank(),
        panel.grid.major = element_line(colour=&quot;grey90&quot;, size=0.1))
  
    if (ifTrend) {
      gRna &lt;- gRna + geom_smooth(data =filter(plotRnaTab), 
                mapping = aes(y=meanExpr, x= mid_position,
                              color = group), 
                formula = y ~ x, method = &quot;loess&quot;, se=FALSE, span=0.2,
                size =0.2, alpha=0.5)
    }
    #for legend
    ## if the patient has CNV data
    lgTab &lt;- tibble(x= seq(6),y=seq(6),
                    Expression = c(rep(&quot;yes&quot;,3), rep(&quot;no&quot;,3)))
  
    lg &lt;- ggplot(lgTab, aes(x=x,y=y)) +
        geom_point(aes(fill = Expression), shape =22,size=3) +
        scale_fill_manual(values = c(yes = &quot;darkred&quot;, no = &quot;darkgreen&quot;), name = gene) +
        theme(legend.position = &quot;bottom&quot;)
    
    lg &lt;- get_legend(lg)
    
    return(list(plotPro = gPro, plotRNA = gRna, legend = lg))
}</code></pre>
<p>Normalize protein and RNA expression</p>
<pre class="r"><code>normalized &lt;- TRUE
#if perform normalization
if (normalized) {
  #for protein
  exprMat &lt;- select(allProtTab,patID, id,expr) %&gt;% 
    spread(key = patID, value =expr) %&gt;% data.frame() %&gt;% 
    column_to_rownames(&quot;id&quot;) %&gt;% as.matrix()
  qm &lt;- jyluMisc::mscale(exprMat, useMad = F)
  normTab &lt;- data.frame(qm) %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
    gather(key = &quot;patID&quot;, value = &quot;expr&quot;, -id)
  allProtTab &lt;- select(allProtTab, -expr) %&gt;% left_join(normTab, by = c(&quot;patID&quot;,&quot;id&quot;))


  #for RNA
  exprMat &lt;- select(allRnaTab,patID, id,expr) %&gt;% 
    spread(key = patID, value =expr) %&gt;% data.frame() %&gt;% 
    column_to_rownames(&quot;id&quot;) %&gt;% as.matrix()
  qm &lt;- jyluMisc::mscale(exprMat, useMad = F)
  normTab &lt;- data.frame(qm) %&gt;% rownames_to_column(&quot;id&quot;) %&gt;%
    gather(key = &quot;patID&quot;, value = &quot;expr&quot;, -id)
  allRnaTab &lt;- select(allRnaTab, -expr) %&gt;% left_join(normTab, by = c(&quot;patID&quot;,&quot;id&quot;))
}</code></pre>
<pre class="r"><code>g &lt;- plotExprVar(&quot;trisomy12&quot;,&quot;chr12&quot;,patBack,allBand, allLine, 
                 allProtTab, allRnaTab, ifTrend = TRUE)
plot_grid(g$plotRNA, g$plotPro, g$legend, ncol = 1, rel_heights = c(1,1,0.2))</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_geneCoord-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="protein-complex-analysis" class="section level1">
<h1>Protein complex analysis</h1>
<div id="preprocessing" class="section level2">
<h2>Preprocessing</h2>
<p>Processing protein complex data</p>
<pre class="r"><code>int_pairs &lt;- read_delim(&quot;../data/proteins_in_complexes&quot;, delim = &quot;\t&quot;) %&gt;%
  mutate(Reactome = grepl(&quot;Reactome&quot;,Evidence_supporting_the_interaction),
         Corum = grepl(&quot;Corum&quot;,Evidence_supporting_the_interaction)) %&gt;%
  filter(ProtA %in% rownames(protSub) &amp; ProtB %in% rownames(protSub)) %&gt;%
  mutate(pair=map2_chr(ProtA, ProtB, ~paste0(sort(c(.x,.y)), collapse = &quot;-&quot;))) %&gt;%
  mutate(database = case_when(
    Reactome &amp; Corum ~ &quot;both&quot;,
    Reactome &amp; !Corum ~ &quot;Reactome&quot;,
    !Reactome &amp; Corum ~ &quot;Corum&quot;,
    TRUE ~ &quot;other&quot;
  )) %&gt;% mutate(inComplex = &quot;yes&quot;)</code></pre>
</div>
<div id="construct-protein-protein-interaction-network-by-connecting-chr12-proteins-and-non-chr12-protein" class="section level2">
<h2>Construct protein-protein interaction network by connecting chr12 proteins and non-chr12 protein</h2>
<pre class="r"><code>comTab &lt;- int_pairs %&gt;% select(ProtA, ProtB, database) %&gt;%
  mutate(chrA = rowData(protCLL[ProtA,])$chromosome_name,
         chrB = rowData(protCLL[ProtB,])$chromosome_name) %&gt;%
  filter(!is.na(chrA), !is.na(chrB)) %&gt;%
  filter((chrA == &quot;12&quot; &amp; chrB != &quot;12&quot;) | (chrA !=&quot;12&quot; &amp; chrB == &quot;12&quot;)) %&gt;%
  mutate(source = ifelse(chrA == 12, ProtA, ProtB),
         target = ifelse(chrA == 12, ProtB, ProtA)) %&gt;%
  select(source, target, database)</code></pre>
<pre class="r"><code>fdrCut &lt;- 0.1
resTab &lt;- select(allRes, name, uniprotID, chrom, padj, padj.rna, logFC,log2FC.rna) %&gt;%
  mutate(sigProt = padj &lt;= fdrCut,
         sigRna = padj.rna &lt;=fdrCut,
         upProt = sigProt &amp; logFC &gt; 0,
         upRna = sigRna &amp; log2FC.rna &gt; 0)
comTab &lt;- comTab %&gt;% 
  left_join(resTab, by = c(source = &quot;uniprotID&quot;)) %&gt;%
  left_join(resTab, by = c(target = &quot;uniprotID&quot;)) %&gt;%
  rename_all(funs(str_replace(., &quot;x&quot;, &quot;source&quot;))) %&gt;%
  rename_all(funs(str_replace(., &quot;y&quot;, &quot;target&quot;))) </code></pre>
<pre class="r"><code>comTab.filter &lt;- filter(comTab, sigProt.source, sigProt.target, !sigRna.target, upProt.source, upProt.target)</code></pre>
<pre class="r"><code>#get node list
allNodes &lt;- union(comTab.filter$name.source, comTab.filter$name.target) 

nodeList &lt;- data.frame(id = seq(length(allNodes))-1, name = allNodes, stringsAsFactors = FALSE) %&gt;%
  mutate(onChr12 = ifelse(rowData(protCLL[match(name, rowData(protCLL)$hgnc_symbol),])$chromosome_name %in% &quot;12&quot;, 
                          &quot;chr12&quot;,&quot;otherChr&quot;))

#get edge list
edgeList &lt;- select(comTab.filter, name.source, name.target, database) %&gt;%
  dplyr::rename(Source = name.source, Target = name.target) %&gt;% 
  mutate(Source = nodeList[match(Source,nodeList$name),]$id,
         Target = nodeList[match(Target, nodeList$name),]$id) %&gt;%
  data.frame(stringsAsFactors = FALSE)

net &lt;- graph_from_data_frame(vertices = nodeList, d=edgeList, directed = FALSE)</code></pre>
<pre class="r"><code>tidyNet &lt;- as_tbl_graph(net)
complexNet &lt;- ggraph(tidyNet, layout = &quot;igraph&quot;, algorithm = &quot;nicely&quot;) + 
  geom_edge_link(color = colList[3], width=1) + 
  geom_node_point(aes(color =onChr12, shape = onChr12), size=6) + 
  geom_node_text(aes(label = name), repel = TRUE, size=6) +
  scale_color_manual(values = c(chr12 = colList[1],otherChr = colList[2])) +
  scale_edge_color_brewer(palette = &quot;Set2&quot;) +
  theme_graph(base_family = &quot;sans&quot;) + theme(legend.position = &quot;none&quot;) 
complexNet</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/unnamed-chunk-22-1.png" width="1440" style="display: block; margin: auto;" /></p>
<div id="heatmap-of-bcr-complex-proteins" class="section level3">
<h3>Heatmap of BCR complex proteins</h3>
<pre class="r"><code>seedProt &lt;- c(&quot;CD79B&quot;,&quot;CD22&quot;,&quot;GRB2&quot;,&quot;PLCG1&quot;)
seedID &lt;- rownames(protCLL[match(seedProt,rowData(protCLL)$hgnc_symbol),])
complexID &lt;- filter(int_pairs, ProtA %in% seedID | ProtB %in% seedID)
resList.BCR &lt;- filter(resList, id %in% c(complexID$ProtA, complexID$ProtB), Gene == &quot;trisomy12&quot;,
                      adj.P.Val &lt;= 0.1)
protList &lt;- resList.BCR$id
plotMat &lt;- assays(protCLL)[[&quot;QRILC&quot;]][protList,]
rownames(plotMat) &lt;- rowData(protCLL[protList,])$hgnc_symbol

colAnno &lt;- filter(patMeta, Patient.ID %in% colnames(protCLL)) %&gt;%
  select(Patient.ID, trisomy12, IGHV.status) %&gt;%
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;) %&gt;%
  arrange(trisomy12)
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

plotMat &lt;- jyluMisc::mscale(plotMat, censor = 5)
plotMat &lt;- plotMat[,rownames(colAnno)]


rowAnno &lt;- rowData(protCLL)[,c(&quot;chromosome_name&quot;,&quot;hgnc_symbol&quot;),drop=FALSE] %&gt;% 
  data.frame(stringsAsFactors = FALSE) %&gt;%
  mutate(onChr12 = ifelse(chromosome_name == &quot;12&quot;,&quot;yes&quot;,&quot;no&quot;)) %&gt;%
  select(hgnc_symbol, onChr12) %&gt;% distinct(hgnc_symbol, .keep_all = TRUE) %&gt;%
  data.frame() %&gt;% remove_rownames() %&gt;%
  column_to_rownames(&quot;hgnc_symbol&quot;)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV.status = c(M = colList[4], U = colList[3]),
                onChr12 = c(yes = colList[1],no = &quot;white&quot;))

#pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = &quot;none&quot;,
#                   annotation_row = rowAnno, cluster_cols = FALSE,
#                   clustering_method = &quot;ward.D2&quot;,
#                   color = colorRampPalette(c(colList[2],&quot;white&quot;,colList[1]))(100),
#                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
#                   show_rownames = TRUE, show_colnames = FALSE,
#                   treeheight_row = 0)

haCol &lt;- ComplexHeatmap::HeatmapAnnotation(df = colAnno, col=annoCol, which = &quot;column&quot;,annotation_name_gp = gpar(fontface = &quot;bold&quot;))
haRow &lt;- ComplexHeatmap::HeatmapAnnotation(df = rowAnno[rownames(plotMat),,drop=FALSE], col=annoCol, which = &quot;row&quot;, annotation_name_gp = gpar(fontface = &quot;bold&quot;))

labelCol &lt;- rep(&quot;black&quot;,nrow(plotMat))
highlight &lt;- c(seedProt, c(&quot;PTPN6&quot;,&quot;PTPN11&quot;,&quot;PTK2B&quot;))
labelCol[rownames(plotMat) %in% highlight] &lt;-&quot;red&quot;


ComplexHeatmap::Heatmap(plotMat, col = colorRampPalette(c(colList[2],&quot;white&quot;,colList[1]))(100),name = &quot;z-score&quot;,
                        top_annotation = haCol, left_annotation = haRow, show_column_names = FALSE,
                        cluster_columns  = FALSE, clustering_method_rows = &quot;ward.D2&quot;,
                        row_names_gp = gpar(col = labelCol), show_row_dend = FALSE,
                        column_title_gp = gpar(cex= 1.5, fontface = &quot;bold&quot;))</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/BCR_complex_heatmap-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="validation-on-peptide-level" class="section level1">
<h1>Validation on peptide level</h1>
<p>Load peptide quantification data</p>
<pre class="r"><code>load(&quot;../data/pepCLL_lumos_enc.RData&quot;)</code></pre>
<pre class="r"><code>stratifier &lt;- &quot;trisomy12&quot;
plotList &lt;- lapply(nameList, function(n) {
 mutStatus &lt;- as.character(patMeta[match(colnames(pepCLL), patMeta$Patient.ID),][[stratifier]])
 names(mutStatus) &lt;- colnames(pepCLL)
 plotPep(pepCLL, n, type = &quot;count&quot;, stratifier = stratifier, mutStatus = mutStatus)
})
cowplot::plot_grid(plotlist = plotList, ncol=1)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_peptide-1.png" width="1440" style="display: block; margin: auto;" /></p>
</div>
<div id="validation-using-timstof-data" class="section level1">
<h1>Validation using timsTOF data</h1>
<p>Load timsTOF data</p>
<pre class="r"><code>load(&quot;../data/proteomic_timsTOF_enc.RData&quot;)
load(&quot;../output/deResList_timsTOF.RData&quot;)</code></pre>
<pre class="r"><code>resList &lt;- filter(resList, Gene == &quot;trisomy12&quot;) %&gt;%
  mutate(adj.P.Val = adj.P.IHW) %&gt;% #use IHW corrected P-value
  mutate(Chr = rowData(protCLL[id,])$chromosome_name)</code></pre>
<pre class="r"><code>protTab &lt;- sumToTidy(protCLL, rowID = &quot;uniprotID&quot;, colID = &quot;patID&quot;)
plotTab &lt;- protTab %&gt;% filter(hgnc_symbol %in% nameList) %&gt;%
  mutate(trisomy12 = patMeta[match(patID, patMeta$Patient.ID),]$trisomy12) %&gt;%
  mutate(status = ifelse(trisomy12 %in% 1,&quot;trisomy12&quot;,&quot;WT&quot;),
         name = hgnc_symbol) %&gt;%
    mutate(status = factor(status, levels = c(&quot;WT&quot;,&quot;trisomy12&quot;)))
pList &lt;- plotBox(plotTab, pValTabel = resList)
cowplot::plot_grid(plotlist= pList, ncol=2)</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/trisomy12_selected_protein_timsTOF-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="assemble-figures" class="section level1">
<h1>Assemble figures</h1>
<div id="main-text-figure-2" class="section level2">
<h2>Main text figure 2</h2>
<pre class="r"><code>leftCol &lt;- plot_grid(tri12Vocano, tri12Enrich,protBoxplot, ncol=1,
                     rel_heights = c(0.35,0.25,0.4),
                     labels = c(&quot;A&quot;,&quot;B&quot;,&quot;C&quot;), label_size = 20,
                     vjust = c(1.5,1.5,0))
rightCol &lt;- plot_grid(dosagePlot, 
                      plot_grid(bufferPlot, bufferEnrich,ncol=2, 
                                rel_widths = c(0.3,0.7), labels = c(&quot;E&quot;,&quot;F&quot;), label_size = 20,
                                vjust = c(0,0)),
                      complexNet, ncol=1,
                      rel_heights = c(0.2,0.2,0.6), labels = c(&quot;D&quot;,&quot;&quot;,&quot;G&quot;), label_size = 20)

plot_grid(leftCol, rightCol, rel_widths = c(0.4,0.6))</code></pre>
<p><img src="figure/manuscript_S3_trisomy12.Rmd/Figure2-1.png" width="1920" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.0 (2019-04-26)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS  10.15.6

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
 [1] grid      parallel  stats4    stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] piano_2.2.0                 latex2exp_0.4.0            
 [3] forcats_0.5.0               stringr_1.4.0              
 [5] dplyr_1.0.0                 purrr_0.3.4                
 [7] readr_1.3.1                 tidyr_1.1.0                
 [9] tibble_3.0.3                tidyverse_1.3.0            
[11] ggbeeswarm_0.6.0            ComplexHeatmap_2.2.0       
[13] pheatmap_1.0.12             ggraph_2.0.3               
[15] ggplot2_3.3.2               igraph_1.2.5               
[17] cowplot_1.0.0               tidygraph_1.2.0            
[19] DESeq2_1.26.0               SummarizedExperiment_1.16.1
[21] DelayedArray_0.12.3         BiocParallel_1.20.1        
[23] matrixStats_0.56.0          Biobase_2.46.0             
[25] GenomicRanges_1.38.0        GenomeInfoDb_1.22.1        
[27] IRanges_2.20.2              S4Vectors_0.24.4           
[29] BiocGenerics_0.32.0         limma_3.42.2               

loaded via a namespace (and not attached):
  [1] shinydashboard_0.7.1   tidyselect_1.1.0       RSQLite_2.2.0         
  [4] AnnotationDbi_1.48.0   htmlwidgets_1.5.1      maxstat_0.7-25        
  [7] munsell_0.5.0          codetools_0.2-16       DT_0.14               
 [10] miniUI_0.1.1.1         withr_2.2.0            colorspace_1.4-1      
 [13] knitr_1.29             rstudioapi_0.11        ggsignif_0.6.0        
 [16] labeling_0.3           git2r_0.27.1           slam_0.1-47           
 [19] GenomeInfoDbData_1.2.2 KMsurv_0.1-5           polyclip_1.10-0       
 [22] bit64_0.9-7            farver_2.0.3           rprojroot_1.3-2       
 [25] vctrs_0.3.1            generics_0.0.2         TH.data_1.0-10        
 [28] xfun_0.15              sets_1.0-18            R6_2.4.1              
 [31] clue_0.3-57            graphlayouts_0.7.0     locfit_1.5-9.4        
 [34] fgsea_1.12.0           bitops_1.0-6           assertthat_0.2.1      
 [37] promises_1.1.1         scales_1.1.1           multcomp_1.4-13       
 [40] nnet_7.3-14            ggExtra_0.9            beeswarm_0.2.3        
 [43] gtable_0.3.0           sandwich_2.5-1         workflowr_1.6.2       
 [46] rlang_0.4.7            genefilter_1.68.0      GlobalOptions_0.1.2   
 [49] splines_3.6.0          rstatix_0.6.0          acepack_1.4.1         
 [52] broom_0.7.0            checkmate_2.0.0        yaml_2.2.1            
 [55] abind_1.4-5            modelr_0.1.8           crosstalk_1.1.0.1     
 [58] backports_1.1.8        httpuv_1.5.4           Hmisc_4.4-0           
 [61] relations_0.6-9        tools_3.6.0            ellipsis_0.3.1        
 [64] gplots_3.0.4           RColorBrewer_1.1-2     Rcpp_1.0.5            
 [67] visNetwork_2.0.9       base64enc_0.1-3        zlibbioc_1.32.0       
 [70] RCurl_1.98-1.2         ggpubr_0.4.0           rpart_4.1-15          
 [73] GetoptLong_1.0.2       viridis_0.5.1          zoo_1.8-8             
 [76] haven_2.3.1            ggrepel_0.8.2          cluster_2.1.0         
 [79] exactRankTests_0.8-31  fs_1.4.2               magrittr_1.5          
 [82] data.table_1.12.8      openxlsx_4.1.5         circlize_0.4.10       
 [85] survminer_0.4.7        reprex_0.3.0           mvtnorm_1.1-1         
 [88] shinyjs_1.1            hms_0.5.3              mime_0.9              
 [91] evaluate_0.14          xtable_1.8-4           XML_3.98-1.20         
 [94] rio_0.5.16             jpeg_0.1-8.1           readxl_1.3.1          
 [97] gridExtra_2.3          shape_1.4.4            compiler_3.6.0        
[100] KernSmooth_2.23-17     crayon_1.3.4           htmltools_0.5.0       
[103] mgcv_1.8-31            later_1.1.0.1          Formula_1.2-3         
[106] geneplotter_1.64.0     lubridate_1.7.9        DBI_1.1.0             
[109] tweenr_1.0.1           dbplyr_1.4.4           MASS_7.3-51.6         
[112] jyluMisc_0.1.5         Matrix_1.2-18          car_3.0-8             
[115] cli_2.0.2              marray_1.64.0          gdata_2.18.0          
[118] km.ci_0.5-2            pkgconfig_2.0.3        foreign_0.8-71        
[121] xml2_1.3.2             annotate_1.64.0        vipor_0.4.5           
[124] XVector_0.26.0         drc_3.0-1              rvest_0.3.5           
[127] digest_0.6.25          fastmatch_1.1-0        rmarkdown_2.3         
[130] cellranger_1.1.0       survMisc_0.5.5         htmlTable_2.0.1       
[133] curl_4.3               shiny_1.5.0            gtools_3.8.2          
[136] rjson_0.2.20           nlme_3.1-148           lifecycle_0.2.0       
[139] jsonlite_1.7.0         carData_3.0-4          viridisLite_0.3.0     
[142] fansi_0.4.1            pillar_1.4.6           lattice_0.20-41       
[145] fastmap_1.0.1          httr_1.4.1             plotrix_3.7-8         
[148] survival_3.2-3         glue_1.4.1             zip_2.0.4             
[151] png_0.1-7              bit_1.1-15.2           ggforce_0.3.2         
[154] stringi_1.4.6          blob_1.2.1             latticeExtra_0.6-29   
[157] caTools_1.18.0         memoise_1.1.0         </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
