<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2020-10-09" />

<title>Section 2: Identifying proteins associated disease drivers in CLL</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CLL Proteome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.embl.de/jlu/Proteomics">
    <span class="fas fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Section 2: Identifying proteins associated disease drivers in CLL</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2020-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-05-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>CLLproteomics_publish_revision/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
unnamed-chunk-24
</li>
<li>
unnamed-chunk-26
</li>
<li>
unnamed-chunk-27
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory <code>manuscript_S2_genomicAssociation_cache</code> and re-run the analysis. To have workflowr automatically delete the cache directory prior to building the file, set <code>delete_cache = TRUE</code> when running <code>wflow_build()</code> or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/manuscript_S1_Overview_cache/
    Ignored:    analysis/manuscript_S2_genomicAssociation_cache/
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/.trisomy12_norm.pdf
    Untracked:  analysis/cohortComposition_all.pdf
    Untracked:  analysis/manuscript_S1_Overview.Rmd
    Untracked:  analysis/manuscript_S2_genomicAssociation.Rmd
    Untracked:  analysis/manuscript_S3_trisomy12.Rmd
    Untracked:  analysis/manuscript_S4_trisomy19.Rmd
    Untracked:  analysis/manuscript_S5_IGHV.Rmd
    Untracked:  analysis/manuscript_S6_del11q.Rmd
    Untracked:  analysis/manuscript_S7_SF3B1.Rmd
    Untracked:  analysis/manuscript_S8_drugResponse_Outcomes.Rmd
    Untracked:  analysis/manuscript_S9_STAT2.Rmd
    Untracked:  analysis/plot_PC1_PC2.pdf
    Untracked:  analysis/timsTOF_validate.Rmd
    Untracked:  code/utils.R
    Untracked:  data/Annotation file March 2021.xlsx
    Untracked:  data/CAS9results.xlsx
    Untracked:  data/CNV_onChrom.RData
    Untracked:  data/ComplexParticipantsPubMedIdentifiers_human.txt
    Untracked:  data/Fig1A.png
    Untracked:  data/IGLV321_SupplementalTables_R2.xlsx
    Untracked:  data/MOFAout.RData
    Untracked:  data/MOFAout_atLeast3.RData
    Untracked:  data/STATexprPCR.xlsx
    Untracked:  data/Western_blot_results_20210309_short.csv
    Untracked:  data/Western_blot_results_separate_blots.xlsx
    Untracked:  data/allComplexes.txt
    Untracked:  data/ddsrna_enc.RData
    Untracked:  data/exprCNV_enc.RData
    Untracked:  data/geneAnno.RData
    Untracked:  data/gmts/
    Untracked:  data/ic50.RData
    Untracked:  data/mofaIn.RData
    Untracked:  data/mofaIn_atLeast3.RData
    Untracked:  data/patMeta_enc.RData
    Untracked:  data/pepCLL_lumos_enc.RData
    Untracked:  data/protMOFA.RData
    Untracked:  data/proteins_in_complexes
    Untracked:  data/proteomic_LUMOS_2pep_enc.RData
    Untracked:  data/proteomic_explore_enc.RData
    Untracked:  data/proteomic_independent_all_enc.RData
    Untracked:  data/proteomic_independent_enc.RData
    Untracked:  data/proteomic_timsTOF_enc.RData
    Untracked:  data/screenData_enc.RData
    Untracked:  data/setToPathway.txt
    Untracked:  data/survival_enc.RData
    Untracked:  output/MSH6_splicing.svg
    Untracked:  output/SUGP1_splicing.svg
    Untracked:  output/deResList.RData
    Untracked:  output/deResListBatch2.RData
    Untracked:  output/deResListRNA.RData
    Untracked:  output/deResListRNA_allGene.RData
    Untracked:  output/deResList_WBC.RData
    Untracked:  output/deResList_batch1.RData
    Untracked:  output/deResList_batch3.RData
    Untracked:  output/deResList_independent.RData
    Untracked:  output/deResList_timsTOF.RData
    Untracked:  output/dxdCLL.RData
    Untracked:  output/dxdCLL2.RData
    Untracked:  output/exprCNV.RData
    Untracked:  output/geneAnno.RData
    Untracked:  output/int_pairs.csv
    Untracked:  output/lassoResults_CPS.RData
    Untracked:  output/resOutcome_batch1.RData
    Untracked:  output/resOutcome_batch13.RData
    Untracked:  output/resOutcome_batch2.RData
    Untracked:  output/resOutcome_batch3.RData

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/analysisSF3B1.Rmd
    Deleted:    analysis/comparePlatforms.Rmd
    Deleted:    analysis/compareProteomicsRNAseq.Rmd
    Deleted:    analysis/correlateCLLPD.Rmd
    Deleted:    analysis/correlateGenomic.Rmd
    Deleted:    analysis/correlateGenomic_removePC.Rmd
    Deleted:    analysis/correlateMIR.Rmd
    Deleted:    analysis/correlateMethylationCluster.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/predictOutcome.Rmd
    Deleted:    analysis/processProteomics_LUMOS.Rmd
    Deleted:    analysis/processProteomics_timsTOF.Rmd
    Deleted:    analysis/qualityControl_LUMOS.Rmd
    Deleted:    analysis/qualityControl_timsTOF.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
<pre class="r"><code>library(limma)
library(DESeq2)
library(qvalue)
library(proDA)
library(IHW)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load(&quot;../data/patMeta_enc.RData&quot;)
load(&quot;../data/ddsrna_enc.RData&quot;)
load(&quot;../data/proteomic_explore_enc.RData&quot;)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE,dev = c(&quot;png&quot;,&quot;pdf&quot;))
#protCLL &lt;- protCLL[,colnames(protCLL) %in% patMeta$Patient.ID]</code></pre>
</div>
<div id="preprocessing" class="section level1">
<h1>Preprocessing</h1>
<div id="process-proteomics-data" class="section level2">
<h2>Process proteomics data</h2>
<pre class="r"><code>#protCLL &lt;- protCLL[rowData(protCLL)$uniqueMap,]
protMat &lt;- assays(protCLL)[[&quot;count&quot;]] #without imputation
protMatLog &lt;- assays(protCLL)[[&quot;log2Norm&quot;]]</code></pre>
</div>
<div id="prepare-genomic-background" class="section level2">
<h2>Prepare genomic background</h2>
<div id="get-mutations-with-at-least-5-cases" class="section level3">
<h3>Get mutations with at least 5 cases</h3>
<pre class="r"><code>geneMat &lt;-  patMeta[match(colnames(protMat), patMeta$Patient.ID),] %&gt;%
  select(Patient.ID, IGHV.status, del11q:U1) %&gt;%
  mutate_if(is.factor, as.character) %&gt;% mutate(IGHV.status = ifelse(IGHV.status == &quot;M&quot;, 1,0)) %&gt;%
  mutate_at(vars(-Patient.ID), as.numeric) %&gt;% #assign a few unknown mutated cases to wildtype
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)


geneMat &lt;- geneMat[,apply(geneMat,2, function(x) sum(x %in% 1, na.rm = TRUE))&gt;=5]</code></pre>
<p><strong>Mutations that will be tested</strong></p>
<pre class="r"><code>colnames(geneMat)</code></pre>
<pre><code> [1] &quot;IGHV.status&quot; &quot;del11q&quot;      &quot;del13q&quot;      &quot;del17p&quot;      &quot;trisomy12&quot;  
 [6] &quot;trisomy19&quot;   &quot;NOTCH1&quot;      &quot;ATM&quot;         &quot;BRAF&quot;        &quot;DDX3X&quot;      
[11] &quot;EGR2&quot;        &quot;MED12&quot;       &quot;SF3B1&quot;       &quot;TP53&quot;       </code></pre>
</div>
</div>
</div>
<div id="differential-protein-expression-using-proda-lumos-dataset" class="section level1">
<h1>Differential protein expression using proDA (LUMOS dataset)</h1>
<p>We will use proDA, which is based on a linear model that considers the missing values using a probabilistic drop out model, to identify protein expression changes related to genotypes.<br />
To avoid potential confounding effect of IGHV and trisomy12, which are main drivers in CLL proteomic profile, we will block for IGHV status when we are testing trisomy12 and block trisomy12 when testing for IGHV status. For other genotypes, we will block for both IGHV status and trisomy12.</p>
<div id="identify-protein-expression-changes-related-to-ighv-and-trisomy12" class="section level2">
<h2>Identify protein expression changes related to IGHV and trisomy12</h2>
<div id="fit-the-probailistic-dropout-model" class="section level3">
<h3>Fit the probailistic dropout model</h3>
<pre class="r"><code>designMat &lt;- geneMat[  ,c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)]
designMat[,&quot;batch&quot;] &lt;- factor(protCLL[,rownames(designMat)]$batch)
fit &lt;- proDA(protMat, design = ~ . ,
             col_data = designMat)</code></pre>
</div>
<div id="limma-for-calculating-log2-fold-change" class="section level3">
<h3>Limma for calculating log2 fold change</h3>
<pre class="r"><code>lmDesign &lt;- model.matrix(~., designMat)
lmFit &lt;- lmFit(protMatLog, design = lmDesign)
fit2 &lt;- eBayes(lmFit)</code></pre>
</div>
<div id="test-for-differentially-expressed-proteins" class="section level3">
<h3>Test for differentially expressed proteins</h3>
<pre class="r"><code>resList.ighvTri12 &lt;- lapply(c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;), function(n) {
  contra &lt;- n
  resTab &lt;- test_diff(fit, contra) %&gt;%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %&gt;% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;%
    select(name, id, logFC, t, P.Value, adj.P.Val, n_obs) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()
  
  foldTab &lt;- topTable(fit2, coef = n, number = &quot;all&quot;) %&gt;%
    as_tibble(rownames = &quot;id&quot;) %&gt;% select(id, logFC) %&gt;%
    dplyr::rename(log2FC = logFC)
  
  resTab &lt;- left_join(resTab, foldTab, by = &quot;id&quot;)
  resTab
  
}) %&gt;% bind_rows()</code></pre>
</div>
</div>
<div id="test-for-other-variantions-blocking-for-ighv-and-trisomy12" class="section level2">
<h2>Test for other variantions (blocking for IGHV and trisomy12)</h2>
<p>Fit the probailistic dropout model and test for differentially expressed proteins</p>
<pre class="r"><code>otherGenes &lt;- colnames(geneMat)[!colnames(geneMat)%in% c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)]
resList &lt;- lapply(otherGenes, function(n) {
  designMat &lt;- geneMat[,c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;,n)]
  designMat[,&quot;batch&quot;] &lt;- factor(protCLL[,rownames(designMat)]$batch)
  designMat &lt;- designMat[!is.na(designMat[[n]]),]
  testMat &lt;- protMat[,rownames(designMat)]
  
  fit &lt;- proDA(testMat, design = ~ .,
             col_data = designMat)
  
  contra &lt;- n
  resTab &lt;- test_diff(fit, contra) %&gt;%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %&gt;% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;%
    select(name, id, logFC, t, P.Value, adj.P.Val, n_obs) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()

  #calculte log2 fold change
  lmDesign &lt;- model.matrix(~., designMat)
  protMatTest &lt;- protMatLog[,rownames(lmDesign)]
  lmFit &lt;- lmFit(protMatTest, design = lmDesign)
  fit2 &lt;- eBayes(lmFit)
  foldTab &lt;- topTable(fit2, coef = n, number = &quot;all&quot;) %&gt;%
    as_tibble(rownames = &quot;id&quot;) %&gt;% select(id, logFC) %&gt;%
    dplyr::rename(log2FC = logFC)
  resTab &lt;- left_join(resTab, foldTab, by = &quot;id&quot;)

  resTab
}) %&gt;% bind_rows()</code></pre>
<p>Combine the results</p>
<pre class="r"><code>resList &lt;- bind_rows(resList.ighvTri12, resList)

#Adjusting p values

#using BH
resList &lt;- mutate(resList, adj.P.global = p.adjust(P.Value, method = &quot;BH&quot;))

#using IHW
ihwRes &lt;- ihw(P.Value ~ factor(Gene), data= resList, alpha=0.1)
resList &lt;- mutate(resList, adj.P.IHW = adj_pvalues(ihwRes))</code></pre>
<p>Save the results for re-using</p>
<pre class="r"><code>save(resList, file = &quot;../output/deResList.RData&quot;)</code></pre>
</div>
<div id="bar-plot-of-number-of-significant-associations-with-proteins-5-fdr" class="section level2">
<h2>Bar plot of number of significant associations with proteins (5% FDR)</h2>
<p>Load the list of differentially expression proteins generated by Section 2</p>
<pre class="r"><code>load(&quot;../output/deResList.RData&quot;)</code></pre>
<pre class="r"><code>plotTab &lt;- resList %&gt;% group_by(Gene) %&gt;%
  summarise(nFDR.local = sum(adj.P.Val &lt;= 0.05))</code></pre>
<p>Individual gene adjusted</p>
<pre class="r"><code>plotTab &lt;- arrange(plotTab, desc(nFDR.local)) %&gt;% mutate(Gene = factor(Gene, levels = Gene))
numCorBar &lt;- ggplot(plotTab, aes(x=Gene, y = nFDR.local)) + geom_bar(stat=&quot;identity&quot;,fill=colList[2]) + 
  geom_text(aes(label = nFDR.local),vjust=-1,col=&quot;black&quot;,size=5) + ylim(0,1200) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab(&quot;Number of associations\n(5% FDR)&quot;) + xlab(&quot;&quot;)

numCorBar</code></pre>
<p><img src="figure/manuscript_S2_genomicAssociation.Rmd/number_of_association_local-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="some-examples-of-protein-gene-associations-that-passed-0.01-p-value-but-not-5-fdr" class="section level1">
<h1>Some examples of protein-gene associations that passed 0.01 p-value but not 5% FDR</h1>
<pre class="r"><code>protTab &lt;- sumToTidy(protCLL, rowID = &quot;uniprotID&quot;, colID = &quot;patID&quot;) %&gt;%
  mutate(count = count_combat)</code></pre>
<div id="del13q" class="section level2">
<h2>Del13q</h2>
<pre class="r"><code>resListSub &lt;- filter(resList, Gene == &quot;del13q&quot;)
nameList &lt;- c(&quot;CD22&quot;,&quot;CD72&quot;)

plotTab &lt;- protTab %&gt;% filter(hgnc_symbol %in% nameList) %&gt;%
  mutate(del13q = patMeta[match(patID, patMeta$Patient.ID),]$del13q) %&gt;%
  mutate(status = ifelse(del13q %in% 1,&quot;del(13)(q14)&quot;,&quot;other&quot;),
         name = hgnc_symbol) %&gt;%
  mutate(status = factor(status, levels = c(&quot;other&quot;,&quot;del(13)(q14)&quot;)))
pList &lt;- plotBox(plotTab, pValTabel = resListSub, y_lab = &quot;Protein expression&quot;)
del13qBox &lt;- cowplot::plot_grid(plotlist= pList, ncol=2)
del13qBox</code></pre>
<p><img src="figure/manuscript_S2_genomicAssociation.Rmd/del13q_example-1.png" width="864" style="display: block; margin: auto;" /></p>
</div>
<div id="tp53" class="section level2">
<h2>TP53</h2>
<pre class="r"><code>resListSub &lt;- filter(resList, Gene == &quot;TP53&quot;)
nameList &lt;- c(&quot;BCAS2&quot;)


plotTab &lt;- protTab %&gt;% filter(hgnc_symbol %in% nameList) %&gt;%
  mutate(TP53 = patMeta[match(patID, patMeta$Patient.ID),]$TP53) %&gt;%
  mutate(status = ifelse(TP53 %in% 1,&quot;Mut&quot;,&quot;WT&quot;),
         name = hgnc_symbol) %&gt;%
  mutate(status = factor(status, levels = c(&quot;WT&quot;,&quot;Mut&quot;)))
pList &lt;- plotBox(plotTab, pValTabel = resListSub, y_lab = &quot;Protein expression&quot;)
tp53Box &lt;- cowplot::plot_grid(plotlist= pList, ncol=1)
tp53Box</code></pre>
<p><img src="figure/manuscript_S2_genomicAssociation.Rmd/TP53_BCAS2-1.png" width="384" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="association-test-in-timstof-data" class="section level1">
<h1>Association test in timsTOF data</h1>
<p>The same procedure as for the LUMOS dataset will be used.</p>
<div id="for-ighv-and-trisomy12" class="section level2">
<h2>For IGHV and trisomy12</h2>
<p>Load timsTOF data</p>
<pre class="r"><code>load(&quot;../data/proteomic_timsTOF_enc.RData&quot;)
protMat &lt;- assays(protCLL)[[&quot;count&quot;]] #without imputation</code></pre>
<p>Genetic data</p>
<pre class="r"><code>geneMat &lt;-  patMeta[match(colnames(protMat), patMeta$Patient.ID),] %&gt;%
  select(Patient.ID, IGHV.status, trisomy12, SF3B1, trisomy19, del11q, del13q) %&gt;%
  mutate_if(is.factor, as.character) %&gt;% mutate(IGHV.status = ifelse(IGHV.status == &quot;M&quot;, 1,0)) %&gt;%
  mutate_at(vars(-Patient.ID), as.numeric) %&gt;% #assign a few unknown mutated cases to wildtype
  data.frame() %&gt;% column_to_rownames(&quot;Patient.ID&quot;)</code></pre>
<p>Fit the probailistic dropout model</p>
<pre class="r"><code>designMat &lt;- geneMat[  ,c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)]
fit &lt;- proDA(protMat, design = ~ .,
             col_data = designMat)</code></pre>
<p>Test for differentially expressed proteins</p>
<pre class="r"><code>resList.ighvTri12 &lt;- lapply(c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;), function(n) {
  contra &lt;- n
  resTab &lt;- test_diff(fit, contra) %&gt;%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %&gt;% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;%
    select(name, id, logFC, t, P.Value, adj.P.Val) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()
}) %&gt;% bind_rows()</code></pre>
</div>
<div id="test-for-other-variantions-blocking-for-ighv-and-trisomy12-1" class="section level2">
<h2>Test for other variantions (blocking for IGHV and trisomy12)</h2>
<p>Fit the probailistic dropout model and test for differentially expressed proteins</p>
<pre class="r"><code>otherGenes &lt;- colnames(geneMat)[!colnames(geneMat)%in% c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)]
resList &lt;- lapply(otherGenes, function(n) {
  designMat &lt;- geneMat[,c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;,n)]
  designMat &lt;- designMat[!is.na(designMat[[n]]),]
  testMat &lt;- protMat[,rownames(designMat)]
  
  fit &lt;- proDA(testMat, design = ~ .,
             col_data = designMat)
  
  contra &lt;- n
  resTab &lt;- test_diff(fit, contra) %&gt;%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %&gt;% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;%
    select(name, id, logFC, t, P.Value, adj.P.Val) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()
  resTab
}) %&gt;% bind_rows()</code></pre>
<p>Combine the results</p>
<pre class="r"><code>resList &lt;- bind_rows(resList.ighvTri12, resList)

#Adjusting p values

#using BH
resList &lt;- mutate(resList, adj.P.global = p.adjust(P.Value, method = &quot;BH&quot;))

#using IHW
ihwRes &lt;- ihw(P.Value ~ factor(Gene), data= resList, alpha=0.1)
resList &lt;- mutate(resList, adj.P.IHW = adj_pvalues(ihwRes))</code></pre>
<pre class="r"><code>save(resList, file = &quot;../output/deResList_timsTOF.RData&quot;)</code></pre>
</div>
<div id="bar-plot-of-number-of-significant-associations-with-proteins-5-fdr-1" class="section level2">
<h2>Bar plot of number of significant associations with proteins (5% FDR)</h2>
<p>Load the list of differentially expression proteins generated by Section 2</p>
<pre class="r"><code>load(&quot;../output/deResList_timsTOF.RData&quot;)</code></pre>
<pre class="r"><code>plotTab &lt;- resList %&gt;% group_by(Gene) %&gt;%
  summarise(nFDR.local = sum(adj.P.Val &lt;= 0.05))</code></pre>
<pre class="r"><code>plotTab &lt;- arrange(plotTab, desc(nFDR.local)) %&gt;% mutate(Gene = factor(Gene, levels = Gene))
numCorBar &lt;- ggplot(plotTab, aes(x=Gene, y = nFDR.local)) + geom_bar(stat=&quot;identity&quot;,fill=colList[2]) + 
  geom_text(aes(label = nFDR.local),vjust=-1,col=&quot;black&quot;,size=5) + ylim(0,1000) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab(&quot;Number of associations\n(5% FDR)&quot;) + xlab(&quot;&quot;)

numCorBar</code></pre>
<p><img src="figure/manuscript_S2_genomicAssociation.Rmd/number_of_association_local_tims-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="identify-assocations-between-rna-expression-and-genotypes" class="section level1">
<h1>Identify assocations between RNA expression and genotypes</h1>
<p>DEseq2 will be used to identify RNA expression changes related to genotypes. The same blocking strategy as used for the proteomic data will also be used for RNAseq data.</p>
<div id="prepare-rna-seq-data" class="section level2">
<h2>Prepare RNA seq data</h2>
<p>Subset for samples with proteomics</p>
<pre class="r"><code>ddsSub &lt;- dds[,dds$PatID %in% colnames(protCLL)]
#how many samples?

ddsSub &lt;- ddsSub[rownames(ddsSub) %in% rowData(protCLL)$ensembl_gene_id,]

#how many genes without any RNA expression detected?
#table(rowSums(counts(ddsSub)) &gt; 0)

ddsSub &lt;- ddsSub[rowSums(counts(ddsSub)) &gt; 0, ]
colData(ddsSub) &lt;- cbind(colData(ddsSub), geneMat[colnames(ddsSub),])</code></pre>
<div id="test-for-ighv-and-trisomy12" class="section level3">
<h3>Test For IGHV and trisomy12</h3>
<pre class="r"><code>design(ddsSub) &lt;- ~ IGHV.status + trisomy12
deRes &lt;- DESeq(ddsSub)</code></pre>
<p>Test for differentially expressed proteins</p>
<pre class="r"><code>resList.ighvTri12 &lt;- lapply(c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;), function(n) {
  resTab &lt;- results(deRes, name = n, tidy = TRUE) %&gt;%
    dplyr::rename(id = row, log2FC = log2FoldChange, t=stat,
                  P.Value = pvalue, adj.P.Val = padj) %&gt;% 
    mutate(name = rowData(ddsSub[id,])$symbol) %&gt;%
    select(name, id, log2FC, t, P.Value, adj.P.Val) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()
}) %&gt;% bind_rows()</code></pre>
</div>
<div id="test-for-other-variantions-blocking-for-ighv-and-trisomy12-2" class="section level3">
<h3>Test for other variantions (blocking for IGHV and trisomy12)</h3>
<pre class="r"><code>otherGenes &lt;- colnames(geneMat)[!colnames(geneMat)%in% c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)]
resList &lt;- lapply(otherGenes, function(n) {
  
  ddsTest &lt;- ddsSub[,!is.na(ddsSub[[n]])]
  design(ddsTest) &lt;- as.formula(paste0(&quot;~ IGHV.status + trisomy12 + &quot;,n))
  deRes &lt;- DESeq(ddsTest, betaPrior = FALSE)

  resTab &lt;- results(deRes, name = n, tidy = TRUE) %&gt;%
    dplyr::rename(id = row, log2FC = log2FoldChange, t=stat,
                  P.Value = pvalue, adj.P.Val = padj) %&gt;% 
    mutate(name = rowData(ddsSub[id,])$symbol) %&gt;%
    select(name, id, log2FC, t, P.Value, adj.P.Val) %&gt;%  
    arrange(P.Value) %&gt;% mutate(Gene = n) %&gt;%
    as_tibble()
  
  resTab
}) %&gt;% bind_rows()</code></pre>
<p>Combine the results</p>
<pre class="r"><code>resListRNA &lt;- bind_rows(resList.ighvTri12, resList)</code></pre>
<p>Save the results for re-using</p>
<pre class="r"><code>save(resListRNA, file = &quot;../output/deResListRNA.RData&quot;)</code></pre>
<p>Load the pre-calculated results (differential expression tests take long time.)</p>
<pre class="r"><code>load(&quot;../output/deResListRNA.RData&quot;)</code></pre>
</div>
<div id="bar-plot-of-number-of-significant-associations-5-fdr" class="section level3">
<h3>Bar plot of number of significant associations (5% FDR)</h3>
<pre class="r"><code>fdrCut = 0.05
plotTab &lt;- resListRNA %&gt;% group_by(Gene) %&gt;%
  summarise(nFDR.local = sum(adj.P.Val &lt;= fdrCut, na.rm=TRUE),
            nP = sum(P.Value &lt; 0.05))</code></pre>
<p>P values adjusted for each variant</p>
<pre class="r"><code>#local adjusted P-values

plotTab &lt;- arrange(plotTab, desc(nFDR.local)) %&gt;% mutate(Gene = factor(Gene, levels = Gene))
ggplot(plotTab, aes(x=Gene, y = nFDR.local)) + geom_bar(stat=&quot;identity&quot;,fill=colList[2]) + 
  geom_text(aes(label = paste0(nFDR.local)),vjust=-1,col=&quot;black&quot;) + 
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab(&quot;Number of associations\n(5% FDR)&quot;) + xlab(&quot;&quot;)</code></pre>
<p><img src="figure/manuscript_S2_genomicAssociation.Rmd/unnamed-chunk-31-1.png" width="864" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] ggbeeswarm_0.6.0            latex2exp_0.4.0            
 [3] forcats_0.5.1               stringr_1.4.0              
 [5] dplyr_1.0.5                 purrr_0.3.4                
 [7] readr_1.4.0                 tidyr_1.1.3                
 [9] tibble_3.1.0                ggplot2_3.3.3              
[11] tidyverse_1.3.0             IHW_1.16.0                 
[13] proDA_1.2.0                 qvalue_2.20.0              
[15] DESeq2_1.28.1               SummarizedExperiment_1.18.2
[17] DelayedArray_0.14.1         matrixStats_0.58.0         
[19] Biobase_2.48.0              GenomicRanges_1.40.0       
[21] GenomeInfoDb_1.24.2         IRanges_2.22.2             
[23] S4Vectors_0.26.1            BiocGenerics_0.34.0        
[25] limma_3.44.3               

loaded via a namespace (and not attached):
 [1] colorspace_2.0-0       ellipsis_0.3.1         rprojroot_2.0.2       
 [4] XVector_0.28.0         fs_1.5.0               rstudioapi_0.13       
 [7] farver_2.1.0           bit64_4.0.5            AnnotationDbi_1.50.3  
[10] fansi_0.4.2            lubridate_1.7.10       xml2_1.3.2            
[13] codetools_0.2-18       splines_4.0.2          cachem_1.0.4          
[16] geneplotter_1.66.0     knitr_1.31             jsonlite_1.7.2        
[19] workflowr_1.6.2        broom_0.7.5            annotate_1.66.0       
[22] dbplyr_2.1.0           compiler_4.0.2         httr_1.4.2            
[25] backports_1.2.1        assertthat_0.2.1       Matrix_1.3-2          
[28] fastmap_1.1.0          cli_2.3.1              later_1.1.0.1         
[31] htmltools_0.5.1.1      tools_4.0.2            gtable_0.3.0          
[34] glue_1.4.2             GenomeInfoDbData_1.2.3 reshape2_1.4.4        
[37] Rcpp_1.0.6             slam_0.1-48            cellranger_1.1.0      
[40] jquerylib_0.1.3        vctrs_0.3.6            xfun_0.21             
[43] rvest_1.0.0            lifecycle_1.0.0        XML_3.99-0.5          
[46] zlibbioc_1.34.0        scales_1.1.1           hms_1.0.0             
[49] promises_1.2.0.1       RColorBrewer_1.1-2     yaml_2.2.1            
[52] memoise_2.0.0          sass_0.3.1             stringi_1.5.3         
[55] RSQLite_2.2.3          highr_0.8              genefilter_1.70.0     
[58] BiocParallel_1.22.0    rlang_0.4.10           pkgconfig_2.0.3       
[61] bitops_1.0-6           evaluate_0.14          lattice_0.20-41       
[64] lpsymphony_1.16.0      labeling_0.4.2         cowplot_1.1.1         
[67] bit_4.0.4              tidyselect_1.1.0       plyr_1.8.6            
[70] magrittr_2.0.1         R6_2.5.0               generics_0.1.0        
[73] DBI_1.1.1              pillar_1.5.1           haven_2.3.1           
[76] withr_2.4.1            survival_3.2-7         RCurl_1.98-1.2        
[79] modelr_0.1.8           crayon_1.4.1           fdrtool_1.2.16        
[82] utf8_1.1.4             rmarkdown_2.7          locfit_1.5-9.4        
[85] grid_4.0.2             readxl_1.3.1           blob_1.2.1            
[88] git2r_0.28.0           reprex_1.0.0           digest_0.6.27         
[91] xtable_1.8-4           extraDistr_1.9.1       httpuv_1.5.5          
[94] munsell_0.5.0          beeswarm_0.3.1         vipor_0.4.5           
[97] bslib_0.2.4           </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
