<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Junyan Lu" />

<meta name="date" content="2020-10-09" />

<title>Section 1: Overview of CLL proteomic dataset</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">CLL Proteome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://git.embl.de/jlu/Proteomics">
    <span class="fas fa-gitlab"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Section 1: Overview of CLL proteomic dataset</h1>
<h4 class="author">Junyan Lu</h4>
<h4 class="date">2020-10-09</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-15
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>CLLproteomics_publish/analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200227code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200227)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200227code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200227)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgitembldejluProteomicstree3fb50c57fa6132d3779996e8364369f6fb4aea70targetblank3fb50c5a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://git.embl.de/jlu/Proteomics/tree/3fb50c57fa6132d3779996e8364369f6fb4aea70" target="_blank">3fb50c5</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rhistory
    Ignored:    code/.DS_Store
    Ignored:    code/.Rhistory
    Ignored:    data/.DS_Store
    Ignored:    output/.DS_Store

Untracked files:
    Untracked:  analysis/.trisomy12_norm.pdf
    Untracked:  analysis/manuscript_S1_Overview.Rmd
    Untracked:  analysis/manuscript_S2_genomicAssociation.Rmd
    Untracked:  analysis/manuscript_S3_trisomy12.Rmd
    Untracked:  analysis/manuscript_S4_trisomy19.Rmd
    Untracked:  analysis/manuscript_S5_IGHV.Rmd
    Untracked:  analysis/manuscript_S6_del11q.Rmd
    Untracked:  analysis/manuscript_S7_SF3B1.Rmd
    Untracked:  analysis/manuscript_S8_drugResponse_Outcomes.Rmd
    Untracked:  analysis/manuscript_S9_STAT2.Rmd
    Untracked:  analysis/test.pdf
    Untracked:  code/utils.R
    Untracked:  data/Fig1A.png
    Untracked:  data/ddsrna_enc.RData
    Untracked:  data/exprCNV_enc.RData
    Untracked:  data/gmts/
    Untracked:  data/patMeta_enc.RData
    Untracked:  data/pepCLL_lumos_enc.RData
    Untracked:  data/pheno1000_enc.RData
    Untracked:  data/proteins_in_complexes
    Untracked:  data/proteomic_LUMOS_enc.RData
    Untracked:  data/proteomic_timsTOF_enc.RData
    Untracked:  data/survival_enc.RData
    Untracked:  output/MSH6_splicing.svg
    Untracked:  output/SUGP1_splicing.svg
    Untracked:  output/deResList.RData
    Untracked:  output/deResList_timsTOF.RData
    Untracked:  output/dxdCLL.RData
    Untracked:  output/dxdCLL2.RData

Unstaged changes:
    Modified:   analysis/_site.yml
    Deleted:    analysis/analysisSF3B1.Rmd
    Deleted:    analysis/comparePlatforms.Rmd
    Deleted:    analysis/compareProteomicsRNAseq.Rmd
    Deleted:    analysis/correlateCLLPD.Rmd
    Deleted:    analysis/correlateGenomic.Rmd
    Deleted:    analysis/correlateGenomic_removePC.Rmd
    Deleted:    analysis/correlateMIR.Rmd
    Deleted:    analysis/correlateMethylationCluster.Rmd
    Modified:   analysis/index.Rmd
    Deleted:    analysis/predictOutcome.Rmd
    Deleted:    analysis/processProteomics_LUMOS.Rmd
    Deleted:    analysis/processProteomics_timsTOF.Rmd
    Deleted:    analysis/qualityControl_LUMOS.Rmd
    Deleted:    analysis/qualityControl_timsTOF.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-datasets" class="section level1">
<h1>Load packages and datasets</h1>
<pre class="r"><code>library(limma)
library(DESeq2)
library(cowplot)
library(proDA)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load(&quot;../data/patMeta_enc.RData&quot;)
load(&quot;../data/ddsrna_enc.RData&quot;)
load(&quot;../data/proteomic_LUMOS_enc.RData&quot;)
source(&quot;../code/utils.R&quot;)
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, dev = c(&quot;png&quot;,&quot;pdf&quot;))</code></pre>
</div>
<div id="rna-protein-associations" class="section level1">
<h1>RNA-protein associations</h1>
<div id="preprocess-transcriptomic-and-proteomic-data" class="section level2">
<h2>Preprocess transcriptomic and proteomic data</h2>
<pre class="r"><code>protCLL &lt;- protCLL[rowData(protCLL)$uniqueMap,]

colnames(dds) &lt;- dds$PatID
dds &lt;- estimateSizeFactors(dds)
sampleOverlap &lt;- intersect(colnames(protCLL), colnames(dds))
geneOverlap &lt;- intersect(rowData(protCLL)$ensembl_gene_id, rownames(dds))
ddsSub &lt;- dds[geneOverlap, sampleOverlap]
protSub &lt;- protCLL[match(geneOverlap, rowData(protCLL)$ensembl_gene_id), sampleOverlap]

#how many gene don&#39;t have RNA expression at all?
noExp &lt;- rowSums(counts(ddsSub)) == 0

#remove those genes in both datasets
ddsSub &lt;- ddsSub[!noExp,]
protSub &lt;- protSub[!noExp,]

#remove proteins with duplicated identifiers
protSub &lt;- protSub[!duplicated(rowData(protSub)$name)]

geneOverlap &lt;- intersect(rowData(protSub)$ensembl_gene_id, rownames(ddsSub))

ddsSub.vst &lt;- varianceStabilizingTransformation(ddsSub)</code></pre>
</div>
<div id="calculate-correlations-between-protein-abundance-and-rna-expression" class="section level2">
<h2>Calculate correlations between protein abundance and RNA expression</h2>
<pre class="r"><code>rnaMat &lt;- assay(ddsSub.vst)
proMat &lt;- assays(protSub)[[&quot;count&quot;]]
rownames(proMat) &lt;- rowData(protSub)$ensembl_gene_id

corTab &lt;- lapply(geneOverlap, function(n) {
  rna &lt;- rnaMat[n,]
  pro.raw &lt;- proMat[n,]
  res.raw &lt;- cor.test(rna, pro.raw, use = &quot;pairwise.complete.obs&quot;)
  tibble(id = n,
         p = res.raw$p.value,
         coef = res.raw$estimate)
}) %&gt;% bind_rows() %&gt;%
  arrange(desc(coef)) %&gt;% mutate(p.adj = p.adjust(p, method = &quot;BH&quot;),
                                 symbol = rowData(dds[id,])$symbol,
                                 chr = rowData(dds[id,])$chromosome)</code></pre>
</div>
<div id="plot-the-distribution" class="section level2">
<h2>Plot the distribution</h2>
<pre class="r"><code>corHistPlot &lt;- ggplot(corTab, aes(x=coef)) + geom_histogram(position = &quot;identity&quot;, col = colList[2], alpha =0.3, bins =50) +
  geom_vline(xintercept = 0, col = colList[1], linetype = &quot;dashed&quot;) + xlim(-0.7,1) +
      xlab(&quot;Pearson&#39;s correlation coefficient&quot;) + theme_half +
  ggtitle(&quot;Correlation between mRNA and protein expression&quot;)
corHistPlot</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/overall_protein_rna_correlation-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>Median Pearson’s correlation coefficient</p>
<pre class="r"><code>median(corTab$coef)</code></pre>
<pre><code>[1] 0.1718669</code></pre>
</div>
<div id="influence-of-overall-proteinrna-abundance-on-correlation" class="section level2">
<h2>Influence of overall protein/RNA abundance on correlation</h2>
<pre class="r"><code>medProt &lt;- rowMedians(proMat,na.rm = T)
names(medProt) &lt;- rownames(proMat)
medRNA &lt;- rowMedians(rnaMat, na.rm = T)
names(medRNA) &lt;- rownames(rnaMat)

plotTab &lt;- corTab %&gt;% mutate(rnaAbundance = medRNA[id], protAbundance = medProt[id])
plotList &lt;- list()

plotList[[&quot;rna&quot;]] &lt;- plotCorScatter(plotTab,&quot;coef&quot;,&quot;rnaAbundance&quot;,
                                    showR2 = FALSE, annoPos = &quot;left&quot;,
                                    x_lab =&quot;Correlation coefficient&quot;,
                                    y_lab = &quot;Median RNA expression&quot;,
                                    title = &quot;&quot;, dotCol = colList[5], textCol = colList[1])

plotList[[&quot;protein&quot;]] &lt;- plotCorScatter(plotTab,&quot;coef&quot;,&quot;protAbundance&quot;,
                                    showR2 = FALSE, annoPos = &quot;left&quot;,
                                    x_lab =&quot;Correlation coefficient&quot;,
                                    y_lab = &quot;Median protein expression&quot;,
                                    title = &quot;&quot;, dotCol = colList[6], textCol = colList[1])

cowplot::plot_grid(plotlist = plotList, ncol =2)</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/corProtRNAabundance-1.png" width="1536" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-protein-rna-correlation-for-selected-genes" class="section level2">
<h2>Plot protein RNA correlation for selected genes</h2>
<div id="good-correlations" class="section level3">
<h3>Good correlations</h3>
<pre class="r"><code>geneList &lt;- c(&quot;ZAP70&quot;,&quot;CD22&quot;,&quot;CD38&quot;,&quot;CD79A&quot;)
plotList &lt;- lapply(geneList, function(n) {
  geneId &lt;- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab &lt;- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef &lt;- cor(plotTab$x, plotTab$y, use=&quot;pairwise.complete&quot;)
  annoPos &lt;- ifelse (coef &gt; 0, &quot;left&quot;,&quot;right&quot;)
  plotCorScatter(plotTab, &quot;x&quot;,&quot;y&quot;, showR2 = FALSE, annoPos = annoPos, x_lab = &quot;RNA expression&quot;, shape = &quot;IGHV&quot;,
                 y_lab =&quot;Protein expression&quot;, title = n,dotCol = colList[4], textCol = colList[1], legendPos=&quot;none&quot;)
})
goodCorPlot &lt;- cowplot::plot_grid(plotlist = plotList, ncol =2)
goodCorPlot</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/RNA_protein_cor_markers-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
<div id="bad-correlations" class="section level3">
<h3>Bad correlations</h3>
<pre class="r"><code>geneList &lt;- c(&quot;DTNBP1&quot;,&quot;PRPF19&quot;)
plotList &lt;- lapply(geneList, function(n) {
  geneId &lt;- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab &lt;- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef &lt;- cor(plotTab$x, plotTab$y, use=&quot;pairwise.complete&quot;)
  annoPos &lt;- ifelse (coef &gt; 0, &quot;left&quot;,&quot;right&quot;)
  plotCorScatter(plotTab, &quot;x&quot;,&quot;y&quot;, showR2 = FALSE, annoPos = annoPos, x_lab = &quot;RNA expression&quot;,
                 y_lab =&quot;Protein expression&quot;, title = n,dotCol = colList[4], textCol = colList[1],
                 shape = &quot;IGHV&quot;, legendPos = &quot;none&quot;)
})
badCorPlot &lt;- cowplot::plot_grid(plotlist = plotList, ncol =2)
badCorPlot</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/RNA_protein_cor_markers_bad-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
</div>
<div id="principal-component-analysis" class="section level1">
<h1>Principal component analysis</h1>
<div id="calculate-pca" class="section level2">
<h2>Calculate PCA</h2>
<pre class="r"><code>#remove genes on sex chromosomes
protCLL.sub &lt;- protCLL[!rowData(protCLL)$chromosome_name %in% c(&quot;X&quot;,&quot;Y&quot;),]
plotMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
sds &lt;- genefilter::rowSds(plotMat)
plotMat &lt;- as.matrix(plotMat[order(sds,decreasing = TRUE),])
colAnno &lt;- colData(protCLL)[,c(&quot;gender&quot;,&quot;IGHV.status&quot;,&quot;trisomy12&quot;)] %&gt;%
  data.frame()
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

pcOut &lt;- prcomp(t(plotMat), center =TRUE, scale. = TRUE)
pcRes &lt;- pcOut$x
eigs &lt;- pcOut$sdev^2
varExp &lt;- structure(eigs/sum(eigs),names = colnames(pcRes))</code></pre>
<p>All proteins are included for PCA analysis</p>
</div>
<div id="plot-pc1-and-pc2" class="section level2">
<h2>Plot PC1 and PC2</h2>
<pre class="r"><code>plotTab &lt;- pcRes %&gt;% data.frame() %&gt;% cbind(colAnno[rownames(.),]) %&gt;%
  rownames_to_column(&quot;patID&quot;) %&gt;% as_tibble()

plotPCA12 &lt;- ggplot(plotTab, aes(x=PC1, y=PC2, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf(&quot;PC1 (%1.2f%%)&quot;,varExp[[&quot;PC1&quot;]]*100)) +
  ylab(sprintf(&quot;PC2 (%1.2f%%)&quot;,varExp[[&quot;PC2&quot;]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  theme_full + theme(legend.position = &quot;none&quot;)

plotPCA12</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/plot_PC1_PC2-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="plot-pc3-and-pc4" class="section level2">
<h2>Plot PC3 and PC4</h2>
<pre class="r"><code>plotPCA34 &lt;- ggplot(plotTab, aes(x=PC3, y=PC4, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf(&quot;PC3 (%1.2f%%)&quot;,varExp[[&quot;PC3&quot;]]*100)) +
  ylab(sprintf(&quot;PC4 (%1.2f%%)&quot;,varExp[[&quot;PC4&quot;]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  theme_full
plotPCA34</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/plot_PC2_PC4-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="correlation-test-between-pcs-and-ighv.status" class="section level2">
<h2>Correlation test between PCs and IGHV.status</h2>
<pre class="r"><code>corTab &lt;- lapply(colnames(pcRes),  function(pc) {
  ighvCor &lt;- t.test(pcRes[,pc] ~ colAnno$IGHV.status, var.equal=TRUE)
  tri12Cor &lt;- t.test(pcRes[,pc] ~ colAnno$trisomy12, var.equal=TRUE)
  tibble(PC = pc, 
         feature=c(&quot;IGHV&quot;, &quot;trisomy12&quot;),
         p = c(ighvCor$p.value, tri12Cor$p.value))
}) %&gt;% bind_rows() %&gt;% mutate(p.adj = p.adjust(p)) %&gt;%
  filter(p &lt;= 0.05) %&gt;% arrange(p)
corTab</code></pre>
<pre><code># A tibble: 5 x 4
  PC    feature          p        p.adj
  &lt;chr&gt; &lt;chr&gt;        &lt;dbl&gt;        &lt;dbl&gt;
1 PC3   trisomy12 1.09e-10 0.0000000107
2 PC4   IGHV      2.84e- 6 0.000275    
3 PC49  IGHV      4.97e- 3 0.477       
4 PC2   trisomy12 2.13e- 2 1           
5 PC1   IGHV      4.81e- 2 1           </code></pre>
</div>
<div id="pathway-enrichment-on-pc1-and-pc2" class="section level2">
<h2>Pathway enrichment on PC1 and PC2</h2>
<p>PC1</p>
<pre class="r"><code>enRes &lt;- list()
gmts = list(H= &quot;../data/gmts/h.all.v6.2.symbols.gmt&quot;,
            KEGG = &quot;../data/gmts/c2.cp.kegg.v6.2.symbols.gmt&quot;,
            C6 = &quot;../data/gmts/c6.all.v6.2.symbols.gmt&quot;)

proMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
iPC &lt;- &quot;PC1&quot;
pc &lt;- pcRes[,iPC][colnames(proMat)]
designMat &lt;- model.matrix(~1+pc)
fit &lt;- limma::lmFit(proMat, designMat)
fit2 &lt;- eBayes(fit)
corRes &lt;- topTable(fit2, &quot;pc&quot;, number = Inf) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;)

inputTab &lt;- corRes %&gt;% filter(adj.P.Val &lt; 0.1) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)

enRes[[&quot;Proteins associated with PC1&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)</code></pre>
<p>PC2</p>
<pre class="r"><code>proMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
iPC &lt;- &quot;PC2&quot;
pc &lt;- pcRes[,iPC][colnames(proMat)]
designMat &lt;- model.matrix(~1+pc)
fit &lt;- limma::lmFit(proMat, designMat)
fit2 &lt;- eBayes(fit)
corRes &lt;- topTable(fit2, &quot;pc&quot;, number = Inf) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;)

inputTab &lt;- corRes %&gt;% filter(adj.P.Val &lt; 0.1) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)

enRes[[&quot;Proteins associated with PC2&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)</code></pre>
<pre class="r"><code>cowplot::plot_grid(plotEnrichmentBar(enRes[[1]], ifFDR = TRUE, pCut = 0.05, setName = &quot;&quot;,title = &quot;Proteins associated with PC1&quot;),
                   plotEnrichmentBar(enRes[[2]], ifFDR = TRUE, pCut = 0.05, setName = &quot;&quot;, title = &quot;Proteins associated with PC2&quot;),ncol=1)</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/enrich_PC1PC2-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
<div id="pathway-enrichment-on-pc1-and-pc2-1" class="section level2">
<h2>Pathway enrichment on PC1 and PC2</h2>
<p>PC3</p>
<pre class="r"><code>enRes &lt;- list()
gmts = list(H= &quot;../data/gmts/h.all.v6.2.symbols.gmt&quot;,
            KEGG = &quot;../data/gmts/c2.cp.kegg.v6.2.symbols.gmt&quot;,
            C6 = &quot;../data/gmts/c6.all.v6.2.symbols.gmt&quot;)

proMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
iPC &lt;- &quot;PC3&quot;
pc &lt;- pcRes[,iPC][colnames(proMat)]
designMat &lt;- model.matrix(~1+pc)
fit &lt;- limma::lmFit(proMat, designMat)
fit2 &lt;- eBayes(fit)
corRes &lt;- topTable(fit2, &quot;pc&quot;, number = Inf) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;)

inputTab &lt;- corRes %&gt;% filter(adj.P.Val &lt; 0.1) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)

enRes[[&quot;Proteins associated with PC3&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)</code></pre>
<p>PC4</p>
<pre class="r"><code>proMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
iPC &lt;- &quot;PC4&quot;
pc &lt;- pcRes[,iPC][colnames(proMat)]
designMat &lt;- model.matrix(~1+pc)
fit &lt;- limma::lmFit(proMat, designMat)
fit2 &lt;- eBayes(fit)
corRes &lt;- topTable(fit2, &quot;pc&quot;, number = Inf) %&gt;%
  data.frame() %&gt;% rownames_to_column(&quot;id&quot;)

inputTab &lt;- corRes %&gt;% filter(adj.P.Val &lt; 0.1) %&gt;%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %&gt;% filter(!is.na(name)) %&gt;%
  distinct(name, .keep_all = TRUE) %&gt;%
  select(name, t) %&gt;% data.frame() %&gt;% column_to_rownames(&quot;name&quot;)

enRes[[&quot;Proteins associated with PC4&quot;]] &lt;- runGSEA(inputTab, gmts$H, &quot;page&quot;)</code></pre>
<pre class="r"><code>cowplot::plot_grid(plotEnrichmentBar(enRes[[1]], ifFDR = TRUE, pCut = 0.05, setName = &quot;&quot;,title = &quot;Proteins associated with PC3&quot;, removePrefix = &quot;HALLMARK_&quot;),
                   plotEnrichmentBar(enRes[[2]], ifFDR = TRUE, pCut = 0.05, setName = &quot;&quot;, title = &quot;Proteins associated with PC4&quot;,removePrefix = &quot;HALLMARK_&quot;),ncol=1)</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/enrich_PC3PC4-1.png" width="1152" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="hierarchical-clustering" class="section level1">
<h1>Hierarchical clustering</h1>
<pre class="r"><code>protCLL.sub &lt;- protCLL[!rowData(protCLL)$chromosome_name %in% c(&quot;X&quot;,&quot;Y&quot;),]
plotMat &lt;- assays(protCLL.sub)[[&quot;QRILC&quot;]]
colAnno &lt;- colData(protCLL)[,c(&quot;IGHV.status&quot;,&quot;trisomy12&quot;)] %&gt;%
  data.frame()
colAnno$trisomy12 &lt;- ifelse(colAnno$trisomy12 %in% 1, &quot;yes&quot;,&quot;no&quot;)

plotMat &lt;- mscale(plotMat, center = 5)

annoCol &lt;- list(trisomy12 = c(yes = &quot;black&quot;,no = &quot;grey80&quot;),
                IGHV.status = c(M = colList[3], U = colList[4]))
pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = &quot;none&quot;,
                   clustering_method = &quot;ward.D2&quot;,
                   color = colorRampPalette(c(colList[2],&quot;white&quot;,colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = FALSE, show_colnames = FALSE,
                   treeheight_row = 0)</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/HC_heatmap_origin-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="summarise-significant-associations" class="section level1">
<h1>Summarise significant associations</h1>
<div id="bar-plot-of-number-of-significant-associations-10-fdr" class="section level2">
<h2>Bar plot of number of significant associations (10% FDR)</h2>
<p>Load the list of differentially expression proteins generated by Section 2</p>
<pre class="r"><code>load(&quot;../output/deResList.RData&quot;)</code></pre>
<pre class="r"><code>plotTab &lt;- resList %&gt;% group_by(Gene) %&gt;%
  summarise(nFDR.local = sum(adj.P.Val &lt;= 0.1),
            nFDR.global = sum(adj.P.global &lt;= 0.1),
            nFDR.IHW = sum(adj.P.IHW &lt;= 0.1),
            nP = sum(P.Value &lt; 0.05))

#use IHW for adjusting p-values
resList &lt;- mutate(resList, adj.P.Val = adj.P.IHW)</code></pre>
<pre class="r"><code>plotTab &lt;- arrange(plotTab, desc(nFDR.IHW)) %&gt;% mutate(Gene = factor(Gene, levels = Gene))
numCorBar &lt;- ggplot(plotTab, aes(x=Gene, y = nFDR.IHW)) + geom_bar(stat=&quot;identity&quot;,fill=colList[2]) + 
  geom_text(aes(label = paste0(&quot;n=&quot;, nFDR.IHW)),vjust=-1,col=colList[1]) + ylim(0,1200) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab(&quot;Number of associations\n(10% FDR)&quot;) + xlab(&quot;&quot;)

numCorBar</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/number_of_association-1.png" width="768" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="assemble-figure" class="section level1">
<h1>Assemble figure</h1>
<pre class="r"><code>designPlot &lt;- draw_image(&quot;../data/Fig1A.png&quot;)
p &lt;- ggdraw() + designPlot

leftCol &lt;- plot_grid(p, 
                     plot_grid(plotPCA12, plotPCA34, ncol=2, rel_widths = c(.45,0.55)),
                     plot_grid(numCorBar,NULL,rel_widths = c(0.8,0.2),ncol=2), ncol = 1,
                     labels = c(&quot;A&quot;,&quot;E&quot;,&quot;F&quot;), label_size = 20, vjust = c(1.5, 0,-0.1),
                     rel_heights = c(1.2,1,1))


rightCol &lt;- plot_grid(corHistPlot,
                       goodCorPlot,
                       badCorPlot, ncol=1, rel_heights = c(0.28,0.48,0.24), labels = c(&quot;B&quot;,&quot;C&quot;,&quot;D&quot;), label_size = 20)
#pdf(&quot;test.pdf&quot;, height = 13, width = 18)
plot_grid(leftCol, rightCol, rel_widths = c(0.6, 0.4))</code></pre>
<p><img src="figure/manuscript_S1_Overview.Rmd/Figure1-1.png" width="1728" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dev.off()</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS  10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] piano_2.4.0                 latex2exp_0.4.0            
 [3] forcats_0.5.1               stringr_1.4.0              
 [5] dplyr_1.0.5                 purrr_0.3.4                
 [7] readr_1.4.0                 tidyr_1.1.3                
 [9] tibble_3.1.0                ggplot2_3.3.3              
[11] tidyverse_1.3.0             proDA_1.2.0                
[13] cowplot_1.1.1               DESeq2_1.28.1              
[15] SummarizedExperiment_1.18.2 DelayedArray_0.14.1        
[17] matrixStats_0.58.0          Biobase_2.48.0             
[19] GenomicRanges_1.40.0        GenomeInfoDb_1.24.2        
[21] IRanges_2.22.2              S4Vectors_0.26.1           
[23] BiocGenerics_0.34.0         limma_3.44.3               

loaded via a namespace (and not attached):
  [1] readxl_1.3.1           backports_1.2.1        fastmatch_1.1-0       
  [4] workflowr_1.6.2        igraph_1.2.6           shinydashboard_0.7.1  
  [7] splines_4.0.2          BiocParallel_1.22.0    digest_0.6.27         
 [10] htmltools_0.5.1.1      magick_2.7.0           fansi_0.4.2           
 [13] magrittr_2.0.1         memoise_2.0.0          cluster_2.1.1         
 [16] annotate_1.66.0        modelr_0.1.8           colorspace_2.0-0      
 [19] blob_1.2.1             rvest_1.0.0            haven_2.3.1           
 [22] xfun_0.21              crayon_1.4.1           RCurl_1.98-1.2        
 [25] jsonlite_1.7.2         genefilter_1.70.0      survival_3.2-7        
 [28] glue_1.4.2             gtable_0.3.0           zlibbioc_1.34.0       
 [31] XVector_0.28.0         scales_1.1.1           pheatmap_1.0.12       
 [34] DBI_1.1.1              relations_0.6-9        Rcpp_1.0.6            
 [37] xtable_1.8-4           bit_4.0.4              DT_0.17               
 [40] htmlwidgets_1.5.3      httr_1.4.2             fgsea_1.14.0          
 [43] gplots_3.1.1           RColorBrewer_1.1-2     ellipsis_0.3.1        
 [46] pkgconfig_2.0.3        XML_3.99-0.5           farver_2.1.0          
 [49] sass_0.3.1             dbplyr_2.1.0           locfit_1.5-9.4        
 [52] utf8_1.1.4             tidyselect_1.1.0       labeling_0.4.2        
 [55] rlang_0.4.10           later_1.1.0.1          AnnotationDbi_1.50.3  
 [58] munsell_0.5.0          cellranger_1.1.0       tools_4.0.2           
 [61] visNetwork_2.0.9       cachem_1.0.4           cli_2.3.1             
 [64] generics_0.1.0         RSQLite_2.2.3          broom_0.7.5           
 [67] evaluate_0.14          fastmap_1.1.0          yaml_2.2.1            
 [70] knitr_1.31             bit64_4.0.5            fs_1.5.0              
 [73] caTools_1.18.1         nlme_3.1-152           mime_0.10             
 [76] slam_0.1-48            xml2_1.3.2             compiler_4.0.2        
 [79] rstudioapi_0.13        marray_1.66.0          reprex_1.0.0          
 [82] geneplotter_1.66.0     bslib_0.2.4            stringi_1.5.3         
 [85] highr_0.8              lattice_0.20-41        Matrix_1.3-2          
 [88] shinyjs_2.0.0          vctrs_0.3.6            pillar_1.5.1          
 [91] lifecycle_1.0.0        jquerylib_0.1.3        data.table_1.14.0     
 [94] bitops_1.0-6           httpuv_1.5.5           R6_2.5.0              
 [97] promises_1.2.0.1       KernSmooth_2.23-18     gridExtra_2.3         
[100] codetools_0.2-18       gtools_3.8.2           assertthat_0.2.1      
[103] rprojroot_2.0.2        withr_2.4.1            GenomeInfoDbData_1.2.3
[106] mgcv_1.8-34            hms_1.0.0              grid_4.0.2            
[109] rmarkdown_2.7          git2r_0.28.0           sets_1.0-18           
[112] shiny_1.6.0            lubridate_1.7.10      </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("show" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
