---
title: "The Protein Landscape of Chronic Lymphocytic Leukemia (CLL)"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: false
editor_options:
  chunk_output_type: console
---

<font size="4">This is the workflow that generates the data analysis results and figures presented in the manuscript "The Protein Landscape of Chronic Lymphocytic Leukemia (CLL)" by Meier-Abt and Lu et al. Users can also use the associated source code to reproduce all results and figures in the manuscript.</font> 


### [Section 1: Overview of CLL proteomic dataset](./manuscript_S1_Overview.html)  
In this part, we will explore the overall structures of our CLL proteomic dataset, by correlating protein expression with RNA expression and by unsupervised clustering, including principal component analysis (PCA) and hierarchical clustering.

### [Section 2: Identifying proteins associated disease drivers in CLL](./manuscript_S2_genomicAssociation.html)  
In this part, we will identify proteins whose expression levels are associated with certain genotypes.


### [Section 3: Proteomic signatures of trisomy12](./manuscript_S3_trisomy12.html) 
In this part, we will explore the proteomic signatures of trisomy12. We will identify differentially expressed proteins in samples with trisomy12 compared to samples without trisomy12, and use gene set enrichment analysis to characterize the potential pathway activity changes related to trisomy12. We will particularly focus on the gene dosage effect and protein level buffering in trisomy12 samples. We will also use protein complex network to illustrate the potential mechanism for the connection between the cis- and trans-effect of trisomy12. 

### [Section 4: Proteomic signature of IGHV mutation status](./manuscript_S4_IGHV.html) 
In this part, we will explore the proteomic signatures related to IGHV mutational status in CLL. Differential protein expression analysis and gene set enrichment analysis will be performed. 


### Section 5-7: 
In Section 5 to 7, we will characterize proteomic signatures for trisomy19, deletion of 11q22.3 and SF3B1 mutations in CLL.   

[Section 5: Proteomic signatures of trisomy19](./manuscript_S5_trisomy19.html)   

[Section 6: Proteomic signature of del(11)(q22.3)](./manuscript_S6_del11q.html)    

[Section 7: Proteomic signature of SF3B1](./manuscript_S7_SF3B1.html)    



### [Section 8: Identifying protein biomarkers drug responses and clinical outcomes](./manuscript_S8_drugResponse_Outcomes.html)  
In this part, we will integrate the protein expression data, genomic data, and patient clinical phenotypes to identify protein biomarkers for drug response and outcome prediction. 


### [Section 9: Factors affecting STAT2 expression in CLL](./manuscript_S9_STAT2.html)  
In this part, we will characterize the expression of one protein, STAT2, in detail. We will integrate multi-omic datasets to identify the factors that contribute to the variance of STAT2 protein expression among CLL samples, and explore the down-stream effect related to STAT2 protein expression. 
