---
title: "Section 8: Identifying proteomic biomarkers drug responses and clinical outcomes"
author: "Junyan Lu"
date: "2020-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages and datasets

```{r, message=FALSE, warning=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, dev = c("png","pdf"))
library(limma)
library(pheatmap)
library(jyluMisc)
library(survival)
library(survminer)
library(maxstat)
library(glmnet)
library(SummarizedExperiment)
library(cowplot)
library(tidyverse)

load("../data/patMeta_enc.RData")
load("../data/pheno1000_enc.RData")
load("../data/survival_enc.RData")
load("../data/proteomic_LUMOS_enc.RData") #using the LUMOS data
protCLL$trisomy12 <- patMeta[match(colnames(protCLL),patMeta$Patient.ID),]$trisomy12
protCLL <- protCLL[rowData(protCLL)$uniqueMap,]
source("../code/utils.R")
```


# Correlations between protein abundance and drug response

## Preprocess datasets

Drug screening data from 1000CPS: low quality samples after QC are removed; the edge effect corrected viability values are used
```{r}
viabMat <- pheno1000_main %>% filter(!lowQuality, ! Drug %in% c("DMSO","PBS"), patientID %in% colnames(protCLL)) %>%
  group_by(patientID, Drug) %>% summarise(viab = mean(normVal.adj.cor_auc)) %>%
  spread(key = patientID, value = viab) %>%
  data.frame(stringsAsFactors = FALSE) %>% column_to_rownames("Drug") %>%
  as.matrix()
```

Proteomics data
```{r}
proMat <- assays(protCLL)[["count"]]
proMat <- proMat[,colnames(viabMat)]

#Remove proteins without much variance (to lower multi-testing burden)

sds <- genefilter::rowSds(proMat,na.rm=TRUE)
proMat <- proMat[sds > genefilter::shorth(sds),]
```

How many samples have both proteomics data and CPS1000 screen data
```{r}
ncol(proMat)
```

## Association test

```{r}
resTab.auc <- lapply(rownames(viabMat),function(drugName) {
  viab <- viabMat[drugName, ]
  designMat <- model.matrix(~1+viab)
  fit <- lmFit(proMat,  designMat)
  fit2 <- eBayes(fit)
  corRes <- topTable(fit2, number ="all", adjust.method = "BH", coef = "viab") %>% rownames_to_column("id") %>%
    mutate(symbol = rowData(protCLL[id,])$hgnc_symbol, Drug = drugName)
}) %>% bind_rows() %>% mutate(adj.P.Val = p.adjust(P.Value, method = "BH")) %>% arrange(P.Value)
```

### Bar plot to show the number of associations (10% FDR)

```{r, fig.width=10, fig.height=8}
#Select significant associations (10% FDR)
resTab.sig <- filter(resTab.auc, adj.P.Val < 0.1) %>% 
  select(Drug, symbol, id,logFC, P.Value, adj.P.Val)

plotTab <- resTab.sig %>% group_by(Drug) %>%
  summarise(n = length(id)) %>% ungroup()
ordTab <- group_by(plotTab, Drug) %>% summarise(total = sum(n)) %>%
  arrange(desc(total))
plotTab <- mutate(plotTab, Drug = factor(Drug, levels = ordTab$Drug)) %>%
  filter(n>0)

drugBar <- ggplot(plotTab, aes(x=Drug, y = n)) + geom_bar(stat="identity",fill=colList[4]) + 
  geom_text(aes(label = paste0("n=", n)),vjust=-1,col=colList[1]) +
  ylim(0,15)+ annotate("text", label = "Number of associations (10% FDR)", x=Inf, y=Inf,hjust=1, vjust=1, size=6)+
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab("Number") + xlab("")

drugBar
```

### Table of significant associations (10% FDR)
```{r}
resTab.sig %>% mutate_if(is.numeric, formatC, digits=2, format= "e") %>%
  DT::datatable()
```

### Correlation plot of selected candidates

```{r}
plotDrugCorScatter <- function(inputTab, x, y, x_lab = "X", y_lab = "Y", title = "",
                           col = NULL, showR2 = TRUE, annoPos = "right",
                           dotCol = colList, textCol="darkred") {

  #prepare table for plotting
  plotTab <- tibble(x = inputTab[[x]],y=inputTab[[y]])
  if (!is.null(col)) plotTab <- mutate(plotTab, status = inputTab[[col]])
  plotTab <- filter(plotTab, !is.na(x), !is.na(y))

  #prepare annotation values
  corRes <- cor.test(plotTab$x, plotTab$y)
  pval <- formatNum(corRes$p.value, digits = 1, format = "e")
  Rval <- formatNum(corRes$estimate, digits = 1, format = "e")
  R2val <- formatNum(corRes$estimate^2, digits = 1, format = "e")
  Nval <- nrow(plotTab)
  annoP <- bquote(italic("P")~"="~.(pval))

  if (showR2) {
    annoCoef <-  bquote(R^2~"="~.(R2val))
  } else {
    annoCoef <- bquote(R~"="~.(Rval))
  }
  annoN <- bquote(N~"="~.(Nval))

  corPlot <- ggplot(plotTab, aes(x = x, y = y))

  if (!is.null(col)) {
    corPlot <- corPlot + geom_point(aes(fill = status), shape =21, size =3) +
      scale_fill_manual(values = dotCol)
  } else {
    corPlot <- corPlot + geom_point(fill = dotCol[1], shape =21, size=3)
  }

  corPlot <- corPlot +   geom_smooth(formula = y~x,method = "lm", se=FALSE, color = "grey50", linetype ="dashed" )

  if (annoPos == "right") {

    corPlot <- corPlot + annotate("text", x = max(plotTab$x), y = Inf, label = annoN,
                                  hjust=1, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = max(plotTab$x), y = Inf, label = annoP,
               hjust=1, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = max(plotTab$x), y = Inf, label = annoCoef,
               hjust=1, vjust =6, size = 5, parse = FALSE, col= textCol)

  } else if (annoPos== "left") {
    corPlot <- corPlot + annotate("text", x = min(plotTab$x), y = Inf, label = annoN,
                                  hjust=0, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = min(plotTab$x), y = Inf, label = annoP,
               hjust=0, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = min(plotTab$x), y = Inf, label = annoCoef,
               hjust=0, vjust =6, size = 5, parse = FALSE, col= textCol)
  }
  corPlot <- corPlot + ylab(y_lab) + xlab(x_lab) + ggtitle(title) +
    scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
    scale_x_continuous(labels = scales::number_format(accuracy = 0.1)) +
    theme_full
  corPlot
}
```

```{r, fig.width=12, fig.height=4}
pairList <- list(c("Cobimetinib","STAT2"),c("Venetoclax","PGD"),c("Idelalisib", "HEBP2"))
plotList <- lapply(pairList, function(pair) {
  textCol <- "darkred"
  drugName <- pair[1]
  proteinName <- pair[2]
  id <- rownames(protCLL)[match(proteinName, rowData(protCLL)$hgnc_symbol)]
  plotTab <- tibble(patID = colnames(viabMat), 
                    viab = viabMat[drugName,],
                    expr = proMat[id,]) %>%
    mutate(IGHV = protCLL[,patID]$IGHV.status,
           trisomy12 = protCLL[,patID]$trisomy12) %>%
    mutate(trisomy12 = ifelse(trisomy12 ==1,"yes","no")) %>%
    filter(!is.na(viab),!is.na(expr))
  
  pval <- formatNum(filter(resTab.sig, Drug == drugName, symbol == proteinName)$P.Value, digits = 1, format = "e")
  Rval <- sprintf("%1.2f",cor(plotTab$viab, plotTab$expr))
  Nval <- nrow(plotTab)
  annoP <- bquote(italic("P")~"="~.(pval))
  annoN <- bquote(N~"="~.(Nval))
  annoCoef <- bquote(R~"="~.(Rval))

  corPlot <- ggplot(plotTab, aes(x = viab, y = expr)) + 
    geom_point(aes(col = trisomy12, shape = IGHV), size=5) +
    scale_shape_manual(values = c(M = 19, U = 1)) + 
    scale_color_manual(values = c(yes = colList[2], no = colList[3])) +
    geom_smooth(formula = y~x,method = "lm", se=FALSE, color = "grey50", linetype ="dashed" ) +
    ggtitle(sprintf("%s ~ %s", drugName, proteinName)) +
    ylab("Protein expression") + xlab("Viability after treatment") +
    theme_full +
    theme(legend.position = "bottom") 

  if (Rval < 0) annoPos <- "right" else annoPos <- "left"
  
  if (annoPos == "right") {

    corPlot <- corPlot + annotate("text", x = max(plotTab$viab), y = Inf, label = annoN,
                                  hjust=1, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = max(plotTab$viab), y = Inf, label = annoP,
               hjust=1, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = max(plotTab$viab), y = Inf, label = annoCoef,
               hjust=1, vjust =6, size = 5, parse = FALSE, col= textCol)

  } else if (annoPos== "left") {
    corPlot <- corPlot + annotate("text", x = min(plotTab$viab), y = Inf, label = annoN,
                                  hjust=0, vjust =2, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = min(plotTab$viab), y = Inf, label = annoP,
               hjust=0, vjust =4, size = 5, parse = FALSE, col= textCol) +
      annotate("text", x = min(plotTab$viab), y = Inf, label = annoCoef,
               hjust=0, vjust =6, size = 5, parse = FALSE, col= textCol)
  }
  
  corPlot <- corPlot + ylab("Protein expression") + xlab("Viability after treatment") + 
    scale_y_continuous(labels = scales::number_format(accuracy = 0.1)) +
    scale_x_continuous(labels = scales::number_format(accuracy = 0.1))

  corPlot
})


drugCor <- cowplot::plot_grid(plotlist = plotList, ncol =3)
drugCor
```

### Association test with blocking for IGHV and trisomy12

```{r, cache=TRUE, autodep=TRUE}
testList <- filter(resTab.auc, adj.P.Val < 0.1)
resTab.auc.block <- lapply(seq(nrow(testList)),function(i) {
  pair <- testList[i,]
  expr <- proMat[pair$id,]
  viab <- viabMat[pair$Drug, ]
  ighv <- protCLL[,colnames(viabMat)]$IGHV.status
  tri12 <- protCLL[,colnames(viabMat)]$trisomy12
  res <- anova(lm(viab~ighv+tri12+expr))
  data.frame(id = pair$id, P.Value = res["expr",]$`Pr(>F)`, symbol = pair$symbol,
             Drug = pair$Drug,
             P.Value.IGHV = res["ighv",]$`Pr(>F)`,P.Value.trisomy12 = res["tri12",]$`Pr(>F)`,
             P.Value.noBlock = pair$P.Value,
             stringsAsFactors = FALSE)
  
}) %>% bind_rows() %>% mutate(adj.P.Val = p.adjust(P.Value, method = "BH")) %>% arrange(P.Value)
```

Assocations that are still significant
```{r}
resTab.sig %>% mutate_if(is.numeric, formatC, digits=2, format= "e") %>%
  DT::datatable()
```
The above mentioned pairs are still significant. 

## Variance explained

**Test whether the protein expression can explain additional variance in drug response compared to genetic alone**

Prepare genomics
```{r}
geneMat <-  patMeta[match(colnames(proMat), patMeta$Patient.ID),] %>%
  select(Patient.ID, IGHV.status, del11p:U1) %>%
  mutate_if(is.factor, as.character) %>% mutate(IGHV.status = ifelse(IGHV.status == "M", 1,0)) %>%
  mutate_at(vars(-Patient.ID), as.numeric) %>% #assign a few unknown mutated cases to wildtype 
  mutate_all(replace_na,0) %>%
  data.frame() %>% column_to_rownames("Patient.ID")


geneMat <- geneMat[,apply(geneMat,2, function(x) sum(x %in% 1, na.rm = TRUE))>=5] %>% as.matrix()
```

**Genes that will be included in the multivariate model**
```{r}
colnames(geneMat)
```

```{r}
compareR2 <- function(Drug, protName, geneMat) {
  viab <- viabMat[Drug,]
  protID <- unique(filter(resTab.sig, symbol == protName)$id)
  expr <- proMat[protID,]

  tabGene <- data.frame(geneMat)
  tabGene[["viab"]] <- viab
  tabCom <- tabGene
  tabCom[[protName]] <- expr
  
  r2Prot <- summary(lm(viab~expr))$r.squared
  r2Gene <- summary(lm(viab~., data=tabGene))$r.squared
  r2Com <- summary(lm(viab~., data=tabCom))$r.squared
  
  plotTab <- tibble(model = c("genetics",paste0(protName, " expression"),sprintf("genetics + \n%s expression",protName)),
                    R2 = c(r2Gene, r2Prot, r2Com))
  
  ggplot(plotTab, aes(x=model, y = R2)) + geom_bar(aes(fill = model),stat="identity", width=0.6) + coord_flip() +
     ggtitle(Drug) + ylab("Variance expalained (R2)") + xlab("") +
    scale_fill_manual(values = colList[4:6]) + theme_full +theme(legend.position = "none") 
}
``` 

```{r, fig.height=10, fig.width=6}
plotList <- lapply(pairList, function(p) {
  compareR2(p[1],p[2],geneMat)
})
plot_grid(plotlist= plotList, ncol= 1)
```
Based on the plot of R2 values, including protein expression in multi-variate model could explain additional variance in drug responses compared to genetic alone. For Venetoclax, using a single protein expression value of PGD already explain more variance than genetics.

```{r, fig.width=5, fig.height=3}
stat2Bar <- plotList[[1]]
stat2Bar
```

# Protein markers for clincial outcomes

## Uni-variate model to identify proteins associated with outcomes

```{r}
protCLL.sub <- protCLL[!rowData(protCLL)$chromosome_name %in% c("X","Y"),]
protMat <- assays(protCLL.sub)[["count"]]

survTab <- survT %>% filter(patID %in% colnames(protCLL)) %>%
  select(patID, OS, died, TTT, treatedAfter) %>%
  dplyr::rename(patientID = patID)
```

```{r, cache=TRUE, autodep=TRUE}
uniRes.ttt <- lapply(rownames(protMat), function(n) {
  testTab <- mutate(survTab, expr = protMat[n, patientID])
  com(testTab$expr, testTab$TTT, testTab$treatedAfter, TRUE) %>%
    mutate(id = n)
}) %>% bind_rows() %>% mutate(p.adj = p.adjust(p, method = "BH")) %>%
  arrange(p) %>% mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
  mutate(outcome = "TTT")

uniRes.os <- lapply(rownames(protMat), function(n) {
  testTab <- mutate(survTab, expr = protMat[n, patientID])
  com(testTab$expr, testTab$OS, testTab$died, TRUE) %>%
    mutate(id = n)
}) %>% bind_rows() %>% mutate(p.adj = p.adjust(p, method = "BH")) %>%
  arrange(p) %>% mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
  mutate(outcome = "OS")

uniRes <- bind_rows(uniRes.ttt, uniRes.os) %>%
  mutate(p.adj = p.adjust(p, method = "BH"))
```

### A table showing significant associations
```{r}
uniRes %>% filter(p.adj < 0.1) %>% mutate_if(is.numeric, formatC, digits=2,format="e") %>%
  select(name, p, HR, p.adj, outcome) %>% DT::datatable()
```

## Selection protein markers independent of known risks using multi-vairate model

Prepare data
```{r}
#table of known risks
riskTab <- select(survTab, patientID) %>%
  left_join(patMeta[,c("Patient.ID","IGHV.status","TP53","trisomy12","del17p","gender")], by = c(patientID = "Patient.ID")) %>% 
  mutate(TP53 = as.numeric(as.character(TP53)),
         del17p = as.numeric(as.character(del17p))) %>%
  mutate(`TP53.del17p` = as.numeric(TP53 | del17p),
         IGHV = factor(ifelse(IGHV.status %in% "U",1,0))) %>%
  select(-TP53, -del17p,-IGHV.status) %>%
  mutate_if(is.numeric, as.factor) %>%
  dplyr::rename(sex=gender) %>%
  mutate(age = survT[match(patientID, survT$patID),]$age) %>%
  mutate(age = age/10) %>%
  select(patientID, sex, age, trisomy12,`TP53.del17p`, IGHV)
```

Multi-variate test
```{r, cache=TRUE, autodep=TRUE, warning=FALSE}
cTab.ttt <- lapply(filter(uniRes, outcome == "TTT", p.adj <=0.1)$id, function(n) {
  risk0 <- riskTab
  expr <- protMat[n,]
  expr <- (expr - mean(expr,na.rm=TRUE))/sd(expr,na.rm = TRUE)
  risk1 <- riskTab %>% mutate(protExpr = expr[patientID])
  res0 <- summary(runCox(survTab, risk0, "TTT","treatedAfter"))
  fullModel <- runCox(survTab, risk1, "TTT","treatedAfter")
  res1 <- summary(fullModel)
  tibble(id = n, c0 = res0$concordance[1], c1 = res1$concordance[1],
         se0 = res0$concordance[2],se1 = res1$concordance[2],
         ci0 = se0*1.96, ci1 = se1*1.96,
         p = res1$coefficients["protExpr",5],
         fullModel = list(fullModel))
}) %>% bind_rows() %>% mutate(diffC = c1-c0) %>%
  arrange(desc(diffC)) %>%
  mutate(name=rowData(protCLL[id,])$hgnc_symbol,
         outcome = "TTT")

cTab.os <- lapply(filter(uniRes, outcome == "OS", p.adj<=0.1)$id, function(n) {
  risk0 <- riskTab
  expr <- protMat[n,]
  expr <- (expr - mean(expr,na.rm=TRUE))/sd(expr,na.rm = TRUE)
  risk1 <- riskTab %>% mutate(protExpr = expr[patientID])
  res0 <- summary(runCox(survTab, risk0, "OS","died"))
  fullModel <- runCox(survTab, risk1, "OS","died")
  res1 <- summary(fullModel)
  
  tibble(id = n, c0 = res0$concordance[1], c1 = res1$concordance[1],
         se0 = res0$concordance[2],se1 = res1$concordance[2],
         ci0 = se0*1.96, ci1 = se1*1.96,
         p = res1$coefficients["protExpr",5],
         fullModel = list(fullModel))
}) %>% bind_rows() %>% mutate(diffC = c1-c0) %>%
  arrange(desc(diffC)) %>%
  mutate(name=rowData(protCLL[id,])$hgnc_symbol,
         outcome = "OS")

cTab <- bind_rows(cTab.ttt, cTab.os) %>%
  mutate(p.adj = p.adjust(p, method = "BH")) %>%
  arrange(p)
```

### A table showing idependent protein markers
```{r}
cTab %>% filter(p.adj < 0.1) %>% mutate_if(is.numeric, formatC, digits=2,format="e") %>%
  select(name, p, p.adj, outcome) %>% DT::datatable()
```

### Forest plot of selected markers

```{r}
plotHazard <- function(survRes, protName, title = "") {
  sumTab <- summary(survRes)$coefficients
  confTab <- summary(survRes)$conf.int
  #correct feature name
  nameOri <- rownames(sumTab)
  nameMod <- substr(nameOri, 1, nchar(nameOri) -1)
  plotTab <- tibble(feature = rownames(sumTab),
                    nameMod = substr(nameOri, 1, nchar(nameOri) -1),
                    HR = sumTab[,2],
                    p = sumTab[,5],
                    Upper = confTab[,4],
                    Lower = confTab[,3]) %>%
    mutate(feature = ifelse(nameMod %in% names(survRes$xlevels), nameMod, feature)) %>%
    mutate(feature = str_replace(feature, "[.]","/")) %>%
    mutate(feature = str_replace(feature, "[_]","-")) %>%
    mutate(feature = str_replace(feature, "IGHV","IGHV-U")) %>%
    mutate(candidate = ifelse(feature == "protExpr", "yes","no")) %>%
    mutate(feature = ifelse(feature == "protExpr", protName, feature)) %>%
    #arrange(desc(abs(p))) %>% 
    mutate(feature = factor(feature, levels = feature)) #%>%
    #mutate(type = ifelse(HR >1 ,"up","down")) %>%
   # mutate(Upper = ifelse(Upper > 10, 10, Upper))

  p <- ggplot(plotTab, aes(x=feature, y = HR, color = candidate)) +
    geom_hline(yintercept = 1, linetype = "dotted") +
    geom_point(position = position_dodge(width=0.8), size=3) +
    geom_errorbar(aes(ymin = Lower, ymax = Upper), width = 0.3, size=1) +
    geom_text(position = position_nudge(x = 0.3),
              aes(y = HR, label =  sprintf("italic(P)~'='~'%s'",
                                           formatNum(p, digits = 1))),
              color = "black", size =5, parse = TRUE) +
    #expand_limits(y=c(-0.5,0))+
    scale_color_manual(values = c(yes = "darkred", no = "black")) +
    ggtitle(title) + scale_y_log10() +
    ylab("Hazard ratio") +
    coord_flip() +
    theme_full +
    theme(legend.position = "none", axis.title.y = element_blank())
  return(p)
}
```



**PRMT5**
```{r, fig.width=5, fig.height=6}
protName <- "PRMT5"
outcomeName <- "TTT"
survRes <- filter(cTab, outcome == outcomeName , name == protName)$fullModel[[1]]
hr.prmt5 <- plotHazard(survRes, protName, outcomeName)
hr.prmt5
```

**PYGB**
```{r, fig.width=5, fig.height=6}
protName <- "PYGB"
outcomeName <- "TTT"
survRes <- filter(cTab, outcome == outcomeName , name == protName)$fullModel[[1]]
hr.pygb <- plotHazard(survRes, protName, outcomeName)
hr.pygb
```

**PES1**
```{r,fig.width=5, fig.height=6}
protName <- "PES1"
outcomeName <- "TTT"
survRes <- filter(cTab, outcome == outcomeName , name == protName)$fullModel[[1]]
hr.pes1 <- plotHazard(survRes, protName, outcomeName)
hr.pes1
```


**GNAI2**
```{r,fig.width=5, fig.height=6}
protName <- "GNAI2"
outcomeName <- "OS"
survRes <- filter(cTab, outcome == outcomeName , name == protName)$fullModel[[1]]
plotHazard(survRes, protName, outcomeName)
```

**RRAS2**
```{r,fig.width=5, fig.height=6}
protName <- "RRAS2"
outcomeName <- "OS"
survRes <- filter(cTab, outcome == outcomeName , name == protName)$fullModel[[1]]
plotHazard(survRes, protName, outcomeName)
```


### Bar plots comparing C-index

**TTT**
```{r, fig.height=4, fig.width=12}
nameList <- c("PRMT5","PYGB","PES1")
cTab.sub <- filter(cTab, outcome == "TTT", name %in% nameList)
plotList <- lapply(seq(nrow(cTab.sub)), function(i) {
  plotTab <- tibble(model = c("knownRisks","plusProtein"),
                    cindex = c(cTab.sub[i,]$c0, cTab.sub[i,]$c1),
                    ci = c(cTab.sub[i,]$ci0,cTab.sub[i,]$ci1))
  ggplot(plotTab, aes(x=model, y=cindex, fill = model)) + 
    geom_bar(stat="identity",width=0.6) + 
    geom_errorbar(aes(ymin=cindex + ci, ymax = cindex-ci), width=0.5) +
    theme_full+ scale_fill_manual(values = colList[6:7]) +
    ggtitle(cTab.ttt[i,]$name) + theme(legend.position = "none") +
    
    ylab("Harrell's C-index")
})
cTTT <- plot_grid(plotlist = plotList, ncol =3)
cTTT
```

**OS**
(The associations with OS are not very reliable as we don't have enough events)
```{r, fig.height=4, fig.width=12}
nameList <- c("GNAI2","RRAS2")
cTab.sub <- filter(cTab, outcome == "OS", name %in% nameList)
plotList <- lapply(seq(nrow(cTab.sub)), function(i) {
  plotTab <- tibble(model = c("knownRisks","plusProtein"),
                    cindex = c(cTab.sub[i,]$c0, cTab.sub[i,]$c1),
                    ci = c(cTab.sub[i,]$ci0,cTab.sub[i,]$ci1))
  ggplot(plotTab, aes(x=model, y=cindex, fill = model)) + 
    geom_bar(stat="identity",width=0.6) + 
    geom_errorbar(aes(ymin=cindex + ci, ymax = cindex-ci), width=0.5) +
    theme_full+ scale_fill_manual(values = colList[6:7]) +
    ggtitle(cTab.ttt[i,]$name) + theme(legend.position = "none") +
    
    ylab("Harrell's C-index")
})
plot_grid(plotlist = plotList, ncol =3)
```


# Assemble figure

## Main text figure 
```{r Figure5, fig.width=18, fig.height=13}

leftCol <- plot_grid(cTTT, drugBar, drugCor, ncol =1, rel_heights = c(0.3,0.4,0.4), labels = c("A","C","D"), label_size =22)
rightCol <- plot_grid(hr.prmt5, stat2Bar, ncol =1, rel_heights = c(0.6,0.4), align = "v", labels = c("B","E"),label_size =22)

plot_grid(leftCol, rightCol, ncol=2, rel_widths = c(0.7,0.3))

```

### Supplementary
```{r suppl_multiCox, fig.width=10, fig.height=8}
plot_grid(hr.pygb, hr.pes1, ncol=2)
```

