---
title: "Section 1: Overview of CLL proteomic dataset"
author: "Junyan Lu"
date: "2020-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages and datasets
```{r, message=FALSE, warning=FALSE}
library(limma)
library(DESeq2)
library(cowplot)
library(proDA)
library(pheatmap)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load("../data/patMeta_enc.RData")
load("../data/ddsrna_enc.RData")
load("../data/proteomic_explore_enc.RData")
source("../code/utils.R")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, dev = c("png","pdf"))
```

# Overview of the patient characteristics in our study cohort

## A table shown the clinical characteristics
(all cohorts combined)
```{r}
labelBatch <- c(batch1 = "batch1", batch2 = "batch3", batch3 = "batch2")
patInfo <- sampleTab %>% 
  filter(!lowQuality, !duplicatedPat) %>%
  select(encID, leukCount, cohort,batch) %>%
  mutate(batch = labelBatch[batch]) %>%
  arrange(batch, encID) %>%
  left_join(select(patMeta, Patient.ID, IGHV.status, trisomy12), by = c(encID = "Patient.ID")) %>%
  left_join(select(survT, patID, OS, died, TTT, treatedAfter, TTT, age, sex, pretreat), by = c(encID = "patID")) %>%
  mutate(`No.` = seq(nrow(.))) %>%
  select(No., encID, age, sex, IGHV.status, trisomy12, leukCount, OS, died, TTT, treatedAfter, pretreat, cohort, batch)

patInfoTab <- patInfo %>% #format
  mutate(trisomy12 = ifelse(trisomy12 %in% "1", "yes", "no"),
         died = ifelse(is.na(OS),NA, ifelse(died,"yes","no")),
         treatedAfter = ifelse(is.na(TTT), NA, ifelse(treatedAfter, "yes","no")),
         pretreat = ifelse(pretreat %in% 1, "yes","no"),
         age = as.integer(age),
         OS = formatC(OS, digits=1),
         TTT = formatC(TTT, digits=1)) %>%
  mutate_all(replace_na,"NA") %>%
  #arrange(cohort, encID) %>%
  dplyr::rename(ID = encID,
                IGHV = IGHV.status,
                `WBC count` = leukCount,
                `Survival time (years)` = OS,
                Died = died,
                `Time to treatment (years)` = TTT,
                `Treatment after sampling` = treatedAfter,
                `Treatment before sampling` = pretreat)

patInfoTab %>% DT::datatable()
```


## Prepare summary matrix for genomics

### Get mutations with at least 5 cases
```{r}
geneMat <-  patMeta[match(patInfo$encID, patMeta$Patient.ID),] %>%
  select(-Methylation_Cluster) %>%
  mutate(IGHV.status = ifelse(!is.na(IGHV.status), ifelse(IGHV.status == "M",1,0),NA)) %>%
  mutate(cohort = sampleTab[match(Patient.ID, sampleTab$encID),]$cohort) %>%
  mutate(cohort = ifelse(cohort == "exploration",1,0)) %>%
  mutate_if(is.factor, as.character) %>%
  mutate_at(vars(-Patient.ID), as.numeric) %>% #assign a few unknown mutated cases to wildtype
  data.frame() %>% column_to_rownames("Patient.ID")


geneMat <- geneMat[,apply(geneMat,2, function(x) sum(x %in% 1, na.rm = TRUE))>=5]
```

```{r}
#Remove some dubious annotations
geneMat <- geneMat[,!colnames(geneMat) %in% c("del5IgH","gain2p","IgH_break")]
useGeneForComposition <- colnames(geneMat)

useGeneForComposition <- unique(c(useGeneForComposition,"U1","cohort","IGHV.status"))
geneMat <- geneMat[,useGeneForComposition]
```

### Plot to summarise genomic background

Separate CNV table and mutation table
```{r}
cnvCol <- colnames(geneMat)[grepl("del|trisomy|IGHV|cohort",colnames(geneMat))]
cnvMat <- geneMat[,cnvCol]
mutMat <- geneMat[,!colnames(geneMat) %in% cnvCol]
cnvMat <- cnvMat[,names(sort(colSums(cnvMat == 1,na.rm=TRUE)))]

#Manually assign CNV feature order for better visualization
cnvMat <- cnvMat[,c("del17p","del11q","del13q","trisomy19","trisomy12","IGHV.status","cohort")]


mutMat <- mutMat[,names(sort(colSums(mutMat == 1, na.rm=TRUE)))]
geneMat <- cbind(mutMat,cnvMat)
geneMat[is.na(geneMat)] <- -1
```

```{r}
sortTab <- function(sumTab) {
  i <- ncol(sumTab)
  #print(i)
  if (i == 1) {
    return(rownames(sumTab)[order(sumTab[,i])])
  }
  allLevel <- sort(unique(sumTab[,i]))
  orderRow <- lapply(allLevel, function(n) {
    sortTab(sumTab[sumTab[,i] %in% n, seq(1,i-1), drop = FALSE])  
  }) %>% unlist() %>% c()
  return(orderRow)
}

sortedPat <- rev(sortTab(geneMat))
geneMat <- geneMat[,!colnames(geneMat) %in% c("IGHV.status","cohort")]
```

```{r}

plotTab <- geneMat %>% as_tibble(rownames="patID") %>% mutate_all(as.character) %>%
  pivot_longer(-patID, names_to = "var", values_to = "value") %>%
  mutate(status = case_when(
    value == -1 ~ "NA",
    value == 0 ~ "WT",
    value == 1 & var %in% cnvCol ~ "CNA",
    value == 1 & !var %in% cnvCol ~ "gene mutation"
  )) %>% 
  mutate(var = factor(var, levels = c(colnames(mutMat),colnames(cnvMat))), 
         patID = factor(patID, levels = sortedPat),
         status = factor(status, levels =c("WT","CNA","gene mutation","NA")))

# get number of mutations
sumMutTab <- group_by(plotTab, var) %>%
  summarise(num=sum(value %in% 1))

formatedName <- lapply(levels(plotTab$var), function(n) {
  num <- filter(sumMutTab, var == n)$num
  if(n %in% cnvCol) {
    nameCNV <- c(del17p = "del(17)(p13)", del11q = "del(11)(q22.3)", del13q = "del(13)(q14)")
    if (n %in% names(nameCNV)) {
      n <- nameCNV[n]
    }
    sprintf("%s [%s]",n, num)
  } else {
    bquote(italic(.(n))~"["*.(num)*"]")
  }
})
```

```{r , fig.height=6, fig.width=12}
pMain <- ggplot(plotTab, aes(x=patID, y = var, fill = status)) + 
  geom_tile(color = "grey80") +
  theme_void() + 
  scale_fill_manual(values = c("gene mutation" = colList[5],
                               "CNA"= colList[4],
                               "WT" ="white", 
                               "NA" = "grey80"),
                    name = "aberrations") +
  scale_y_discrete(labels = formatedName) +
  theme(axis.text.x =  element_blank(),
        axis.text.y = element_text(size=11, face = "bold", hjust = 1),
        axis.ticks.length.y = unit(0.05,"npc")) +
  ylab("") + xlab("")
```

## Column annotations for other characteristics

### Cohort 
```{r}
cohortTab <- select(patInfo, encID, cohort) %>%
  mutate(patID = encID, status = cohort, type = "cohort") %>%
  mutate(status = ifelse(status == "exploration","main","additional")) %>%
  mutate(status = factor(status, levels = c("main","additional"))) %>%
  filter(patID %in% sortedPat) %>%
  mutate(patID = factor(patID, levels = sortedPat)) %>%
  select(patID, type, status)

pCohort <- ggplot(cohortTab, aes(x=patID, y = type, fill = status)) + 
  geom_tile(color = NA) +
  theme_void() + xlab("") + ylab("") +
  coord_cartesian(expand = FALSE) +
  scale_fill_manual(values = c(main=colList[3],additional = colList[2]), name = "cohort") +
  theme(axis.text.y = element_text(face = "bold", size=11),
        axis.ticks.length.y = unit(0.05,"npc"))

```


### IGHV status
```{r}
ighvTab <- select(patMeta, Patient.ID, IGHV.status) %>%
  mutate(patID = Patient.ID, status = IGHV.status, type = "IGHV") %>%
  filter(patID %in% sortedPat) %>%
  mutate(patID = factor(patID, levels = sortedPat)) %>%
  select(patID, type, status) %>%
  mutate(status = ifelse(is.na(status),"NA",status)) %>%
  mutate(status = factor(status, levels = c("M","U","NA")))

pIGHV <- ggplot(ighvTab, aes(x=patID, y = type, fill = status)) + 
  geom_tile(color = NA) +
  theme_void() + xlab("") + ylab("") +
  coord_cartesian(expand = FALSE) +
  scale_fill_manual(values = c(M="black",U="white","NA" = "grey80"), name = "IGHV") +
  theme(axis.text.y = element_text(face = "bold", size=11),
        axis.ticks.length.y = unit(0.05,"npc"))

#pIGHV
```

### Sex
```{r}
sexTab <- select(survT, patID, sex) %>%
  mutate(status = as.character(sex), type = "sex") %>%
  filter(patID %in% sortedPat) %>%
  mutate(patID = factor(patID, levels = sortedPat),
         status = case_when(status %in% "m" ~ "male",
                            status %in% "f" ~ "female")) %>%
  select(patID, type, status)

pSex <- ggplot(sexTab, aes(x=patID, y = type, fill = status)) + 
  geom_tile(color = NA) +
  theme_void() + xlab("") +  ylab("") +
  coord_cartesian(expand = FALSE) +
  scale_fill_manual(values = c(male=colList[7],female=colList[5]), name = "sex") +
  theme(axis.text.y = element_text(face = "bold",size=11),
        axis.ticks.length.y = unit(0.05,"npc"))
#pSex
```

### Age
```{r}
agePlotTab <- survT %>% filter(patID %in% sortedPat) %>%
  select(patID, age) %>%
  mutate( status = age, type = "age") %>%
  mutate(patID = factor(patID, levels = sortedPat)) %>%
  select(patID, type, status)


pAge <- ggplot(agePlotTab, aes(x=patID, y = type, fill = status)) + 
  geom_tile(color = NA) +
  theme_void() + xlab("") +  ylab("") +
  coord_cartesian(expand = FALSE) +
  scale_fill_viridis_b(name = "age") +
  theme(axis.text.y = element_text(face = "bold",size=11),
        axis.ticks.length.y = unit(0.05,"npc"))
#pAge
```

## Generate combined summarisation plots

Main plot
```{r cohortSummary_all, fig.height=4, fig.width=6}
lMain <- get_legend(pMain + geom_tile(color = "black") )
lAge <- get_legend(pAge + geom_tile(color = "black") )
lSex <- get_legend(pSex+ geom_tile(color = "black") )
lIGHV <- get_legend(pIGHV+ geom_tile(color = "black") )
lCohort <- get_legend(pCohort + geom_tile(color = "black"))

noLegend <- theme(legend.position = "none")
mainPlot <- plot_grid(pAge + noLegend, pSex + noLegend, 
                      pCohort + noLegend,
                      pIGHV + noLegend,
                     pMain + noLegend, ncol=1, align = "v",
                     rel_heights = c(rep(1,4),18))


legendPlot <- plot_grid(lAge, lSex, lIGHV, lMain,ncol=1, align = "hv")

plot_grid(mainPlot)
```

Figure legend
```{r cohortSummary_legend_all, fig.height=4, fig.width=5}


legendPlot <- plot_grid(lAge, lSex, lCohort, lIGHV, lMain,nrow=1, align = "hv")

plot_grid(legendPlot, ncol=1, align = "hv")
```





# Number of detected proteins

## Proteins identified in at least in 50% of the samples
```{r}
dim(protCLL)
```


# RNA-protein associations

## Preprocess transcriptomic and proteomic data

Preprocessing RNA sequencing data
```{r, cache=TRUE, autodep=TRUE}
dds <- estimateSizeFactors(dds)
sampleOverlap <- intersect(colnames(protCLL), colnames(dds))
geneOverlap <- intersect(rowData(protCLL)$ensembl_gene_id, rownames(dds))
ddsSub <- dds[geneOverlap, sampleOverlap]
protSub <- protCLL[match(geneOverlap, rowData(protCLL)$ensembl_gene_id), sampleOverlap]

#how many gene don't have RNA expression at all?
noExp <- rowSums(counts(ddsSub)) == 0

#remove those genes in both datasets
ddsSub <- ddsSub[!noExp,]
protSub <- protSub[!noExp,]

protSub <- protSub[!duplicated(rowData(protSub)$name)]

geneOverlap <- intersect(rowData(protSub)$ensembl_gene_id, rownames(ddsSub))

ddsSub.vst <- varianceStabilizingTransformation(ddsSub)
```

## Calculate correlations between protein abundance and RNA expression

```{r, warning=FALSE, cache=TRUE, autodep=TRUE}
rnaMat <- assay(ddsSub.vst)
proMat <- assays(protSub)[["count_combat"]]
rownames(proMat) <- rowData(protSub)$ensembl_gene_id

corTab <- lapply(geneOverlap, function(n) {
  rna <- rnaMat[n,]
  pro.raw <- proMat[n,]
  res.raw <- cor.test(rna, pro.raw, use = "pairwise.complete.obs")
  tibble(id = n,
         p = res.raw$p.value,
         coef = res.raw$estimate)
}) %>% bind_rows() %>%
  arrange(desc(coef)) %>% mutate(p.adj = p.adjust(p, method = "BH"),
                                 symbol = rowData(dds[id,])$symbol,
                                 chr = rowData(dds[id,])$chromosome)
```

## Plot the distribution 
```{r overall_protein_rna_correlation, fig.width=6.2, fig.height=4}
corHistPlot <- ggplot(corTab, aes(x=coef)) + geom_histogram(position = "identity", col = colList[2], alpha =0.3, bins =50) +
  geom_vline(xintercept = 0, col = colList[1], linetype = "dashed") + xlim(-0.7,1) +
      xlab("Pearson's correlation coefficient") + theme_half +
  ggtitle("Correlation between mRNA and protein expression") +
  theme(axis.text = element_text(size=18), axis.title = element_text(size=18))
corHistPlot
```

Median Pearson's correlation coefficient
```{r}
median(corTab$coef)
```

## Influence of overall protein/RNA abundance on correlation
```{r corProtRNAabundance, fig.height=8, fig.width=16}
medProt <- rowMedians(proMat,na.rm = T)
names(medProt) <- rownames(proMat)
medRNA <- rowMedians(rnaMat, na.rm = T)
names(medRNA) <- rownames(rnaMat)

plotTab <- corTab %>% mutate(rnaAbundance = medRNA[id], protAbundance = medProt[id])
plotList <- list()

plotList[["rna"]] <- plotCorScatter(plotTab,"coef","rnaAbundance",
                                    showR2 = FALSE, annoPos = "left",
                                    x_lab ="Correlation coefficient",
                                    y_lab = "Median RNA expression",
                                    title = "", dotCol = colList[5], textCol = colList[1])

plotList[["protein"]] <- plotCorScatter(plotTab,"coef","protAbundance",
                                    showR2 = FALSE, annoPos = "left",
                                    x_lab ="Correlation coefficient",
                                    y_lab = "Median protein expression",
                                    title = "", dotCol = colList[6], textCol = colList[1])

cowplot::plot_grid(plotlist = plotList, ncol =2)
```


## Plot protein-RNA correlation for selected genes

### Good correlations
```{r RNA_protein_cor_markers, fig.height=4, fig.width=12}
geneList <- c("ZAP70","CD22","CD79A")
plotList <- lapply(geneList, function(n) {
  geneId <- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab <- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef <- cor(plotTab$x, plotTab$y, use="pairwise.complete")
  annoPos <- ifelse (coef > 0, "left","right")
  plotCorScatter(plotTab, "x","y", showR2 = FALSE, annoPos = annoPos, x_lab = "RNA expression", shape = "IGHV",
                 y_lab ="Protein expression", title = n,dotCol = colList[4], textCol = colList[1], legendPos="none")
})
goodCorPlot <- cowplot::plot_grid(plotlist = plotList, ncol =3)
goodCorPlot
```

### Bad correlations
```{r RNA_protein_cor_markers_bad, fig.height=4, fig.width=8}
geneList <- c("DTNBP1","PRPF19")
plotList <- lapply(geneList, function(n) {
  geneId <- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab <- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef <- cor(plotTab$x, plotTab$y, use="pairwise.complete")
  annoPos <- ifelse (coef > 0, "left","right")
  plotCorScatter(plotTab, "x","y", showR2 = FALSE, annoPos = annoPos, x_lab = "RNA expression",
                 y_lab ="Protein expression", title = n,dotCol = colList[4], textCol = colList[1],
                 shape = "IGHV", legendPos = "none")
})
badCorPlot <- cowplot::plot_grid(plotlist = plotList, ncol =2)
badCorPlot
```

# Principal component analysis

## Calculate PCA

We will perform principal component analysis to identify and annotate the major dimensions in our dataset.

```{r,fig.width=6, fig.height=5}
#remove genes on sex chromosomes
protCLL.sub <- protCLL[!rowData(protCLL)$chromosome_name %in% c("X","Y"),]
plotMat <- assays(protCLL.sub)[["QRILC_combat"]]
sds <- genefilter::rowSds(plotMat)
plotMat <- as.matrix(plotMat[order(sds,decreasing = TRUE),])
colAnno <- colData(protCLL)[,c("gender","IGHV.status","trisomy12")] %>%
  data.frame()
colAnno$trisomy12 <- ifelse(colAnno$trisomy12 %in% 1, "yes","no")

pcOut <- prcomp(t(plotMat), center =TRUE, scale. = TRUE)
pcRes <- pcOut$x
eigs <- pcOut$sdev^2
varExp <- structure(eigs/sum(eigs),names = colnames(pcRes))
```
All proteins are included for PCA analysis

## Correlation test between PCs and IGHV.status/trisomy12
```{r}
corTab <- lapply(colnames(pcRes),  function(pc) {
  ighvCor <- t.test(pcRes[,pc] ~ colAnno$IGHV.status, var.equal=TRUE)
  tri12Cor <- t.test(pcRes[,pc] ~ colAnno$trisomy12, var.equal=TRUE)
  tibble(PC = pc, 
         feature=c("IGHV", "trisomy12"),
         p = c(ighvCor$p.value, tri12Cor$p.value))
}) %>% bind_rows() %>% mutate(p.adj = p.adjust(p)) %>%
  filter(p <= 0.05) %>% arrange(p)
corTab
```
The first three components are shown to be correlated with trisomy12 and IGHV status.

## Plot PC1 and PC2
```{r plot_PC1_PC2,fig.height=7, fig.width=6}
plotTab <- pcRes %>% data.frame() %>% cbind(colAnno[rownames(.),]) %>%
  rownames_to_column("patID") %>% as_tibble()

plotPCA12 <- ggplot(plotTab, aes(x=PC1, y=PC2, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf("PC1 (%1.2f%%)",varExp[["PC1"]]*100)) +
  ylab(sprintf("PC2 (%1.2f%%)",varExp[["PC2"]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  geom_hline(yintercept = 0, linetype ="dashed", color = "grey50") +
  geom_vline(xintercept = 0, linetype ="dashed", color = "grey50") +
  theme_full + theme(legend.position = "bottom", legend.text = element_text(size =15), legend.title = element_text(size=15))

plotPCA12
ggsave("plot_PC1_PC2.pdf", height = 4, width = 5.5)
```

## Plot PC3 and PC4
```{r plot_PC2_PC4,fig.height=7, fig.width=6}
plotPCA34 <- ggplot(plotTab, aes(x=PC3, y=PC4, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf("PC3 (%1.2f%%)",varExp[["PC3"]]*100)) +
  ylab(sprintf("PC4 (%1.2f%%)",varExp[["PC4"]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  geom_hline(yintercept = 0, linetype ="dashed", color = "grey50") +
  geom_vline(xintercept = 0, linetype ="dashed", color = "grey50") +
  xlim(-60,60) + ylim(-60,60) +
  theme_full + theme(legend.position = "bottom", legend.text = element_text(size =15), legend.title = element_text(size=15))

plotPCA34
```

## Plot PC1 and PC3
```{r plot_PC1_PC5,fig.height=7, fig.width=6}
plotTab <- pcRes %>% data.frame() %>% cbind(colAnno[rownames(.),]) %>%
  rownames_to_column("patID") %>% as_tibble()

plotPCA13 <- ggplot(plotTab, aes(x=PC1, y=PC3, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf("PC1 (%1.2f%%)",varExp[["PC1"]]*100)) +
  ylab(sprintf("PC3 (%1.2f%%)",varExp[["PC3"]]*100)) +
  scale_color_manual(values = colList) +
  geom_hline(yintercept = 0, linetype ="dashed", color = "grey50") +
  geom_vline(xintercept = 0, linetype ="dashed", color = "grey50") +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  theme_full + theme(legend.position = "bottom", legend.text = element_text(size =15), legend.title = element_text(size=15))
plotPCA13
```


## Pathway enrichment on PC1 and PC2

We will use gene set enrichment analysis to characterize the principal components (PCs) on pathway level.

### PC1
```{r }
enRes <- list()
gmts = list(H= "../data/gmts/h.all.v6.2.symbols.gmt",
            KEGG = "../data/gmts/c2.cp.kegg.v6.2.symbols.gmt",
            C6 = "../data/gmts/c6.all.v6.2.symbols.gmt")

proMat <- assays(protCLL.sub)[["QRILC_combat"]]
iPC <- "PC1"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.05) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC1"]] <- runGSEA(inputTab, gmts$H, "page")
```

### PC2
```{r}
proMat <- assays(protCLL.sub)[["QRILC_combat"]]
iPC <- "PC2"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.05) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC2"]] <- runGSEA(inputTab, gmts$H, "page")
```


### PC3
```{r}
proMat <- assays(protCLL.sub)[["QRILC_combat"]]
iPC <- "PC3"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.05) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC3"]] <- runGSEA(inputTab, gmts$H, "page")
```


```{r enrich_PC1PC2, fig.height=10, fig.width=12}
cowplot::plot_grid(plotEnrichmentBar(enRes[[1]], ifFDR = TRUE, pCut = 0.05, setName = "",title = "Proteins associated with PC1", removePrefix = "HALLMARK_", setMap = setMap),
                   plotEnrichmentBar(enRes[[2]], ifFDR = TRUE, pCut = 0.05, setName = "", title = "Proteins associated with PC2", removePrefix = "HALLMARK_", setMap = setMap),
                   plotEnrichmentBar(enRes[[3]], ifFDR = TRUE, pCut = 0.05, setName = "", title = "Proteins associated with PC3", removePrefix = "HALLMARK_", setMap = setMap),
                   ncol=1,
                   align = "hv",
                   rel_heights = c(9,3,6))
```


# Hierarchical clustering

## Heatmap of expression of top 1000 most variant proteins in all samples
```{r HC_heatmap_origin, fig.height=10, fig.width=10}
protCLL.sub <- protCLL[!rowData(protCLL)$chromosome_name %in% c("X","Y"),]
plotMat <- assays(protCLL.sub)[["QRILC_combat"]]
sds <- rowSds(plotMat) 
plotMat <- plotMat[order(sds, decreasing = T)[1:1000],]
colAnno <- colData(protCLL)[,c("IGHV.status","trisomy12")] %>%
  data.frame()
colAnno$trisomy12 <- ifelse(colAnno$trisomy12 %in% 1, "yes","no")

plotMat <- mscale(plotMat, center = TRUE, scale = TRUE)

annoCol <- list(trisomy12 = c(yes = "black",no = "grey80"),
                IGHV.status = c(M = colList[3], U = colList[4]))
pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = "none",
                   clustering_method = "average", clustering_distance_cols = "correlation",
                   color = colorRampPalette(c(colList[2],"white",colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = FALSE, show_colnames = FALSE,
                   treeheight_row = 0)
```

## Heatmap of sample-sample correlation matrix based on protein expression
```{r, fig.height=4, fig.width=6}
callback = function(hc, mat){
   sv = svd(t(mat))$v[,1]
   dend = reorder(as.dendrogram(hc), wts = sv)
   as.hclust(dend)
}

breaks <- c(seq(-1,-0.39, length.out=20), seq(-0.4,0.4, length.out=40), seq(0.41, 1, length.out =20))
colorList <- c(rep(colList[2],20), colorRampPalette(c(colList[2],"white",colList[1]))(40), rep(colList[1], 20))
#colList <- colorRampPalette(c(colList[2],"white",colList[1]))(100)
corMat <- cor(plotMat)
#corMat <- mscale(corMat, center=FALSE, scale=FALSE, censor = 0.5)
pheatmap(corMat, annotation_col = colAnno, clustering_method = "ward.D2", annotation_colors = annoCol,
         color =  colorList, border_color = NA,
           breaks = breaks,show_rownames = FALSE, show_colnames = FALSE,
          treeheight_col = 0, treeheight_row = 20)
```


