---
title: "Section 2: Identifying proteins associated disease drivers in CLL"
author: "Junyan Lu"
date: "2020-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages and datasets
```{r, echo=TRUE, message=FALSE, warning=FALSE}
library(limma)
library(DESeq2)
library(proDA)
library(IHW)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load("../data/patMeta_enc.RData")
load("../data/ddsrna_enc.RData")
load("../data/proteomic_LUMOS_enc.RData")
source("../code/utils.R")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE,dev = c("png","pdf"))
```


# Preprocessing

## Process proteomics data
```{r}
protCLL <- protCLL[rowData(protCLL)$uniqueMap,]
protMat <- assays(protCLL)[["count"]] #without imputation
```

## Prepare genomic background

### Get mutations with at least 5 cases
```{r}
geneMat <-  patMeta[match(colnames(protMat), patMeta$Patient.ID),] %>%
  select(Patient.ID, IGHV.status, del11p:U1) %>%
  mutate_if(is.factor, as.character) %>% mutate(IGHV.status = ifelse(IGHV.status == "M", 1,0)) %>%
  mutate_at(vars(-Patient.ID), as.numeric) %>% #assign a few unknown mutated cases to wildtype
  data.frame() %>% column_to_rownames("Patient.ID")


geneMat <- geneMat[,apply(geneMat,2, function(x) sum(x %in% 1, na.rm = TRUE))>=5]
```

**Mutations that will be tested**
```{r}
colnames(geneMat)
```

### Plot to summarise genomic background
```{r}
sortTab <- function(sumTab) {
  i <- ncol(sumTab)
  #print(i)
  if (i == 1) {
    #print(rownames(sumTab)[order(sumTab[,i])])
    return(rownames(sumTab)[order(sumTab[,i])])
  }
  orderRow <- c(sortTab(sumTab[sumTab[,i]==0, seq(1,i-1), drop = FALSE]), sortTab(sumTab[sumTab[,i]==1, seq(1,i-1), drop = FALSE]))
  return(orderRow)
}

geneMat.sort <- geneMat
geneMat.sort[is.na(geneMat.sort)] <- 0
sortedGene <- names(sort(colSums(geneMat.sort)))
geneMat.sort <- geneMat.sort[,sortedGene]
sortedPat <- sortTab(geneMat.sort)
```

```{r geneSummary, fig.height=6, fig.width=8}
plotTab <- geneMat %>% rownames_to_column("patID") %>% mutate_all(as.character) %>%
  gather(key = "gene", value = "status", -patID) %>%
  mutate(gene = factor(gene, levels = sortedGene), patID = factor(patID, levels = sortedPat)) %>%
  mutate(status = ifelse(is.na(status),"NA",status))

ggplot(plotTab, aes(x=patID, y = gene, fill = status)) + 
  geom_tile(color = "black") +
  theme_minimal() + 
  scale_fill_manual(values = c("1" = colList[4],"0" ="white", "NA" = "grey80")) +
  theme(axis.text.x = element_text(angle = 90, hjust =1, vjust=0.5),
        panel.grid = element_blank(), legend.position = "none") +
  ylab("") + xlab("Patient ID")
```

# Differential protein expression using proDA

## For IGHV and trisomy12

Fit the probailistic dropout model
```{r, eval=FALSE}
designMat <- geneMat[  ,c("IGHV.status","trisomy12")]
fit <- proDA(protMat, design = ~ .,
             col_data = designMat)
```

Test for differentially expressed proteins
```{r, eval=FALSE}
resList.ighvTri12 <- lapply(c("IGHV.status","trisomy12"), function(n) {
  contra <- n
  resTab <- test_diff(fit, contra) %>%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %>% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
    select(name, id, logFC, t, P.Value, adj.P.Val) %>%  
    arrange(P.Value) %>% mutate(Gene = n) %>%
    as_tibble()
}) %>% bind_rows()
```

## Test for other variantions (blocking for IGHV and trisomy12)
Fit the probailistic dropout model and test for differentially expressed proteins
```{r, eval=FALSE}
otherGenes <- colnames(geneMat)[!colnames(geneMat)%in% c("IGHV.status","trisomy12")]
resList <- lapply(otherGenes, function(n) {
  designMat <- geneMat[,c("IGHV.status","trisomy12",n)]
  designMat <- designMat[!is.na(designMat[[n]]),]
  testMat <- protMat[,rownames(designMat)]
  
  fit <- proDA(testMat, design = ~ .,
             col_data = designMat)
  
  contra <- n
  resTab <- test_diff(fit, contra) %>%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %>% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
    select(name, id, logFC, t, P.Value, adj.P.Val) %>%  
    arrange(P.Value) %>% mutate(Gene = n) %>%
    as_tibble()
  resTab
}) %>% bind_rows()
```

Combine the results
```{r, eval=FALSE}
resList <- bind_rows(resList.ighvTri12, resList)

#Adjusting p values

#using BH
resList <- mutate(resList, adj.P.global = p.adjust(P.Value, method = "BH"))

#using IHW
ihwRes <- ihw(P.Value ~ factor(Gene), data= resList, alpha=0.1)
resList <- mutate(resList, adj.P.IHW = adj_pvalues(ihwRes))

```

Save the results for re-using
```{r, eval=FALSE}
save(resList, file = "../output/deResList.RData")
```

Load the pre-calculated results (differential expression tests take long time.)
```{r}
load("../output/deResList.RData")
```

# Summarise significant associations

## Bar plot of number of significant associations (10% FDR)

```{r}
plotTab <- resList %>% group_by(Gene) %>%
  summarise(nFDR.local = sum(adj.P.Val <= 0.1),
            nFDR.global = sum(adj.P.global <= 0.1),
            nFDR.IHW = sum(adj.P.IHW <= 0.1),
            nP = sum(P.Value < 0.05))

#use IHW for adjusting p-values
resList <- mutate(resList, adj.P.Val = adj.P.IHW)
```

```{r number_of_association, fig.height=5, fig.width=8}
plotTab <- arrange(plotTab, desc(nFDR.IHW)) %>% mutate(Gene = factor(Gene, levels = Gene))
ggplot(plotTab, aes(x=Gene, y = nFDR.IHW)) + geom_bar(stat="identity",fill=colList[2]) + 
  geom_text(aes(label = paste0("n=", nFDR.IHW)),vjust=-1,col=colList[1]) + ylim(0,1200) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab("Number of associations\n(10% FDR)") + xlab("")
```

## Bar plot of number of raw P-value < 0.05 

```{r number_of_association_pval, fig.height=5, fig.width=8}
plotTab <- arrange(plotTab, desc(nP)) %>% mutate(Gene = factor(Gene, levels = Gene))
ggplot(plotTab, aes(x=Gene, y = nP)) + geom_bar(stat="identity",fill=colList[2]) + 
  geom_text(aes(label = paste0("n=", nP)),vjust=-1,col=colList[1]) + ylim(0,1500) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab("Number of associations\n(nominal P-value < 0.05)") + xlab("")
```

# Association test in timsTOF data

## For IGHV and trisomy12

Load timsTOF data
```{r}
load("../data/proteomic_timsTOF_enc.RData")
protMat <- assays(protCLL)[["count"]] #without imputation
```

Genetic data
```{r}
geneMat <-  patMeta[match(colnames(protMat), patMeta$Patient.ID),] %>%
  select(Patient.ID, IGHV.status, trisomy12, SF3B1, trisomy19, del11q) %>%
  mutate_if(is.factor, as.character) %>% mutate(IGHV.status = ifelse(IGHV.status == "M", 1,0)) %>%
  mutate_at(vars(-Patient.ID), as.numeric) %>% #assign a few unknown mutated cases to wildtype
  data.frame() %>% column_to_rownames("Patient.ID")
```

Fit the probailistic dropout model
```{r, eval=FALSE}
designMat <- geneMat[  ,c("IGHV.status","trisomy12")]
fit <- proDA(protMat, design = ~ .,
             col_data = designMat)
```

Test for differentially expressed proteins
```{r, eval=FALSE}
resList.ighvTri12 <- lapply(c("IGHV.status","trisomy12"), function(n) {
  contra <- n
  resTab <- test_diff(fit, contra) %>%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %>% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
    select(name, id, logFC, t, P.Value, adj.P.Val) %>%  
    arrange(P.Value) %>% mutate(Gene = n) %>%
    as_tibble()
}) %>% bind_rows()
```

## Test for other variantions (blocking for IGHV and trisomy12)
Fit the probailistic dropout model and test for differentially expressed proteins
```{r, eval=FALSE}
otherGenes <- colnames(geneMat)[!colnames(geneMat)%in% c("IGHV.status","trisomy12")]
resList <- lapply(otherGenes, function(n) {
  designMat <- geneMat[,c("IGHV.status","trisomy12",n)]
  designMat <- designMat[!is.na(designMat[[n]]),]
  testMat <- protMat[,rownames(designMat)]
  
  fit <- proDA(testMat, design = ~ .,
             col_data = designMat)
  
  contra <- n
  resTab <- test_diff(fit, contra) %>%
    dplyr::rename(id = name, logFC = diff, t=t_statistic,
                  P.Value = pval, adj.P.Val = adj_pval) %>% 
    mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>%
    select(name, id, logFC, t, P.Value, adj.P.Val) %>%  
    arrange(P.Value) %>% mutate(Gene = n) %>%
    as_tibble()
  resTab
}) %>% bind_rows()
```

Combine the results
```{r, eval=FALSE}
resList <- bind_rows(resList.ighvTri12, resList)

#Adjusting p values

#using BH
resList <- mutate(resList, adj.P.global = p.adjust(P.Value, method = "BH"))

#using IHW
ihwRes <- ihw(P.Value ~ factor(Gene), data= resList, alpha=0.1)
resList <- mutate(resList, adj.P.IHW = adj_pvalues(ihwRes))

```

Save the results for re-using
```{r, eval=FALSE}
save(resList, file = "../output/deResList_timsTOF.RData")
```
