---
title: "Section 1: Overview of CLL proteomic dataset"
author: "Junyan Lu"
date: "2020-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages and datasets
```{r, message=FALSE, warning=FALSE}
library(limma)
library(DESeq2)
library(cowplot)
library(proDA)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load("../data/patMeta_enc.RData")
load("../data/ddsrna_enc.RData")
load("../data/proteomic_LUMOS_enc.RData")
source("../code/utils.R")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, dev = c("png","pdf"))
```


# RNA-protein associations

## Preprocess transcriptomic and proteomic data

```{r, cache=TRUE, autodep=TRUE}
protCLL <- protCLL[rowData(protCLL)$uniqueMap,]

colnames(dds) <- dds$PatID
dds <- estimateSizeFactors(dds)
sampleOverlap <- intersect(colnames(protCLL), colnames(dds))
geneOverlap <- intersect(rowData(protCLL)$ensembl_gene_id, rownames(dds))
ddsSub <- dds[geneOverlap, sampleOverlap]
protSub <- protCLL[match(geneOverlap, rowData(protCLL)$ensembl_gene_id), sampleOverlap]

#how many gene don't have RNA expression at all?
noExp <- rowSums(counts(ddsSub)) == 0

#remove those genes in both datasets
ddsSub <- ddsSub[!noExp,]
protSub <- protSub[!noExp,]

#remove proteins with duplicated identifiers
protSub <- protSub[!duplicated(rowData(protSub)$name)]

geneOverlap <- intersect(rowData(protSub)$ensembl_gene_id, rownames(ddsSub))

ddsSub.vst <- varianceStabilizingTransformation(ddsSub)
```

## Calculate correlations between protein abundance and RNA expression
```{r, warning=FALSE, cache=TRUE, autodep=TRUE}
rnaMat <- assay(ddsSub.vst)
proMat <- assays(protSub)[["count"]]
rownames(proMat) <- rowData(protSub)$ensembl_gene_id

corTab <- lapply(geneOverlap, function(n) {
  rna <- rnaMat[n,]
  pro.raw <- proMat[n,]
  res.raw <- cor.test(rna, pro.raw, use = "pairwise.complete.obs")
  tibble(id = n,
         p = res.raw$p.value,
         coef = res.raw$estimate)
}) %>% bind_rows() %>%
  arrange(desc(coef)) %>% mutate(p.adj = p.adjust(p, method = "BH"),
                                 symbol = rowData(dds[id,])$symbol,
                                 chr = rowData(dds[id,])$chromosome)
```

## Plot the distribution 
```{r overall_protein_rna_correlation, fig.width=5, fig.height=3}
corHistPlot <- ggplot(corTab, aes(x=coef)) + geom_histogram(position = "identity", col = colList[2], alpha =0.3, bins =50) +
  geom_vline(xintercept = 0, col = colList[1], linetype = "dashed") + xlim(-0.7,1) +
      xlab("Pearson's correlation coefficient") + theme_half +
  ggtitle("Correlation between mRNA and protein expression")
corHistPlot
```

Median Pearson's correlation coefficient
```{r}
median(corTab$coef)
```

## Influence of overall protein/RNA abundance on correlation
```{r corProtRNAabundance, fig.height=8, fig.width=16}
medProt <- rowMedians(proMat,na.rm = T)
names(medProt) <- rownames(proMat)
medRNA <- rowMedians(rnaMat, na.rm = T)
names(medRNA) <- rownames(rnaMat)

plotTab <- corTab %>% mutate(rnaAbundance = medRNA[id], protAbundance = medProt[id])
plotList <- list()

plotList[["rna"]] <- plotCorScatter(plotTab,"coef","rnaAbundance",
                                    showR2 = FALSE, annoPos = "left",
                                    x_lab ="Correlation coefficient",
                                    y_lab = "Median RNA expression",
                                    title = "", dotCol = colList[5], textCol = colList[1])

plotList[["protein"]] <- plotCorScatter(plotTab,"coef","protAbundance",
                                    showR2 = FALSE, annoPos = "left",
                                    x_lab ="Correlation coefficient",
                                    y_lab = "Median protein expression",
                                    title = "", dotCol = colList[6], textCol = colList[1])

cowplot::plot_grid(plotlist = plotList, ncol =2)
```


## Plot protein RNA correlation for selected genes

### Good correlations
```{r RNA_protein_cor_markers, fig.height=8, fig.width=8}
geneList <- c("ZAP70","CD22","CD38","CD79A")
plotList <- lapply(geneList, function(n) {
  geneId <- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab <- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef <- cor(plotTab$x, plotTab$y, use="pairwise.complete")
  annoPos <- ifelse (coef > 0, "left","right")
  plotCorScatter(plotTab, "x","y", showR2 = FALSE, annoPos = annoPos, x_lab = "RNA expression", shape = "IGHV",
                 y_lab ="Protein expression", title = n,dotCol = colList[4], textCol = colList[1], legendPos="none")
})
goodCorPlot <- cowplot::plot_grid(plotlist = plotList, ncol =2)
goodCorPlot
```

### Bad correlations
```{r RNA_protein_cor_markers_bad, fig.height=4, fig.width=8}
geneList <- c("DTNBP1","PRPF19")
plotList <- lapply(geneList, function(n) {
  geneId <- rownames(dds)[match(n, rowData(dds)$symbol)]
  stopifnot(length(geneId) ==1)
  plotTab <- tibble(x=rnaMat[geneId,],y=proMat[geneId,], IGHV=protSub$IGHV.status)
  coef <- cor(plotTab$x, plotTab$y, use="pairwise.complete")
  annoPos <- ifelse (coef > 0, "left","right")
  plotCorScatter(plotTab, "x","y", showR2 = FALSE, annoPos = annoPos, x_lab = "RNA expression",
                 y_lab ="Protein expression", title = n,dotCol = colList[4], textCol = colList[1],
                 shape = "IGHV", legendPos = "none")
})
badCorPlot <- cowplot::plot_grid(plotlist = plotList, ncol =2)
badCorPlot
```

# Principal component analysis

## Calculate PCA

```{r,fig.width=6, fig.height=5}
#remove genes on sex chromosomes
protCLL.sub <- protCLL[!rowData(protCLL)$chromosome_name %in% c("X","Y"),]
plotMat <- assays(protCLL.sub)[["QRILC"]]
sds <- genefilter::rowSds(plotMat)
plotMat <- as.matrix(plotMat[order(sds,decreasing = TRUE),])
colAnno <- colData(protCLL)[,c("gender","IGHV.status","trisomy12")] %>%
  data.frame()
colAnno$trisomy12 <- ifelse(colAnno$trisomy12 %in% 1, "yes","no")

pcOut <- prcomp(t(plotMat), center =TRUE, scale. = TRUE)
pcRes <- pcOut$x
eigs <- pcOut$sdev^2
varExp <- structure(eigs/sum(eigs),names = colnames(pcRes))
```
All proteins are included for PCA analysis

## Plot PC1 and PC2
```{r plot_PC1_PC2,fig.height=5, fig.width=6}
plotTab <- pcRes %>% data.frame() %>% cbind(colAnno[rownames(.),]) %>%
  rownames_to_column("patID") %>% as_tibble()

plotPCA12 <- ggplot(plotTab, aes(x=PC1, y=PC2, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf("PC1 (%1.2f%%)",varExp[["PC1"]]*100)) +
  ylab(sprintf("PC2 (%1.2f%%)",varExp[["PC2"]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  theme_full + theme(legend.position = "none")

plotPCA12
```

## Plot PC3 and PC4
```{r plot_PC2_PC4,fig.height=5, fig.width=6}
plotPCA34 <- ggplot(plotTab, aes(x=PC3, y=PC4, col = trisomy12, shape = IGHV.status)) + geom_point(size=4) +
  xlab(sprintf("PC3 (%1.2f%%)",varExp[["PC3"]]*100)) +
  ylab(sprintf("PC4 (%1.2f%%)",varExp[["PC4"]]*100)) +
  scale_color_manual(values = colList) +
  scale_shape_manual(values = c(M = 16, U =1)) +
  xlim(-60,60) + ylim(-60,60) +
  theme_full
plotPCA34
```

## Correlation test between PCs and IGHV.status
```{r}
corTab <- lapply(colnames(pcRes),  function(pc) {
  ighvCor <- t.test(pcRes[,pc] ~ colAnno$IGHV.status, var.equal=TRUE)
  tri12Cor <- t.test(pcRes[,pc] ~ colAnno$trisomy12, var.equal=TRUE)
  tibble(PC = pc, 
         feature=c("IGHV", "trisomy12"),
         p = c(ighvCor$p.value, tri12Cor$p.value))
}) %>% bind_rows() %>% mutate(p.adj = p.adjust(p)) %>%
  filter(p <= 0.05) %>% arrange(p)
corTab
```


## Pathway enrichment on PC1 and PC2

PC1
```{r }
enRes <- list()
gmts = list(H= "../data/gmts/h.all.v6.2.symbols.gmt",
            KEGG = "../data/gmts/c2.cp.kegg.v6.2.symbols.gmt",
            C6 = "../data/gmts/c6.all.v6.2.symbols.gmt")

proMat <- assays(protCLL.sub)[["QRILC"]]
iPC <- "PC1"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.1) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC1"]] <- runGSEA(inputTab, gmts$H, "page")
```

PC2
```{r}
proMat <- assays(protCLL.sub)[["QRILC"]]
iPC <- "PC2"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.1) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC2"]] <- runGSEA(inputTab, gmts$H, "page")
```

```{r enrich_PC1PC2, fig.height=10, fig.width=12}
cowplot::plot_grid(plotEnrichmentBar(enRes[[1]], ifFDR = TRUE, pCut = 0.05, setName = "",title = "Proteins associated with PC1"),
                   plotEnrichmentBar(enRes[[2]], ifFDR = TRUE, pCut = 0.05, setName = "", title = "Proteins associated with PC2"),ncol=1)
```

## Pathway enrichment on PC1 and PC2

PC3
```{r }
enRes <- list()
gmts = list(H= "../data/gmts/h.all.v6.2.symbols.gmt",
            KEGG = "../data/gmts/c2.cp.kegg.v6.2.symbols.gmt",
            C6 = "../data/gmts/c6.all.v6.2.symbols.gmt")

proMat <- assays(protCLL.sub)[["QRILC"]]
iPC <- "PC3"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.1) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC3"]] <- runGSEA(inputTab, gmts$H, "page")
```

PC4
```{r}
proMat <- assays(protCLL.sub)[["QRILC"]]
iPC <- "PC4"
pc <- pcRes[,iPC][colnames(proMat)]
designMat <- model.matrix(~1+pc)
fit <- limma::lmFit(proMat, designMat)
fit2 <- eBayes(fit)
corRes <- topTable(fit2, "pc", number = Inf) %>%
  data.frame() %>% rownames_to_column("id")

inputTab <- corRes %>% filter(adj.P.Val < 0.1) %>%
  mutate(name = rowData(protCLL[id,])$hgnc_symbol) %>% filter(!is.na(name)) %>%
  distinct(name, .keep_all = TRUE) %>%
  select(name, t) %>% data.frame() %>% column_to_rownames("name")

enRes[["Proteins associated with PC4"]] <- runGSEA(inputTab, gmts$H, "page")
```

```{r enrich_PC3PC4, fig.height=10, fig.width=12}
cowplot::plot_grid(plotEnrichmentBar(enRes[[1]], ifFDR = TRUE, pCut = 0.05, setName = "",title = "Proteins associated with PC3", removePrefix = "HALLMARK_"),
                   plotEnrichmentBar(enRes[[2]], ifFDR = TRUE, pCut = 0.05, setName = "", title = "Proteins associated with PC4",removePrefix = "HALLMARK_"),ncol=1)
```


# Hierarchical clustering

```{r HC_heatmap_origin, fig.height=10, fig.width=10}
protCLL.sub <- protCLL[!rowData(protCLL)$chromosome_name %in% c("X","Y"),]
plotMat <- assays(protCLL.sub)[["QRILC"]]
colAnno <- colData(protCLL)[,c("IGHV.status","trisomy12")] %>%
  data.frame()
colAnno$trisomy12 <- ifelse(colAnno$trisomy12 %in% 1, "yes","no")

plotMat <- mscale(plotMat, center = 5)

annoCol <- list(trisomy12 = c(yes = "black",no = "grey80"),
                IGHV.status = c(M = colList[3], U = colList[4]))
pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = "none",
                   clustering_method = "ward.D2",
                   color = colorRampPalette(c(colList[2],"white",colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = FALSE, show_colnames = FALSE,
                   treeheight_row = 0)
```

# Summarise significant associations

## Bar plot of number of significant associations (10% FDR)

Load the list of differentially expression proteins generated by Section 2
```{r}
load("../output/deResList.RData")
```

```{r}
plotTab <- resList %>% group_by(Gene) %>%
  summarise(nFDR.local = sum(adj.P.Val <= 0.1),
            nFDR.global = sum(adj.P.global <= 0.1),
            nFDR.IHW = sum(adj.P.IHW <= 0.1),
            nP = sum(P.Value < 0.05))

#use IHW for adjusting p-values
resList <- mutate(resList, adj.P.Val = adj.P.IHW)
```

```{r number_of_association, fig.height=5, fig.width=8}
plotTab <- arrange(plotTab, desc(nFDR.IHW)) %>% mutate(Gene = factor(Gene, levels = Gene))
numCorBar <- ggplot(plotTab, aes(x=Gene, y = nFDR.IHW)) + geom_bar(stat="identity",fill=colList[2]) + 
  geom_text(aes(label = paste0("n=", nFDR.IHW)),vjust=-1,col=colList[1]) + ylim(0,1200) +
  theme_half + theme(axis.text.x = element_text(angle = 90, hjust=1, vjust=0.5)) +
  ylab("Number of associations\n(10% FDR)") + xlab("")

numCorBar
```

# Assemble figure

```{r Figure1,fig.height=15, fig.width=18}
designPlot <- draw_image("../data/Fig1A.png")
p <- ggdraw() + designPlot

leftCol <- plot_grid(p, 
                     plot_grid(plotPCA12, plotPCA34, ncol=2, rel_widths = c(.45,0.55)),
                     plot_grid(numCorBar,NULL,rel_widths = c(0.8,0.2),ncol=2), ncol = 1,
                     labels = c("A","E","F"), label_size = 20, vjust = c(1.5, 0,-0.1),
                     rel_heights = c(1.2,1,1))


rightCol <- plot_grid(corHistPlot,
                       goodCorPlot,
                       badCorPlot, ncol=1, rel_heights = c(0.28,0.48,0.24), labels = c("B","C","D"), label_size = 20)
#pdf("test.pdf", height = 13, width = 18)
plot_grid(leftCol, rightCol, rel_widths = c(0.6, 0.4))
#dev.off()
```

