---
title: "Section 7: Proteomic signature of SF3B1"
author: "Junyan Lu"
date: "2020-10-09"
output: workflowr::wflow_html
editor_options:
  chunk_output_type: inline
---

# Load packages and datasets
```{r, message=FALSE, warning=FALSE}
library(limma)
library(DESeq2)
library(proDA)
library(cowplot)
library(pheatmap)
library(ggbeeswarm)
library(SummarizedExperiment)
library(tidyverse)

#load datasets
load("../data/patMeta_enc.RData")
load("../data/ddsrna_enc.RData")
load("../data/proteomic_explore_enc.RData")
load("../output/deResList.RData") #precalculated differential expression


#protCLL <- protCLL[rowData(protCLL)$uniqueMap,]
source("../code/utils.R")
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE,dev = c("png","pdf"))

```


# Overview of differentially expressed proteins

## A table of associations with 5% FDR
```{r}
resList <- filter(resList, Gene == "SF3B1") %>%
  #mutate(adj.P.Val = adj.P.global) %>% #use IHW corrected P-value
  mutate(Chr = rowData(protCLL[id,])$chromosome_name)
resList %>% filter(adj.P.Val <= 0.05) %>% 
  select(name, Chr,logFC, P.Value, adj.P.Val) %>%
  mutate_if(is.numeric, formatC, digits=2) %>%
  DT::datatable()
```

## Heatmap of differentially expressed proteins (6% FDR)
```{r, fig.height=8, fig.width=10}
proList <- filter(resList, !is.na(name), adj.P.Val < 0.05) %>% distinct(name, .keep_all = TRUE) %>% pull(id)
plotMat <- assays(protCLL)[["QRILC_combat"]][proList,]
rownames(plotMat) <- rowData(protCLL[proList,])$hgnc_symbol

colAnno <- filter(patMeta, Patient.ID %in% colnames(protCLL)) %>%
  select(Patient.ID, SF3B1, IGHV.status) %>% 
  arrange(SF3B1) %>%
  data.frame() %>% column_to_rownames("Patient.ID")
colAnno$SF3B1 <- ifelse(colAnno$SF3B1 %in% 1, "yes","no")

plotMat <- jyluMisc::mscale(plotMat, censor = 5)
plotMat <- plotMat[,rownames(colAnno)]
annoCol <- list(SF3B1 = c(yes = "black",no = "grey80"),
                IGHV.status = c(M = colList[3], U = colList[4]))

pheatmap::pheatmap(plotMat, annotation_col = colAnno, scale = "none", cluster_cols = FALSE,
                   clustering_method = "ward.D2",
                   color = colorRampPalette(c(colList[2],"white",colList[1]))(100),
                   breaks = seq(-5,5, length.out = 101), annotation_colors = annoCol, 
                   show_rownames = TRUE, show_colnames = FALSE,
                   treeheight_row = 0)
```

## Volcano plot

```{r SF3B1_volcano, fig.height=12, fig.width=12}
plotTab <- resList 
nameList <- c("SUGP1")
sf3b1Volcano <- plotVolcano(plotTab, fdrCut =0.05, x_lab="log2FoldChange", posCol = colList[1], negCol = colList[2],
            plotTitle = "SF3B1 (Mutants versus WT)", ifLabel = TRUE, labelList = nameList)
sf3b1Volcano
```


## Boxplot plot of selected proteins

```{r SF3B1_selected_protein, fig.height=4, fig.width=5}
nameList <- c("SUGP1")
protTab <- sumToTidy(protCLL, rowID = "uniprotID", colID = "patID") %>%
  mutate(count = count_combat)
plotTab <- protTab %>% filter(hgnc_symbol %in% nameList) %>%
  mutate(SF3B1 = patMeta[match(patID, patMeta$Patient.ID),]$SF3B1) %>%
  mutate(status = ifelse(SF3B1 %in% 1,"Mutated","WT"),
         name = hgnc_symbol) %>%
  mutate(status = factor(status, levels = c("WT","Mutated")))
pList <- plotBox(plotTab, pValTabel = resList, y_lab = "Protein expression")
sf3b1Box <- cowplot::plot_grid(plotlist= pList, ncol=1)
sf3b1Box
```


# Differential splicing

Processing splicing dataset
```{r,eval=FALSE}
library(DEXSeq)
dxdCLL <- dxdCLL[,dxdCLL$diag %in% "CLL"]
dxdCLL$SF3B1 <- factor(patMeta[match(dxdCLL$patID, patMeta$Patient.ID),]$SF3B1)
dxdCLL$trisomy12 <- factor(patMeta[match(dxdCLL$patID, patMeta$Patient.ID),]$trisomy12)
dxdCLL$IGHV <- factor(patMeta[match(dxdCLL$patID, patMeta$Patient.ID),]$IGHV.status)
dxdCLL.sub <- dxdCLL[rowData(dxdCLL)$symbol %in% filter(resList, adj.P.Val < 0.1)$name, 
                     !is.na(dxdCLL$SF3B1) & !is.na(dxdCLL$trisomy12) & !is.na(dxdCLL$IGHV)]

```


